//code
lastupdate='Sep 26, 2013'
jmolversion='Jmol Version 13.2'
newCmd('.[Jmol command syntax]','','','*v+11.8 -- introduces multi-line free-format commands and brace syntax;math','In general, commands in Jmol start with a command word and continue with a set of parameters separated by white space and terminated by an end of line character or a semicolon. A backslash just prior to the end of a line indicates a line continuation. However, commands can "wrap" to the next line in a more JavaScript-like or Java-like fashion. In general, any unclosed (, [, or {{ on a line indicates that the command continues on to the next line. In addition, with {#.print~} and {#.set~}, lines can continue just after or before a mathematical operator such as +, -, * or /. This results in a more natural line formatting, which has strong similarities to Java and JavaScript. For example:<br /><br />function processInfo(f, i) {{<br />&nbsp;&nbsp;&nbsp;&nbsp;var pdbid = (i ? f[i] : f)<br />&nbsp;&nbsp;&nbsp;&nbsp;load @{{"="+pdbid}}<br />  <br />&nbsp;&nbsp;&nbsp;&nbsp;var molinfo = ({{protein}} == 0 ? "nonprotein"<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;: {{protein and not *.ca}} == 0 ? "alpha carbon model"<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;: {{*}}[0].model != {{*}}[1].model ? "multi-model"<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;: "OK")<br />&nbsp;&nbsp;&nbsp;&nbsp;<br />&nbsp;&nbsp;&nbsp;&nbsp;if (molinfo != "OK") {{<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print pdbid+"\t"+molinfo; <br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return;<br />&nbsp;&nbsp;&nbsp;&nbsp;}}<br /><br />&nbsp;&nbsp;&nbsp;&nbsp;var chaincount = script("select protein; show chain").trim().lines.length<br />&nbsp;&nbsp;&nbsp;&nbsp;var helixcount = script("show structure").lines.find("HELIX").length<br />&nbsp;&nbsp;&nbsp;&nbsp;var sheetcount = script("show structure").lines.find("SHEET").length<br />&nbsp;&nbsp;&nbsp;&nbsp;print pdbid+"\t"+molinfo<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+"\t"+{{*.ca}}.size<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+"\t"+chaincount<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+"\t"+helixcount<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;+"\t"+sheetcount<br />}}<br /><br />Strings in Jmol can be delimited as in JavaScript -- accepting \' \' or " " for all math expressions and most script commands. The exceptions for commands are the five commands that take an implicit string parameter -- that is, everything after the command name is considered a single parameter. These are specifically: {#.echo~}, {#.hover~}, {#.label~}, {#.message~}, and {#.pause~}.','0|1','','')
newCmd('.[Jmol math]','','','*v+11.2 -- new;math','Jmol incorporates a rich JavaScript-like math environment including multiple variable types and a wide variety of {#.functions~}. This section of the documentation details the use of variables and the sorts of operations that are allowed with them. ','0|1','','')
newCmd('','','','*v+11.2;','Variables may be assigned using standard mathematical expressions and used throughout Jmol in virtually any script command to substitute for parameters using the syntax @varName or @{{math expression}}:<br /><br /> <b>x = 0.2; wireframe @x</b> <br /> <b>zoom @{{x * 3}};</b> <br /><br />Variables may be combined with {#.set~"settings"} to adjust parameters: <br /><br /><b>x = bondModeOR; set bondModeOr  !x</b> <br /><br />or<br /><br /> <b>if (x) {{set showBoundBox x}}</b><br /><br />Variables may be used to extract information from a model: <br /><br /><b>x = {{carbon}}.bonds.length.min</b><br /><br />Variables may be used to introduce atom-related property data from external files into the model: <br /><br /><b>x = load("chargeData.txt");select 2.1;data "property_charges @x";select 2.1 and property_charges < 0.5</b><br /><br />Variables may be inspected using <b>show x</b>, <b>show @x</b>,  or <b>message @x</b> where <b>x</b> is a variable name. You can use {#.message~} or {#.echo~} to transmit this information to the user or to the web page via a JavaScript {#.set(callback)~callback} function or use the Jmol.js function jmolEvaluate(). If using the Jmol stand-alone application or the signed applet, you can write a variable to a file using the {#.write~} command. Variables and math expressions can be checked using the {#.print~} command.<br /><br />Standard C++/Java in-place assignments such as <b>x++</b>, <b>x--</b>, <b>++x</b>, <b>--x</b> are allowed, along with assignments  <b>+=</b>, <b>-=</b>, <b>*=</b>, <b>/=</b>, <b>\\=</b> (integer division), <b>|=</b>, and <b>&=</b>.','TEXT','Jmol Variables','')
newCmd('','','','*v+12.0;','Variables in Jmol have one of two scopes -- <b>global</b> or <b>local</b>. Global variables hold values that need to persist across scripts or functions. Local variables override the values previously assigned to the same name for some limited extent. In general, variables defined in Jmol are global variables. A major difference between local and global variables is that only global variables are recorded in the {#.write~state}. It is good programming practice to define variables locally as much as possible, but there are times when global assignments are needed. Just be aware that any global variables created and not destroyed using the {#.reset~} command consume memory and may slow script processing. The following rules govern the scope of local variables: [|| applet localization | all variables in Jmol are localized to a specific applet. Note that this is not necessarily true for {#.functions~}. Specifically, functions beginning with <b>static_</b> are common to all applets. These static functions, though able to be used by any applet, will not share variables. ||script localization <br/><br/>Variables defined using the keyword VAR within a script file that is read by the {#.script~} command are localized to that script and scripts that that script calls.| var x = 30<br />var&nbsp;d&nbsp;=&nbsp;{{atomno=1}}.xyz.distance({{atomno=2}}.xyz) || function localization <br /><br /> The parameters of a function and any variables defined within a function using VAR are local to that function. The Jmol script on the right will print<br /><br /><b>a=1 b=2<br />testing now</b>| a = "testing"; c = "now"<br />function checkX(a, b) {{<br />&nbsp;&nbsp;&nbsp;var c = 15;<br />&nbsp;&nbsp;&nbsp;print "a="+a + " b=" + b<br />}}<br />checkX(1,2)<br />print "" + a + " " + c||FOR/WHILE localization <br /><br /> Variables defined using VAR within the context of a {#.FOR~} or {#.WHILE~} loop will be local to that loop. (Jmol 12.0) The script on the right  will print<br /><b><br />5<br />6<br />7<br />3</b> |<br />x = 3<br />for (var x = 5; x < 8; x++) {{<br />&nbsp;&nbsp;&nbsp;print x<br />}}<br />print x|| {{ ... }} localization <br /><br /> Variables defined using VAR can be localized to a section of a script by bracketing that section of the script with {{ and }}. (Jmol 12.0) This script will print<br /><br /><b>10<br />3</b> |<br />x = 3<br />{{<br />&nbsp;&nbsp;&nbsp;var x = 10;print x<br />}}<br />print x ||]','TEXT','Global and Local Scope','')
newCmd('','','','*v+11.2;','Some variables have preset meanings, as shown in the table below.  A subset of these variables can be used in math expressions. If you create your own variables, their  names must not begin with an underscore. Variables starting with underscore are defined by Jmol and can be used but not set in a script. To check the settings of all available read-only variables, use <b>show _?</b>. Read-only variables include: <br />[||<b>_animating</b>|whether or not Jmol is currently running through the frames as a result of <b>animation on</b> or <b>animation play</b> (true or false)||<b>_animTimeSec</b>| The time in seconds required for an animation to complete one cycle. (Jmol 12.3.10) ||<b>_applet</b>|whether or not Jmol is running as an applet||<b>_atomHovered</b>|the overall atom index of the atom that was most recently hovered over (or -1)||<b>_atomPicked</b>|the overall atom index of the atom that was most recently picked (or -1). Can be used, for example, with <b>select atomIndex = _atomPicked</b>||<b>_currentFileNumber</b>|the number of the currently displayed file, starting with 1 (value will be 0 if more than one file is displayed) ||<b>_currentModelNumberInFile</b>|the number of the currently displayed model in its file (or 0 if more than one model is displayed) ||<b>_depthPlane</b>|The plane defining the back limit of the model for the current orientation and slabbing. See also <b>_slabPlane</b>||<b>_fileType</b>|The file type of the most recently loaded file||<b>_firstFrame</b>|The first frame in the current animation frame range expressed in x.y notation ||<b>_frameID</b>| A numerical frame number equal to 1000000 times the file number plus the model index in the file||<b>_height</b>|the height of the applet or application window in pixels ||<b>_lastFrame|The last frame in the current animation frame range expressed in x.y notation||<b>_loadPoint</b>| a number indicating the load status, usually 3 indicating successful load, possibly 0 indicating load failure. ||<b>_memory</b>|the amount of memory allocated to the applet or application ||<b>_modelFile</b>|the filename of the model (or "" if more than one model is displayed)  ||<b>_modelName</b>|the name of the model (or "" if more than one model is displayed)  ||<b>_modelNumber</b>|the current model number <b>as a string</b> in file.model notation (or "file1.model1 - file2.model2" if more than one model is currently displayed)  ||<b>_modelTitle</b>|information from the file reader interpreted as a title||<b>_mouseAction</b>| a number associated with the current mouse/key state ||<b>_mouseModifiers</b>| a number associated with the current key combination associated with the mouse ||<b>_mouseX</b>| the current X-position of the mouse ||<b>_mouseY</b>| the current Y-position of the mouse ||<b>_multiTouchClient</b>|indicates if Jmol is operating as a Sparsh-UI client and has connected with the Sparsh-UI server (possibly itself).||<b> _multiTouchServer</b>| indicates whether Jmol is functioning as a {http://code.google.com/p/sparsh-ui/~Sparsh-UI} server. (Requires a specialized Jmol Sparsh-UI driver) and has successfully connected to a multi-touch device. (Jmol 12.0)||<b>_navigating</b>| TRUE if navigation is on</b>||<b>_nProcessors</b>|the number of available processors||<b>_pickInfo</b>|information about the last atom picked, including a description of the atom, its atom number, and its x y z coordinates. For example: <br /><b>[GLN]25:A.O/2.1 #175 40.271 8.524 2.615</b> ||<b>_signedApplet</b>|whether or not Jmol is running as an signed applet||<b>_slabPlane</b>|The plane defining the front limit of the model for the current orientation and slabbing. See also <b>_depthPlane</b>||<b>_spinning</b>|whether or not the model is currently spinning (true or false). The _spinning variable can be used, for example, to toggle spinning on and off: <br /><b>if (_spinning);spin off;else;spin on;endif;</b> ||<b>_useCommandThread</b>|TRUE if Jmol is using an independent command thread.||<b>_version</b>|the version of Jmol expressed as an integer: vvrrxxx for Jmol vv.rr.xxx. For example, for Jmol 11.2.38, _version = 1102038||<b>_width</b>|the width of the applet or application window in pixels ||]','TEXT','Read-Only Variables','')
newCmd('','','','*v+11.4;','Jmol math allows for several distinct variable types, some of which are common types (boolean, integer, decimal, string, serial array, associative array), and some of which are special types of particular use in molecular calculations (point, plane, quaternion, 3x3 matrix, 4x4 matrix, atom bitset, bond bitset). Array types may include any number of any combination of these data types. Examples include: <br /><br />[||array| xlist = array(true,3,3.5,"testing",{{2.3,3.4,4.5}});<br />xlist = [true,3,3.5,"testing",{{2.3, 3.4, 4.5}}]<br /><br />Arrays may contain a mix of any variable type. Arrays are stored, saved, and manipulable as objects, and all array assignments are by reference, as for JavaScript and Java, meaning  that if x = ["a", "b", [3, 4, 5]], and we assign y = x[3], then y <i>is</i> x[3], not a copy of it, and if we later assign y[4] = "new", then at x[3] = ["a", "b", [3, 4, 5], "new"] as well.||associative array|    b = {{"test" : 34, "test2" : 45}}<br />   b["test"] = 35<br />   b["test3"] =["one", 3, 4, 5]<br />   print b["test3"][1]<br /><br /> one<br /><br />   show b<br /><br /> b = {{ "test3":["one", 3, 4, 5],"test2":45,"test":35 }}<br /> <br />   b -= "test2"<br /> show b<br /> <br />  b = {{ "test3":["one", 3, 4, 5],"test":35 }}<br /><br />Associative arrays store information retrievable by string-based keys. The information may be any type, including another associative array. When an element is retrieved, a copy of that array element is retrieved.   ||atom bitset| x = {{atomno < 30}};<br /> x = ({{0:28 45  56:62}})||bond bitset| x = {{atomno < 30}}.bonds;<br /> x = [{{0:4 6 9:12}}]|| boolean |isOK = TRUE; if(isOK);print "all is good";end if;||decimal|x = {{atomno < 10}}.x * 10.0; || integer| for(var i = 1; i < 10; i = i + 1); print i;end for;|| 3x3 matrix|m = [[1.0,2.0,3.0],[4.0,5.0,6.0],[7.0,8.0,9.0]];<br />m = axisAngle({{1,0,0}},30)%-9.|| 4x4 matrix|m = [[1.0,2.0,3.0,4.0],[4.0,5.0,6.0,7.0],[7.0,8.0,9.0, 10.0]];.||point | pt = {{3.2,3.3,3.4}};|| plane | xyPlane = {{0 0 1 0}};<br /><br /> Planes are defined as {{a b c d}}, where the equation of the plane is <b>ax + by + cz + d = 0</b>.||quaternion | q = quaternion({{1 0 0}}, 30);<br /> q = {{0.25881904 0.0 0.0 0.9659258}};<br /><br />(see next section) Quaternions are saved internally in Jmol using the same format as for planes, as single-precision floating point four-element vectors, with parameter order {{x y z w}} or {{q1 q2 q3 q0}}. Ordering the parameters in this way is consistent with {http://java.sun.com/javase/technologies/desktop/java3d/forDevelopers/J3D_1_3_API/j3dapi/javax/vecmath/Quat4f.html~Java Quat4f} format and allows both quaternions and planes to contain axis information in the first three parameters. The common storage format for planes and quaternions works because typical quaternion operations are not common to operations involving planes.||string| myLabel = "this is a test"; <br /><br />Strings may be enclosed in either single or double quotes.||]<br /><br />Variable types may be combined in mathematical expressions. In general, expressions are evaluated from left to right using standard operator precedence (*,/,\\ before +/-; +/- before AND/OR/NOT). ','TEXT','Jmol Math Variable Types','')
newCmd('','','','*v+11.4;','All quaternions in Jmol are <b>unit</b> quaternions, which are four-dimensional vectors that can be used to define the relative rotational aspects of a protein or nucleic acid structure as well as overall molecular orientation. This means that they can be used in a variety of commands, including {#.navigate~}, {#.moveto~}, and {#.rotate~} to rotate the model or selected atoms of the model around specific axes and to specific orientations. Each quaternion can be thought of as representing a unique axis and angle which will transform a reference frame (either the molecular reference frame or the window reference plane to a given orientation. If <b>n</b> is the axis and <b>theta</b> is the angle (measured in a right-handed, counter-clockwise direction), then <b>q</b>(<b>theta</b>/2, <b>n</b>) = (cos(<b>theta</b>/2), <b>n</b> sin(<b>theta</b>/2)). For storage, the vector <b>n</b> is broken out into its x, y, z components, giving a total of four numbers, (q0, q1, q2, q3), where q0 = cos(<b>theta</b>/2), q1 = <b>n<sub>x</sub></b> sin(<b>theta</b>/2), q2 = <b>n<sub>y</sub></b> sin(<b>theta</b>/2), and q3 = <b>n<sub>z</sub></b> sin(<b>theta</b>/2). Jmol reports a quaternion as a four-vector with q0 listed last: {{q1, q2, q3, q0}} in order to be consistent with Java\'s {{x, y, z, w}} notation. <br /><br />The {#.functionsx=fyfunctions~quaternion()} function constructs quaternions using a variety of starting points, including the four numbers q0-q3, <b>quaternion(q0, q1, q2, q3)</b>, axis-angle information, <b>quaternion({{0 0 01}}, 30)</b>, and 3x3 rotation matrices, <b>quaternion(mat)</b>. In addition, quaternions representing the orientation of specific amino acid residues and nucleic acid bases can be constructed automatically based on the setting of the Jmol parameter {#.set (misc)~quaternionFrame}, and certain commands and functions such as {#.show~show rotation} and <b>script("show rotation")</b> deliver quaternions.  These quaternion values can be depicted visually using the {#.draw~} and {#.plot~} commands and can be listed or saved to a file using the {#.show~} and {#.write~} commands, respectively. <br /><br />An interesting feature of quaternions that makes them different from the more common 3x3 rotation matrices is that they can encode rotations up to 720 degrees. For example, quaternion({{0 0 1}}, 30) = {{0.0 0.0 0.25881904 0.9659258}} (a 30-degree CCW rotation around the Z axis), while quaternion({{0 0 1}}, 390) = {{0.0 0.0 -0.25881904 -0.9659258}} (because sin(195) and cos(195) are both negative). While rotations of 0 degrees, {{0 0 1 1}}, and 360 degrees, {{0 0 -1 -1}} represent the same final state, in certain cases they can represent different <i>processes</i> and can thus be significantly different. In addition, quaternion differences (or "derivatives") can be quantified in ways that cannot be done for 3x3 rotation matrices. For example, the mean and standard deviation of a set of quaternions can be determined: <b>print [a,b,c,d,e].average</b>, <b>print [a,b,c,d,e].stddev</b>, where a-e are quaternions, providing, for example, the average helical axis for a protein helix structure, or a measure of how "ideal" that helix is.  <br /><br />The quaternion <b>q = quaternion(script("show rotation"))</b> (or, in Jmol 12.2, just <b>q = quaternion()</b>) measures how much the model has been rotated from its reference position (with <b>x</b> to the right, <b>y</b> to the top, and <b>z</b> toward the user). It can be used to align any axis or plane with any other "window-frame" axis. For example, here is a quaternion recipe for the shortest rotation that takes atom "X" and puts it between the center of the model and the user,  that is, "in the forefront:"<br /><br /><b>q = quaternion(script("show rotation"))<br />toAtom = {{C1}}.xyz - {{*}}.xyz<br />toUser = (!q)%{{0 0 1}}<br />axis = cross(toAtom, toUser)<br />a =  angle(toAtom, {{0 0 0}}, axis, toUser)<br />rotate MOLECULAR {{0 0 0}} @axis @a</b><br /><br />The idea here is to apply the <i>reverse</i> model rotation quaternion to the z axis, {{0 0 1}} to give a vector that is pointing straight at the user. Then we get an axis that is perpendicular to both that and the vector we want to assume this position. The directed angle around that axis is obtained as a dihedral angle relating our atom, the origin, that axis, and the point in space directed toward the user. Then we rotate in a MOLECULAR frame to get the desired effect.<br /><br />For another example, let\'s line up the plane through atoms C1, C5, and C8 such that, relative to the user, C1 is directly to the right of C5, and C8 is in the plane of the screen, above the other two atoms. This is trivial using quaternions:<b><br /><br />x = quaternion({{c5}}, {{C1}}, {{c8}})<br />q = quaternion(script("show rotation"))<br />rotate molecular quaternion @{{(!q) * (!x)}}</b><br /><br />This basically says, "Move this plane into the position currently occupied by the xy plane (!x), then move that plane back to the original position of the xy plane (!q)."','TEXT','Jmol Quaternion Math','')
newCmd('','','','*v+11.4;','Jmol expressions can include standard operators: +, -, *, /, **(exponentiation, Jmol 12.0) %(modulus), and (&, &&, and), or (&brvbar;, &brvbar;&brvbar;, or), not (!, not), and all standard comparison operators. For example:  <br /><br /><b>twoPI = 3.14159 * 2;<br />minBondDistance = minBondDistance * 0.5;</b><br /><br /> In addition, you can set variables to be the number of atoms that match an atom expression. For example: <br /><br /><b>nNC = {{_N and */1 and connected(_C)}}.size<br />nAtoms = {{*}}.size;<br />nCH = {{_H}} + {{_C}};</b><br /><br />Since Jmol math does not include strict typecasting, it uses a relatively complex set of rules to determine the result of operations on mixed variable types. When two different variables are operated upon, the resulting variable type depends upon the operator used, the order of the variables, and sometimes the value of the variables. See {misc/operations.txt~misc/operations.txt} for details. In general, when conversion is required for a string, point, plane, bitset, or array, Jmol will attempt to convert it to a variable type compatible with the left-hand operand prior to operating. These conversions generally involve:<br />[||string            |to boolean|The strings "FALSE", "0", and decimal strings such as "0E1" and "0.0" that equal 0 are converted to FALSE; all other strings are converted to TRUE.||string|to integer|A string evaluating to an integer is converted to that integer; all other strings are converted to 0.||string|to decimal|A string evaluating to a number is converted to that number; all other strings are converted to the decimal value "not a number", or "NaN".||string|to other|Jmol saves all states as simple strings, so certain character sequences are automatically turned back into other variable types when operated upon. Jmol automatically converts:<br/>[||{{x y z}}| to a point ||{{x y z w}}| to a plane or quaternion||({{i j k l:m ... }})| to an atom bitset ||[{{i j k l:m ... }}]| to a bond bitset||]||point|to integer|The distance from the point to {{0 0 0}} rounded DOWN to the nearest integer. Note that this allows rounding a positive decimal number x down to the nearest integer (a "floor" operation) using <b>0 + point(x, 0, 0)</b>.||point|to decimal|The distance from the point to {{0 0 0}}; same as x.distance({{0 0 0}})||plane|to integer|The distance from the plane to {{0 0 0}} rounded DOWN to the nearest integer||plane|to decimal|The distance from the plane to {{0 0 0}}; same as x.distance({{0 0 0}})||quaternion|to decimal|cotangent(abs(theta)/2), where theta is the angle of the rotation associated with this quaternion ||bitset|to decimal or integer|The number of selected atoms or bonds in the bitset; same as x.size||array|to decimal or integer|The number of elements in the array; same as x.size||]','TEXT','Operators and Operands','')
newCmd('','','','*v+11.4;','Rules for operations with the given types include:<br />[||             <br />+|addition|a + b produces a decimal number EXCEPT:<br />[||array + b|array|b added to end of a ||a + array|array|b added to beginning of a ||integer + b|integer| unless b is a decimal or an array ||x + plane or quaternion|varies|{{x y z}} are extracted from the plane or quaternion and then added to x ||matrix3x3 + matrix3x3 | matrix3x3| sum of individual elements of the two matrices||matrix3x3 + point | matrix4x4| adds a translation vector to a rotation matrix to give a 4x4 matrix||quaternion + (decimal)x|quaternion| addition of x to the angle of rotation associated with this quaternion||string + b|string| unless b is an array ||point + b|point| unless b is an array||]||                                                            <br />                  -|subtraction|a - b produces a decimal number EXCEPT:<br />[||associative array - x| associative arrray|removes key "x" from the associative array||integer - b|integer| unless b is a decimal ||matrix3x3 - matrix3x3 | matrix3x3| difference of individual elements of the two matrices||x - plane or quaternion|varies|{{x y z}} are extracted from the plane or quaternion and then subtracted from x ||quaternion - (decimal)x|quaternion| subtraction of x to the angle of rotation associated with this quaternion||string - integer|integer| string a is converted to integer, then b is subracted ||point - b|point| subtraction of {{b b b}} from point a||]||                         <br />*|multiplication|a * b produces a decimal number EXCEPT:<br />[||integer * b|integer | unless b is a decimal ||point * b|point | unless b is a point (dot product, producing a decimal)||quaternion * quaternion|quaternion|quaternion multiplication q2 * q1 results in a composite rotation resulting from first rotating by q1, then by q2.||quaternion * (decimal)x|quaternion| multiplication of the angle associated with this quaternion by x ||]||                  <br />                       /|division|a / b produces a decimal number EXCEPT:<br />[||integer / integer|integer ||point / b|point| if b is a point, then a is scaled by the magnitude of b; thus <b>a/a</b> when a is a point produces a normalized vector in the direction from {{0 0 0}} to point a||quaternion / (decimal)x| quaternion| division of the angle associated with this quaternion by x ||quaternion / quaternion|quaternion| q2 / q1 = q2 * q1<sup>-1</sup> (absolute difference in rotation, in the molecular frame)||]||                  <br />                       \\|left division|a \\ b produces integer division EXCEPT:<br />[||quaternion \\ quaternion|quaternion| q1 \\ q2 = q1<sup>-1</sup> * q2 (relative difference in rotation, in the q1 frame)||]||                  <br />                       **|exponentiation|a**b takes a to the power of b; if both a and b are integers, the result is an integer, otherwise the result is a decimal number.||                    <br />                  %|modulus|a % b is fully defined only for integer b and produces an integer EXCEPT:<br />[||decimal % 0|integer | decimal a rounded to nearest integer, with n.5 rounding to (n + 1) and -n.5 rounding to -(n + 1)||decimal % b|string | decimal a rounded to b digits after the decimal point when b > 0; decimal a rounded to b significant digits in scientific notation when b < 0||matrix4x4 % 1| matrix3x3 | extract the 3x3 rotation matrix from a 4x4 rotation/translation matrix||matrix4x4 % 2|point| extract the translational vector from a 4x4 rotation/translation matrix||quaternion % 0|decimal|extract q0||quaternion % 1|decimal|extract q1||quaternion % 2|decimal|extract q2||quaternion % 3|decimal|extract q3||quaternion (or plane) % 4|point|plane normal or quaternion axis; in the case of a plane, the vector from this point to {{0 0 0}} is directed toward the plane; for quaternions, the axis is defined such that the angle would be between -180 and 180 degrees. ||quaternion % -1|point|extract the rotational axis ({{q1 q2 q3}} or {{x y z}}) as a point (vector from {{0 0 0}})||quaternion % -2|decimal|extract the angle in degrees for the rotation associated with this quaternion||quaternion % -3|point|extract the first column of the rotation matrix associated with this quaternion (what {{1 0 0}} is transformed to)||quaternion % -4|point|extract the second column of the rotation matrix associated with this quaternion (what {{0 1 0}} is transformed to)||quaternion % -5|point|extract the third column of the rotation matrix associated with this quaternion (what {{0 0 1}} is transformed to)||quaternion % {{x y z}}|point|transform {{x y z}} by the rotation associated with this quaternion||string % b|string| when b > 0, right-justified in a field b characters wide; when b < 0, left-justified in a field b characters wide. Special values 9999 and -9999 return upper case and lower case, respectively. b = 0 trims the string of white space. (Jmol 12.2)||point % b|point|generates the unitcell coordinate corresponding to the point, offset from {{0 0 0}} by {{b/1 b/1 b/1}}. ||bitset % b|bitset|a truncated to first b items; same as a[1][b]||array % b|array|each element treated as a string and justified individually||]||                           <br />&amp;&amp;&nbsp;&brvbar;&brvbar;&nbsp;! |AND/OR/NOT|In Jmol versions before 12.0, both sides of all AND/OR operations are evaluated prior to checking. Starting with Jmol 12.0, in a logical AND operation (as opposed to a bitset AND operation) if the operand on the left is FALSE, the operand on the right will not be evaluated; likewise for logical OR -- if the left operand evaluates to TRUE, then the right operand is not evaluated. This is the same as in most programming languages, including JavaScript and Java. The following are equivalent: [|| AND | &amp; | &amp;&amp;||OR | &brvbar;&brvbar;| &brvbar;||NOT | &nbsp;!&nbsp;||] <br />a AND/OR b as well as NOT b produce a boolean except for the following cases:<br />[||array OR x | array | adds the element <b>x</b> to the array. If x itself is an array, the operation appends the elements of x as individual elements, concatenating the two arrays. To append one array onto another as a single element, simply surround the array being added with brackets: <b>a = [1,2,3]; a &brvbar;= [4, 5 6]</b> leaves <b>a</b> as <b>[1,2,3,4,5,6]</b>;   <b>a = [1,2,3]; a &brvbar;= [ [4, 5 6] ]</b> leaves <b>a</b> as <b>[1,2,3,[4,5,6] ]</b>. (Jmol 12.2) || bitset AND bitset|bitset| the intersection of the two bitsets ||bitset OR bitset|bitset| including all selections from both bitsets ||NOT bitset|bitest|the inverse of the bitset, based on the total atom count for an atom bitset or the total bond count for a bond bitset||!q|quaternion|quaternion inverse, {{-x -y -z w}}||]||                         <br />==&nbsp;&nbsp;!=|equal/not equal |== (or just "=") and != generally convert values to decimal values and then test these values are within 1E-6 of each other. The following cases are exceptions: <br />[||point == point| true if the distance between the points is less than 1E-6. ||plane == plane </ br>quaternion == quaternion| true if the four-vector distance between the quaternions or planes is less than 1E-6. ||]||]','TEXT','Operation Rules','')
newCmd('.[Jmol parameters]','','','*v+11.2 -- new;math','Many parameters in Jmol can be {#.set~} and may also be checked using Jmol math. ','0|1','','')
newCmd('','','','*v+11.2;','TABLE1','TEXT','General Parameters','')
newCmd('','','','*v+11.2;','TABLE2','TEXT','Set-Only Parameters','')
newCmd('','','','*v+11.2;','TABLE3','TEXT','Deprecated Parameters','')
newCmd('','','','*v+11.2;','TABLE4','TEXT','Reserved Names','')
newCmd('.[using the clipboard]',' ','','*v+11.0 -- new;clip','Model data that has been clipped from local sources, such as a return from {http://www.rcsb.org/pdb/files/1crn.pdb~the RCSB PDB file directory}, can be loaded into Jmol directly. From the application, simply use Edit…Paste; if using the applet, simply right-click on the applet and select Show...Console. Then paste the data into the lower (input) frame and click <b>Load</b>.','1|2','','')
newCmd('.[Jmol SMARTS/SMILES]','','','*v+12.0 -- new;','Jmol 12.0 supports a full implementation of {http://www.daylight.com/dayhtml/doc/theory/theory.smiles.html~SMILES} and {http://www.daylight.com/dayhtml/doc/theory/theory.smarts.html~SMARTS} along with powerful extensions (see {http://jmol.svn.sourceforge.net/viewvc/jmol/trunk/Jmol/src/org/jmol/smiles/package.html~}) that allow the SMARTS syntax to be used for biomolecular substructure searching and three-dimensional conformation searching. The bioSMILES, bioSMARTS, and 3D-SMARTS syntaxes are relatively simple extensions of the Daylight definitions, including all stereochemistry and "primitive" syntax in those definitions. Importantly, Jmol can search SMILES strings themselves (independent of any loaded structure) and find matches. This feature allows, for example, checking of student answers to questions that require the entry of structures using 2D drawing programs such as JSDraw and JME. Examples of using SMILES and SMARTS include:<br />[||<b>select search("[r4]")</b>| search for all atoms in 4-membered rings||<b>x = {{*}}.find("SMARTS","{{C}}C=O")</b>|load the variable x with all alpha-carbons.||<b>x = {{1.1}}.find("[!H](.t:-55,-65)CC[!H] &brvbar;&brvbar; [!H](.t:55,65)CC[!H]", true)</b>|fill array x with all non-hydrogen "gauche" interactions (torsions between -55 and -65 degrees or between 55 and 65 degrees). ||<b>x = {{*}}.find("@TILDEr@TILDE{{*:1}}[$(*())2-4]{{*:1}}", true)</b>|create an array variable x that lists all RNA (@TILDEr@TILDE) loops ( {{*:1}}...{{*:1}} ) that involve from 2 to 4 un-paired bases ( [$(*())2-4] ).||<b>select 1.1; show SMILES</b>|Show the SMILES string for model 1.1; same as <b>print {{1.1}}.find("SMILES")</b>. Note that when creating a SMILES string from a 3D model, Jmol may "over-express" chirality centers. The SMILES algorithm identifies <i>local</i> stereochemistry; it does not run down two paths to see if they are the same or not. Jmol will check only for the following, and only for carbon: (a) two atoms that are identical and have only one connection (H, F, Cl, Br, for example) and (b) two CH3 groups. Thus, for example, Jmol 12.0 will not tag an isopropyl group as R[C@H](C)C, but it will tag RCHEt<sub>2</sub> as R[C@H](CC)CC. This is intentional.||<b>print {{*:A}}.find("SMILES",true);</b>| display the sequence for chain A in bioSMILES format:<pre>//* Jmol bioSMILES 12.0.RC25   2010-07-14 15:28 1 *//<br />//* chain A protein 1 *// @TILDEp@TILDEGRRIQGQRRGRGTSTFRAPSHRYKADLEHRKVEDGDV //* 37 *//</pre>||<b>print {{selected}}.find("SEQUENCE")</b>|returns the Jmol bioSMILES sequence for the specified atoms (same as .find("SMILES",true), but without the header comment. Adding an the optional second parameter TRUE adds crosslinking:<pre>//* chain A protein 1 *//@TILDEp@TILDETTC:1C:2PSIVARSNFNVC:3RLPGTPEAIC:3ATYTGC:2IIIPGA<br />  TC:1PGDYAN //* 46 *//</pre>||<b>jmolEvaluate("\'" + stringAnswer + "\'.find("\'SMILES\',\'" + stringKey + "\') > 0")|JavaScript call to a Jmol applet to find out if a given SMILES string that a student has entered using a 2D drawing object matches the key. If required stereochemistry indicated in the key is missing in the answer, the result will be FALSE. If the student\'s answer has unnecessary stereochemistry indicated, the result will be TRUE. ||] ','1|2','','')
newCmd('.[Jmol/JSpecView/MagresView]','12.3.16','','*v+13.0;load;sync','Jmol 13.0 <image src="img/Jmol-JspecView.png" width="400" height="250" align="right" />marks a new  collaboration with the {jspecview.sourceforge.net~JSpecView project} with the incorporation of JSpecView into the Jmol application and synchronized communication between the Jmol applet and the JSpecView applet. This is most evident in that there is a new menu  item Tools...JSpecView, which opens a new JSpecView frame that includes the current model. Work is in progress in the design and implementation of a new JCAMP-DX format that incorporates models and assignments  into JDX files using custom ##$MODELS and ##$PEAKS records. See {misc/Jmol-JSpecView-specs.pdf~Jmol-JSpecView-specs.pdf} for proposed specification. Progress can be monitored at {http://chemapps.stolaf.edu/jmol/jspecview~}. The Jmol application command <b>SYNC ON; sync * "JSpecView:..."</b> sends commands to JSpecView. <br /><br />Starting with Jmol 13.4, the Jmol application can access simulated NMR specta using the {#.show_NMR~show NMR} command and display them along with their correlations. <br /><br />Also starting with Jmol 13.4, functionality is added for interacting with {http://www.ccpnc.ac.uk/pmwiki.php/CCPNC/MagresView~MagresView} in relation to calculation and visualization of solid-state NMR constants: [||{{*}}.ms and %[ms] (for labels) | magnetic shielding tensor isotropic values||{{*}}.cs and %[cs] (for labels) | chemical shift tensor isotropic values||set shift_XX xxx | sets chemical shift offset, in ppm||getProperty nmrInfo | delivers the table of isotopes|| print getproperty("nmrinfo.isotopes")["2H"] | an array for deuterium including [n,dc,qc] where n is the isotope number, (negative if the most abundant isotope), dc is the dipolar coupling constant, and qc is the quadrupolar coupling constant.||measure({{a}} {{b}} "isc_hz") | function returning Magres J-coupling value||measure({{a}} {{b}} "dc_khz") | function returning Magres dipolar constant|| MEASURE {{a}} {{b}} "2://dc_hz" | measurement line showing dipolar constant, Hz||  MEASURE {{a}} {{b}} "2:%3.2VALUE//dc_khz" | measurement line showing dipolar constant, kHz||  MEASURE {{a}} {{b}} "2://khz" | measurement defaults to dc_khz||  MEASURE {{a}} {{b}} "2://hz" | measurement defaults to isc_hz||  MEASURE {{a}} {{b}} "2://isc_1hz" | measurement line showing specifically "isc_1" values in Magres file|| {{xxx}}.tensor(type,what) | full access to tensor quantities, including j, chi, dc, eigenvalues, eigenvectors, value, asymMatrix, symMatrix, isotropy, anisotropy, asymmetry, eulerzyz, euelerzxz, quaternion, indices, type id, span, and skew.||]','0|1','','')
newCmd('.animation',' ','','*v+11.0 -- adds greatly expanded frame animation control;anim;image','Sets selected animation parameters or turns animation on or off. Note that there are four distinct animation types that can be employed using Jmol: (1) files may contain multiple structures that are "played" sequencially, (2) models may contain vibrational modes that can be animated, (3) certain Jmol script commands (namely {#.move~}, {#.moveTo~}, {#.navigate~}, {#.restore~restore ORIENTATION} and {#.zoomTo~}), can create the illusion of motion over a period of time, and (4) Jmol scripts can be run through in a predefined way, involving {#.loop~} and {#.delay~}. The "animation" command only refers to method (1).  See also {#.set(misc)~set backgroundModel}.','1|2|4','','')
newCmd('',' ','','*v+11.0;','Turns on or off animation. With <b>animation ON</b>, the {#.frame~} range is also reset to all frames. (An implicit <b>frame range ALL</b> command is executed. This functionality is for backward compatibility with Chime.) If this resetting is not desired because the frame range has been set, then <b>frame PLAY</b> or <b>frame PLAYREV</b> should be used instead of <b>animation ON</b>.','*10','._on_off{"ON"}','')
newCmd('',' ','','*v+11.0;','Sets the animation direction to be from last frame to first frame.','*21','.direction','-1')
newCmd('','13.1.13','','*v+13.2;','Applies a filter to a running animation to display only a certain set of atoms.','*30','DISPLAY','{{atom set}}')
newCmd('',' ','','*v+11.0;','Sets the animation direction to be first frame to last frame.','*20','.direction','+1')
newCmd('',' ','','*v+11.0;','Sets the animation frames per second (maximum 50).','*40','.fps','._anim_fps')
newCmd('',' ','','*v+11.0;','<b>animation frame</b>, <b>frame</b>, and <b>model</b> are synonymous. See options at the {#frame~} command. ','*50','.frame','')
newCmd('',' ','','*v+11.0;','Sets the animation mode to restart the sequence automatically when the last frame has played.','*90','.mode','LOOP')
newCmd('',' ','','*v+11.0;','Allows for a time delay at the start and end of the loop.','*91','','LOOP','._time_delay1','._time_delay2')
newCmd('',' ','','*v+11.0;','Sets the animation to play once through and then stop (the default mode).','*92','','ONCE')
newCmd('',' ','','*v+11.0;','Sets the animation to play forward and back endlessly.','*93','','PALINDROME')
newCmd('',' ','','*v+11.0;','Allows for a time delay at the start and end of the palindrome.','*94','','PALINDROME','._time_delay1','._time_delay2')
newCmd('','13.1.13','','*v+13.2;','The <b>animation MORPH</b> option requires a previous LOAD TRAJECTORY. It creates a linear morph consisting of the designated number of representations inserted between trajectories. For example, <b>load test.pse;animatino morph 3</b>. It can also be used between just two models: <b>load trajectory "test1.pdb" "test2.pdb"; animation morph 32</b>. ','*95','MORPH','(integer)')
newCmd('.[atom expressions]','','','*v+11.8;atoms;math;prop','An increasing number of commands, including {#.center~}, {#.connect~}, {#.define~}, {#.dipole~}, {#.display~}, {#.draw~}, {#.isosurface~}, {#.measure~}, {#.polyhedra~}, {#.restrict~}, and {#.select~} take for parameters one or more expressions that represent collections of atoms in one or more models. All terms can be preceeded by the keyword NOT and joined by AND, OR, or XOR. [||general terms|all, bonded, clickable, none, selected, visible||file.model|as, for example, <b>select 3.2</b>, a specific model in a specific file. Note that <b>select 3.0</b> selects all atoms in all models of the third file of the most recent {#.load~}. ||subset|the currently defined {#.subset~}. Note that if a subset is currently defined, then <b>select/display all</b> is the same as <b>select/display subset</b>, <b>restrict none</b> is the same as <b>restrict not subset</b>. In addition, <b>select not subset</b> selects nothing. ||unitcell|atoms within the current {#.unitcell~}, which may be offset. This includes atoms on the faces and at the vertices of the unitcell. ||chemical elements|element_name (including "deuterium and tritium"), _Xx (an element symbol preceeded by underscore, possibly with isotope number listed, such as _Cu, _Fe, _2H, _31P)||isaromatic|atoms connected with the AROMATIC, AROMATICSINGLE, or AROMATICDOUBLE bond types||non-protein groups|carbohydrate, dna, hetero, ions (specifically the PDB designations "PO4" and "SO4"), ligand (originally "hetero and not solvent"; changed to "!(protein,nucleic,water,UREA)" for Jmol 12.2), nucleic, purine, pyrimidine, rna, sidechain||protein residues|[||acidic|ASP, GLU||acyclic| amino and not cyclic||aliphatic|ALA GLY ILE LEU VAL||amino|all twenty standard amino acids, plus ASX, GLX, UNK||aromatic|HIS PHE TRP TYR (see also "isaromatic" for aromatic bonds)||basic|ARG, HIS, LYS||buried| ALA CYS ILE LEU MET PHE TRP VAL||charged|acidic or basic||cyclic|HIS PHE PRO TRP TYR||helix, helixalpha, helix310, helixpi|secondary structure-related. (Note that helixalpha, helix310, and helixpi are subtypes of helix introduced in Jmol 12.0.)||hetero|PDB atoms designated as HETATM||hydrophobic| ALA GLY ILE LEU MET PHE PRO TRP TYR VAL||large |ARG GLU GLN HIS ILE LEU LYS MET PHE TRP TYR||medium|ASN ASP CYS PRO THR VAL||negative|acidic||neutral| amino and not (acidic or basic)||polar| amino and not hydrophobic||positive|basic||protein| defined as a group that contains PDB atom designations C, N, and CA||sheet|secondary structure-related||small|ALA GLY SER||surface| amino and not buried||turn|secondary structure-related||]||protein-related|alpha, backbone, base, mainchain, sidechain (backbone and mainchain are synonymous), spine (*.CA, *.N, *.C for proteins; *.P, *.O3*, *.O5*, *.C3*, *.C4*, *.C5 for nucleic acids; Jmol 12.0)||solvent-related|solvent, PDB "HOH", water, also the connected set of H-O-H in any model||]<br/>The comparison operators &lt;, &lt;=, =, &gt;, &gt;=, and != operate with may keywords. These are summarized under {#.atomproperties|atom properties}.','0','','')
newCmd('','11.1.14','','*v+11.2;','An atom expression is simply a list of atoms. You can select a single atom or a range of atoms from an atom expression. The way to do this is simply to suround the atom expression with parentheses and follow it with one or two numbers in brackets: <b>select (carbon)[3][5]</b>.This says, "Select the third through fifth carbon atoms." If the second selector is not present, then only a single atom is selected; the selector [0] indicates the last atom in the set, and negative numbers count back from that atom. Thus, <b>select (*)[0]</b> selects the last atom, and <b>select (carbon and 2.3)[-1][0]</b> selects the last two carbon atoms in model 2.3. Atom selectors can be used for any expression embedded in another command. In that case an additional set of parentheses or braces is required around the whole expression: <b>measure {{(_O)[1]}}  {{(_O)[2]}}</b>.','TEXT','Atom selectors','')
newCmd('','11.1.12','','*v+11.0;','The following functions are also supported.[||CONNECTED()| allows for selection of specific atoms based on their connectivity to other atoms. The general format is: <b>connected([optional min # bonds], [optional max # bonds], [optional bond type], [optional atom expression])</b> Bond type may be any described for {#.connect~}. See {misc/groups.txt~groups.txt} for many examples of using connected() with {#.define~} || SUBSTRUCTURE()| atoms within a given substructure of the model. The <b>substructure()</b> function takes a quoted {http://www.daylight.com/smiles~smiles string} for its argument. For Jmol 12.0, see also the math function <b>within("SMILES",...)</b>. (Note: aromatic ring SMILES were not supported until Jmol 12.0.) ||WITHIN(setName,atomExpression)|any atom within a given set. The setName can be any one of the words <b>BOUNDBOX</b>, <b>CHAIN</b>, <b>ELEMENT</b>, <b>GROUP</b>, <b>MODEL</b>, <b>MOLECULE</b>, <b>POLYMER</b> (Jmol 12.0), <b>SITE</b>, or <b>STRUCTURE</b>, or it can be a protein or nucleic acid sequence expressed in single-letter notation surrounded by quotation marks as, for example, "GGCCCTT" or "MAACYXV" (in which case the sequence is found within the expression).  (SITE refers to all crystallographic sites common to the specified atom set; BOUNDBOX refers to the smallest box containing the atom set.) Additional options, including "BASEPAIR", "SMILES", and "SMARTS", are discussed below.||WITHIN(distance, withinAllModels, atomExpression)|any atom within the specified distance of any atom in the atomExpression. The optional TRUE/FALSE flag withinAllModels (by default FALSE) may be set TRUE to allow finding atoms in one model that may be within some distance of another model. If the distance is negative, then the operation applies to all atoms having normalized unit cell coordinates within the given distance of the designated atoms.|| WITHIN(distance, {{x y z}})|any atom within the specified distance of the given fractional or Cartesian coordinate.  If the distance is negative, then the operation applies to all atoms having normalize unit cell coordinates within -distance of the designated atoms.||  WITHIN(distance, $surfaceObject) | any atom within the given distance of a point on the specified surface. (Jmol 12.2) || WITHIN(x.x, VDW, {{atomset}})| Selects atoms that have overlapping van der Waals surfaces with {{atomset}}. The distance parameter is optional, and its meaning depends upon its magnitude. If it is &gt; 10, then it is assumed to be a percent, such as 110%, 100%, 90%; otherwise it is assumed to be a distance to be added to the van der Waals radii of both atoms under consideration. So, for example, <b>select GROUPS within(1.4, VDW, {{*:A}})</b> selects all groups with solvent-accessible surfaces overlapping with chain A, and <b>select GROUPS within(100, VDW, {{ligand}})</b> selects all groups "clashing" with any ligand.||WITHIN(nResidues,GROUP,{{atoms}})| groups that are within a given number of residues of a specified group of atoms. (Jmol 12.0) || WITHIN(0,planeType, planeDesignation)| selects for any atoms within 0.01 Angstroms of a plane. If planeType is HKL, then planeDesignation is in the form {{h k l}}, where h, k, and l are Miller indices. If planeType is PLANE, then planeDesignation should be of the form @{{plane(a,b,c)}}, where a, b, and c are atom expressions or coordinates. ||WITHIN(distance,planeType, planeDesignation)|selects for atoms within the given distance in Angstroms from the plane. Positive distances are on one side; negative distances are on the other side. Experimentation may be necessary to determine which side is which for these purposes. In all cases the atoms in the plane itself (within 0.01 Angstroms of the plane on either side) are included. ||WITHIN(ATOMNAME,"aa,bb,ccc")|any atom having a listed atom name ||WITHIN(ATOMTYPE, "atomType,atomType,..."| selects for atoms of one or more atom type. Atom type is defined in certain file types, including MOL2 model files and AMBER topology files. For other file types, atom types are the same as atom names. For example, <b>select within(ATOMTYPE,"HW,OW")</b> selects all water atoms an AMBER topology file.||WITHIN(BASEPAIR("XY...")| (Jmol 12.0) finds all atoms within hydrogen-bonded DNA or RNA basepairs. Any number of pairs can be indicated. For example, <b>display within(BASEPAIR,"GCAU")<b> would select only G-C and A-U pairs. (Note that the RasMol-derived predefined sets "gc" and "at" refer simply to "G or C" and "A or T", respectively, and do not relate to base pairing.) || WITHIN(BOUNDBOX)| selects all atoms within the currently defined {#.boundbox~}||WITHIN(BRANCH,{{first atom}}, {{second atom}})| selects the second atom and all atoms in the molecular branch starting with the second atom but not including the first atom.||WITHIN(HELIX)|Selects groups  that would be selected using <b>select helix</b> but are not at either end of a helix section.||WITHIN(SEQUENCE,"sequence")| a protein or nucleic acid sequence expressed in single-letter notation surrounded by quotation marks as, for example, "GGCCCTT" or "MAACYXV" (the entire sequence must be found; as indicated above, the keyword SEQENCE is optional). || WITHIN(SHEET)|Selects groups that would be selected using <b>select sheet</b> but are not at either end of a sheet section. || WITHIN(SMARTS,"smartsString") | all atoms that conform to the given SMARTS string are found. When used as a math function, this method returns a list of all matching sets of atoms; when used in a selection context (SELECT, DISPLAY, HIDE, etc.), all matching atoms are returned. Only hydrogen atoms that are explicitly indicated as [H] are returned. Extensive details on Jmol 3D-SEARCH SMARTS capability may be found {http://jmol.svn.sourceforge.net/viewvc/jmol/trunk/Jmol/src/org/jmol/smiles/package.html~on-line}. (Jmol 12.0)|| WITHIN(SMILES,"smilesString") | all atoms that conform to the given {http://www.daylight.com/dayhtml/doc/theory/theory.smiles.html~SMILES} string are found. When used as a math function, this method returns a list of all matching sets of atoms, including any indicated hydrogen atoms or hydrogen atoms required to complete the valence on an atom.  When used in a selection context (SELECT, DISPLAY, HIDE, etc.), all matching atoms are returned. Note that for substructure searches, WITHIN(SMARTS,"smartsString") is recommended. (Jmol 12.0)||]','TEXT','Functions','')
newCmd('',' ','','*v+11.0;','The general specification of atoms in PDB "residues" follows the method used in RasMol. While the order of specifiers is somewhat flexible, the following order is generally applicable:<br /><br />[residueType]seqRange ^insertionCode :chainLetter .atomName  %altLoc /modelNumber<br /><br />[||[residueType]|[ALA] [G] [2E1] [L??] When used without any other specifiers it is possible <b>in some but not all cases</b> to leave off the brackets around the residue type. However, leaving off the brackets is not recommended and is <b>known to fail</b> when the residue type begins with a number.||seqRange| 1  1-30  40-   Note that ranges refer to physical ranges of data in the file. If residues corresponding to both the starting and ending residue numbers are not present in the file, selection returns no atoms. If residues with numbers between the starting and ending numbers are out of place in the file -- not physically between those two file positions -- they will not be included in the selection. If there is a desire to include such residues, or the selection should allow starting or ending residues to not be present, then use the <b>resno</b> comparison method instead. In this case, for example: <b>select resno >=1 and resno <=30</b> or <b>select resno >= 40</b>. ||^insertionCode | ^A ^B ^? ||:chainLetter | :A :B :? ||.atomName |.Ca .C? .? .?? ||%altLoc | %1 %A %? ||/modelNumber| /1 /2 /* refer to the number on the MODEL record in multimodel PDB files or the sequential number of the model in the file otherwise. When multiple files are loaded, these numbers refer to the file number, indicating "all models in that file." Specific models in specific files can be specified using a decimal notation: <b>file.model</b> as, for example, <b>select *.CA/2.1</b> -- all alpha carbons in the first model of the second file listed in the {#.load~} command.||]','TEXT','RasMol biomolecular residue specifications','')
newCmd('',' ','','*v+11.0;','Unspecified components of the atom specification are indicated in some cases using a question mark and in others using an asterisk. The wildcard * can be used in place of [residueType]seqRange to indicate "any." For example: <b>select *.CA</b>. Wildcards can be used elsewhere in the specification, but it is preferred simply to not include a specifier altogether. Thus, <b>select [ALA].*</b> is the same as <b>select [ALA]</b>. Note that in the case of PDB files and MOL2 files with residues indicated, * may be used in the form x* only in the case of residue names, not atom names. Thus, <b>select AS*</b> selects aspartate and asparagine. When used for an atom, for example, with the unremediated PDB file 1bkx <b>select A.O?*</b> the * is not wild and selects atoms A.O1* and A.O4*. (In remediated PDB files, this * becomes a single quote or "prime" character -- AO1\', AO4\'.) For other file types, * can be used at the end of an atom name fragment. Asterisks cannot be used in place of insertionCode or altLoc.  <br /><br />Question marks are used to indicate "some character": <b>select *.C??</b>. Note that the number of question marks is significant. ".?" only finds atoms with single-letter names such as "O" and "C"; ".??" finds atoms with single-letter or double-letter names.  The specification :?, ^?, and %? mean SOME chain, SOME insertion code or SOME alternate location; use :, ^, and % alone to indicate "atoms without chain indication," "atoms without insertion code," and "atoms without alternate location," respectively.<br /><br /> You can use \\? to match an actual ? in an atom name. For instance, if there are two atoms, one with the name "O1" and one with the name "O1?" then <b>select O1?</b> will select both atoms, but <b>select O1\\?</b> will select only the second atom. You cannot use \\* to escape an actual * in an atom name.','TEXT','Wildcards','')
newCmd('',' ','','*v+11.0;','Atom names can also be used for some non-PDB file types. For example, in CIF files, the atom_site_label field is used for the atom name. If an atom has the label "C34" you can select it using <b>select *.C34</b> or <b>select C34</b> or even <b>select C*</b>. Note that in this case, the wildcard * is no problem, since non-PDB file types do not include residue names, which might conflict with atom names. Similarly, Jaguar, NWChem, Tripos MOL2, Wavefunction Odyssey, SHELX, and Wavefunction Spartan files list atoms as "H3" and "O2". Atoms for these file types can be selected using these names, and the names can be displayed in {#.label~labels} using the format code %a. For file types such as XYZ that do not indicate a number with the atom symbol, Jmol constructs an atom name from the element symbol and the sequential number of the atom in the file.','TEXT','Atom names for other file types','')
newCmd('.[atom properties]',' ','','*v+11.8;prop;math','Well over 80 atom properties can be selected or retrieved from model data, and most of these can be set as well. The older, more limited Rasmol notation <b>[group].atomName^insertion:chain%altloc</b> can still be used, but equally well one can combine any subsets of those using a more natural notation. For example, <b>select group="ARG" and atomname="CA" and chain="A"</b> is equivalent to <b>select [ARG].CA:A</b>. Mostly, the newer <b>xxxx=y</b> notation generalizes better in terms of additional parameters not unique to Rasmol. Atom properties may also be set directly using <b>{{atom expression}}.xxxx = y</b>. An associative array of all matching atom properties can be created using <b>{{atom expression}}.xx?</b>,  where "xx" is the beginning of a property name, or even just <b>{{atom expression}}.?</b> for all atom properties. (for example, <b>print {{atomno=3}}.a?</b> or <b>print @3.?</b>). For {#.label~labels}, the <b>%[xxx]</b> notation extents the original single-character %x code of early Jmol versions. Within labels, <b>%%</b> can be used to "escape" the percent sign and have it not be a special character. The full list of atom properties is given below.<br />[||<b>property</b>|<b>select xxx=y</b>|<b>label %[xxx]</b>|<b>label %x</b>|<b>print {{*}}.xxx</b>|<b>{{*}}.xxx = y</b>|<b>description</b>||adpmax|yes|yes|&nbsp;|yes|&nbsp;|the maximum anisotropic displacement parameter for the selected atom||    adpmin|yes|yes|&nbsp;|yes|&nbsp;|the minimum anisotropic displacement parameter for the selected atom||              altloc|yes|yes|A|yes|&nbsp;|PDB alternate location identifier||altname|yes|yes|&nbsp;|yes|no|an alternative name given to atoms by some file readers (for example, P2N)||atomID|yes|yes|&nbsp;|yes|yes|special atom IDs for PDB atoms assigned by Jmol||atomIndex|yes|yes|D|yes|&nbsp;|atom 0-based index; a unique number for each atom regardless of the number of models loaded||atomName|yes|yes|a|yes|yes|atom name||atomno|yes|yes|i|yes|&nbsp;|sequential number -- starting with Jmol 12.2, you can use "@" instead of "atomno=" -- for example, <b>select @33</b> or <b>Var x = @33 or @35</b>||atomType|yes|yes|B|yes|yes|atom type (mol2, AMBER files) or atom name (other file types)||atomX|yes|yes|x|yes|yes|Cartesian X coordinate (or just X [Jmol 12.0])||atomY|yes|yes|y|yes|yes|Cartesian Y coordinate (or just Y [Jmol 12.0])||atomZ|yes|yes|z|yes|yes|Cartesian Z coordinate (or just Z [Jmol 12.0])||                   bondcount|yes|yes|&nbsp;|yes|&nbsp;|covalent bond count||cell|yes|&nbsp;|&nbsp;|&nbsp;|&nbsp;|crystallographic unit cell, expressed either in lattice integer notation (111-999) or as a coordinate in ijk space, where {{1 1 1}} is the same as 555. The positioning is either absolute, based on the original file coordinates, or relative to the current setting of {#.unitcell~}, as determined by <b>set fractionalRelative</b>. ANDing two cells, for example <b>select cell=555 and cell=556</b>, selects the atoms on the common face. (Note: in the specifc case of CELL, only "=" is allowed as a comparator.)||configuration|yes|&nbsp;|&nbsp;|&nbsp;|&nbsp;|Only in the context <b>{{configuration=n}}</b>, this option selects the set of atoms with either no ALTLOC specified or those atoms having this index into the array of altlocs within its model. So, for example, if the model has altloc "A" and "B", <b>select configuration=1</b> is equivalent to <b>select altloc="" or altloc="A"</b>, and <b>print {{configuration=2}}</b> is equivalent to <b>print {{altloc="" or altloc="B"}}</b>. Configuration 0 is "all atoms in a model having configurations", and an invalid configuration number gives no atoms.  (Note: in the specifc case of CONFIGURATION, only "=" is allowed as a comparator.)||<b>property</b>|<b>select xxx=y</b>|<b>label %[xxx]</b>|<b>label %x</b>|<b>print {{*}}.xxx</b>|<b>{{*}}.xxx = y</b>|<b>description</b>||chain|yes|yes|c/s|yes|&nbsp;|protein chain. For newer CIF files allowing multicharacter chain specifications, use quotations marks: <b>select chain="AA"</b>. For these multicharacter desigations, case is not checked unless the CIF file has lower-case chain designations. ||color|yes|yes|&nbsp;|yes|yes|the atom color||covalent|yes|yes|&nbsp;|yes|&nbsp;|covalent bonding radius||cs|yes|yes|&nbsp;|yes|&nbsp;|chemical shift calculated using computational results that include magnetic shielding tensors. [Jmol 13.2]||element|yes|yes|e|yes|yes|element symbol. The value of this parameter depends upon the context. Used with <b>select structure=x</b>, x can be either the quoted element symbol, "H", "He", "Li", etc. or atomic number. In all other contexts, the value is the element symbol. When the atom is a specific isotope, the string will contain the isotope number -- "13C", for example.||elemno|yes|yes|l (<em>el</em>)|yes|yes|atomic element number||         eta/theta|yes|yes|&nbsp;|yes|&nbsp;|Based on Carlos M. Duarte, Leven M. Wadley, and Anna Marie Pyle, <i>RNA structure comparison, motif search and discovery using a reduced representation of RNA conformational space</i>, <b>Nucleic Acids Research</b>, <b>2003</b>, Vol. 31, No. 16 4755-4761. The parameter eta is the C4\'[i-1]-P[i]-C4\'[i]-P[i+1] dihedral angle; theta is the P[i\]-C4\'[i]-P[i+1]-C4\'[i+1] dihedral angle. Both are measured on a 0-360 degree scale because they are commonly near 180 degrees. Using the commands {#.plot~plot PROPERTIES eta theta resno; select visible;wireframe only} one can create these authors\' "RNA worm" graph. ||file|yes|yes|&nbsp;|yes|&nbsp;|file number containing this atom||formalCharge|yes|yes|C|yes|yes|formal charge||format|yes|yes|&nbsp;|yes|yes|format (label) of the atom. ||fXyz|&nbsp;|yes|&nbsp;|yes|yes|fractional XYZ coordinates||fX|yes|yes|X|yes|yes|fractional X coordinate||fY|yes|yes|Y|yes|yes|fractional Y coordinate||fZ|yes|yes|Z|yes|yes|fractional Z coordinate||fuxyz|&nbsp;|yes|&nbsp;|yes|yes|fractional XYZ coordinates in the {#.unitcell~} coordinate system||fux|yes|yes|&nbsp;|yes|yes|fractional X coordinate in the unitcell coordinate system||fuy|yes|yes|&nbsp;|yes|yes|fractional Y coordinate in the unitcell coordinate system||fuz|yes|yes|&nbsp;|yes|yes|fractional Z coordinate in the unit cell coordinate system||group|yes|yes|n|yes|&nbsp;|3-letter residue code||group1|yes|yes|m|yes|&nbsp;|single-letter residue code (amino acids only)||groupID|yes|yes|&nbsp;|yes|&nbsp;|group ID number: A unique ID for each amino acid or nucleic acid residue in a PDB file. <br /> [|| 0 | noGroup||  1-5  |ALA, ARG, ASN, ASP, CYS|| 6-10|  GLN, GLU, GLY, HIS, ILE|| 11-15|  LEU, LYS, MET, PHE, PRO|| 16-20 | SER, THR, TRP, TYR, VAL || 21-23 | ASX, GLX, UNK||  24-29|  A, +A, G, +G, I, +I || 30-35|  C, +C, T, +T, U, +U||]<br />Additional unique numbers are assigned arbitrarily by Jmol and cannot be used reproducibly. ||<b>property</b>|<b>select xxx=y</b>|<b>label %[xxx]</b>|<b>label %x</b>|<b>print {{*}}.xxx</b>|<b>{{*}}.xxx = y</b>|<b>description</b>||groupindex|yes|yes|G|yes|&nbsp;| overall group index ||identify|yes|yes|U|yes|&nbsp;|for a  PDB/mmCIF file, same as [%[group]]%r:%[chain] %%%[altloc]/%[model]  #%[atomno]. For non-PDB data, same as %[atomName]/%[model]  #%[atomno]||insertion|yes|yes|E|yes|&nbsp;|protein residue insertion code||ionic|yes|yes|I|yes|yes|radius used for bonding (ionic radius when a formal charge is defined); synonymous with <b>ionicRadius</b> and settable starting in Jmol 12.0||label|yes|yes|&nbsp;|yes|yes|current atom label (same as <b>format</b>)||mass|yes|yes|&nbsp;|yes|no|atomic mass -- especially useful with appended <b>.max</b> or <b>.sum</b>||model|yes|yes|M|yes|&nbsp;|model number||modelindex|yes|yes|&nbsp;|yes|&nbsp;|a unique number for each model, starting with 0 and spanning all models in all files ||molecule|yes|yes|N|yes|&nbsp;|molecule number||ms|yes|yes|&nbsp;|yes|&nbsp;|magnetic shielding calculated from file-loaded tensors. [Jmol 13.2]||occupancy|yes|yes|q/Q|yes|yes| CIF file site occupancy. In SELECT command comparisons ("select occupancy < 90"), an integer n implies measurement on a 0-100 scale; also, in the context %[occupancy] or %q for a label, the reported number is a percentage. In all other cases, such as when %Q is used in a label or when a decimal number is used in a comparison, the scale is 0.0 - 1.0.||partialCharge|yes|yes|P|yes|yes|partial charge||phi|yes|yes|f|yes|&nbsp;|protein group PHI angle for atom\'s residue  ||polymer|yes|yes|&nbsp;|yes|yes|sequetial polymer number in a model, starting with 1. (Jmol 12.0)||polymerLength|yes|yes|L|yes|&nbsp;|polymer length||property_xx|yes|yes|&nbsp;|yes|yes|a property created using the DATA command||psi|yes|yes|p|yes|&nbsp;|protein group PSI angle for the atom\'s residue  ||radius|yes|yes|I|yes|yes|currently displayed radius -- In SELECT command comparisons ("select radius=n"), integer n implies Rasmol units 1/250 Angstroms; in all other cases or when a decimal number is used, the units are Angstroms.||<b>property</b>|<b>select xxx=y</b>|<b>label %[xxx]</b>|<b>label %x</b>|<b>print {{*}}.xxx</b>|<b>{{*}}.xxx = y</b>|<b>description</b>||resno|yes|yes|R|yes|&nbsp;|PDB residue number, not including insertion code||selected|yes|&nbsp;|&nbsp;|yes|yes|1.0 if atom is selected; 0.0 if not [Jmol 12.0]||sequence|yes|yes|&nbsp;|yes|&nbsp;|PDB one-character sequence code, as a string of characters, with "?" indicated where single-character codes are not available||shape|yes|yes|&nbsp;|yes|&nbsp;|hybridization geometry such as "tetrahedral -- AX4" (Jmol 12.2) ||site|yes|yes|S|yes|&nbsp;|crystallographic site number||spacefill|yes|yes|&nbsp;|yes|yes|currently displayed radius||straightness|yes|yes|T|yes|&nbsp;|quaternion-derived straightness (second derivative of the quaternion describing the orientation of the residue. This quantity will have different values depending upon the setting of <b>quaternionFrame</b> as "A" (alpha-carbon/phosphorus atom only), "C" (alpha-carbon/pyrimidine or purine base based), "P" (carbonyl-carbon peptide plane/phosphorus tetrahedron based), or "N" (amide-nitrogen based). The default is alpha-carbon based, which corresponds closely to the following combination of Ramachandran angles involving three consecutive residues i-1, i, and i+1: -psi<sub>i-1</sub> - phi<sub>i</sub> + psi<sub>i</sub> + phi<sub>i+1</sub>.||strucno|yes|yes|&nbsp;|yes|&nbsp;|a unique number for each helix, sheet, or turn in a model, starting with 1.||structure|yes|yes|&nbsp;|yes|&nbsp;|The value of this parameter depends upon the context. Used with <b>select structure=x</b>, x can be either the quoted keyword "none", "turn", "sheet", "helix", "dna", "rna", or "carbohydrate" or a respective number 0-6. In the context <b>{{*}}.structure</b>, the return value is a number; in the context <b>label %[structure]</b>, the return is one of the six keywords. ||substructure|yes|yes|&nbsp;|yes|&nbsp;|like <b>structure</b>, the value of this parameter depends upon the context. Used with <b>select substructure=x</b>, x can be either the quoted keyword "none", "turn", "sheet", "helix", "dna", "rna", "carbohydrate", "helix310", "helixalpha", or "helixpi", or the respective number 0-9. In the context <b>{{*}}.substructure</b>, the return value is a number; in the context <b>label %[substructure]</b>, the return is one of the nine keywords. (Jmol 12.0) ||surfacedistance|yes|yes|u|yes|&nbsp;|A value related to the distance of an atom to a nominal molecular surface. 0 indicates at the surface. Positive numbers are minimum distances in Angstroms from the given atom to the surface. ||symop|yes|&nbsp;|&nbsp;|yes|&nbsp;|symmetry operation code that generated this atom by Jmol; an integer starting with 1. This operator is only present if the file contains space group information and the file was loaded using the {{i, j, k}} option so as to generate symmetry-based atoms. To select only the original atoms prior to application of symmetry, you can either use "SYMOP=n", where n is the symmetry operator corresponding to "x,y,z", or you can specify instead simply "NOT symmetry" the way you might specify "NOT hydrogen". Note that atoms in special positions will have multiple operator matches. These atoms can be selected using the keyword SPECIALPOSITION.  The special form <b>select SYMOP=nijk</b> selects a specific translation of atoms from the given crystallographic symmetry operation. Comparators  &lt;, &lt;=, >, >=, and != can be used and only apply to the ijk part of the designation. The ijk are relative, not absolute. Thus, <b>symop=2555</b> selects for atoms that have been transformed by symop=2 but not subjected to any further translation. <b>select symop=1555</b> is identical to <b>select not symmetry</b>. All other ijk are relative to these selections for 555. If the model was loaded using <b>load "filename.cif" {{444 666 1}}</b>, where the 1 indicates that all symmetry-generated atoms are to be packed within cell 555 and then translated to fill the other 26 specified cells, then <b>select symop=3555</b> is nearly the same as <b>select symop=3 and cell=555</b>. (The difference being that cell=555 selects for all atoms that are on any edge of the cell, while symop=3555 does not.) However, the situation is different if instead the model was loaded using <b>load "filename.cif" {{444 666 0}}</b>, where the 0 indicates that symmetry-generated atoms are to be placed exactly where their symmetry operator would put them (x,-y,z being different then from x, 1-y, z). In that case, <b>select symop=3555</b>  is for all atoms that have been generated using symmetry operation 3 but have not had any additional translations applied to the x,y,z expression found in the CIF file. If, for example, symmetry operation 3 is -x,-y,-z, then <b>load "filename.cif" {{444 666 0}}</b> will place an atom originally at {{1/2, 1/2, 1/2}} at positions {{-1/2, -1/2, -1/2}} (symop=3555) and {{-3/2, -3/2, -3/2}} (symop=3444) and 24 other sites.||<b>property</b>|<b>select xxx=y</b>|<b>label %[xxx]</b>|<b>label %x</b>|<b>print {{*}}.xxx</b>|<b>{{*}}.xxx = y</b>|<b>description</b> || symmetry|&nbsp;|yes|o|yes|&nbsp;|list of crystallographic symmetry operators generating this atom||temperature|yes|yes|b/t|yes|yes|temperature factor (B-factor)||unitXyz|&nbsp;|yes|&nbsp;|yes|&nbsp;|unit cell XYZ coordinates||uX|yes|yes|&nbsp;|yes|&nbsp;|unit cell X coordinate normalized to [0,1)||uY|yes|yes|&nbsp;|yes|&nbsp;|unit cell Y coordinate normalized to [0,1)||uZ|yes|yes|&nbsp;|yes|&nbsp;|unit cell Z coordinate normalized to [0,1)||valence|yes|yes|&nbsp;|yes|yes|the valence of an atom (sum of bonds, where double bond counts as 2 and triple bond counts as 3||vanderwaals|yes|yes|V|yes|yes|van der Waals radius ||vectorScale|yes|yes|&nbsp;|yes|no|vibration vector scale|| volume|yes|yes|&nbsp;|yes|&nbsp;|approximate van der Waals volume for this atom. Note, {{*}}.volume gives an average; use {{*}}.volume.sum to get total volume. [Jmol 12.0]||vXyz|&nbsp;|yes|v|yes|yes|vibration vector, or individual components as %vx %vy %vz||vX|yes|yes|&nbsp;|yes|yes|vibration vector X coordinate||vY|yes|yes|&nbsp;|yes|yes|vibration vector Y coordinate||vZ|yes|yes|&nbsp;|yes|yes|vibration vector Z coordinate ||x|yes|yes|x|yes|yes|Cartesian X coordinate [Jmol 12.0]||y|yes|yes|y|yes|yes|Cartesian Y coordinate [Jmol 12.0]||z|yes|yes|z|yes|yes|Cartesian Z coordinate [Jmol 12.0]||xyz|&nbsp;|yes|&nbsp;|yes|yes+I54|Cartesian XYZ coordinates||&nbsp;|&nbsp;|&nbsp;|g|&nbsp;|&nbsp;|group index in chain||&nbsp;|&nbsp;|&nbsp;|r|&nbsp;|&nbsp;|PDB residue number, including insertion code||&nbsp;|&nbsp;|&nbsp;|W|&nbsp;|&nbsp;|PDB residue designator with x, y, z included: [%n]%r %x %y %z ||]','0|1','','')
newCmd('.axes',' ','','*v+11.0 -- new;set','Turns on or off displayed axes, and determines their line style and line width (as a decimal number, in Angstroms).','0|1','','')
newCmd('',' ','','*v+11.0;','Turns axes on or off','*1','._on_off{"ON"}','')
newCmd('',' ','','*v+11.0;','Sets the axes diameter in Angstroms. ','*20','(decimal)','')
newCmd('','12.0.RC23','','*v+12.0;','Sets the axes origin to the specified point, which may be an atom expression such as {{*}} or {{rna}}.','*21','CENTER','{{x y z}}')
newCmd('',' ','','*v+11.0;','Sets the axes style to a thin dotted line.','*3','DOTTED','')
newCmd('',' ','','*v+12.0;*v-13.2;','Sets the labels for the positive X, Y, and Z axes. Negative-directed axes are hidden.','*51','LABELS','&quot;x-label&quot;','&quot;y-label&quot;','&quot;z-label&quot;')
newCmd('','13.1.12','','*v+13.2;','Sets the labels for the positive and negative X, Y,  and Z axes, as well as the origin. ','*52','LABELS','&quot;x-label&quot;','&quot;y-label&quot;','&quot;z-label&quot; &quot;-x-label&quot; &quot;-y-label&quot; &quot;-z-label&quot;+&quot;-origin-label&quot;')
newCmd('',' ','','*v+11.0;','Sets the axes diameter in pixels (1 to 19).','*4','(integer)','')
newCmd('','','','*v+11.2;','Sets the axes to be based on the molecular coordinate {{0 0 0}}','*53','MOLECULAR','')
newCmd('','','','','Sets the axes to align with the a, b, and c axes of the unit cell (default if fractional coordinates)','*62','SCALE','(decimal)')
newCmd('','11.9.12','','*v+12.0;ticks','Sets the parameters for ticks along the axes. An optional specific axis (X, Y, or Z) can be indicated. There are three levels of ticks - major, minor, and "subminor." Only the major ticks have labels. Which of these tick levels are displayed and the distance between ticks depends upon the parameter that takes the form of a point. This point may be in fractional form, {{1/2 0 0}}. The optional keyword FORMAT allows formating of the labels for the major ticks. These are based on an array of strings given after the FORMAT keyword. If the array is shorter than the number of ticks, the formats in the array are repeated. Following that, the optional keyword SCALE allows setting the scale either for each axis direction independently {{scaleX, scaleY, scaleZ}} or overall (as a decimal number). ','*63','TICKS X|Y|Z','{{major,minor,subminor}} FORMAT ["%0.2f", ...] ','SCALE {{scaleX, scaleY, scaleZ}} | x.xx','')
newCmd('','','','*v+11.2;','Sets the axes to be based on the center of the bounding box (default if not fractional coordinates)','*8','WINDOW','')
newCmd('','','','*v+11.2;','Sets the axes to align with the a, b, and c axes of the unit cell (default if fractional coordinates)','*7','UNITCELL','')
newCmd('','','','*v+11.8;','Sets the axes to be positioned at a specific screen coordinate or fractional position along the horizontal and vertical screen dimensions. Must be preceded by <b>axes on</b>.','*61','POSITION','[x y] or [x y %]')
newCmd('.backbone',' ','~structure','disp','Shows the backbone of a protein or nucleic acid macromolecule by connecting the alpha carbons. The selection of backbone units to display depends upon the currently selected atoms and the {#.set_bondmode~bondmode} setting.','0|1','','')
newCmd('',' ','','','Turns the backbone on or off','*10','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns backbone rendering on and all other rendering (including labels) off.','*11','ONLY','')
newCmd('',' ','','','Backbone radius can be specified in angstroms using a decimal number (1.0, 2.0, etc.). Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._backbone_radius','')
newCmd('.background',' ','','*v+11.6 -- image support;disp;color','Sets color of the background or sets the background image. For color specifications, see {#.color~}.','1|2','','')
newCmd('',' ','','','Sets the background color of the applet/application window.','*1','._colorRGB','')
newCmd('',' ','','','Sets the background of the most recently defined {#.echo~} text and subsequent user-defined echo text to be the given color.','*21','ECHO','._color_or_none')
newCmd('',' ','','','Sets the background of the applet/application window to the specified image file, which can be of format JPG, PNG, or GIF. The image is stretched to fit the size of the window.','*23','IMAGE','&quot;filename&quot;')
newCmd('',' ','','','Sets the background color of the atom labels that appear with the "label" command. "NONE" results in there being no label background. Operates globally, not on selected atoms.','*4','LABELS','._color_or_none')
newCmd('',' ','','','Sets the background color for the pop-up label box that appear when the mouse "hovers" over an atom.  "NONE" results in there being no hover backgrounds. Operates globally, not on selected atoms.','*3','HOVER','._color_or_none')
newCmd('.bind','11.9.9','','*v+12.0;bind','The <b>bind</b> and {#.unbind~} commands allow users to customize the effects of mouse actions. When a mouse action is bound to a user-defined script, Jmol will replace _X, _Y, _DELTAX, _DELTAY, _TIME, _ACTION, and _MODE (Jmol 12.2.13 adds _ATOM, _BOND, and _POINT) in the script with appropriate values. Capitalization is significant. The _ACTION variable encodes the mouse/key combination involved as well as the click count (1 SHIFT, 2 CTRL, 4 RIGHT, 8 ALT or MIDDLE, 16 LEFT, 32 WHEEL all ORed with 64*clickCount) . The _MODE variable indicates the mouse action that occurred (1 dragged, 2 clicked, 3 wheeled, 4 pressed, or 5 drag-released). Starting with Jmol 13.2, an added "+:" at the beginning of the binding name indicates to NOT unbind Jmol action while still defining  a user action.','1|2','','')
newCmd('','11.9.9','','*v+12.0;','Ties a specific mouse action to a specific Jmol action.','*1','._mouse_action','._jmol_action')
newCmd('','11.9.9','','*v+12.0;','Ties a specific mouse action to a script defined by the user. ','*2','._mouse_action','&quot;script&quot;')
newCmd('.break','','','*v+11.4;flow','{#.while~} and {#.for~} loops may be exited using <b>break</b> and truncated using <b>continue</b>. A number following <b>break</b> or <b>continue</b> indicates how many levels of for/while loops  beyond the innermost one to break out of or continue.<br /><br />n = {{*}}.length<br />for (var i = 1; i <= n; i++) {{<br />&nbsp;&nbsp;for (var j = i + 1;j <= n; j++) {{<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;var dis = {{*}}[i].distance({{*}}[j]);<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if (dis < 1.23) {{<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print "short i-j: " + i + "," + j + " " + dis%2<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;measure {{atomno=i}} {{atomno=j}}<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;continue;<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}} else if (dis < 1.77) {{<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print "medium i-j: " + i + "," + j + " " + dis%2<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;line = "line"+i+"_"+j<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;draw @line {{atomno=i}} {{atomno=j}}<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;break 1;<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}}<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print "long";<br />&nbsp;&nbsp;}}<br />}}','0|1','','')
newCmd('.continue','','','*v+11.4;flow','See {#.break~}.','0|1','','')
newCmd('.bondorder','','','*v+11.4 -- new;bond','Sets the bond order of the selected atoms or bonds. An alternative to the  {#.connect~} command.','0|1','','')
newCmd('','','','','Sets the bond order to the specified order. Values -1, -1.5, and -2.5 specify HBOND, PARTIALDOUBLE (reverse solid/dash of AROMATIC), and PARTIALTRIPLE2, respectively. ','*1','0.5, 1, 1.5, 2, 2.5, 3, 4, -1, -1.5, -2.5','')
newCmd('','','','','Sets the bond order to the specified type. See details at {#.connect~} for PARTIAL N.M .','*2','._connection_options','')
newCmd('.boundbox','11.3.44','','*v+11.4 -- expanded capability;set;boundbox','Turns on or off a wire-frame box that contains the model or a designated subset of the model or a designated region of space, and determines the line style and line width (as a decimal number, in Angstroms) of that box. If the atom set is not indicated, the boundbox is drawn around the entire model set (including all models in frames). A decimal number specifies the boundbox line diameter in Angstroms; DOTTED specifies a fine dotted line. For an explanation of center, corners, and vectors, see {#.show~show boundbox}.','0|1','','')
newCmd('','','','*v+11.4;','Turns the boundbox on or off  around the specified set of atoms. In a multi-model context, if a set of atoms is specified and that set of atoms is within a subset of the models, then the boundbox is only displayed only when the {#.model~} command has made that subset of models displayable. ','*1','._atom_expression{"*"}','._axes_type{"ON"}')
newCmd('',' ','','*v+11.4;','Sets the boundbox to be centered on the atom set or coordinate given as the first parameter and extending by a vector to a corner specified by the second parameter. If the third parameter is not given, no change in the visibility of the boundbox or its line style is made.','*2','._atom_expression_or_coord','._coordxyz','._axes_type{"unchanged"}','')
newCmd('',' ','','*v+11.4;','Sets the boundbox based on two corners, each specified by the center off an atom set or a coordinate. If the fourth parameter is not given, no change in the visibility of the boundbox or its line style is made','*3','CORNERS','._atom_expression_or_coord','._atom_expression_or_coord','._axes_type{"unchanged"}')
newCmd('','11.9.12','','*v+12.0;ticks','Sets the parameters for ticks along the first three boundbox edges. The parameters are similar to those of {#.axes~axes TICKS}, but unit cell scaling is not an option.','*41','TICKS X|Y|Z','{{major,minor,subminor}} FORMAT ["%0.2f", ...] ','SCALE {{scaleX, scaleY, scaleZ}} | x.xx','')
newCmd('','11.9.19','','*v+12.0;','With any of the above parameters, specifies to scale the boundbox by (a) multiplying by the scaling factor if positive or (b) adding to the size if negative. (Jmol 12.0)','*42','SCALE','x.xx')
newCmd('','11.9.19','','*v+12.0;','Sets the current boundbox to match the extent of a given isosurface. (Jmol 12.0)','*5','$isosurfaceID','')
newCmd('.cache','13.1.2','','*v+13.2;cache;load','Establishes a memory-based cache of data. A summary of the contents of the cache can be obtained with <b>show cache</b>.','0|1','','')
newCmd('','13.1.2','','*v+13.2;','Clears the cache','*2','CLEAR','')
newCmd('','13.1.2','','*v+13.2;','Adds a file to the memory cache as a set of bytes.','*1','ADD','&quot;filename&quot;')
newCmd('','13.1.2','','*v+13.2;','Removes the specified file from the cache. <b>cache REMOVE ALL</b> is the same as <b>cache CLEAR</b>.','*3','REMOVE','&quot;filename&quot;')






newCmd('.calculate',' ','','*v+13.0 -- adds calculation of partial charges using the MMFF94 charge model;calc','Calculates specific quantities.','0|1','','')
newCmd('.calculate',' ','','*v+12.0 -- adds calculations for standard hydrogen bonds, hydrogens, pointgroup, straightness, DSSP-based structure, and struts;*v-13.0;calc','Calculates specific quantities.','0|1','','')
newCmd('','11.3.29','','*v+11.2;','Calculates alternating single and double aromatic bonds for all bonds of type AROMATIC. If just one bond of a conjugated system is specified as AROMATICSINGLE or AROMATICDOUBLE, then all bonds of that system will be consistent with that bond. . For example: <br /><br />reset aromatic;<br />connect (atomno=3) (atomno=4) AROMATICDOUBLE;<br />calculate aromatic;<br /> <br/>or<br /><br />select carbon and within(1.6, {{0 0 0}});<br />connect (selected) aromatic modify;<br />calculate aromatic;<br /><br />','*11','AROMATIC','')
newCmd('','13.3.1','','*v+13.4;','Calculates formal charges based on hybridization and bonding.','*12','FORMALCHARGE','')
newCmd('','12.0.RC1','','*v+12.0;','If no atom sets are indicated, performs the same as {#.hbonds~hbonds calculate}. Three types of hydrogen bond calculation are available. <br />[||RasMol-type pseudo-hbonds (PDB/mmCIF files only)|<b>select not hydrogen; set hbondsRasmol TRUE; calculate HBONDS</b> | The default calculation when no hydrogen atoms are selected. Creates pseudo-hydrogen bonds only between protein amide and carbonyl groups or between nucleic acid base pairs using a RasMol-like (DSSP) calculation. Bonds can be differentiated using {#.color bond object~color hbonds TYPE} (see {http://jmol.sourceforge.net/jscolors/#Hydrogen%20bonds~} or  {#.color bond object~color hbonds ENERGY} (red - high energy, less stable; blue - low energy, more stable). This energy is calculated according to the equation given at {http://en.wikipedia.org/wiki/DSSP_%28protein%29~http://en.wikipedia.org/wiki/DSSP_(protein)}||Creates Jmol-type pseudo-hbonds|<b>select not hydrogen; set hbondsRasmol FALSE; calculate HBONDS </b> | pseudo-hydrogen bonds between O or N atoms on proteins and nucleic acid (but not limited to backbone atoms), and other atoms, including water oxygens. Psuedo-hydrogen bonds are not calculated between two water molecules. Uses an algorithm developed for Jmol involving parameters (see below). These hbonds are not colorable by energy.|| Standard hydrogen bonds|all cases where hydrogens are selected followed by <b>calculate HBONDS</b>| Creates bonds from hydrogen atoms on O or N to other O and N atoms, regardless of whether they are in macromolecules. When two atom sets are used, hydrogen atoms must be present in the first set; other atoms in that set will be ignored. Coloring of standard hydrogen bonds by "energy" is possible, but the value associated with that color should be be taken as a true energy. Note that you can add hydrogen atoms to a PDB file using <b>set pdbAddHydrogens</b> prior to loading a PDB file.||] If two atom sets are provided, hydrogen bonds will be between atoms in the first atom set and atoms in the second atom set. These sets may overlap, and a common invocation of the command is simply <b>calculate HBONDS</b>, which uses the currently selected atoms for both sets. When the RasMol calculation is not used, the parameters <b>hbondsAngleMinimum</b> (default value 90) and <b>hbondsDistanceMaximum</b> (default 3.25 for pseudo-hydrogen bonds; 2.5 for standard hydrogen bonds) are used. The angle referred to is the bond angle between the candidate hydrogen (or pseudo-hydrogen) bond and the covalent bonds to the atoms involved. The idea is that all bond angles that include  hydrogen bonds should be greater than some designated value. Note that the default value of 90 degrees may be too large for the general case. Values as low as 70 or 80 may be approriate in some cases, and experimentation may be necessary. The distance maximum is used only for pseudo-hydrogen bonds when its value is set larger than 2.5.    ','*22','HBONDS','._atom_expression','._atom_expression','')
newCmd('','12.0.18','','*v+12.0;','Calculates the hydrogen bonds that define the DSSP structures generated with {#.calculate~calculate STRUCTURE}. These are not all of the "hydrogen bonds" identified by DSSP; rather they are just the ones that define the helixes, sheets, and bridges. (Jmol 12.0 and Jmol 12.2)','*23','HBONDS','STRUCTURE')
newCmd('','11.9.19','','*v+12.0;;','Adds hydrogens at calculated positions based on bonding patterns at the designated atoms or, if no atoms are specified, at all atoms. Note that this command is not intended to be used for the assignment of hydrogen atoms in standard PDB files. All atoms must have any formal charges already designated, and all multiple bonds must be in place. If you want to add hydogen atoms to a standard PDB structure, use <b>set pdbAddHydrogens</b> prior to loading the file. Jmol will then do a proper hydrogen atom addition, to proteins, nucleic acids, carbohydrates, and all ligands. See {#set_pdbaddhydrogens~set pdbAddHydrogens}, below.','*240','HYDROGENS','._atom_expression{"*"}')
newCmd('','12.3.24','','*v+11.6;;','Calculates relatively reasonable partial charges using the MMFF94 charge model. Multiple bonding is required. The currently selected atom set is used or, if no atoms are selected, the currently visible atom set.','*241','PARTIALCHARGE','')
newCmd('','11.5.52','','*v+11.6;;','Calculates the point group symmetry for a symmetrical or nearly symmetrical molecule. The calculation is carried out only on the currently selected atoms and is limited to at most 100 selected atoms. The symmetry-determining algorithm looks for proper and improper rotation axes using a variety of methods. In each case, a test is made as to whether all atoms subjected to a specific rotation or reflection map onto the positions of some other atom.The extent to which imperfections in symmetry will be tolerated depends upon two adjustable paremeters. <b>pointGroupDistanceTolerance</b> (default 0.2 Angstroms) determines the maximum distance between a rotated atom and another atom in the molecule. <b>pointGroupLinearTolerance</b> (default 8.0 degrees) determines whether a potential axis matches one that has already been discovered. Setting these values to higher numbers allows more flexibility in terms of atom positions, but also may result in molecules being reported with higher symmetry than they really have.','*25','POINTGROUP','')
newCmd('','11.3.47','','*v+11.4;','Calculates for each atom the property surfaceDistance, which is the distance of the atom to a "shrink-wrap" surface surrounding the set of preselected atoms (usually the entire model, often without solvent molecules). Values for atoms OUTSIDE this surface are not generally valid due to the nature of the calculation. Use <b>calculate surfacedistance FROM {{atom expression}}</b> instead for calculating distances outside of a van der Waals surface surrounding a subset of atoms in a model. {#.isosurface~isosurface sasurface map property surfaceDistance} then creates an isosurface colored by distance from the specified subset; or either {#.color~color surfaceDistance}  or {#.color~color property surfaceDistance} colors the atoms along the same lines.','*9','SURFACEDISTANCE','WITHIN','._atom_expression','')
newCmd('','11.3.47','','*v+11.4;','Calculates for each atom the property surfaceDistance, which is the distance of the atom to a van der Waals surface surrounding the specified subset of atoms of the model. {#.isosurface~isosurface sasurface map property surfaceDistance} then creates an isosurface colored by distance from the specified subset; or either {#.color(color scheme)~color surfaceDistance}  or {#.color(color scheme)~color property surfaceDistance} colors the atoms along the same lines.','*8','SURFACEDISTANCE','FROM','._atom_expression','')
newCmd('','11.5.43','','*v+11.6;','Calculates a value for "straightness" (ranging from -1 to 1) within a biomolecular polymer (protein or nucleic) as defined by the following equation: <br /> <br /> straightness =   1 - 2 * (acos(q<sub>i</sub> / q<sub>i-1</sub> * q<sub>i+1</sub> / q<sub>i</sub>) / PI)<br /> <br />where q<sub>i</sub> is the {#.quaternion~} defining the frame of the i<sup>th</sup> amino acid or nucleic acid. A value of 1 for straightness indicates that the three amino acids (i-1, i, i+1) are perfect rotations around the same helical axis. The straightness can then be displayed as part of a label using the %T format code or as a color using {#.color~color straightness.','*3','STRAIGHTNESS','')
newCmd('','10.9.62','','*v+11.4;','Recalculates the secondary structure of proteins based on Jmol\'s implementation of {http://en.wikipedia.org/wiki/Secondary_structure~DSSP} (starting with Jmol 12.0) or a Ramachandran-angle-based method intrinsic to Jmol (prior to Jmol 12.0 or with the added parameter RAMACHANDRAN in Jmol 12.0). The default DSSP setting is to ignore any file-based backbone amide hydrogens, but this can be changed using <b>set dsspCalculateHydrogenAlways FALSE</b>. The calculation is performed for all models containing any currently selected atoms. For these models all cartoons and other biomolecular shapes are turned off. The next <b>cartoons on</b> command will then show a complete set of cartoons.  This command also recalculates the polymer chains making up a protein or nucleic acid. The results are affected by any bonding that has changed via the {#.connect~} command. A typical use is for PDB files that load with incomplete bonding (because the author specified only a fraction of the bonds). After loading, one can issue <b>connect</b> to use Jmol\'s autobonding feature, then <b>calculate structure</b> to regenerate the secondary structure. ','*4','STRUCTURE','')
newCmd('','11.9.22','','*v+12.0;','Generates {#.struts~}. Three parameters are used (defaults given): <b>set strutSpacing 6</b> sets the minimum spacing between struts, <b>set strutLengthMaximum 7.0</b> sets the maximum length that is allowed for a strut, and <b>strutsMultiple</b> when set TRUE allows multiple struts on a given atom. In addition, <b>set strutDefaultRadius 0.3</b> sets the default radius for struts.','*5','STRUTS','')



newCmd('.capture','13.3.6','','*v+13.3;image','Captures animations and mouse action results as animated GIF files. Application and signed applet only. [||<img src=img/caffeine.png title=\'click to animate\' onclick="this.src=\'img/caffeine.ani.gif\'" />||click image to animate||]','0|1','','')
newCmd('','13.3.6','','','CAPTURE alone closes the current capture file.','*1','','')
newCmd('','13.3.6','','','starts capturing to the designated GIF file.  Whether the animation loops or not depends upon the value of {#.animation~animation mode}: PALINDROME or LOOP results in a looping GIF (no distinction, as there is no PALINDROME option in GIF animation) or ONCE (no looping). Frame rate is determined by {#.animation~animation fps}, which has a maximum of 50 frames per second.','*2','&quot;filename.gif&quot;','')
newCmd('','13.3.6','','','creates a looping GIF that rocks around the specified axis (x, y, or z; default y) the specified number of degrees (default 5). Both the axis and degrees are optional.','*3','&quot;filename.gif&quot;','ROCK','axis','degrees')
newCmd('','13.3.6','','','creates a looping GIF that spins around the specified axis (x, y, or z; default y).','*4','&quot;filename.gif&quot;','SPIN','axis','')

newCmd('','13.3.6','','','temporarily turns capturing off; file is still open at this point.','*6','OFF','')
newCmd('','13.3.6','','','re-starts capturing.','*7','ON','')
newCmd('.cartoon',' ','structure','disp','Cartoons are the classic shapes the are used to depict alpha helices and beta-pleated sheets. A combination of cartoons and {#.rockets~} can be displayed using cartoons along with {#.set_cartoonRockets~}. The {#.set_rocketBarrels~} option removes the arrow heads from cartoon rockets. Jmol 13.2 adds {#.set_cartoonLadders~}, which removes the bases from nucleic acid cartoons but leaves the ladder "rungs". The {#.set_hermitelevel~hermite level} determines the overall cross-section of the helix and sheet renderings. Starting with Jmol 13, hermite levels 6 or higher produces a ribbon cross-section in the shape of an ellipse. Use {#.set_ribbonaspectratio~set ribbonAspectRatio 4} rather than the default value of 16 for a more rounded ellipse.','0|1','','')
newCmd('',' ','','','','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns cartoon rendering on and all other rendering (including labels) off.','*13','ONLY','')
newCmd('',' ','','',' Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._cartoon_radius','')
newCmd('.case','','','*v+12.0;math;flow','See {#.switch~}.','0|1','','')
newCmd('.cd',' ','','*v+11.8;cd','Changes the default directory or, with no parameters, displays the default directory. This command sets the Jmol parameter <b>defaultDirectory</b> and also, if a local directory, sets the default directory for writing files (application and signed applet). The directory can be local or it can have a URL prefix such as http:// or ftp://, and it may or may not be enclosed in quotes. Note that <b>x = file("") </b> sets x to the full path to the current default directory while <b>x = defaultDirectory</b> sets x to the current default directory as set by the user by the <b>cd</b> command (full path) or simply by setting the <b>defaultDirectory</b> parameter (which may or may not be the full path).','0|1','','')
newCmd('',' ','','*v+11.8;','Displays the default directory.','*1','','')
newCmd('',' ','','*v+11.8;','Resets the default directory to the base directory of the application or applet.','*2','&quot;&quot;','')
newCmd('',' ','','*v+11.8;','Changes the default directory to that specified in the directory name. Quotes are optional. The standard notation of two periods indicates "up one level" as in <b>cd ../temp</b>. Note that forward slash, not back slash, should be used to separate directory names along a path.','*3','&quot;directoryName&quot;','')
newCmd('',' ','','*v+11.8;','(signed applet or Jmol application only) Displays a file dialog box allowing for creating and changing directories on the local system.','*4','?','')
newCmd('',' ','','*v+11.8;','Changes the default directory to that specified in the URL directory specified in the {#.set(files and scripts)~loadFormat} variable.','*5','=','')
newCmd('.default','','','*v+12.0;math;flow','See {#.switch~}.','0|1','','')
newCmd('.center',' ','','centering','Sets the center of rotation to be the center of the set of atoms defined by the {#.atom expressions~atom expression}. This is calculated as the mean value of the coordinates of the selected atoms along each of the respective axes. If no atoms are selected then the center is set to the center of the bounding box (the default). The three values can be written using braces as a single coordinate, {{x y z}}, if desired.','0|1','','')
newCmd('',' ','','','Recenters the model on the default center.','1','','')
newCmd('',' ','','','Centers the model on the specified atom set. Along with {#.show~show center}, allows for the reading of the coordinated position of a specific atom using, for example, <b>center [CYS]4.O;show center;center</b>. See also {#.centerAt~}','*1','._atom_expression','')
newCmd('',' ','','','Centers the model on a specified model-frame coordinate','*2','._coordxyz','')
newCmd('',' ','','','Centers the model on a specified {#.draw~} object','*3','._draw_object','')
newCmd('.centerAt',' ','','centering','The <b>centerAt</b> command allows centering of the model in one of three different ways: based on an absolute coordinate position, based on an offset relative to the center of the boundbox (the overall application default), or based on an offset relative to the average position of the currently selected atoms. Centering on a specific atom or atom set without first selecting it is also available using the {#.center~} command. If the three numerical values are omitted, they default to 0.0 0.0 0.0. The numbers can be in the form of a coordinate (with  braces), {{x y z}}, if desired.','1|2','','')
newCmd('',' ','','','specifies an absolute coordinate for the center, in Angstroms','2','ABSOLUTE','x y z {0.0 0.0 0.0}')
newCmd('',' ','','','relative to the center of the boundbox, which is defined by the minimum and maximum atom center coordinates along each of the cartesian axes','2','BOUNDBOX','x y z {0.0 0.0 0.0}')
newCmd('',' ','','','relative to the average atom position (also known as the "unweighted center of gravity")','1','AVERAGE','x y z {0.0 0.0 0.0}')


newCmd('.color',' ','','*v+11.4 -- adds fully customizable color schemes;','In general, the color command takes the following syntax:<br /><br /><b>color [object] [translucent/opaque] [color, property, or color scheme]</b><br /> <br /> Colors can be designated as one of the {http://jmol.sourceforge.net/jscolors/#JavaScript%20colors~[standard JavaScript colors]}, as a red-green-blue triplet in square brackets, [255, 0, 255], as a red-green-blue triplet expressed as a six-digit hexidecimal number in brackets, [xFF00FF], as a triplet expressed as a point in red-green-blue color space  -- {{255,0,255}} or fractional {{0.5, 0.5, 1}} (Jmol 12.0). To specifiy a set of atoms to color, you can either select them first -- <b>select *.N?; color green</b> -- or specify them using {#.atomExpressions~atom expression} notation: <b>color {{*.N?}} green</b>.','1|2','','')
newCmd('',' ','','*v+11.4;','The color command takes several forms, depending upon the type of object being colored: an atom object, a bond object, a chemical element, or a model object. This section of the guide discusses each of these in turn:<br /><ul><li>{#.color (atom object)~}</li><li>{#.color (bond object)~}</li><li>{#.color (element)~}</li><li>{#.color (model object)~}</li><li>{#.color (named object)~}</li><li>{#.color (scheme)~}</li></ul> Additional information external to this documentation can be found in relation to {http://jmol.sourceforge.net/jscolors/~[Jmol color schemes]} and {http://jmol.sourceforge.net/jscolors/#JavaScript colors~[standard JavaScript color names and codes]}. In addition, a page is available that lays out the {misc/color.htm~[Jmol color command matrix]}. Color schemes may be customized using the {#.set userColorScheme~} command.','TEXT','[object]','')
newCmd('',' ','','','The color command allows an optional color modifier of TRANSLUCENT or OPAQUE, which can be used with any object, either alone or with a color. TRANSLUCENT can take an integer in the range 0-8 (indicating eighths -- 0, 1/8, 2/8, etc.), 32-255 (indicating fraction of 256), or a decimal in the range 0.0 to 1.0. Larger numbers are more translucent -- dimmer. Currently implemented transclucencies are -1 (Jmol 10 translucency), 0 (opaque), 0.125 (1, 32), 0.25 (2, 64), 0.375 (3, 96), 0.5 (4, 128), 0.625 (5, 160), 0.75 (6, 192), 0.875 (7, 224). Future versions of Jmol may include more options, so it is recommended that the decimal numbers be used. The default is TRANSLUCENT 0.5, which can be set using "defaultTranslucent = x.xx", where x.xx is a decimal number. For example:<br /><br /> color atoms TRANSLUCENT orange<br /> color ribbons TRANSLUCENT 0.5 [255, 165, 0]<br />select oxygen; color opaque.  <br /><br />If neither TRANSLUCENT nor OPAQUE is specified, OPAQUE is assumed. Thus, <b>color atoms red</b> and <b>color atoms OPAQUE red</b> are synonymous. ','TEXT','[translucent/opaque]','')
newCmd('','11.1.31','','*v+11.2;','Besides the standard ways of representing a color, Jmol allows coloring based on properties or by one of the known {http://jmol.sourceforge.net/jscolors/~Jmol color schemes}. If <b>color TEMPERATURE</b> is used, the color range will depend upon the setting of the {#.set rangeSelected~rangeSelected} flag. Coloring can also be carried out by SURFACEDISTANCE. <br /> <br />You can also color atom-based objects based on user-defined properties read from an external file or standard Jmol atom properties such as partialCharge that Jmol has read from the model file.  In the case of Jmol {#atomproperties~atom properties}, use the keyword PROPERTY followed by the name the property, as in <b>color atoms PROPERTY partialCharge</b>. In the case of user-defined properties (created using the {#.data~} command), which always start with "PROPERTY_", simply give the property name: <b>x = load("mydata.txt");data "property_mydata @x";select model=2;color atoms property_mydata</b>. When using either PROPERTY or PROPERTY_xxx, you can set the absolute range of values that will span the full spectrum of colors of the propertyColorScheme you have chosen. Simply add the keyword ABSOLUTE followed by the minimum and maximum values you wish to use for the two ends of the spectrum. So, for example: <b>color atoms property temperature ABSOLUTE 0.0 30.0</b>. ','TEXT','[color schemes and color PROPERTY xxx]','')
newCmd('',' ','','','Many objects inherit both color and opacity from the underlying associated atom (which, themselves "inherit" their color by default from their associated chemical element). For example, by default a bond will inherit the two colors+translucencies of its endpoint atoms. If you simply \'color atoms translucent\', then both the atoms and the bonds will be translucent. But if you \'color bonds opaque\' or \'color bonds red\' and also \'color atoms translucent\' only the atoms will be translucent. <br /><br />The level of \'translucent\' can be controlled. The algorithm allows for proper mixing of colors for two translucent objects and approximately correct mixing for more than two overlapping translucent objects. ','TEXT','Color Inheritance','')
newCmd('',' ','colorrasmol','','Sets the previously selected atom set to a color based on a color name or value or one of the {http://jmol.sourceforge.net/jscolors/~[Jmol color schemes]}, namely one of "roygb" (default rainbow), "bwr" (blue-white-red), "rwb" (red-white-blue), "low" (red-green), or "high" (green-blue). Jmol 12.0 adds  "bw" (black-white) and "wb" (white-black). ','*1','._color_name_or_scheme','')
newCmd('',' ','colorrasmol','*v+12.0;','The additional parameter TRANSLUCENT creates a gradient of translucency from transparent to opaque across the color scheme. (Jmol 12.0)','*1','._color_name_or_scheme','TRANSLUCENT')
newCmd('.color (atom object)',' ','','color;latom;','Sets the color of atom-related objects (atoms, backbone, cartoons, dots, halos, labels, meshribbon, polyhedra, rockets, stars, strands, struts (Jmol 12.0), trace, and vibration vectors).','1|2','','')
newCmd('',' ','coloratoms','','Sets the color of atom-related objects based on a previously selected atom set to a specific color, a  color scheme, or back to its default color (CPK), or to inherit the color of its associated atom (NONE). In the case of "color atoms", CPK and NONE both revert to the default color. In the case of "color labels", coloring a label to the background color automatically uses the background contrast color, not the actual background color (white or black, depending upon the background color). If it is desired for some reason to color a label the background color, then the label should be colored instead a color very close to the background color but not it exactly. For instance, to color labels black with a black background, use [0,0,1] instead of [0,0,0].','*2','._atom_object','._color_name_or_scheme')
newCmd('','13.1.16','','*v+13.2;','Sets the color of the space-fill sphere of an atom without coloring the bonds leading to that atom or other atom-based objects such as stars and polyhedra.','*3','BALLS','._color_name_or_scheme')
newCmd('.color (bond object)',' ','','color;hbondc','Three types of bonds are distinguished by Jmol for coloring purposes: regular bonds, disulfide bonds, and hydrogen bonds. Each can be colored independently, and hydrogen bond colors in proteins can take on a special coloring scheme based on their connectivity.','1|2','','')
newCmd('',' ','','*v+11.4;','Colors selected bonds a specific color or resets them to inherit the color of their associated atoms.','*1','BONDS','._color_or_none_or_CPK')
newCmd('',' ','','*v+11.4;','Colors disulfide bonds a specific color or resets them to inherit their color from their associated atoms.','*2','SSBONDS','._color_or_none_or_CPK')
newCmd('',' ','','*v+11.4;','Colors hydrogen bonds a specific color or resets them to inherit their color from their associated atoms. Additional HBOND color options include TYPE  and ENERGY','*3','HBONDS','._color_or_none_or_CPK')
newCmd('',' ','','','Colors hydrogen bonds based on DSSP "energy" (see {#.calculate~calculate HBONDS} for details).','*4','HBONDS','ENERGY')
newCmd('',' ','','','Colors hydrogen bonds specifically in proteins based on how many residues one end is from the other. Note that to get this effect, one must first execute "hbonds ON" and then issue "color hbonds TYPE".'+' The colors assigned are based on the number of redidues between the interacting H atoms. This TENDS to indicate secondary structure, but is not perfect.The correlation between color and offset are as follows:'+'[||<b>Color</b>|<b>Offset</b>||green|-4||cyan|-3||white|+2||magenta|+3 (turns)||red|+4 (alpha-helix)||orange|+5||yellow|other (e.g. beta-pleated sheet)||]','*4','HBONDS','TYPE')
newCmd('.color (element)',' ','','*v+10.2 -- new;color','You can use the \'color\' command to specify customized default colors that are used for elements. {http://jmol.sourceforge.net/jscolors/#Atoms%20(\'CPK%20colors\')~[default Jmol element colors]} <br /><br /><b>color carbon limegreen<br />color hydrogen [x32CD32];</b><br /><br />These changes are not molecule-specific; they will continue in effect even if new molecules are loaded. However, in a page with multiple applets, each applet will have its own set of element colors.<br /><br />If you choose to use this feature, you should consider encapsulating your favorite colors into a script and then executing that script as a subroutine. For example:<br /><br /><b>script LoadMyFavoriteColors.txt;<br />load foo.xyz;<br />load bar.xyz;</b><br /><br />Note:<br /><ol><li>Custom element colors are independent of and are not affected by the currently selected set of atoms.</li><li>To reset custom element colors, use {#.set%20(default%20color%20scheme)~\'set defaultColors Jmol\'} or \'set defaultColors Rasmol\'.</li><li>\'translucent\' or \'opaque\' cannot be specified as part of the element color specification. (You cannot \'color carbon transparent green\', for instance.)</li><li>At this time only elements can be custom colored. There is no support for customizing other color palettes such as those used for protein chains or groups.</li></ol>','2','._element_name','._colorRGB')
newCmd('.color measures',' ','','*v+11.0 -- adds selective measurement coloring;color','Colors the measurement numbers and dotted lines.  In Jmol 10.2, "color measures" change all measurement colors at once. In version 11.0, "color measures" acts on all future measures, allowing for selective coloring of measurements. Thus, "color measure" in 11.0 acts on (a) any measures currently with no color assigned and (b) on any future measures. If measurement colors have already been set, then "color measures NONE" needs to be invoked to turn off measurement colors prior to resetting them.','2','._colorRGB','')
newCmd('.color (other)',' ','','*v+11.0 -- new;','Additional colorable objects include selectionHalos, highlights (Jmol 12.0), and objects that have IDs such as isosurfaces and objects created with the draw command.','1|2','','')
newCmd('','12.0.RC15','','*v+12.0;color','Sets the color of the {#.set(highlights)~highlight} ring.','*1','HIGHLIGHT','._colorRGB')
newCmd('','12.1.50','','*v+12.2;color','Colors the section of the isosurface nearest the given atomSet with the given color. Similar to the PyMOL command "set surface_color, (color), (selection)". For example, <b>isosurface sasurface; color isosurface {{hydrophobic}} white</b>','*2','ISOSURFACE','{{atomSet}}','._colorRGB','')
newCmd('','12.1.51','','*v+12.2;color','Colors an isosurface with a range of <b>n</b> colors from one color to another color. For example, <b>color isosurface red to white 30</b>. ','*3','ISOSURFACE','._colorRGB','TO','._colorRGB;+ (integer)')
newCmd('','12.3.5','','*v+12.2;color','Colors an isosurface such as an atomic or molecular orbital with two colors after it has been created.  (Jmol 12.3.5)','*4','ISOSURFACE','PHASE','._colorRGB','._colorRGB;+ (integer)')
newCmd('',' ','','*v+11.0;color','Sets the default color of the halos displayed by {#.selectionHalos~}. The <i>default</i> default selection halo color is GOLD. To assign colors based on the underlying atoms, use <b>color selectionHalos NONE</b>. This command setting persists for the life of the applet or application, like element colors. If the atom\'s halo color has been set using <b>select ...; color halos ...</b>, then <b>color selectionHalo</b> has no effect until those colors are returned to their default settings with <b>color halos none</b>.','*5','SELECTIONHALOS','._color_or_none')
newCmd('','','','','Sets the color of an object created using {#.draw~}, {#.isosurface~}, {#.pmesh~} or {#.polyhedra~} using the name identifier preceded by a dollar sign ($). ','*6','._draw_object','._colorRGB')
newCmd('.color (model object)',' ','colortext','color','Sets the color of various model objects.','2','._model_object','._colorRGB')

newCmd('.color (scheme)','11.3.13','','*v+11.4 -- new;','Sets the working color scheme and range for determining correlations between property values and color values. The names "user" and its reverse, "resu", refer to a {#.set userColorScheme~user-defined color scheme}. The RANGE values indicate what parameter values correspond to the first and last color values. The range values are required unless they have been set already in a previous command such as {#.isosurface~}. Once set, calculations such as <b>x = {{atomno=3}}.partialcharge.color</b> return a color value based on this color scheme and range. You can implement any number of your own color schemes simply by including "=" after the name and adding a set of hexadecimal color values: <b>color "myScheme=[x00FF00] [xFFFF00] [x00FFFF] [x0000FF]"</b>, for example. Then, <b>color atoms property atomno "myScheme"</b> will do just that. You can redefine the Jmol coloring schemes as well simply by using "jmol=[x......]...." or "rasmol=[x......]...." as the name. To return these to their default values, include the equal sign but no values. Four additional built-in color schemes include "byElement_Jmol", "byElement_Rasmol", "byResidue_Jmol" (corresponding to <b>color shapely</b>), and "byResidue_Rasmol" (corresponding to <b>color amino</b>). Setting the range has no effect on these color schemes, as they are intended to correspond with specific elements (elemno) and residues (groupID). You can implement your own byElement or byResidue color schemes simply by using those keywords at the beginning of a color scheme definition. byElement color schemes start with "unknown", then run through the periodic table; byResidue color schemes start with "no group", then the amino acids and nucleic acids in order: ALA, ARG, ASN, ASP, CYS, GLN, GLU, GLY, HIS, ILE, LEU, LYS, MET, PHE, PRO, SER, THR, TRP, TYR, VAL, ASX, GLX, UNK, A, +A, G, +G, I, +I, C, +C, T, +T, U, +U.','2','&quot;colorSchemeName&quot;','RANGE','[min]','[max]')
newCmd('.[comment (#)]','11.3.42','comment','*v+11.4 -- adds /**   **/ and ****Jmol Embedded Script****;','Comments in Jmol are preceded by a number sign, \'#\'. The pattern /* comment text  */ may also be used, and /** ... **/ can be used as a "super-comment" to comment out whole blocks of script that might contain /* ... */ as well. // at any point in a line acts as in JavaScript to indicate the beginning of comments for that line. If the character string <b>****Jmol Embedded Script****</b> is found within a comment or super-comment block, then ONLY the text following that string and carrying through the rest of the comment or super-comment is processed, and all other text in the file is ignored. [||`valign="top"><b>#</b>| Anything following \'#\' up until the end of a statement is ignored by Jmol with the following three exceptions. (A statement is terminated by a semicolon ";" or a newline.)||`valign="top"><b>#jx</b>|Commands prefixed with #jx <b>will be executed by Jmol</b>||`valign="top"><b>#jc</b>| If the string \'#jc\' appears <b>anywhere</b> within a statement, then that <b>entire statement</b> will be assumed to be a comment and will be completely ignored by the Jmol interpreter.||`valign="top">state definitions| The {#.write~write state} command produces a script that uses standard-looking Jmol comments appended to some commands in order to supply necessary information for defining the exact state that was present when the command was issued. Generally this involves what atoms are to be selected or ignored, and what files and models are involved. The formatting of these extended script "comments" is very precise; these comments should not be manipulated.||]CHIME NOTE: Similar comment controls exist in Chime. Commands prefixed with #! will be executed in Chime but not in RasMol. Commands containing ## will be ignored by Chime, but the portion preceding the ## will be executed in RasMol. Thus we have: [||# | not read by Jmol, Chime, or Rasmol||#jx [commands here] | Jmol excecution only||#! [commands here]  | Chime execution only||[commands here] ## #jc |Rasmol execution only||[commands here] #jc | Chime and Rasmol only || [commands here] ## | Jmol and Rasmol only||]','0','','')

newCmd('.[immediate (!)]','11.1.12','','*v+11.2;pause','Generally Jmol runs with a script "queue" meaning that as you enter script commands they are stacked up and run in succession. However, there are times when it is desirable to temporarily interrupt a running script. From the applet or application console, or from JavaScript using Jmol buttons or the jmolScript() function, including an exclamation point (!) at the VERY beginning of a script serves as a flag to tell Jmol to temporarily interrupt any running script in order to execute the script that follows. For example, <b>!spin off</b> will turn spinning off immediately, even if another script is running. The interruption occurs just prior to the next command in the running script, never during an actual command execution. In effect, the interruption script is simply slipped into the running script at the next available point, and then the running script continues. In addition, <b>!quit</b> and <b>!exit</b> are slightly different than {#.quit~} and {#.exit~} in that they not only stop the currently running script (or all scripts in the case of <b>!exit</b>), they also stop spinning, animation, vibration, and minimization that might be in process.','0','','')
newCmd('.compare','11.9.35','','*v+12.0 -- new;','Compares two models and optionally reorients the first model relative to the second based on a given atom-atom coordinate pairing or quaternion-based group-group orientation pairing. References to the atom-atom correlation algorithm can be found in the literature [1] and [2]. Quaternion-based orientation pairing is an unpublished technique specific to Jmol at this point. It minimizes the standard deviation of the correlated quaternion frames for groups in the two models using spherical averaging. (Quaternion comparison is based on minimizing the difference in <i>orientation</i>, not <i>position</i>. Results of this option depend upon the setting of <b>set quaternionFrame</b>).<br /><br />Except for the BONDS option, by default the command does not move any atoms and just reports RMSD.  A return of "NaN" indicates that the specified atom sets could not be matched because there was not a 1:1 correlation of atoms.  The independent options ROTATE and TRANSLATE allow the option to do just rotation, do just center-of-mass translation, or do both. An optional number of seconds can be added and can be 0 to effect no animation. (The default is 1 second.) For example:[||`style="width:500px"> <b>compare {{2.1}} {{1.1}} SUBSET {{*.CA}}</b> | reports RMSD for the comparison of two models based on alpha carbons only. || <b>compare {{2.1}} {{1.1}} ROTATE TRANSLATE 2.0</b> | Compares the second model to the first and rotates and translates it into superposition with the first model over the course of 2 seconds.||<b>load files "$tyrosine" "$lysergamide";frame *;<br />compare {{1.1}} {{2.1}} BONDS "c1ccccc1CCN" rotate translate</b> | flexibly fits tyrosine onto lysergamide based on the aromatic ring and C-C-N atoms only, overlaying the structures over 1 second. Note that the required initial internal bond rotations are carried out even if ROTATION and TRANSLATION are not indicated.||] <br /><br />See also the {#functionsx=fyfunctions~compare()} function. [1] {http://www.opticsinfobase.org/viewmedia.cfm?uri=josaa-4-4-629&seq=0~Berthold K. P. Horn, "Closed-form solution of absolute orientation using unit quaternions"  J. Opt. Soc. Amer. A, 1987, Vol. 4, pp. 629-64.} [2] G. R. Kramer, "Superposition of Molecular Structures Using Quaternions" Molecular Simulation, 1991, Vol. 7, pp. 113-119.','1|2','','')
newCmd('','11.9.35','','*v+12.0;','Compares the specified subset of atoms of two models, minimizing the RMSD of paired atom coordinates. If the SUBSET option is not specified, the models are matched atom-for-atom based on "SPINE" atoms (*.CA, *.N, *.C for proteins; *.P, *.O3*, *.O5*, *.C3*, *.C4*, *.C5 for nucleic acids). The keyword ATOMS is optional and may be followed by any number of specific pairs of atom sets to be correlated. For example, <b>compare {{2.1}} {{1.1}} SUBSET {{*.CA}} ATOMS {{10-20}} {{50-60}} {{30-40}} {{20-30}} ROTATE TRANSLATE</b> will correlate the positions of alpha carbon atoms in groups 10-20 and 30-40 of model 2.1 with corresponding atoms in groups 50-60 and 20-30 of model 1.1 and move all atoms in model 2.1 in the process. The result of this atom-atom pairing comparison is essentially the same as the PyMol pair_fit command, though easier to implement and using an exact form of the structure-structure correlation rather than an iterative process.','*1','{{model1}} {{model2}}','SUBSET','{{atomSet}} ATOMS [paired atom list]','')
newCmd('','11.9.35','','*v+12.0;','Compares the specified subset of atoms of two models, minimizing the RMSD of paired group quaternion orientations. If no other parameters are included, the models are matched group for group based on the current setting of <b>quaternionFrame</b>. The keyword ORIENTATIONS may be followed by any number of specific pairs of atom sets to be correlated. For example, <b>set quaternionFrame "C"; compare {{2.1}} {{1.1}}  QUATERNIONS {{10-20 or 30-40}} {{50-60 or 80-90}} ROTATE TRANSLATE</b> will correlate the orientations of alpha carbon atoms in groups 10-20 and 30-40 of model 2.1 with corresponding orientations in groups 50-60 and 80-90 of model 1.1 and move all atoms in model 2.1 in the process. The result of this orientation pairing comparison gives the best fit of orientations and the best translation, but not necessarily the best rotation to fit coordinate positions. The ORIENTATION option has no corresponding PyMol command.','*8','{{model1}} {{model2}}','ORIENTATIONS','[paired atom list]','')
newCmd('','11.9.35','','*v+12.0;','The ORIENTATION option allows for explicit comparison of quaternion arrays rather than atom lists. The result is independent of the setting of  <b>quaternionFrame</b>. These arrays can be specified in terms of variables. For example: <b>qset1 = quaternion({{10-20:A/1.1}}); qset2 = quaternion({{20-30:D/2.1}}); compare {{2.1}} {{1.1}} ORIENTATIONS @qset2 @qset1</b>.','*9','{{model1}} {{model2}}','ORIENTATIONS','[paired quaternion array list]','')
newCmd('','12.0.RC12','','*v+12.0;','This option allows the  alignment of two structures based on SMILES or SMARTS atom matching. The basic idea is to use a SMILES (whole molecule) or SMARTS (substructure) description to find the atoms in one structure that correlate one-for-one with atoms in the second structure, then find the rotation and translation that best aligns them. ','*4','{{model1}} {{model2}}','SMARTS or SMILES','&quot;smartsString&quot;','')
newCmd('','13.3.5','','*v+13.4;','Compares a  set of atom coordinates to a corresponding ordered set of points.','*5','{{atomSet}}','[array of coordinates]')
newCmd('','13.3.5','','*v+13.4;','Compares a subset of atom coordinates to a corresponding ordered set of points.','*6','{{atomSet}}','ATOMS','{{subset1}} [coords1] {{subset2}} [coords2]. . .','')
newCmd('','13.3.6','','*v+14.4;','The BONDS option  does a flexible fit by first constructing a list of matching dihedrals based on the  SMARTS (substructure) string, rotating the bonds into position, then doing a COMPARE operation based on the matching atoms. The preliminary rotation is carried out even if ROTATE and TRANSLATE are not present. In  the case of <b>compare {{1.1}} {{2.1}} BONDS "c1ccccc1CCN"</b>, the initial internal rotation can be done in a prelminary step using the following command: <b>rotate branch @{{compare({{1.1}},{{2.1}},"c1ccccc1CCN","BONDS")}} 0</b>. See {#.rotate_branch~rotate BRANCH} for details.','*7','{{model1}} {{model2}}','BONDS','&quot;smartsString&quot;','')




newCmd('.configuration',' ','','*v+11.0 -- new;','File types PDB, mmCIF, and CIF allow for the designation of certain atoms to be in "alternative locations" or in "disorder groups". This leads to two or more possible structures. While full treatment of this issue is not possible, Jmol can display the different possible "configurations" described in these files. for PDB and mmCIF files. This command selects the speficied configration (but does not [#.restrict] or [#.display] it, and also resets all biomolecular shapes (cartoons, traces, etc.) to use the values of that configuration. See also  the atom property {#.atomProperties~configuration}, which allows selection of configurations without updating shapes (Jmol 12.0). ','1','[configuration number]','')
newCmd('.connect',' ','','*v+11.4 adds "_1" option;bond','The <b>connect</b> command allows real-time bond manipulation, allowing the user or application to connect and disconnect specific atom sets.  The general syntax is as follows:  <br />  <br /><b>connect [minimum and maximum distances] [source and target atom sets]  [bond type] [radius option] [color option] [modify/create option] </b> <br /> <br />(<b>connect</b> by itself deletes all bonds and then creates bonds based on Jmol default bond-generating algorithms, all as single bonds, without respect to what bonding patterns might have been indicated in the model file.)','0|2','','')
newCmd('',' ','','','Distances are given in Angstroms, either as decimals or integers. If only one distance parameter is given, it represents a maximum distance. If neither the minimum nor the maximum distance parameter is given, all connections between the two atom sets are made, regardless of distance.  Starting in Jmol 12.0, the option to connect atoms based on ranges of percentage of bonding/ionic radii instead of fixed values can be specified using a % sign after a distance.','TEXT','[minimum and maximum distances]','')
newCmd('',' ','','','The source and target atom sets are embedded {#.atom expressions~atom expressions} and therefore must be enclosed in parentheses. If the source atom set is not given, it is taken to be the currently selected atom set, "(selected)". If neither atom set is given, "(selected) (selected)" is assumed. The atom expression "_1" in the second selection set signifies "the atom selected in the first set". Thus, it is possible to select something like <b>(_N) (_O and not within(chain, _1))</b> -- meaning "all nitrogens and all oxygens not in the same chain as the selected nitrogen." Of course, this would be used with a distance qualifier.','TEXT','[source and target atom sets]','')
newCmd('',' ','','hbond','Unless otherwise specified, connections are automatically introduced as single bonds. Any one of the following bond types may be specified: SINGLE, DOUBLE, TRIPLE, QUADRUPLE, AROMATIC, PARTIAL, PARTIALDOUBLE, HBOND, STRUT (Jmol 12.0) or UNSPECIFIED. In appearance, AROMATIC and PARTIALDOUBLE are identical except for which side of the bond is represented by a dashed line. PARTIAL and HBOND are both dashed, but they have different patterns, and newly created hydrogen bonds are only thin lines. Additional bond types include PARTIALTRIPLE, PARTIALTRIPLE2, AROMATICSINGLE, AROMATICDOUBLE, numeric bond order (including -1 for PARTIAL, 1.5 for AROMATIC, -1.5 for PARTIALDOUBLE, 2.5 for PARTIALTRIPLE, and -2.5 for PARTIALTRIPLE2), as well as a fully generalized set of partial bonds indicated with <br /><br />connect ... partial N.M<br /><br />where N is the number of lines (from 1 to 5) and M is a binary mask indicating which lines are dashed:  <br />[||M|binary|meaning||1|00001|first line dashed||2|00010|second line dashed||3|00011|first and second lines dashed||4|00100|third line dashed||...|...|...||31|11111|all lines dashed||] <br /> <br />So, for example, we have: <br /> <br />[||partial 1.0|single||partial 1.1|same as "partial"||partial 2.0|double||partial 2.1|same appearance as "aromatic", though not "aromatic"||partial 2.2|partialDouble||partial 3.0|triple||partial 3.1|partialTriple||partial 3.4|parialTriple2||]','TEXT','[bond type]','')
newCmd('',' ','','*v+11.2;','Addition of the keyword "radius" followed by a distance in angstroms allows definition of the radius of a modified or newly created bond. If the modify/create option is absent, then "modify" is assumed; if the bond type is absent, then bonds of any type are set, but their bond type is not changed.','TEXT','[radius option]','')
newCmd('',' ','','*v+11.2;','Addition of a color name or designation optionally along with the keyword "translucent" or "opaque" allows definition of the color and/or translucency of a modified or newly created bond. If the modify/create option is absent, then "modify" is assumed; if the bond type is absent, then bonds of any type are set.','TEXT','[color option]','')
newCmd('',' ','','*v+11.2;','Four additional mutually exclusive options relate to what sort of connections are made. The default when a radius or color option is present is "Modify"; otherwise the default is "ModifyOrCreate". These include: <br />[||<b>Create</b> | Only new bonds will be created. If a bond of any type already exists between two matching atoms, it will not be affected.||<b>Modify</b> | Only pre-existing bonds will be modified. No new bonds will be created. ||<b>ModifyOrCreate</b> | If the connection fits the parameters, it will be made. Bonds already present between these atoms will be replaced.|| <b>Delete</b> | Delete the specified connections.||]','TEXT','[modify/create option]','')

newCmd('.console',' ','','*v+11.0 new command;','Throws up a console window from which a user can enter script commands and monitor the messages returned by Jmol as, for example, from the {#.show~} or {#.getProperty~} command. ','0','','')
newCmd('.contact','12.2.0','','*v+12.2 -- new;','The <b>contact</b> command, introduced in Jmol 12.2, is a powerful command capable of displaying a wide range of contact surfaces in stunning detail relating to van der Waals interactions, hydrogen bonds, and clashes. The general format of the <b>contact</b> command is <br /><br /><b>contact ID "id" {{atomSet1}} {{atomSet2}} [contactOptions] [colorOption]</b><br /><br /> The first atom set is the "target" atom set, typically a ligand or a subset of a protein. The target atom set is required. Its Van der Waals radius will be used. The second atom set is the interacting atom set and is optional, defaulting to &quot;all other atoms&quot;. For example: <b>contact {{ligand}}</b> or <b>contact {{ligand}} {{oxygen or nitrogen}}</b>. The interacting atom set Van der Waals radius can be adjusted using absolute (<b>1.2</b>), percentage (<b>120%</b>), or Angstrom offset from 100% (<b>+1.2</b>) relative to the Van der Waals radius of atomSet2. Several contact options are available and are listed below. Color options are similar to those available for {#.isosurface~isosurfaces}. The <b>color density</b> option produces a result roughly similar to {http://kinemage.biochem.duke.edu/software/probe.php~Kinemage Probe}. Examples can be found at {http://stolaf.edu/people/hansonr/jmol/contact/gordon.htm~} and {http://stolaf.edu/people/hansonr/jmol/contact/ligand.htm~}. [|| |<center><img src="img/contact1b.jpg" alt="" /></center> |<center><img src="img/contact1.jpg" alt="" /></center> |<center><img src="img/contact2.jpg" alt="" /></center> || |<img src="img/contact3.jpg" alt="" /> |<img src="img/contact5.jpg" alt="" /> |<img src="img/contact4.jpg" alt="" /> ||]<br /><br /><b>contact options</b><br /><br />[|| hbond, vdw, clash | just hydrogen bonds, van der Waals near-misses, or van der Waals overlaps that are not hydrogen bonds (clashes). || NCI | {http://pubs.acs.org/doi/abs/10.1021/ct100641a~NCIPlot} promolecular calculation results.|| surface | van der Waals surface around the atom set colored by contact proximity. || cap | similar to <b>surface</b>, but with clashing sections capped off. (Specifically designed for prototyping.) || trim | similar to <b>surface</b>, but trimmed to only display van der Waals radius overlaps. || full | same as <b>trim</b> but with both overlapping van der Waals surfaces displayed. || plane | approximately planar circles of the voronoi surface between the groups trimmed to display only the desired area. || connect | cigar-shaped objects that connect one interacting atom with another. ||]','0','','')


newCmd('.data',' ','','*v+11.6 adds new capabilities;vdw','The <b>data</b> command allows data to be introduced in-line or via a variable. The command consists of two statements, <b>data "label"</b> and <b>end "label"</b>, between which the data are presented on as many lines as desired. "label" may be any string, though strings starting with "property_" are special (see below). Quotes must be used in both the <b>data</b> line and the <b>end</b> line. The first word of the label defines the data type, but the label itself may be any number of words. If the data type is "model" as in the following example, then the data is interpreted as an in-line model (and loaded using the default lattice, if crystallographic). If the data type is "append", then the data is interpreted as a model, and the model is appended either into a new frame or into the last existing frame, based on the setting of <b>set appendNew</b>. Additional data types may be loaded and later {#.show~shown}, but they will be ignored by Jmol.<b>[||background red;<br />data "model example"<br />2<br />testing<br />C 1 1 1<br />O 2 2 2<br />end "model example";show data||]</b><br />Note that the <b>data</b> statement itself should not include a semicolon at the end. In the specific case of a model file, if it is desired to use no new-line characters, you can start the data with  &brvbar; (vertical bar) and then use a vertical bar to separate all lines: <b>data "model example"&brvbar;2&brvbar;testing&brvbar;C 1 1 1&brvbar;O 2 2 2&brvbar;end "model example";show data</b>. For this option you MUST start the data with a vertical bar immediately following the quotation mark closing the label or on the very next line. If the first character is a vertical bar or a new-line character, it is not part of the model. To include data representing more than one file, first define a data separator using <b>set dataSeparator</b>, for example, <b>set dataSeparator "@TILDE@TILDE@TILDE"</b>. Then use that separator between data sets. The <b>data</b> command thus provides an alternative to the JavaScript Jmol.js (applet-only) <b>loadInline()</b> function. It can be included in any {#.script~}, and commands can come before and after it for further processing. Note that model data in the system clipboard can also be pasted into the applet console or endered into the application using Edit...Paste for direct introduction into Jmol. Starting with Jmol 12.0, using {#.load DATA~load data} instead of just <b>data</b>  you can load model data with all of the loading options of the standard LOAD command. See also {#.show~show data}, {#.getProperty~getProperty data}, and {#.load~load "@x"}. ','1|2','','')
newCmd('',' ','','*v+11.6;props','Jmol 12 Note: If you just want to assign a set of properties to a set of atoms, and you have the properties in a file as a list of numbers, you do not need the DATA command. Simply load the values straight into an atom property directly from the file: <br /><br /><b>{{*}}.property_xxx = load("myfile.dat")</b><br /><br />Atom property data may be loaded into Jmol using the DATA command.  In addition, when reading some model file types (CRYSTAL, specifically) Jmol will create property_xxx data automatically -- check the Java console after file loading to see a report of what properties were assigned. To assign properties using the DATA command, first select the atoms to which data are to be assigned. Then, to assign the data, use <b>DATA "property_xxx"...end "property_xxx"</b>, where "xxx" is any alphanumeric string. Properties are assigned sequentially to the currently selected atom set, and will be assigned to atoms in the selection set one after another until either the data or the atom list is exhausted. If the data are exhausted before the selected atoms, then the value 0 is recorded for each of the remaining selected atoms. In this way, if only a few atoms need data, only a few can be selected and assigned values. Properties can be checked using atom labels: (<b>label %[property_xxx]</b>). The following special values for xxx read data into Jmol exactly as though read from a model file:  atomName, atomType, coord,  element, formalCharge, occupancy (0 - 100), partialCharge, temperature, valence, vanDerWaals, and vibrationVector. Any other label will be saved simply under its property name. <br /><br />To read selected data that is in tab- or space-separated format (for example, from a spreadsheet), specify which column the data is in using <b>set propertyDataField = n</b>, (n = 1 being the first column), prior to the data command. Lines having fewer than <b>n</b> tokens will be ignored. If the data is fixed-column format, then <b>n</b> is the starting column number, and set <b>propertyDataColumnCount</b> to be the number of columns to assign to this field. Lines shorter than the required number of characters will be ignored. Setting <b>propertyDataField</b> to 0 indicates that no data field is present, and data are to be read sequentially. <br /><br />Atom selection need not be contiguous. If you want to associate specific data with specific atom numbers, a column containing these atom numbers (starting with 1 for each model loaded) can be specified using <b>propertyAtomNumberField = m</b>. If the data is fixed-column format, then <b>m</b> is the starting column number, and also set <b>propertyAtomColumnCount</b> to be the number of columns to assign to this field. Specifying 0 for <b>m</b> indicates that the set of currently selected atoms should be assigned values from the data.<br /><br />Atom property data may also be loaded from variables. To do this, use add "@x", where x is a variable name <i>within</i> the quotation marks defining the first parameter of the data command: <b>DATA "property_partialCharge @x"</b> or <b>DATA "property_mydata 66 3 @x"</b>. No <b>end</b> line is required. The variable x should already contain a list of numbers, perhaps from using the x = {#.functions~load}("myfile.dat"); perhaps just by creating a string of numbers: x = "2.3 3.4 5.6 7.8...".  <br /><br />The {#.functions~data()} function also allows direct conversion of data into arrays, which can be directly stored in an atom property using, for example, <br /><br /><b>{{*.CA/2.1}}.temperature = data(load("mydata.dat"),6,0,3)</b><br /><br />meaning, "Read the file mydata.dat and store the sixth free-format field of each line starting from the third line as the temperature of the C-alpha carbons of model 2.1." ','TEXT','Setting atom properties','')
newCmd('',' ','','*v+11.0;','Defines a set of data in line, ending with a matching <b>end "label"</b>, where "label" is any string. Quotes are required. Certain labels have special meaning and are described more fully below.','*11','&quot;label&quot;','')
newCmd('',' ','','*v+11.0;','Clears the data table.','*2','CLEAR','')
newCmd('','','','*v+11.0;','Defines the USER set of van der Waals radii on an element-by element basis. Entries may be separated by semicolons or entered one per line. . In the example given here, carbon (atomic number 6) is given a radius of 1.7, and nitrogen (atomic number 7) is given a radius of 1.8. ','*3','&quot;element_vdw&quot; 6 1.7; 7 1.8 END &quot;element_vdw&quot;','')
newCmd('','','','*v+11.6;','Defines a set of data with the given label from a variable (variable <b>x</b> in this case). Quotes are required, but no <b>end</b> command is required.','*12','&quot;label @x&quot;','')
newCmd('','','','*v+11.6;','Defines a data set to be paired x,y data inline and ending with <b>end data2d_xxx</b>, where xxx can be any alphanumeric string.','*13','&quot;data2d_xxx&quot;','')
newCmd('','','','*v+11.6;','Defines an atom property  based on data provided next inline and ending with <b>end property_xxx</b>, where xxx can be any alphanumeric string. <b>propertyAtomField</b>, <b>propertyAtomColumnCount</b>,  <b>propertyDataField</b>, and <b>propertyDataColumnCount</b> override set values for these parameters.','*16','&quot;property_xxx propertyAtomField propertyAtomColumnCount propertyDataField propertyDataColumnCount&quot;','')
newCmd('','','','*v+11.6;','Defines an atom property  based on data provided next inline and ending with <b>end property_xxx</b>, where xxx can be any alphanumeric string. <b>propertyAtomField</b> and <b>propertyDataField</b> are optional, and if provided override the set values for <b>propertyAtomField</b> and <b>propertyDataField</b>.','*15','&quot;property_xxx propertyAtomField propertyDataField&quot;','')

newCmd('.define',' ','','*v+11.2 -- new DYNAMIC_ prefix;rreset','Defines the specified variable to the the atoms selected by the {#.atom expressions~atom expression}. In general, definitions are "static" in the sense that when they are generated with the <b>define</b> command, they represent the matching set of atoms indefinitely. However, if "DYNAMIC_" is at the beginning of the variable name, as, for example, "DYNAMIC_myatoms", then whenever that variable is used it is recalculated. (When used, the prefix "DYNAMIC_" should be dropped.) <br /><br />IMPORTANT NOTE: The <b>define</b> command should be used with some discretion. If you should define a term that later becomes a reserved keyword to any command in any future version of Jmol, your page may not be compatible with that future version. A simple way to avoid this situation is to put a tilde (@TILDE) in front of any definition you make. <b>Do not use a leading underscore (_)</b>, as there are many "hidden" reserved definitions that start with that character. ','2','._variablename','._atom_expression')

newCmd('.delay',' ','','pause','Causes the screen to refresh and the script to stop executing for the specified number of seconds. See also {#.set_delaymaximumms~set delayMaximumMs}.','1','.on','')
newCmd('',' ','','','','1','._time_delay','')
newCmd('.delete','11.9.19','','calc','The DELETE command deletes atoms or identified objects.  <b>DELETE</b> by itself deletes all atoms, similarly to the way <b>select</b> by itself selects all atoms.','1|2','','')
newCmd('','','','','Deletes the specified atoms. For example: <b>DELETE hydrogen</b> or <b> DELETE atomno > 30 </b>. Note that <b>delete</b> by itself deletes all atoms, similarly to the way <b>select</b> by itself selects all atoms.','2','._atom_expression','')
newCmd('','','','','Deletes the specified objects (from {#.draw~}, {#.isosurface~}, etc. using the ID parameter). Wildcards may used. For example, <b>delete $d*</b> will delete all identified objects starting with the letter d. <b>delete $*</b> deletes all such objects.','3','$','[object id]')


newCmd('.depth','11.1.5','','slabdepth',' Slab and Depth together control the percentage of the molecule to be displayed based on clipping planes. See {{#.slab~}}.','1|2','','')
newCmd('.dipole',' ','','*v+11.0 -- NEW;','The <b>dipole</b> command allows for the drawing of a bond or molecular dipole arrow with or without a cross near the tail. Note that without the cross, since it can be drawn from any point in molecular space to any other, a "dipole" can be used by a web page developer for a simple arrow in order point to some particular aspect of the model having nothing to do whatsoever with dipole moments.The values of each dipole can be set by the user or will be estimated using partial charge data or molecular dipole information if available in the loaded model file. Only a very crude calculation is used to estimate at least the direction of all bond dipoles. The general syntax of the <b>dipole</b> command is as follows:  <br />  <br /><b>dipole [objectID] [modifying parameters] [positions]  </b> <br /> <br />','0|1','','')
newCmd('',' ','','*v+11.6;','The optional identifier such as "bond1" that can be referred to in later scripts as $bond1. These words are arbitrary and if preceded by the optional keyword ID may be any string. The special identifier <b>BONDS</b> (without the keyword ID) refers to the entire collection of bond dipoles -- those dipoled defined specifically as between two atoms. Similarly, the special identifier <b>MOLECULAR</b> (without the keyword ID) is primarily for files for which molecular dipole information is available. The value and placement of this dipole can also be set using the <b>dipole</b> command.','TEXT','ID [object id]','')
newCmd('',' ','','','The dipole is defined using a small set of parameters. These include:<br />[||<b>CROSS</b><br /><b>NOCROSS</b>| include (default) or do not include a 3D cross near the tail of the arrow.   ||<b>DELETE</b>|Deletes the specified dipole if an identifier is given or all dipoles if no identifier is given; not used with any other parameters.||<b>WIDTH x.xx</b>|The width of the dipole in Angstroms. The default value is 0.005 Angstroms.||<b>ON/OFF</b>|Turns on or off the specified dipole or all drawn objects if no identifier is given; not used with any other parameters.||<b>OFFSET x.xx</b><br /><b>OFFSET n</b> | Dipoles are by default centered between the two endpoints. The OFFSET value sets the offset of the dipole from this position along the axis of its endpoints. In Angstroms if a decimal number is given; in percent of the distance between the two endpoints if an integer is used. ||<b>OFFSETSIDE x.xx</b>|The offset of the dipole in Angstroms perpendicular to the axis of its endpoints. The default value is 0.4 Angstroms for atom-based dipoles and 0 for the molecular dipole.||<b>VALUE x.xxx</b> | A decimal number indicates the value of the dipole. Overall scaling is accomplished either by setting this number or using {#.set (misc)~set dipoleScale}. The VALUE keyword is optional.||]','TEXT','[modifying parameters]','')
newCmd('',' ','','','The positions of the endpoints of the dipole are set either using embedded atom expressions in parentheses, such as (atomno=1), or using a specific point in molecular space, {{x y z}}, either as a cartesian coordinate or a {#.fractional coordinates~fractional unit cell coordinate}. If two atoms are designated, then the dipole becomes a member of the "bonds" dipole collection and can be colored with that group. If a set of atoms is indicated, the geometric center is used. Thus, (*) indicates the geometric center of the molecule.','TEXT','[positions]','')
newCmd('.display',' ','','*v+12.2 -- adds ADD, REMOVE, and GROUP keywords;dhide','The opposite of {#.hide~}. <b>Display</b> is similar to {#.restrict~} in its action, but it is far more flexible. Atoms can be added to the displayed set using <b>display displayed or ...</b> or removed from the hidden set using <b>display displayed and not ...</b>. Unlike <b>restrict</b>, the <b>display</b> command does not delete the hidden shapes. Thus, after <b>restrict none</b> all cartoons, traces, spacefill spheres, and bond sticks are effectively deleted. In contrast, <b>display none</b> is easily reversed using <b>display all</b>.  or deselected with <b>select not hidden</b>. Atoms can be added to the hidden set using <b>hide hidden or ...</b> or removed from the hidden set using <b>hide hidden and not ....</b>. In addition, starting with Jmol 12.2, bonds can be hidden or displayed.','0|1','','')
newCmd('.display',' ','','*v-12.2;dhide','The opposite of {#.hide~}. <b>Display</b> is similar to {#.restrict~} in its action, but it is far more flexible. Atoms can be added to the displayed set using <b>display displayed or ...</b> or removed from the hidden set using <b>display displayed and not ...</b>. Unlike <b>restrict</b>, the <b>display</b> command does not delete the hidden shapes. Thus, after <b>restrict none</b> all cartoons, traces, spacefill spheres, and bond sticks are effectively deleted. In contrast, <b>display none</b> is easily reversed using <b>display all</b>.  or deselected with <b>select not hidden</b>. Atoms can be added to the hidden set using <b>hide hidden or ...</b> or removed from the hidden set using <b>hide hidden and not ....</b>. In addition, starting with Jmol 12.2, bonds can be hidden or displayed.','0|1','','')
newCmd('',' ','','','Displays a set of atoms. Displays atoms and associated structures (bonds, halos, stars, cartoons, etc.) and hides all others. Starting with Jmol 12.2, the optional parameters ADD and REMOVE prior to the atom expression allow quick alternatives to "displayed or..." and "displayed and not...". ','1','._atom_expression','')
newCmd('',' ','','','Displays a set of bonds.','2','[{…}]','')
newCmd('',' ','','','Displays all bonds.','3','BONDS','')

newCmd('.dots',' ','','disp','Turns a dotted surface around the currently selected atoms. See also the {#.geoSurface~} command and the settings {#.set_dotdensity~set dotDensity} and  {#.set_dotscale~set dotScale}, {#.set_dotsselectedonly~set dotsSelectedOnly}, and  {#.set_dotsurface~set dotSurface}.','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns dot rendering on and all other rendering (including labels) off.','*13','ONLY','')
newCmd('',' ','','','Draws dots at the van der Waals radius for the selected atoms. See {misc/radii.xls~radii.xls}.','*2','VANDERWAALS','')
newCmd('',' ','','','Draws dots at the (nominal) ionic radius for the selected atoms if the atom\'s formal charge has been read and is nonzero; uses the nominal covalent bonding radius otherwise ({misc/radii.xls~radii.xls}).','*3','IONIC','')
newCmd('',' ','','*v+11.0;','Draws dots at the indicated percent of the van der Waals radius for each atom (maximum value 1000%).','*4','nn%','')
newCmd('',' ','','*v+11.0;','Draws dots at the indicated radius in Angstroms for each atom (maximum value 10.0 Angstroms).  Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*5','(decimal)','')
newCmd('',' ','','','Draws dots at the indicated distance in Angstroms beyond the van der Waals radius for each atom (maximum value 10.0 Angstroms). The "+" sign is required.','*6','+(decimal)','')
newCmd('','','','*v+11.6;','Draws dots at the radius corresponding to the minimum anisotropic displacement parameter for the selected atoms factored by the given percentage. See also {#.ellipsoid~}.','*7','ADPMIN','n%')
newCmd('','','','*v+11.6;','Draws dots at the radius corresponding to the maximum anisotropic displacement parameter for the selected atoms factored by the given percentage. See also {#.ellipsoid~}.','*8','ADPMAX','n%')
newCmd('.draw',' ','','*v+11.6 Jmol 11.6 adds ARROW ARC, ARC, CIRCLE, and ID options, allows for mixed-type point sets, and special options Ramachandran, Quaternion, and PointGroup;iso, plot','The <b>draw</b> command allows for the insertion of points, lines, and planes to a model. The general syntax of the <b>draw</b> command is as follows:  <br />  <br /><b>draw [objectID] [modifying parameters] [positions]  </b> <br /> <br />','0|1','','')
newCmd('',' ','','*v+11.6;','The optional identifier such as "line1" or "plane2" that can be referred to in later scripts as $line1 or $plane2. These words are arbitrary and if preceded by the optional keyword ID may be any string. Specifically for the options ON, OFF, and DELETE, the id may include a wildcard character (asterisk). Thus, <b>draw v* off</b> turns off all drawn objects having an ID starting with the letter "v".','TEXT','ID [object id]','')
newCmd('',' ','','*v+11.0;','Several options allow for a wide variety of simple structures to be drawn. These include:[||<b>"hoverText"</b>|text that will appear after issuing <b>set drawHover TRUE</b>  ||<b>ARROW</b>|Draws a straight (two-point) or curve (more than two-point) arrow. Jmol 12.2 adds the keyword BARB to indicate "fish-hook" half-arrows as for organic chemistry mechanisms involving radicals. ||<b>ARC {{pt1}} {{pt2}} {{ptRef}} {{nDegreesOffset theta fractionalOffset}} SCALE scale DIAMETER diameter</b>|Draws a theta-degree arc in a plane perpendicular to the line pt1--pt2 starting at nDegreesOffset degrees rotation from reference point {{ptRef}} at a point fractionalOffset from pt1 to pt2. The SCALE parameter is used to set the diameter of the overall circle containing the arc; the DIAMETER parameter sets the diameter of the curved arc line itself. <b>ARROW ARC</b> adds an arrow head.||<b>ARC {{pt1}} {{plane}} {{ptRef}} {{nDegreesOffset theta fractionalOffset}}</b>|As above, but uses the plane as a reference to define a perpendicular axis. ||<b>CIRCLE {{pt1}} {{pt2}} SCALE scale DIAMETER diameter</b>|Draws a circle with center at pt1 (which may be an atom expression) in the plane perpendicular to the line between pt1 and pt2. If pt2 is not specified, the circle appears in 2D and remains in the plane of the screen when the model is manipulated. Together, the SCALE and DIAMETER parameters set the overall size of the circle. If no DIAMETER is given and an atom expression is given for pt1, the default diameter is one that includes those atoms; otherwise the default diameter is 1.0 Angstrom. The circle will be filled to form a solid disk unless the MESH NOFILL option is given.||<b>CIRCLE {{pt1}} {{plane}} SCALE scale DIAMETER diameter</b>|Draws a circle around pt1 in the indicated plane.||<b>COLOR (color)</b>|Sets the color of the drawn object at the time it is created. (The {#.color (model object)~color} command can be used retroactively as well.) ||<b>CROSSED</b>|Two lines (already drawn objects) specified next are crossed; switch the order of vertices for defining a plane.||<b>CURVE</b>|Draws a smooth curve through the given positions. ||<b>CYLINDER</b>|Draw CYLINDER creates a cylinder of the designated diameter. End caps can be set to closed/flat (FILL, the default) or open (MESH NOFILL).||<b>DELETE</b>|Deletes the object if an identifier is given or all drawn objects if none is given; not used with any other parameters.||<b>DIAMETER n</b>|Sets the number of pixels for the diameter of points, lines, arrows, and curves. Note that this means that zooming of the model does not change the width of these objects.  <b>n</b> may be a decimal value x.x, same as <b>WIDTH x.x</b>.||<b>FIXED/MODELBASED</b>|Sets whether the surface generated is to be associated with the fixed window -- and thus appear with all frames/models -- or is to be associated with the currently displayed model (the default).||<b>ON/OFF</b>|Turns on or off the identified object or all drawn objects if no identifier is given; not used with any other parameters.||<b>PERP</b> <br /><b>PERPENDICULAR</b>|Draw this object perpendicular to the next indicated object.||<b>PLANE</b>|Create a four-vertex quadrilateral even if only three points are given.||<b>REVERSE</b>|Reverse the order of vertices used if the next object listed is a line.||<b>ROTATE45</b>|Rotate a perpendicular plane to a line by 45 degrees.||<b>LENGTH (decimal)</b>|The length for a line/axis in Angstroms. The keyword LENGTH is optional.||<b>OFFSET {{x y z}}</b>| offsets the object by the given x, y, and z distances.||<b>SCALE (decimal)<br />SCALE (integer)</b>|SCALE with a decimal value indicates a scaling factor for the drawn object. For example, <b>draw SCALE 1.5 (atomno=1) (atomno=2)</b> draws a line with length 1.5 times the distance from atom 1 to atom 2. The line is centered on the two atoms. The keyword SCALE is required in this case. Note that a draw command can consist of just an identifier and a scale. Thus, if $line1 is already defined, <b>draw line1 SCALE 1.3</b> will rescale that line to 130% of the distance between its defining points. SCALE with an integer number indicates a percent scale. The keyword SCALE is optional in this case.||<b>VECTOR</b> | This option is similar to ARROW, and accepts two points. The first point is the origin; the second is of the form {{dx,dy,dz}}, indicating a vector to the next point rather than the point itself.||<b>VERTICES</b> | Generally the geometric center of an atom expression or drawn object is used for positioning. Added just before the atom set or object name reference, VERTICES indicates to use all vertices, not just the center point of the atoms in the expression or the points in the object.||<b>WIDTH x.x</b>|Sets the diameter of points, lines, arrows, and curves to a given width in Angstroms. Objects drawn will be scaled based on perspective and zoom setting .||<b>"text"</b>| A simple text label can accompany drawn objects. The text appears near the first point. Text starting with ">" will be associated with the last point instead of the first. The ">" character will be stripped before the text is displayed.||]','TEXT','[modifying parameters]','')
newCmd('',' ','','*v+11.0;','Positions define position of the point, the endpoints of the line/axis, or the corners of a plane. Positions can be indicated in any combination of any of the following four ways. Mixed types are processed in the order given on the command line. <br />[||{{x, y, z}}|a model-frame cartesian coordinate, in braces,  ||{{x, y, z/}}|for crystal structures, a unit-cell {#.fractional coordinates~fractional coordinate}, in braces,  ||$object|a previously defined drawing object such as $line1 or $plane2, preceded by "$". ||(atom expression)|an atom expression, in parentheses.||@{{ {{atomExpression}}.split()}} | atom expressions split based on model index .||] In addition, if two parameters are given, where the first evaluates to a point and the second is a four-vector {{a b c d}}, a line is drawn from the point to the nearest point on the plane defined by <b>ax + by + cz + d =  0</b>. (Note that because quaternions and axisAngle are stored in quaternion format, which is internally the same as that used for planes, if a quaternion or axisAngle is used in this context, the line will be along the axis of rotation represented by the quaternion or axisAngle <b>q</b> to a point on a plane defined by <b>d = q.w = cos(theta/2)</b>, where <b>theta</b> is the rotation angle).','TEXT','[positions]','')
newCmd('',' ','','*v+11.0;','Display options for DRAW are indicated in the following table. These may be given at the end of the definition or in a later command having just these keywords and the identifier (or "ALL") of the desired draw object.  <br />[||<b>FILL/NOFILL</b>| Display the drawn opject as a smoothly filled surface.||<b>FRONTONLY/NOTFRONTONLY</b>| Display only the front half of the surface. This can be useful when the options <b>mesh nofill</b> are used. ||<b>FRONTLIT /BACKLIT /FULLYLIT</b>| In some cases the object may appear flat. Using BACKLIT will fix this; FULLYLIT makes exterior and interior surfaces bright; FRONTLIT is the default setting. ||<b>MESH/NOMESH</b>| Display a mesh of lines intersecting at the vertexes used to construct the object. For cylinders, the combination <b>MESH NOFILL</b> creates a cylinder with no end caps. ||<b>ON/OFF</b>| Turn the object ON or OFF, but do not delete it. ||<b>OPAQUE/TRANSLUCENT n</b>| Display the object as an opaque or translucent object. Several translucent options are available; see the {#.color~} command. ||]','TEXT','[display options]','')
newCmd('','12.1.48','','*v+12.2;','The <b>DRAW</b> command can be used to draw organic "mechanistic arrows." You need to specify  two atoms, two bonds, an atom and a bond, or a bond and an atom. Atoms are identified using the keyword ATOM followed by an atom expression; bonds are identified using the keyword BOND followed by two atom expressions. For example: <b>draw ID "arrow1" ARROW ATOM @1 BOND @1 @2</b>, which draws a curved arrow from atom 1 (atomno=1) to the middle of the bond connecting that atom to atom 2. These drawn objects are then "connected" to the atoms, so if the atoms are not displayed, the arrow is not displayed either. Of course, individual arrows can be hidden independently of their associated atoms as well. ','*20','ARROW','ATOM/BOND')
newCmd('','11.9.17','','*v+11.0;','Draws the currently defined {#.boundbox~}. Note that by default this is a solid. To show just edges, use options <b>MESH NOFILL</b>.','*30','BOUNDBOX','')
newCmd('',' ','','*v+11.0;','Deletes all draw objects','*31','DELETE','')
newCmd('','11.7.29','','*v+11.8;','Draws a frame (an x,y,z axis set) at the given center with the given quaternion orientation. Quaternions are expressed using the {#.functions~quaternion()} function within a math @{{...}} wrapper. For example, <b>draw ID "q1" frame {{0 0 0}} @{{quaternion(1,0,1,0)}}</b> draws a frame at the origin that has been rotated 90 degrees relative to the Y axis. (Jmol automatically normalizes the quaternion to q0=0.70710677, q1=0, q2=0.70710677, q4=0, which represents a 90-degree rotation about the Y axis.)','*34','FRAME','{{atom expression}}','{{quaternion}}','')
newCmd('','11.7.47','','*v+11.8;','Draws a vector representing the local helical axis for the selected amino acid or nucleic acid residue, connecting it to the previous residue in its chain. The length of the arrow is the vertical height per residue. ','*35','HELIX AXIS','')
newCmd('','11.9.17','','*v+12.0;','Draws the portion of a plane that intersects the current {#.boundbox~} based on a {#.plane expressions~plane expression}.','*36','INTERSECTION','boundBox','(plane expression)','')
newCmd('','11.9.17','','*v+12.0;','Draws the portion of a plane that intersects the current {#.unitcell~unit cell} based on a {#.plane expressions~plane expression}.','*38','INTERSECTION','UnitCell','(plane expression)','')
newCmd('','11.1.28','','*v+11.2;','Lists all draw objects.','*4','LIST','')
newCmd('','11.5.52','','*v+11.6;pg','Calculates and draws point group symmetry planes and axes for a symmetrical or nearly symmetrical molecule. As for {#.calculate~calculate pointgroup}, the calculation is carried out only on the currently selected atoms and is limited to at most 100 selected atoms. Parameters include specification of a subset to draw (Cn, C2, C3, C4, C5, C6, C8, Sn, S3, S4, S5, S6, S8, S10, S12, Cs, Ci) optionally followed by an index (for example, <b>draw POINTGROUP C3 2</b> draws the second C3 axis only). A second option, SCALE x, allows adjusting the scale of the drawn planes and axes. The default scale of 1 puts the edge of planes directly through the outermost atoms. This command automatically sets <b>perspectiveMode OFF</b> so as to properly represent the planes and axes.  ','*50','POINTGROUP','[parameters]')
newCmd('','11.9.19','','*v+12.0;','The POLYGON option allows the ability to draw polygons based on a set of vertices and a set of faces. (Jmol 12.0) This capability allows drawing any number of flat triangular (not quadrilateral, but read on...) faces with or without edges around each face. The description is somewhat like that for PMESH files and involves (a) giving the number of vertices, (b) listing those vertices, (c) giving the number of faces, and (d) listing the faces with a special array syntax. Each face is described as an array indicating the three (0-based) vertex indices followed by a number from 0 to 7 indicating which edges to show a border on when the <b>mesh</b> option is given: [|| 0 | no edge|| 1| edge between first and second vertex|| 2 | edge between second and third vertex||4 | edge between third and first vertex || 3, 5, 6, 7 | combinations of the above.||] For example: <b>draw POLYGON 4 {{0 0 0}} {{1 1 1}} {{1 2 1}} {{0 5 0}} 2 [0 1 2 6] [0 3 2 6] mesh nofill</b>. Starting with Jmol 12.2, the points and faces can be provided as arrays: <b>draw POLYGON @x @y mesh nofill</b>, where in this example, x is an array of the four points, and y is [[0 1 2 6] [0 3 2 6]].','*51','POLYGON','[polygon definition]')
newCmd('','11.5.43','','*v+11.6;','Draws vectors for the previously selected residues representing the quaternion frame and rotational axis for each residue. The parameters for this command are the same as for {#.plot~plot quaternion} (Jmol 12.0) or the <b>quaternion</b> command. Vectors are named based on the axis (x, y, z, q), residue number, and chain. ','*6','QUATERNION','[parameters]')
newCmd('','11.5.47','','*v+11.6;','Draws curved arrows marked with PHI and PSI angles in planes perpendicular to the N-CA and CA-C bonds of the selected amino acids, respectively.','*7','RAMACHANDRAN','')
newCmd('','12.1.26','','*v+12.2;','Creates a DRAW object based on DRAW object $id, which must be of POLYGON type, that is slabbed based on a plane definition. For example: <b>draw p4 polygon 4 {0 0 0} {0 3 0} {3 3 0} {0 3 3} 2 [0 1 2 0] [2 3 0 0]; draw p4 slab $p4 PLANE @{plane({0 0 0}, {5 0 0})}</b>','*8','SLAB','$id','PLANE','(plane expression)')
newCmd('','11.8.1','','*v+11.8;','Draws a graphic representation of a crystallographic space group symmetry operation. Operations include simple rotations, screw rotations, rotation-inversions, mirror planes, and glide planes, Either a number (for one of the model\'s symmetry operations) or a string indicating a Jones-Faithful operation may be used. The position may be an atom expression may be a point. If a point, it can be expressed either as a cartesian coordinate or a fractional coordinate (using a "/" in at least one position -- for example, {{ 1/2 0 0}}. The ID of the draw command is prepended to the drawn object IDs. For example, <b> draw ID "s1" SYMOP "x,1/2-y,z" {{1/2 1/2 1/2}}</b>. See also the {http://chemapps.stolaf.edu/jmol/docs/examples-11/showsym.htm~Jmol Crystal Symmetry Explorer}.','*9','SYMOP','[n or &quot;x,y,z&quot;]','{{atom expression}}','')
newCmd('','11.9.17','','*v+12.0;','Draws the symmetry relations between  any two atoms or groups or any two positions in space. For example, <b>draw SYMOP {{molecule=1}} {{molecule=2}}</b>.','*91','SYMOP','{{atom expression}}','{{atom expression}}','')
newCmd('','11.9.17','','*v+12.0;','Draws the symmetry relation associated with the specified symmetry operator  between  any two atoms or groups or any two positions in space. For example, <b>draw SYMOP {{molecule=1}} {{molecule=2}}</b>.','*92','SYMOP','(integer)','{{atom expression}}','{{atom expression}}')
newCmd('','11.9.3','','*v+12.0;','Draws the symmetry operation associated with the given 4x4 matrix, which most likely comes from a variable, for example here the symmetry operation that is the product of two other symmetry operations: <b>draw SYMOP @{{symop(11)*symop(14)}}</b>.','*93','SYMOP','[matrix]')
newCmd('','11.9.17','','*v+11.0;','Draws the currently defined {#.unitcell~}. Note that by default this is a solid. To show just edges, use options <b>MESH NOFILL</b>.','*99','UNITCELL','')

newCmd('.echo','11.1.6','','*v+11.2 -- adds variable options;label;math','In Jmol models can be annotated in one of three ways. Text can be associated with a specific atom using a {#.label~}, text can appear when the user hovers the mouse over an atom or other user-defined point in space for a designated period of time ({#.hover~}), and text can be placed at a specified position on the window (2D <b>echo</b>) or at a point in space (3D <b>echo</b>). In addition, the text is echoed in the Java console, the Jmol {#.console~}, and the {#.set_messageCallback~} or {#.set_echoCallback~} function, if defined. Multi-line text can be generated using a vertical bar as a line separator.<br /><br /> See also the {#.message~} command for variable-displaying capabilities that send information to the consoles and callback functions without displaying text. One can also place JPEG, PNG, or GIF images at either a 2D screen position or a 3D molecular position. See {#.set echo~} for details. ','1','(string)','')
newCmd('.else','','','*v+11.4;flow','See {#.if~}.','0|1','','')
newCmd('.elseIf','','','*v+11.4;flow','See {#.if~}.','0|1','','')
newCmd('.ellipsoid','11.5.29','','*v+11.6 NEW;disp','The <b>ellipsoid</b> command displays anisotropic displacement parameters (thermal ellipsoids) based on crystallographic Uij data in CIF or PDB files and tensor quantities found in CRYSTAL files. (For PDB files, you must specify a lattice, for example, <b>load =1a6g {{1 1 1}}</b>, in order to have Jmol read the ANISOU records.) The resulting ellipsoids have three mutually perpendicular axes that are not necessarily the same length. Overall ellipsoid size can be set to a percentage value (default 50%). Rendering styles are set globally using the <b>set</b> command and one or more of the settings described below. The diameter of the axis and arc lines can be set using <b>set ellipsoidAxisDiameter x.x</b> where x.x is a distance in Angstroms (default 0.02). Note that the Uij data when read from CIF files are also stored in the .temperature value for atoms as 100 times the geometric mean of the diagonal U-factor parameters, 100 * (U11 * U22 * U33)^0.3333. <br /><br /> User-defined ellipsoids can also be created by specifying the keyword ID followed by a user-named ID. These parameters may be indicated in any order, but the ellipsoid is not defined until the AXES parameter is specified.<br /><br /> [||<b>set ellipsoidAxes</b>|<img align="right" src="img/ellipsoidaxes.jpg">This option may be combined with any other option. When combined with balls, the color of the axes are set to white or black in order to contrast with the background.||<b>set ellipsoidArcs</b>|<img align="right" src="img/ellipsoidarcs.jpg">When combined with balls, the color of the arcs are set to white or black in order to contrast with the background. This option is ignored when dots are displayed.||<b>set ellipsoidBall</b>|<img align="right" src="img/ellipsoidball.jpg">This option is the default option and may be combined with ellipsoidAxes, ellipsoidArcs, or ellipsoidFill (which displays a cut-out octant).||<b>set ellipsoidFill</b>|<img align="right" src="img/ellipsoidfillarcs.jpg"><img align="right" src="img/ellipsoidfillball.jpg"><img align="right" src="img/ellipsoidfillspace.jpg">This option affects the rendering of arcs and balls. In the case of arcs, it fills in the arcs to form a set of three elliptical planes; in the case of balls, it provides a cut-out octant. When combined with {#.spacefill~spacefill ADPMIN n%}, where n is the current percentage size of the ellipsoid, the combination of fill and arcs provides a strikingly unique rendering of the anisotropic displacement parameters for an atom. ||<b>set ellipsoidDots</b>|<img align="right" src="img/ellipsoiddots.jpg">This option renders a random set of dots defining the ellipsoid that sparkle as the model is manipulated. The number of dots can be set using <b>set ellipsoidDotCount n</b> where n is an integer (default 200). This option is ignored when ellipsoid balls are displayed.||]','0|1','','')
newCmd('','','','*v+11.6;','Turns ellipsoids on or off for the currently selected atoms.','*1','._on_off{"ON"}','')
newCmd('','','','*v+11.6;','Sets the size of the ellipsoids for the currently selected atoms.','*2','nn%','')
newCmd('','','','*v+11.6;','Sets the three perpendicular axes for the ellipsoid. These axes should be perpendicular. If they are not, the ellipsoid may not be rendered, or its shape will be unpredictable. The ellipsoid is not displayed until this parameter is set.','*35','ID','[object id]','AXES','{{ax ay az}} {{bx by bz}} {{cx cy cz}}')
newCmd('','','','*v+11.6;','Sets the color of the ellipsoid using parameters described in the {#.color~} command. ','*5','ID','[object id]','COLOR','[color parameters]')
newCmd('','','','*v+11.6;','Sets the scale of the ellipsoid relative to its axes lengths.','*8','ID','[object id]','SCALE','(decimal)')
newCmd('','','','*v+11.6;','Deletes the specified ellipsoid.','*6','ID','[object id]','DELETE','')
newCmd('','13.1.19','','*v+13.2;','Sets one or more options separated by semicolon:   "arcs;arrows;axes;ball;dots;fill;wireframe" with optional "no" in front of one or more. This option overrides the settings of <b>set_ellipsoidball</b> and related settings.','*7','ID','[object id]','OPTIONS','&quot;options&quot;')
newCmd('','','','*v+11.6;','Sets the center for this ellipsoid the specified point (which may be fractional).','*41','ID','[object id]','CENTER','{{x y z}}')
newCmd('','','','*v+11.6;','Sets the center for this ellipsoid to the center of the specified atoms.','*42','ID','[object id]','CENTER','{{ atom expression }}')
newCmd('','','','*v+11.6;','Sets the center for this ellipsoid to the point specified by the object name (draw or ellipsoid)','*43','ID','[object id]','CENTER','$object')
newCmd('','','','*v+11.6;','Turns this ellipsoid on.','*31','ID','[object id]','ON','')
newCmd('','','','*v+11.6;','Turns this ellipsoid off.','*32','ID','[object id]','OFF','')
newCmd('.[export]','','','*v+11.4;','The Jmol application (not the applet) allows export of the currently rendered scene as files that can be read by {http://www.avatech.com/solutions/visualization/product-details.aspx?product=17~Maya}, {http://www.povray.org/~POV-Ray}, and VRML readers. See the {#.write~} command for details.','0','','')

newCmd('.exit',' ','','*v+11.0 -- new;pause;quit','When the {#.set_scriptQueue~} is turned on, each script waits for the previous to complete. Either {#.quit~} or <b>exit</b> at the very beginning of a script command halts any previous still-running script. Processing then continues with the second command on the line. Anywhere else in the command, <b>quit</b> and <b>exit</b> abort that script only. In addition, <b>exit</b> clears the script queue of any remaining scripts, thus stopping all script processing.','0','','')
newCmd('.file',' ','','anim','Carries out a {#.frame~} command bringing into  frame all models in the specified file number.','1','(integer)','')
newCmd('.fix',' ','','','The <b>fix</b> command takes argument like display or select. But ensures no atoms of this set will be moved or dragged anywhere accidentally.','1','._atom_expression','')
newCmd('.font',' ','','label','Sets font size, face (serif, sansSerif), and style (plain, italic, bold, bolditalic) in labels and other text-bearing elements. If only a font size is given and the object is a label, then only the size of the font is changed, not the font face or style. For all other objects, if only a font size is given, the font face and style are changed to sansSerif. For <b>font ECHO</b> you can apply a scaling factor that allows the font to scale with zoom. This scaling factor, in "pixels per micron", determines the absolute size of the font relative to a "standard" window size. (Scaled {#.echo~} fonts can also be created using <b>set fontscaling TRUE</b>, then defining the echo text. The scaling factor also applies to images added with the {#.set echo IMAGE~set echo} command.','5','._object_with_text','._fontsize','._fontface{"SansSerif"}','._fontstyle{"Plain"} [scaling factor]')
newCmd('.for','','','*v+12.2 -- for (var x IN ...);math;flow','Jmol supports both the standard <b>for</b> statement as well as a specialized in-line FOR construct that provides a powerful way to work with atom data.','0|1','','')
newCmd('.for','','','*v+11.4 -- new flow control options;*v-12.2;math;flow','Jmol supports both the standard <b>for</b> statement as well as a specialized in-line FOR construct that provides a powerful way to work with atom data.','0|1','','')
newCmd('','','','*v+11.4;math;flow','<b>for</b> takes three aguments in parentheses and separated by a semicolon. The loop may be exited using {#.break~} and truncated using {#.break~continue}. For example, <b>for(var i=1; i <= 10; i++){{...}}</b> loops through a block of script ten times, incrementing the variable <b>i</b> by one each time. Any variables created with the VAR keyword are local to the <b>for</b> block.','TEXT','for (var i = 1; i < n i++) {...}','')
newCmd('','','','*v+12.2 -- new;math;flow','A second form of <b>for</b> takes just two arguments -- a variable name and either an atom set or an array. For example, <b>for(var i IN {{selected}}) {{print i.label()}}</b>  loops through all atoms that have been selected, assigning <b>i</b> to each one in turn, printing its default label, and <b>var a = [100,150,200]; for(var i IN a) {{print i + 5}}</b>  prints a list that is five more than the array values. The loop may be exited using {#.break~} and truncated using {#.break~continue}. If the looping variable (<b>i</b> in this case) is assigned any value during the process, the for loop will exit at the end of the current loop. Thus, for example, <b>i = 0</b> is a way of saying: "Continue to the end of this loop and then exit."','TEXT','for (var i IN [atom-expression or array]) {...}','')
newCmd('','11.7.40','','*v+11.8;','You can create lists of values using a form of inline FOR function. The syntax is simply <b>xxx = for(dummyVariable;{{atom expression}};math expression)</b>. For example, <br /><br />&nbsp;&nbsp;&nbsp;&nbsp;aniso = for(x;{{*}};x.adpmax - x.adpmin)</br></br>creates an array containing differences between the maximum and minimum anisotropic density parameters for a set of atoms. Arrays such as these can be used to assign properties to atoms or to color them. So, for example, we might want to set the radius of atoms based on temperature, but not the exact value of the temperature: <br /><br />&nbsp;&nbsp;&nbsp;&nbsp;{{*}}.radius = for(x;{{*}};x.temperature/20)<br /><br />Combined with the inline IF function, the inline FOR can be quite useful. For example:<br /><br />&nbsp;&nbsp;&nbsp;{{*}}.color = for(x;{{*}};if(x.temperature > 10;"red";if(x.temperature < 2;"blue";"white")))<br .><br />See also {#.functions~} for how to define a function that operates on atoms one at a time with the syntax </b>{{*}}.myFunction(a, b)</b>.','TEXT','in-line FOR','')
newCmd('.[fractional coordinates]',' ','','*v+11.0 NEW, 12.0 adds set fractionalRelative;','Several Jmol commands, namely {#.center~}, {#.centerAt~}, {#.dipole~}, {#.draw~}, {#.isosurface~}, {#.moveTo~}, {#.rotate~}, {#.spin~}, {#.translateSelected~}, and {#.unitcell~}, accept coordinates in place of atom expressions. These coordinates are introduced using braces: {{x, y, z}} or {{x y z}}. (The commas are optional.) However, when the file data are crystallographic, and the coordinates have been derived by transformation of unit cell coordinates into cartesian coordinates, one can use the unit cell fractional coordinate system instead. The designation of a coordinate as fractional is simplicity itself: just include somewhere in one of the three coordinate values a fraction symbol, "/". Thus, {{1/2, 0, 0}} is a fractional coordinate, and it will be automatically transformed into the correct cartesian point. This allows formation of commands such as <b>set unitCell {{1/2, 1/2, 1/2}}</b> to move the unit cell to a new crystallographic orgin (for display purposes only). Since n/1 is n, one can use decimals as well, writing {{0.5/1, 0, 0}} instead of {{1/2, 0, 0}}. And since "/1" is not particularly informative, the "1" can be left off to give {{0.5/, 0, 0}} or {{0.5, 0, 0/}} as sufficient indication of fractional coordinates. Note that when the {#.unitcell~unit cell} is shifted, an atom\'s fractional coordinates also shift with it. Starting with Jmol 12.0, you can set the meaning of {{1/2 1/2 1/2}} to be relative to the current {#.unitcell~} setting, <b>set fractionalRelative TRUE</b> or to the default absolute definition, using <b>set fractionalRelative FALSE</b>. ','0','','')

newCmd('.frame',' ','models','*v+13.0 -- adds frame names and IDs;anim','Sets the current animation frame. Numbers refer to the physical position of the model in the file (1 being the first). Same as the {#animation~animation frame} command. See also {#.model~}. Note that you can show specific pairs or sets of frames or models by using <b>frame all</b> followed by <b>display (*/n,*/m,*/p)</b>, where n, m, and p are frame numbers. See also {#.set_backgroundModel~}. For the applet, if {#.set_animFrameCallback~AnimFrameCallback} is enabled, a message indicating the frame change is sent to the associated JavaScript function. The simple command <b>frame</b> has no observable effect but forces an animFrameCallback message to be sent, which also serves to update the pop-up context menu. This is sometimes useful if scripting has changed something in the structure such as the addition of vibration vectors that would alter menu options . ','1|1','','')

newCmd('',' ','','*v+11.0;','Go to a specific model in the case of loading a single file with multiple models. In the case of the loading of a single PDB file containing MODEL records, the integer used here corresponds to the number in that record. In all other situations, the number used here is the sequential index of the model in the file, starting with 1. If more than one file is loaded, the number indicates the file, and all models in that file are overlaid.','*10','(integer >= 1)','')
newCmd('','11.1.9','','*v+11.2;','Go to a specific model in a specific file when one or more files are loaded. The format for specifying which model to go to is the same as for {#.select~} or {#.display~}: file.model. For example, <b>frame 2.3</b> goes to the third model in the second file listed in the most recent {#.load~} command. Note that atoms in models chosen with the <b>frame</b> command must also be in the current {#.display~} set in order to be visible. So, for example, <b>display 2.1;frame 2.2</b> will display nothing; <b>display connected(hbond);frame 2.2</b> will display only the hydrogen-bonded atoms in model 2.2. ','*12','(positive decimal)','')
newCmd('','13.1.13','','*v+13.2;','Using a negative decimal indicates that one of the linear morphs between trajectory frames is to be shown. Files must have been loaded using the TRAJECTORY option.','*14','(negative decimal)','')
newCmd('','11.1.9','','*v+11.2;','Sets the animation range and displays a range of models, possibly spanning multiple files. The hyphen is optional. If the hyphen is present but the second model number is missing, then all models from the designated model forward are assumed.','*13','(decimal)','-','(decimal)','')
newCmd('',' ','','*v+11.0;','Same as <b>frame ALL</b>.','*16','0&#46;0','')
newCmd('','12.3.19','','*v+13.0;','This syntax allows setting of a model by an atom set. The first atom in the set determines which model is displayed. For example, <b>model {{within(smarts,"[r3]")}}</b> would bring into frame the first model with a 3-membered ring.  (Jmol 13.0)','*17','{{atomSet}}','')
newCmd('','12.3.26','','*v+13.0;','Recalls a frame with the given ID assigned with the <b>frame ID</b> command option.','*18','&quot;id&quot;','')
newCmd('',' ','','*v+11.0;','Overlay all frames of the current frame range set. Note that this may not be all the models if the frame range has been set to a subset of the models or if multiple files are loaded and only models within one file have been specified with a previous <b>frame</b> command.','*15','0','')
newCmd('',' ','','*v+11.0;',' Provides a way to align structures across a set of frames. This is important for certain animations. The atom expression is evaluated per frame, and the resultant point is aligned in each case.','*19','ALIGN','{{ atom expression }}')
newCmd('',' ','','*v+11.0;','Resets the frame range to all models and overlays them.','*20','ALL','')
newCmd('','13.1.15','','*v+13.2;','Creates one or more "empty" models. If no parameter is given, 1 is assumed.','*22','CREATE','(integer)')
newCmd('','12.3.26','','*v+13.0;','Assigns an ID to a frame that can later be used to recall the frame using <b>FRAME "xxx"</b> where <b>xxx</b> is a frame ID.','*24','ID','&quot;id&quot;')
newCmd('',' ','','*v+11.0;','Go to the last frame in the frame range set.','*25','LAST','')
newCmd('',' ','','*v+11.0;','Go to next frame in the frame range set.','*30','NEXT','')
newCmd('',' ','','*v+11.0;','Pause animation at the current frame.','*40','PAUSE','')
newCmd('',' ','','*v+11.0;','Start playing at the specified frame number (current frame if number is omitted). Direction, speed of play, and mode of animation (ONCE, LOOP, or PALINDROME) are set using {#.animation~animation mode}.','*50','PLAY','(starting frame)')
newCmd('',' ','','*v+11.0;','Start playing at the specified frame number (current frame if number is omitted), reversing the direction.','*60','PLAYREV','(starting frame)')
newCmd('',' ','','*v+11.0;','Go to previous frame in the current frame set.','*70','PREVIOUS','')
newCmd('',' ','','*v+11.0;','Sets the range of frames to play as an animation and sets the current frame to the first number given. If the starting frame number is larger than the ending frame number, then play is in reverse. If only one number is given, then the range is set from that frame through the last frame in the file. If both numbers are omitted, then the range is set to include all frames in the file.','*80','RANGE','(starting frame)','(ending frame)','')
newCmd('',' ','','*v+11.0;','Resume playing from the current frame. (Same as PLAY.)','*90','RESUME','')
newCmd('',' ','','*v+11.0;','Return to the first frame in the frame range set. ','*91','REWIND','')
newCmd('',' ','','*v+11.4;','Sets a title for the frame (or, in Jmol 12.0, all frames currently visible), which appears in the bottom left corner of the applet or application. Starting with Jmol 12.0, if the title includes an expression such as "@{{_modelName}}", then that expression is evaluated whenever the model is rendered (for example, when the frame is changed). To set all titles at once, first make all frames visible. with <b>frame all</b>, then issue the <b>frame title</b> command.','*92','TITLE','&quot;title&quot;')
newCmd('','12.1.40','','*v+12.2;','Sets the title for all visible frames at once using an array of strings. For example, this sequence might give each frame title that shows the minimum energy for the files, converting energies to kJ/mol in the process: <br /><br /> x = getproperty("modelinfo.models.energy")<br /> y = x.sub(x.min).mul(2625.5) # relative energies, in kJ<br />frame *;frame title @y','*93','TITLE','(array)')

newCmd('.frank',' ','frank','*v+11.0 -- new;','Determines whether or not "Jmol" is indicated in the bottom right corner of the window.','0|1','','')
newCmd('',' ','','*v+11.0;','Turning the frank off is disabled for the signed applet running on a web server.','1','._on_off','')
newCmd('.[functions]','','','*v+11.8 -- adds support for quaternion functions;math','Jmol math allows for two types of functions. The first simply operate on their given parameters. For example, <b>x = load("myfile.dat")</b> loads the variable x with the contents of the file "myfile.dat". The second operate on the elements of a variable individually in some way and are referred to here as <i>item selector</i> functions. In the listings below, these functions all begin with a period character. For example, <b> x = {{carbon}}.bonds</b> operates so as to deliver the bond set associated with the carbon atoms; <b> x = "this is a test".replace("s","S")</b> operates on the individual characters of the string "this is a test"; <b> x = {{oxygen}}.label("%U")</b> assigns x the list of labels for the oxygen atoms in the model. Some functions can be used in both contexts. For example, <b> x = distance({{oxygen}}, {{carbon}}) </b> delivers the distance from the CENTER of the oxygens to the CENTER of the carbons. <b> x = {{oxygen}}.distance{{carbon}}</b> delivers the AVERAGE distance of an oxygen atom to the CENTER of the carbons. These are subtly different. Some functions require parentheses; some do not. Basically, if a function CAN have parameters, for example, .join(), .split(), or .label(), then it MUST have at least empty (); if a function CANNOT have parameters, for example, .atoms, .bonds, or .ident, then it NEVER uses parentheses.','0|1','','')

newCmd('','','','*v+11.2;','Several of these functions are described more fully under the heading {#.atomexpressions~atom expressions}, as they can also be used in commands such as {#.select~} and {#.display~}.[||<b> x = acos(y)</b>|the arccosine of y, in degrees in the range 0 to 180 .||<b>x = angle(a,b,c)</b>|the a-b-c angle, where a, b, and c can be points or atom sets.||<b>x = angle(a,b,c,d)</b>| the dihedral angle a-b-c-d is measured.||<b>x = cache()</b>|returns an array describing the contents of the file {#.cache~} in the  format {{filename: nBytes, filename: nbytes,...}}  (Jmol 13.2)||<b>x = color(schemeName, min, max)</b>|  (Note: To get just the {{r g b}} color value of a named color, use the <b>.color</b> function rather than the function described here: <b>print "red".color</b> reports {{255.0 0.0 0.0}}.) Coloring by atom property and mapping of isosurfaces utilize mapped color schemes, which are sets of colors that are associated with given ranges of values. The <b>color</b> function, introduced in Jmol 12.2, returns information relating to these schemes. The basic form of the function, with one or three parameters returns an associatvie array describing a color scheme such as "rwb" or "roygb" that has been mapped from value <b>min</b> to value <b>max</b> with the keys listed below. If <b>schemeName</b> is "", then the current property scheme (default "roygb") is used.) The parameters <b>min</b> and <b>max</b> are optional and, if absent, will take the numbers 1 and N, where N is the number of colors in the color scheme if no property has been mapped, or the most recent range of values, if a property has been mapped.[|| "colors" | an array of points giving {{r g b}} on a 0 to 255 scale||"max" | the maximum value being mapped || "min" | the minimum value being mapped || "name" | the name of the color scheme || "reversed" | TRUE if <b>min</b> &gt; <b>max</b> || "values" | an array of values associated with the two ends and each boundary between individual color of the scheme. (Note that if there are N colors, then there will be N + 1 values.) ||] <br />Thus, for example, <b>print color("rwb", 0, 100)["colors"][1]</b> prints the first color of the color scheme, and <b>print color("rwb", 0, 100)["values"][0]</b> prints the last value (max). ||<b>x = color(schemeName, min, max, value)</b>| Returns a color as {{r g b}} on a 0 to 255 scale from a specified color scheme that has been mapped using the specified minimum and maximum values. For example: <b>print color("rwb", 0, 100, 30)</b> returns <b>{{255.0 144.0 144.0}}</b>.  (Jmol 12.2)|| <b>x = color("$isosurfaceID")</b> | Returns the color scheme associative array (see above) associated with this isosurface. (Jmol 12.2)|| <b>x = color("$isosurfaceID", value)</b> | Returns the color as {{r g b}} associated with the given value of a mapped isosurface. (Jmol 12.2) ||<b>x = compare({{atomset1}}, {{atomset2}})</b>|Returns a 4x4 matrix that gives the optimal  rotation (x%1) and translation (x%2) to superimpose the atoms in {{atomset1}} onto the atoms in {{atomset2}} using the closed-form quaternion method. One or the other or both of the atom sets may be replaced by an array of points. The number of atoms/points must be the same in each set, or "NaN" will be returned instead. (Jmol 12.0)||<b>x =compare({{atomset1}}, {{atomset2}}, "stddev")</b>|Returns the standard deviation of atom positions for the optimal rotation and translation to superimpose the atoms in {{atomset1}} onto the atoms in {{atomset2}}. The number of atoms must be the same in each set, or "NaN" will be returned instead. (Jmol 12.0)  || <b>x =compare({{atomset1}}, {{atomset2}}, "ISOMER")</b>| Carries out a comparisons of atoms in the two atom sets and returns one of the following strings (Jmol 12.2): [||  CONSTITUTIONAL ISOMERS | having the same molecular formula but not the same connectivity || ENANTIOMERS | configurationally enantiomeric (mirror images) || DIASTERIOMERS | differing in one or more stereocenters (including double bonds) || CONFORMATIONAL ISOMERS | having the same constitution and configuration, but differing significantly in conformation (having root-mean-square deviations greater than ?? || IDENTICAL | the same, constitutionally, configurationally, and conformationally || NONE | no relationship  ||] ||<b>x =compare({{atomset1}}, {{atomset2}}, "SMARTS", SMARTSstring)</b>|Uses a SMARTS (substructure) description to find the first instance of atoms in the first structure that correlate one-for-one with atoms in the second structure, then finds the rotation and translation that best aligns them. Returns a 4x4 matrix. (Jmol 12.0)||<b>x =compare({{atomset1}}, {{atomset2}}, "SMARTS", SMARTSstring, "stddev")</b> | Same as above, but returns the standard deviation. (Jmol 12.0) ||<b>x =compare({{atomset1}}, {{atomset2}}, "SMILES", SMILESstring)</b> |Same as "SMARTS", but does a whole-molecule comparison. (Jmol 12.0) ||<b>x =compare({{atomset1}}, {{atomset2}},  "SMILES", SMILESstring, "stddev")</b> |Same as "SMARTS", but does a whole-molecule comparison. (Jmol 12.0) ||<b>x = connected(...{{atomSet}})</b>|See the discussion of connected() at {#.atomexpressions~atom expressions}.||<b>x =connected(...{{atomset1}}, {{atomset2}})</b>| Specifying two atom sets in a CONNECTED function returns the bond set [{{...}}] including all such bonds. This bond set can be selected, colored, sized using {#.wireframe~}, and (starting with Jmol 12.2) hidden or displayed. || <b>x = contact({{atomset1}}, {{atomset2}})</b><br /> <b>x = contact(nPercent, {{atomset1}}, {{atomset2}})</b><br /> <b>x = contact(distance, {{atomset1}}, {{atomset2}})</b>|returns the atoms in {{atomset2}} within van der Waals contact of {{atomset1}}. {{atomset2}} is optional and defaults to "not atomset1". Optional parameters include a percent van der Waals radius expressed as an integer or a distance in Angstroms do be added to the van der Waals radius, expressed as a decimal number (Jmol 12.2)||<b>x = cos(y)|the cosine of y, where y is in degrees  ||<b>x = cross(a,b)</b>| the cross product of two vectors of the form {{x,y,z}} ||<b>x = data({{atomset}},"type")</b>|creates model file data of the type MOL, PDB, or XYZ for the selected atom set. ||<b>x = data("dataset_name")</b>|places the text of the data set created using the {#.data~} command into variable x.||<b>x = data(stringData,fieldOrColumn,columnCount,firstLine)</b>|separates stringData into lines, then reads lines starting from firstLine (1 being the first line). Data are read from free-format field or column fieldOrColumn. If the data are free-format, then set columnCount = 0, otherwise columnCount indicates the number of columns to read for each data point. The data() function returns a newline-separated list, which can be read directly into atomic properties, for example, using <b>{{*}}.partialCharge = data(load("mydata.mol2"),9,0,7)</b>.||<b>x = distance(a,b)</b>| The distance from the geometric center of a to the geometric center of b, where a and b are atom expressions or coordinates.||<b>x = file("filename")</b>| the full path to the indicated file name. Note that <b>file("?")</b> displays a file dialog, which allows the user to navigate to a different directory, and <b>file("")</b> returns the full path to the current default directory ||<b>x = format("sprintf format", a, b, c, ...)</b>|This function creates a string using a format similar to that used in C++ to format a set of variables into a string with special codes that start with %. While not an exact implementation of this format, there are strong similarities. Here a, b, and c are variable names, and "sprintf format" is the format string containing a </b>%n.mX</b> code for each variable. As for {#.labelcolumnformattingandscientificnotation~labels}, <b>n</b> and <b>m</b> indicate column formatting and precisions. Both <b>n</b> and <b>.m</b> are optional. Here, <b>X</b> indicates the variable type , whereas in, for example, {{*}}.label("%a"), <b>X</b> represents an atom property, and the type of formatting is determined automatically from that. The options for <b>X</b> include:<br />[||   <b>i</b> or <b>d</b>| integer. Either i or d can be used synonymously.||   <b>f</b>|float/decimal. A negative value for m indicates to use scientific notation with a total of -m digits. The default is full width, full precision.|| <b>e</b>|exponential (scientific notation). "%8.3e" is equivalent to "%8.-3f". ||   <b>p</b>|point. Each of the three coordinates is formatted according to the specified width and precision. The default is %6.2p. ||   <b>q</b>|quaternion/plane/axisangle. Each of the four elements of the vector {{x y z w}} are formatted according to the specified width and precision. The default is %6.2q||   <b>s</b>|string. Values for precision, m, determine maximum number of characters starting from the left (m > 0) or right (m < 0). So, for example, <b>print format("%0.-3s","testing")</b> prints "ing".  ||]<br />For example: <b>calculate straightness;print format("average straightness = %4.2f", {{*}}.straightness)</b>||<b>x = getProperty("type",parameters...)| The {#.getProperty~property} of the given type is returned as a Jmol math list, <b>print getProperty()</b> by itself giving the list of available types. Each property type has its own intrinsic structure, but in general the parameters may include an initial atom set specification followed by one or more key values and, in the case of arrays, an item selector. For example, <br /><br /><b>print getProperty("boundboxInfo","center")</b><br /><b> x = getProperty("atomInfo",{{atomno=3}})</b> <br /><b>x = getproperty("bondInfo",{{*}},2,"atom1", "sym")</b>||    <b>x = hkl(a,b,c)</b>| generates the plane associated with a given set of Miller plane indices. (Jmol 12.0) || <b>x = intersection(plane1, plane2)</b>|returns the line of intersection between two planes in the form of an array: [point,vector] or "" if the planes are parallel. (Jmol 12.2)||<b>x = intersection(point,plane)</b>|returns the nearest point on a plane to a given point. (Jmol 12.2)||<b>x = intersection(point,vector,plane)</b> |returns the point of intersection of a plane with a line trough the given point with in the direction of the given vector. (Jmol 12.2)||<b>x = intersection(point1, point2, point3)</b>|returns the nearest point to point3 on the line through point1 and point2. (Jmol 12.2)       ||<b>x = javascript("...")</b>|returns the result of evaluating the specified JavaScript. Applet only; disallowed if on the web page _jmol.noEval = true.||<b>x = label(...)</b>| See <b>x = format(...)</b>, above.||<b>x = load("filename")</b>| Load the data from the specified file into variable x.||<b>x = load("filename", nBytesMax)</b>| Load the data from the specified file into variable x, but no more than the specified number of bytes. Since this function returns "java.io.FileNotFoundException: ..." when working from a local drive, the function <b>x = load("filename", 0)</b> can be used to test for the existance of a file. If the function returns the empty string "", then the file exists; if it returns an error message, then the file does not exist. So, for example, the following code loads a PDB file from the RCSB only if not found on the local drive:<b><br /><br />if (load(pdbid + ".pdb", 0) == "") {{load @{{pdbid + ".pdb"}} }} else {{load @{{"=" + pdbid}} }}<br /><br /></b>||<b> x = measure(...)</b>| The <b>measure</b> function requires from two to four atom expressions and/or points in space and returns associated measurements in a string, one measurement per line. Additional optional parameters include minimum and maximum ranges, the designations "connected" or "notConnected", the units "nm", "nanometers", "pm", "picometers", "angstroms", "ang", or "au", and a format string in quotes similar to that used by the {#.measure~} command. (Jmol 12.0) ||<b>x = now()</b>| returns the time in milliseconds since some old date; can be used for timing scripts. <b>now(x)</b> returns the number of milliseconds since time <b>x</b>:  <b>x = now();.......;print now(x);</b>. (Jmol 12.0)|| <b>x = plane(pt1, pt2)</b>| the plane that bisects the line connecting pt1 and pt2. (Jmol 12.2) ||<b>x = plane(pta,ptb,ptc,ptd)</b>|creates an {{x y z w}} plane from the first three points, and assigns the signs of x, y, z, and w to correspond to a positive distance to ptd as measured with the distance() function. Parameters may be mathematical expressions.  ||<b>x = plane(a, b, c, d)</b>|creates the four-vector {{a b c d}}, which represents a plane satisfying the equation <b>ax + by + cz + d = 0</b>||<b>x = plane("{{a b c d}}")</b>|creates an plane satisfying the equation <b>ax + by + cz + d = 0</b> from a string equivalent. As for all Jmol math expressions, parameters may be mathematical expressions.  ||<b>x = plane(pta,ptb,ptc)</b>|creates an {{x y z w}} plane through the three given points, which may themselves be mathematical expressions that evaluate to {{x y z}} points or atom expressions.  || <b>x = plane(r,theta,phi)</b>|generates a plane that is perpendicular to the vector from {{0 0 0}} to the point defined by the given r, theta, and phi. The vector is generated by sequential rotation of a vector along the Z axis first by phi around the X axis, then by theta around the Z axis: <b>v = R[z,theta]R[x,phi]{{0 0 r}}</b>. ||<b>x = point(a,b,c)</b>|Creates an {{x y z}} point. Parameters may be mathematical expressions.||<b>x = point("{{x,y,z}}")</b>|creates an {{x y z}} point from the string equivalent. Parameters may be mathematical expressions.  ||<b>x = point(x)</b>|Rounds x  down (toward 0) to the nearest integer. Note that x%0 rounds to the NEAREST integer value, instead. [12.2] ||<b>x = prompt(message)</b>|displays a pop-up message box and waits for the user to press OK||<b>x = prompt(message,defaultInput)</b>|displays an input dialog allowing the user to enter some text and press OK (or ESCAPE to cancel). If canceled, returns "null". ||<b>x = prompt(message,buttonText,TRUE)|displays a message box with buttons and returns the label of the button that was pressed. The parameter is a list of button labels separated by &brvbar;, for example: "Yes&brvbar;No" or "OK&brvbar;cancel" or "Spacefill&brvbar;Wireframe&brvbar;Ball&Stick".||<b>x = prompt(message,buttonArray)|displays a message box with buttons based on the values in the array parameter and returns an integer indicating which button was pressed (starting with 1, or -1 if ESCAPE was pressed). ||<b>x = quaternion()</b>| The current model rotation in the form of a quaternion. Same as <b>x = quaternion(script("show rotation"))</b> (Jmol 12.2)||<b>x = quaternion("best")</b>| The  {#.rotate_best~"best"} model rotation in the form of a quaternion. Same as <b>x = quaternion(script("show best rotation"))</b> (Jmol 13.4)||<b>x = quaternion({{x y z}},theta)</b>|the quaternion {{x y z w}} associated with a rotation of <b>theta</b> degrees (counter-clockwise) around axis {{x y z}}. {{x y z}} need not be a unit vector -- Jmol will normalize it automatically. ||<b>x = quaternion("{{x y z w}}")</b>|the unit quaternion {{x y z w}} produced by normalizing the specified quaternion. That is, where <b>theta</b> is the rotation angle, and <b>q0 = w = cos(theta/2)</b>, <b>{{x y z}} = sin(theta/2) * unitNormal</b>. ||<b>x = quaternion(q0, q1, q2, q3)</b>|the unit quaternion {{x y z w}} produced by normalizing the specified quaternion. Note that q0 is first in the list of parameters, even though Jmol will store the quaternion in the form {{q1/f, q2/f, q3/f, q0/f}} where <b>f</b> is sqrt(q0*q0 + q1*q1 + q2*q2 + q3*q3).  ||<b>x = quaternion({{m00 m10 m20}}, {{m01 m11 m21}})</b>|the unit quaternion corresponding to the rotation matrix having the first two column vectors indicated. ||<b>x = quaternion({{center}},{{x-axis point}}, {{xy-plane point}})</b>|the unit quaternion associated with a frame that has a center at the first parameter position, an x axis in the direction of the second parameter position, and a y axis in the plane of the three parameter positions. These parameters may be atom expressions. For example, <b>x = quaternion({{215.CA}},{{215.C}},{{215.N}})</b> creates the "standard" quaternion for residue 215. The z axis will be generated using the x and y axes and the right-hand rule. ||<b>x = quaternion({{atom}})</b>| the quaternion associated with the an atom (or the first atom in the atom set) based on the setting of <b>set quaternionFrame</b>.||<b>x = quaternion({{atomset}}, nMax)</b>| an <i>array</i> of quaternions, one per residue, up to nMax long (or all if nMax is &lt;=0). Quaternions are created based on the setting of <b>set quaternionFrame</b>.||<b>x = quaternion({{atom1}}, {{atom2}})</b>| the quaternion difference of the two residues containing the specified atoms (or the first atom in each set, if applicable), created based+I219 on the setting of <b>set quaternionFrame</b>.  An optional last parameter "relative" utilizes quaternion left-division. The quaternion relative difference represents the necessary rotation to get from q1 to q2 within the reference frame of q2 rather than the standard reference frame (Jmol 12.0)||<b>x = quaternion({{atomset1}}, {{atomset2}}, nMax)</b>| an <i>array</i> up to nMax long (or all if nMax is &lt;=0) of quaternion differences, one per residue pair in the two sets, created based on the setting of {#set(misc)~set quaternionFrame}.  An optional last parameter "relative" utilizes quaternion left-division. (Jmol 12.0)||<b>x = quaternion(quaternionArray1, quaternionArray2)</b>| an <i>array</i> of quaternion differences of the two array elements, taken a pair at a time.  An optional last parameter "relative" utilizes quaternion left-division. (Jmol 12.0)||<b>x = script("...")</b>|returns the output from the specified script command; particularly informative script commands include {#.getProperty~} and {#.show~}.||<b>x = script("...", appletName)</b>|returns the output from the specified script command run in one or more applets. For example, <b>print script("show orientation moveto", 2)</b> will print the orientation moveto command for an applet with name "2" or, if that does not exist, "jmolApplet2". See {#.script~} for details. ||<b> x= select(x;{{a}};b)</b>|selects atoms from the atom expression <b>{{a}}</b> based on the boolean expression <b>b</b>.  Note the use of semicolons, not commas, to separate the three components of this function. The variable <b>x</b> is local to the function, and when it appears in the boolean expression in the form <b>x.<i>property</i></b> represents a single atom of the atom expression. For example, <b>x = select(a;{{*}};a.distance({{0 0 0}}) > 3 and a.elemno > 18)</b>. select() functions can be nested -- just use two different variable names: <b>x = select(x;{{*.ca}};x.phi < select(y; {{*.ca}}; y.resno = x.resno + 1).phi)</b>. The select() function provides a powerful selection mechanism that can utilize any Jmol math expressio+I353n involving properties of an atom. (In contrast, {#.select~} command comparisons are limited to =, &lt;, &gt;, &lt;=, and &gt;=, and values are rounded to the nearest 1/100th).   ||<b>x = sin(y)|the sine of y, where y is in degrees ||<b>x = sqrt(y)|the square root of y ||<b>x = substructure("smiles")</b>| find atoms matching the given smiles string, which may include bond types such as = or - between atoms. More powerful SMILES/SMARTS functions include  {#.jmolsmarts/smiles~find() and search()}. (Jmol 12.0)||<b>x = within(...)</b>|returns a matching atom set for a wide variety of atoms that are in some way "within" another atom set or plane or "within some distance" of any one of a set of atoms or a plane.||<b>x = within("SMILES", smilesString,{{searchSet}})</b>|returns any array of atom sets that match the given smiles string. If provided, only atoms in {{searchSet}} will be checked. More powerful SMILES/SMARTS functions include  {#.jmolsmarts/smiles~find() and search()}. (Jmol+I330 12.0) ||<b>x = write(...)</b>|the output of the {#.write~} command is loaded into variable <b>x</b>. The parameters are those of the write command. For example, <b>x = write("PDB")</b> or <b>x = write(quaternion, "r", "difference2")</b>. Note that <b>x = write("image")</b> is not supported.||]','TEXT','x = f(y) functions','')
newCmd('','11.7.37','','*v+11.2;','These functions operate on coordinates, either the geometric center of a set of atoms expressed as {{atom expression}} or a coordinate point expressed as {{x y z}}. The unit cell system used is the currently defined {#.unitcell~} for the current model. If more than one model is visible, all coordinates are considered Cartesian coordinates, and these functions are not distinguishable. <br /><br /> [||<b>x = pt.xyz</b> | the Cartesian coordinates for the point.  For example, if <b>pt = {{1/2 3/2 1}}</b> in an orthonormal unit cell with a = 28.0, b = 5.04, c = 6.04, then pt.xyz would equal {{14.0 7.56 6.04}}.  || <b>x = pt.fxyz</b>| the fractional coordinates of a point. In the same case, <b>pt.fxyz</b> would be {{0.5 1.5 1.0}}.  ||<b>pt.fx</b>, <b>pt.fy</b>, <b>pt.fz</b>| The fractional x, y, and z coordinates, respectively.|| <b>x = pt.uxyz</b>| the unit-cell normalized point in the range [0,1). In this case, <b>pt.uxyz</b> would be {{0.5 0.5 0}}. ||<b>pt.ux</b>, <b>pt.uy</b>, <b>pt.uz</b>| The unit cell x, y, and z coordinates, respectively.  || <b>pt.x</b>, <b>pt.y</b>, <b>pt.z</b>| The x, y, or z component of the point, regardless of the unit system.  ||]<br /><br />Note that the fractional notation {{1 1/2 1}} is immediately converted to Cartesian coordinates, so {{1/2 3/2 1}} = {{14.0 7.56 6.04}} in this case, and {{1/2 3/2 1}}.y = 7.56, whereas {{1/2 3/2 1}}.fy = 1.5, and {{1/2 3/2 1}}.uy = 0.5. Note that if vibration vectors are set using, for example, <b>{{atomno=3}}.vxyz = {{1/2 1/2 1/2}}</b>, then the <b>frame</b> command should be sent so as to update the popup menu so that the <b>vibration</b> menu item is enabled .','TEXT','coordinate transforming .xxx functions','')
newCmd('','','','*v+11.8;','These modifiers can be used with a number of different variable types.<br /> [||<b>x = y.count</b>|Same as <b>y.size</b> -- length of array or string, or number of atoms/bonds in an atom/bond set. But see also .count(), below.||<b>x = y.keys</b>|Returns the set of keys in associative array y. (Jmol 12.0) ||<b>x = y.length</b>|In the case of x a set of bonds, the average length of the bonds. If y is a DRAW or ISOSURFACE id, such as <b>$draw1</b>, then <b>$draw1.length</b> returns a point {{a b c}} where a is the index of the object in the set of DRAW/ISOSURFACE objects, b is the total number of such objects, and c is the number of vertices. (Jmol 12.2) In all other cases, the length of the data. For example: <b>x = {{carbon}}.bonds.length</b>, <b>x = {{*}}.length</b>. To check the size of a bondset, use .size. ||<b>x = y.lines</b>|splits y into a set of lines based on new-line characters, appending a new-line character onto the end if necessary so that there is one new-line character per line.</b>||<b>x = y.size</b>|The nominal "size" of y, which depends upon its data type -- number of characters in a string, number of elements of an array, number of selected atoms or bonds in a bitset. Except for bonds, same as <b>.length</b>. Negative numbers indicate boolean (-1), integer (-2), decimal (-4), point (-8), or plane (-16).||<b>x = y.type</b>|the type of variable; one of "boolean", "integer", "decimal", "string", "point", "point4", "bitset", "array", "matrix3f", "matrix4f" ||]','TEXT','general .xxx functions','')
newCmd('','','','*v+11.8;','Several modifiers can be added to property functions <b>{{atoms}}.y</b> or arrays, such as <b>[1.2, 1.4, 1.6]</b>.<br /> [||<b>.all</b>| If a property, such as <b>{{selected}}.vanderwaals</b>, then appending <b>.all</b> creates a list of those measures. This list can be used to transfer one property to another, as in: <b>{{*}}.partialCharge = {{*}}.temperature.all</b>, which would allow temperature data to be used for partial charges in an <b>isosurface molecular map MEP</b> command, for instance. Note that the "list" is really an array of string values. ||<b>.average</b>|the average value (the default modifier). ||<b>.max</b>|the maximum value, for example: {{*}}.temperature.max</b>||<b>.min</b>|the minimum value, for example: {{*}}.partialCharge.min</b> ||<b>.reverse</b>|reverses an array IN PLACE (does not make a copy). Thus, the preferred command is simply <b>xxx.reverse</b> (Jmol 12.2).||<b>.sort</b>|sorts an array IN PLACE (does not make a copy). Thus, the preferred command is simply <b>xxx.sort</b> (Jmol 12.2). An optional parameter allows for sorting a multidimensional array based on a specific "column" of data: For example, <b>x = [ [1,"orange"], [2,"apple"], [3, "banana"] ]; x.sort(2)</b> sorts this array to read: [[2,"apple"], [3, "banana"],  [1,"orange"]  ] ||<b>.stddev</b>|the standard deviation, for example: <b>print {{helix}}.straightness.stddev</b> ||<b>.sum</b>|the sum of the values (Jmol 12.0; before this version, use <b>.add()</b>)||<b>.sum2</b> |sum of squares ||]','TEXT','[array].xxx or .propertyName.xxx modifiers','')
newCmd('','','','*v+11.8;','These functions operate on the individual elements of some group or listing. In addition to these are all of the atom properties described under the heading {#.atomexpressions~atom expressions}, for example, <b> x = {{oxgyen}}.temperature</b>, in which case they give the average value. [||<b>x = y.atoms</b>|The atoms associated with a set of bonds.||<b>x = y.bonds</b>| The bonds associated with the specified atoms, using the current setting of <b>bondModeOr</b> (true, bonds having one OR the the other atom within this set; or false, bonds having BOTH atoms within the set) ||<b>x = y.boundbox</b>|A Jmol math list containing the boundbox center, vector to corner, and two opposite corners associated with this atom set ||<b>x = y.color</b>|The average color of the atoms in the set y expressed as a {{r,g,b}} coordinate in color space. If y is already a point, then .color converts this to a string of the form [[xRRGGBB]]. If the propertyColorScheme has been set, and it has been used, for example with <b>color atoms property partialcharge "rwb" range -1 1</b> or <b>color "roygb" range 0 10</b>, then y can be a number, as in <b>x = (3).color</b>, in which case the color associated with that value is returned. Using this mechanism, a key can be generated within the Jmol applet using echo text appropriately positioned on the screen. Leaving the echo blank but coloring the background of the echo as well as the echo itself produces a horizontal bar of the desired color.   ||<b> x = y.ident</b>| a list of the standard identity labels for the elements of y, either atoms or bonds.||<b>x = y.length</b>|In the case of x a set of bonds, the average length of the bonds. In all other cases, the length of the data. For example: <b>x = {{carbon}}.bonds.length</b>, <b>x = {{*}}.length</b>. To check the size of a bondset, use .size. ||<b>x = y.size</b>|The number of selected atoms or bonds in a bitset. ||]','TEXT','{atomExpression}.propertyName','')
newCmd('','','','*v+11.2;','These functions operate on the individual elements of some group or listing, returning a modified listing.[||<b>x = data1.add(data2)</b>|Specifically for use with data set lists, adds each element of data1 to its corresponding element in data2 and returns the result. If data2 is a simple number, adds that number to each element of data1.||<b>x = data1.add("\\t",data2)</b>|Specifically for use with data set lists, creates a new column separated from the previous with a tab. ("\\t" may be replaced with whatever separation one desires. ||<b>x = yArrray.bin(low,high,binSize)</b>| Operating on an array, delivers an array that counts the number of values in specific bins defined by a low limit, a high limit, and a bin size. Values less than the low limit are also included in the first bin, and values greater than or equal to the high limit are included in the last bin. Values equal to the lower limit of a bin are included in that bin. ||<b>x = yArray.count()</b>| Operating on an array, <b>.count()</b> returns an array of [value, count] pairs. Thus, for example, <b>{{*}}.radius.all.count()</b> would return a list of all the radii of all the atoms and how many of each radius value  there are. (Jmol 12.2) || <b>x = y.count(z)</b>| returns how many <b>z</b> there are in y, which should be either an array or a string. (Jmol 12.2).     || <b>x = y.distance({{atoms}})</b>|the average distance from elements of y to the CENTER of {{atoms}}||<b>x = {{atomset1}}.distance.min({{atomset2}})</b> | returns an array listing the minimum distance of each atom in atomset1 to any atom in atomset2||<b>x = {{atomset1}}.distance.min({{atomset2}}).min</b> | returns the shortest distance from any atom in atomset1 to any atom in atomset2.||<b>x = {{atomset1}}.distance.max({{atomset2}})</b> | returns an array listing the maximum distance of each atom in atomset1 to any atom in atomset2||<b>x = {{atomset1}}.distance.max({{atomset2}}).max</b> | returns the furthest distance from any atom in atomset1 to any atom in atomset2.||<b>x = y.find("s")</b>|finds the first location of "s" in y or, in the case of y being a set of lines, selects only the lines of y containing "s".||<b>x = y.find("pattern","flags")</b>| Searches string or list y for a {http:<http://java.sun.com/j2se/1.4.2/docs/api/java/util/regex/Pattern.htm~Java regular expression pattern}. The second parameter is a set of flags. This parameter must be included, even if it is the blank string "" so as to distinguish this command from the standard .find() command. Flags include:<br><br> [|| &nbsp;|(no flags) Returns the position in the string containing the pattern, starting with 1, or 0 if the pattern is not found. For lists, returns a sublist containing the elements that match.||<b>i</b> |(case-insensitive) Match upper  or lower case letters.||<b>v</b> |(reverse match) With a string, <b>v</b> returns "false" if the string contains the match or "true" if it does not; with lists, <b>v</b> returns all elements of the list that do not contain the match.|| <b>m</b>|(return match) The <b>m</b> option allows returning only the portion of the string that matches (or, with <b>vm</b>, only the portion of the string that does NOT match). With lists, both <b>m</b> and <b>vm</b> return only elements that contain the match, but, as for strings, each element is returned as just the matching phrase or the reverse. ||]<br><br>Note that special characters such as \\S and \\d must be escaped with two back-slashes, and if they are introduced via JavaScript, they will need double escaping (four back-slashes). Examples include:<br><br>[||"this test is a Test".find("Test","") |<b>16</b>||print "this test is a Test".find("Test","i")|<b>6</b>||print "this test is a Test".find("Test","m")|<b>Test</b>||print "this test is a Test".find(" a test","v")|<b>true</b> (because it was <b>not</b> found)||print "this test is a Test".find(" Test","ivm")|<b>this is a Test</b>||print "this test is a Test".find("\\\\stest","m")|<b> test</b>||print "this test is a Test".find("\\\\stest","vm")|<b>this is a Test</b>||print script("show spacegroup all").split()<br />.find("Hall symbol:")|<b>Hall symbol: P 1<br />primitive Hall symbol: P 1<br />Hall symbol: -P 1<br />primitive Hall symbol: P 1 -1<br />Hall symbol: P 2y<br />primitive Hall symbol: P 2y<br />...</b>||print script("show spacegroup all").split()<br />.find("Hall symbol:").find("primitive","v")|<b>Hall symbol: P 1<br />Hall symbol: -P 1<br />Hall symbol: P 2y<br />...</b>||print script("show spacegroup all").split()<br />.find("Hall symbol:").find("primitive","v")<br />.find("Hall symbol:","vm")[1][3]|<b> P 1<br /> -P 1<br /> P 2y<br /></b>   ||] ||<b>x = {{atomExpression}}.find("smartsString", "BONDS") |  Finds the first match in the structure to the given SMARTS string and returns an array of all the dihedrals involving the matching atoms. Each dihedral is expressed as an ordered list of the four atom indices. ||<b>x = {{atomExpression}}.find("MF")</b>|(Jmol 12.0) finds the molecular formula of the atom expression.||<b>x = {{atomExpression}}.find("SEQUENCE")</b>|(Jmol 12.0) returns the Jmol {#.Jmol SMILES/SMARTS~bioSMILES} sequence for the specified atoms. An optional second parameter TRUE adds crosslinking.||<b>x = smilesString.find("SMILES",pattern)</b>|(Jmol 12.0) Searches a SMILES string for at least one occurance of <b>pattern</b>. For example: <b>"O[C@](F)(Cl)I".find("smiles","[C@](O)(F)(Cl)I")</b> would return the value 6, meaning all 6 atoms of O[C@](F)(Cl)I (including the implied H on the oxygen) were a match. A return value of 0 means the pattern was not found; a return of -1 indicates there was a problem parsing one or the other of the SMILES strings. This function allows Jmol to match two SMILES strings without need of "canonicalization."||<b>x = smilesString.find("SMARTS",pattern)</b>|(Jmol 12.0) Searches a SMILES string for at least one occurance of <b>pattern</b>, where the pattern is in the form of a SMARTS string. For example: <b>"CCCC".find("smarts","CC")</b> would return the value [1, 5, 8, 11], because in the full structure, CH3CH2CH2CH3, the 1st, 5th,8th, and 11th atoms match. A return value of [], with .length = 0, means the pattern was not found; a return of "?" indicates there was a problem parsing one or the other of the strings. Since an array evaluates to its length and "?" is not greater than 0 in a logical context, a simple test of a SMARTS match is <b>if("CCCC".find("smarts","CC") > 0)...</b>.This function allows Jmol to match a SMILES string with a SMARTS pattern.||<b>x = y.find("SMILES","MF")</b>| returns the canonical molecular formula of the SMILES string associated with the SMILES string or atom set y||<b>x = y.find("SMARTS", "MF")|returns the canonical MF for the SMARTS string (thus not including H atoms) of the SMILES string or atom set y. ||<b>x = y.join("s")</b>|joins lines of y using the character or string "s"||<b>x = y.label("format")</b>|A list of labels for atoms or bonds using format strings. (see {#.label~})||<b>x = data1.mul(data2)</b>|See data1.add()||<b>x = y.replace("s1","s2")</b>|replaces all occurances of "s1" with "s2" in y. If y is a number, this function first converts the number to a string, then does the replacement.||<b>x = y.split("s")</b>|splits y into lines by replacing all occurances of "s" with a new-line character and converting the string to a set of lines.||<b>x = data1.sub(data2)</b>|See data1.add()||<b>x = y.substring()</b>| Jmol math does not include a .substring() function. Instead, this is handled using the more general item-selector [i][j] syntax, described below.||<b>x = y.symop(op,atomOrPoint)</b>| Returns the point that is the result of the transformation of <b>atomOrPoint</b> via a crystallographic symmetry operation. The atom set <b>y</b> selects the unit cell and spacegroup to be used. If only one model is present, this can simply be <b>all</b>. Otherwise, it could be any atom or group of atoms from any model, for example <b>{{*/1.2}}</b> or {{atomno=1}}. The first parameter, <b>op</b>, is a symmetry operation. This can be the 1-based index of a symmetry operation in a file (use <b>show spacegroup</b> to get this listing) or a specific Jones-Faithful expression in quotes such as <b>"x,1/2-y,z"</b>. ||<b>x = y.symop(op,"label")</b>| This form of the .symop() function returns a set of {#.draw~} commands that describe the symmetry operation in terms of rotation axes, inversion centers, planes, and translational vectors. The <b>draw</b> objects will all have IDs starting with whatever is given for the label.  ||<b>x = y.tensor("type","property")</b>|returns one of several tensor properties using tensors read by certain file readers (CASTEP, QuantumEspresso, Magres). "type" might be "charge" or "isc" or "ms", for example. Eigenvectors are sorted based on their corresponding eigenvlues, using the convention: &brvbar;v3 - v1&brvbar; >= &brvbar;v3 - v2&brvbar; > &brvbar;v2 - v1&brvbar;. Implemented "property" values include:[||   "anisotropy"|v3 - (v1 + v2)/2||   "asymmetry"|(v2 - v1)/(v3 -v_iso)||   "asymMatrix"|matrix representation of the asymmetric tensor component|| "chi"|specific to electric field gradient (efg) tensors, the quadrupolar constant || "dc" | dipolar coupling constant; tensor type ignored|| "eigenvalues"|[v1,v2,v3]||   "eigenvectors"|[V1,V2,V3]||   "eulerzxz" |[alpha,beta,gamma] for z-x-z convention||   "eulerzyz" |[alpha,beta,gamma] for z-y-z convention||   "indices" |[modelIndex,atomIndex1,atomIndex2], with atomIndex2 = -1 if this tensor is single-atom based||   "isotropy" | (v1 + v2 + v3)/3 || "j" | J-coupling constant;specific to indirect spin coupling (isc) tensors, in hz||  "quaternion" | the quaternion representing the rotation assoicated with the symmetric matrix||  "span" | abs(v2 - v0) || "skew" | (3 (v1 - isotropy) / span)||  "symMatrix" |the symmetric matrix.||   "value" | v3, the eigenvalue furthest from the mean||"string"| a listing of selected readable data||   "type" | the tensor type, such as "charge" or "isc" or "ms" ||] Note that any other property name results in the entire set of properties to be returned in an associative array. ||<b>x = y.trim("chars")</b>|trims any one of the characters specified from both ends of the string y, or from every line of y if y is a set of lines.||<b>x = data1.div(data2)</b>|See data1.add(). Division by 0 results in the value "NaN", meaning "not-a-number".||]','TEXT','x.f(y) functions','')
newCmd('','11.1.49','','*v+11.2;','<br /><br />Specific elements of lists and strings can be selected using a square bracket notation similar to that used for {#.atom expressions~}. Positive numbers select for the specified element. For example, <b>x = "testing"[3]</b> selects for the third letter of the string, "s"; Zero selects for the last element: <b>x = "testing"[0]</b> selects "g". Negative numbers count from the end of the string backward, so <b>x = "testing"[-1]</b> selects "n". Two selectors select an inclusive range of items. <b>x = "testing"[2][4]</b> acts similar to a "substring()" method and selects characters 2 through 4 -- "est"; <b>x = "testing"[-1][0]</b> selects the last two characters, "ng". Similarly, for arrays, <b>array("this", "test", 3)[0]</b> selects the number 3, and <b>x = array("this", "test", 3); print x[1][2]</b> prints <br /><br />this<br />test<br /><br /> ','TEXT','item-selector [i][j]','')
newCmd('','','','*v+11.8;','In addition, Jmol allows for user-defined functions. You can define functions to do simple tasks, include parameters to create subroutines, and add return values to create math functions. With applets, if the function name starts with "static_", then the function will be defined for ALL applets, not just the one running the script. Prior to Jmol 12.0, functions were part of the state and were saved with a state; starting with Jmol 12.0, though, functions are not part of the state. Examples follow. <br /><br />[||function rotateModel {{<br />&nbsp;&nbsp;&nbsp;rotate {{atomno=1}} {{atomno=2}} 10 <br />}}<br /><br />rotateModel|Without any parameters listed, any commands within the function declaration are processed with a single command word.|| function rotateModelX(i,j,n) {{<br />&nbsp;&nbsp;&nbsp;a[2] = getProperty("orientationInfo.moveTo")<br />&nbsp;&nbsp;&nbsp;var x = 10<br />&nbsp;&nbsp;&nbsp;rotate {{atomno=i}} {{atomno=j}} @{{n + x}}<br />&nbsp;&nbsp;&nbsp;a[2] = getProperty("orientationInfo.moveTo") <br />}}<br /><br />a = [];rotateModelX(10,11,30, a);print a[1];|Any number of parameters may be passed into a function. The variable names are local to that function. Note that, like JavaScript, if an array is passed, it is passed "by reference," meaning that if its value is changed within the function, then that change is a global change. Variables preceded by "var" as in this example are local to that function.|| function getDistance(i,j) {{<br />&nbsp;&nbsp;&nbsp;var&nbsp;d&nbsp;=&nbsp;({{atomno=i}}.distance({{atomno=j}})*100)%0 <br />&nbsp;&nbsp;&nbsp;return d<br />}}<br /><br />print "the distance is " + getDistance(3,5)|The <b>return</b> command works as one might expect, returning the value of its expression.|| function getXplusY {{<br />&nbsp;&nbsp;&nbsp;return&nbsp;_x.x&nbsp;+&nbsp;_x.y<br />}}<br /><br />print {{atomno=3}}.getXplusY<br />print {{*}}.getXplusY.min<br />print {{*}}.getXplusY.max<br />print {{*.CA}}.getXplusY.all<br />|You can define atom selector functions. The local variable <b>_x</b> will represent the selected atom within the function. Such function references may include parameters in parentheses provided qualifiers such as .min, .max, or .all are not used within the function itself.||]<br /><br /> Functions must be defined prior to use. Similar to the JavaScript language, a function must be declared prior to its inclusion in a script command. You can redefine a function as many times as you wish. In addition, you can save a function\'s definition using <b>myFunctionDef = script("show function myFunction")</b> where function <b>myFunction</b> has been defined. This can be useful if a function definition is to be replaced and later returned to its original value using <b>script inline @myFunctionDef</b>. ','TEXT','user-defined functions','')
newCmd('','12.1.16','','*v+12.0;','Starting with Jmol 12.0, you can replace standard Jmol commands with commands of your own, and you can call functions just like any Jmol command, without the need for parentheses or commas. For example: <br /><br />[||function measure (a,b) {{<br />&nbsp;&nbsp;&nbsp;prompt "measuring " + a + " " + b<br />&nbsp;&nbsp;&nbsp;measure @a @b <br />}}<br /><br />measure {{atomno=1}} {{atomno=2}} |The function "measure" replaces the Jmol {#.measure~} command, allowing now just two parameters and displaying a message box when executed.||]','TEXT','customized command functions','')
newCmd('.geoSurface',' ','','*v+11.0 -- provides a quickly rendered crude geodesic molecular/solvent-accessible surface.;disp','Turns a crude geodesic molecular surface on or off around the currently selected atoms. If a decimal with an explicit "+" sign is given, or {#.set_solvent~set solvent ON}) is in effect, the resultant surface is a crude solvent-accessible surface.  This command has the same syntax as the {#.dots~} command. To color the surface, use {#.color (model object)~color geosurface}. For a smoother surface, use {#.isosurface~isosurface SASURFACE 1.2}.','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns geosurface rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','*v+11.0;','Draws a geodesic surface at the van der Waals radius for the selected atoms. See {misc/radii.xls~radii.xls}.','*2','VANDERWAALS','')
newCmd('',' ','','*v+11.0;','Draws a geodesic surface at the (nominal) ionic radius for the selected atoms if the atom\'s charge has been read and is nonzero; uses the nominal covalent bonding radius otherwise ({misc/radii.xls~radii.xls}).','*3','IONIC','')
newCmd('',' ','','*v+11.0;','Draws a geodesic surface at the indicated percent of the van der Waals radius for each atom (maximum value 1000%).','*4','(integer)','')
newCmd('',' ','','*v+11.0;','Draws a geodesic surface at the indicated radius in Angstroms for each atom (maximum value 10.0 Angstroms).','*5','(decimal)','')
newCmd('',' ','','*v+11.0;','Draws a geodesic surface at the indicated distance in Angstroms beyond the van der Waals radius for each atom (maximum value 10.0 Angstroms). The "+" sign is required. This surface approximates the solvent-accessible surface with the indicated solvent probe radius. Typically this number is +1.2 or +1.4. This command overrides the {#.set_solvent~set solvent/set radius} method of defining the solvent-accessible surface.','*6','+(decimal)','')
newCmd('.getProperty',' ','','*v+11.0 NEW;javascript','The <b>getProperty</b> command sends information to the {#.console~Jmol console} or message callback function defined for a Jmol applet using the jmolSetCallback("messageCallback", funcName) function in Jmol.js or via the {#.set (callback)~set} command. Either a simple text string in the case of a file property or a valid JSON (JavaScript Object Notation) string in the case of a molecular property is returned.  Used with jmolScriptWait(), the <b>getProperty</b> script command provides a powerful way to interact with the Jmol applet. Even simpler, though, is to use one of the Jmol.js built-in JavaScript commands jmolGetPropertyAsString(), jmolGetPropertyAsJSON(), jmolGetPropertyAsJavaObject(), or (most useful, probably) jmolGetPropertyAsArray().  For example, <code><br /><br />var modelInfo = jmolGetPropertyAsArray("modelInfo")<br />alert(modelInfo.modelCount)<br />for (int i = 0; i &lt; modelInfo.modelCount; i++)<br />&nbsp;&nbsp;alert(modelInfo.models[i].name)</code>. <br /><br />In addition, when using the Jmol application, see the note at {#.show~} regarding setting the output to go directly into a file on your system. This information is also available using the Jmol math {#.functions~getProperty()} function.','0|1','','')
newCmd('',' ','','*v+11.0;','Structure describing the current state of animation. See {misc/animationInfo.txt~animationInfo.txt}.','*10','animationInfo','')
newCmd('',' ','','*v+11.0;','Structure describing the applet, including, for example, the applet version, compile date, Java version, and name of the applet object. See {misc/appletInfo.txt~appletInfo.txt}.','*11','appletInfo','')
newCmd('',' ','','*v+11.0;','Structure describing the atoms in the model. A second (optional) parameter specifies a subset of the atoms. The default is (visible). Parentheses are required. See {misc/atomInfo.txt~atomInfo.txt}.','*12','atomInfo','(atom expression)')

newCmd('',' ','','*v+11.0;','Structure describing auxiliary information that is in the loaded file. This information is file-dependent and might include, for example, symmetry information, molecular orbital coefficients, dipole moments, partial charges, and/or vibrational modes. See {misc/auxiliaryInfo.txt~auxiliaryInfo.txt}.','*14','auxiliaryInfo','')
newCmd('',' ','','*v+11.0;','Structure describing the bonds in the model. A second (optional) parameter specifies a subset of the atoms that are involved in the bonds. The default is (visible). Parentheses are required. See {misc/bondInfo.txt~bondInfo.txt}.','*15','bondInfo','(atom expression)')
newCmd('',' ','','*v+11.0;boundbox','A simple JSON array containing the coordinates of the center and corner of the volume containing the molecule. See {misc/boundBoxInfo.txt~bondBoxInfo.txt}.','*16','boundBoxInfo','')
newCmd('',' ','','*v+11.0;','A single JSON array giving the current center coordinate. See {misc/centerInfo.txt~centerInfo.txt}.','*17','centerInfo','')
newCmd('',' ','','*v+11.0;','Structure describing the chains in a biomodel (PDB or mmCIF, for example). Information for each residue of the chain is provided. A second (optional) parameter specifies a subset of the atoms. The default is (visible). Parentheses are required. See {misc/chainInfo.txt~chainInfo.txt}.','*18','chainInfo','(atom expression)')
newCmd('',' ','','*v+11.0;','Returns an array having two elements. Generally the first element is the data label, and the second element is the {#.data~} itself. <b>getProperty data TYPES</b> returns a first element that is the string "types" and a second element that is a comma-separated list of the available data types. ','*19','dataInfo','type')
newCmd('',' ','','*v+11.0;','The extractModel keyword delivers text in the form of a MOL file, allowing up to 999 atoms and 999 bonds to be "extracted" from the model as an independent structure. ','*20','extractModel','(atom expression)')
newCmd('',' ','','*v+11.0;','The contents of the currently loaded file.','*21','fileContents','')
newCmd('',' ','','*v+11.0;','The contents of ANY file on the web or, if operating locally, any file on the hard drive in the directory containing the JAR file or any directory below that.','*22','fileContents','filepath')
newCmd('',' ','','*v+11.0;','The file header for the file. This will depend upon the file format; some file formats may not have file headers.','*23','fileHeader','')
newCmd('',' ','','*v+11.0;','The file name of the currently loaded file.','*241','fileName','')
newCmd('',' ','','*v+11.0;','An associative array of file information. For most files this is simply the file header lines. In the case of PDB files, this array associates PDB header information with its resource name. For example, <b>getProperty fileInfo.SHEET</b> reports SHEET information from the PDB header.  In the case of CIF files, this information is the entire contents of the file in array format. (See also the function <b>getProperty("fileInfo")</b>, which allows direct access to file information as a Jmol variable. For example, <b>x = getProperty("fileInfo.REMARK800")</b>. ','*243','fileInfo','type')
newCmd('',' ','','*v+11.0;','A base-64 encoded JPEG suitable for writing by the applet to an image tag using &lt;img src="data:image/jpeg;base64,…">" (does not work in Microsoft Explorer).','*251','image','')
newCmd('','13.1.7','','*v+13.2;','Returns information about an isosurface of <b>getProperty isosurfaceInfo</b> along with vertices, vertex values, and polygons.','*252','isosurfaceData','')
newCmd('',' ','','*v+12.2;','Returns information about an isosurface, including haveQuads, haveValues, id, polygonCount, and vertexCount.','*253','isosurfaceInfo','')


newCmd('',' ','','*v+12.2;','Structure indicating ligand information, including and array of "ligands", each containing items "reslist", "atoms", and "groupNames".','*281','ligandInfo','')
newCmd('',' ','','*v+11.0;','Structure describing the currently-defined measurements for the model, including the atoms involved, the measurement type, and the value of the measurement in decimal and in string formats. See {misc/measurementInfo.txt~measurementInfo.txt}.','*282','measurementInfo','')

newCmd('',' ','','*v+11.4;','A tab-separated string defining the current Jmol menu, including what options are currently enabled. See, for example, {misc/menu.tab~misc/menu.tab}. See also {#.show~show MENU}.','*300','menu','')
newCmd('','11.5.23','','*v+11.6;','A summary of the minimization that was carried out. See {misc/minimizationInfo.txt~minimizationInfo.txt}.','*301','minimizationInfo','')
newCmd('',' ','','*v+11.0;','Structure describing each model in the loaded model collection. See {misc/modelInfo.txt~modelInfo.txt}.','*302','modelInfo','')
newCmd('',' ','','*v+11.0;','Structure describing each molecule (covalent set of atoms) in the model, including the number of atoms, the number of elements, and the molecular formula. A second (optional) parameter specifies a subset of the atoms. The default is (visible). Parentheses are required. See {misc/moleculeInfo.txt~moleculeInfo.txt}.','*311','moleculeInfo','(atom expression)')
newCmd('','13.1.19','','*v+13.2;','Structure describing NMR calculation data including (a) a table of isotope masses (negative if these are default parameters for a nucleus), gyromagnetic ratio, and quadrapolar coupling constant, and (b) assigned shift references in PPM. See {misc/nmrInfo.txt~nmrInfo.txt}.','*312','nmrinfo','')
newCmd('',' ','','*v+11.0;','Structure describing the moveTo command required to return to the currently displayed orientation. See {misc/orientationInfo.txt~orientationInfo.txt}.','*32','orientationInfo','')
newCmd('',' ','','*v+11.0;','Structure similar to chainInfo describing the residues in a biomodel. A second (optional) parameter specifies a subset of the atoms. The default is (visible). Parentheses are required. See {misc/polymerInfo.txt~polymerInfo.txt}.','*33','polymerInfo','(atom expression)')
newCmd('',' ','','*v+11.0;','Structure listing a small amount of information relating to shapes displayed with the model (molecular orbitals, isosurfaces, cartoons, rockents, dipoles, etc.) See {misc/shapeInfo.txt~shapeInfo.txt}.','*34','shapeInfo','')
newCmd('',' ','','*v+11.0;','The same report as <b>SHOW STATE</b>; useful especiall as <b>x = getProperty("stateInfo")</b>.','*35','stateInfo','')
newCmd('',' ','','*v+11.0;','Structure representing the current transformation matrix describinng the current orientation of the model. See {misc/transformInfo.txt~transformInfo.txt}.','*36','transformInfo','')
newCmd('.goto','11.1.9','','*v+11.2 -- new;pause;flow','Transfers script execution to the {#.message~} command having the matching text. If an underscore is at the beginning of the text, then the text is not displayed, otherwise the message text is displayed in the console as usual for that command.<br /><br />message _test<br />...<br />...<br />goto _test<br /><br /><br />The <b>goto</b> command is largely deprecated by {#.if_while_for~<b>if</b>, <b>while</b>, and <b>for</b>}. These alternative commands are recommended.','0|1','','')
newCmd('.halos',' ','','*v+11.6 -- adds new capability;halos','Displays a translucent two-dimensional ring around an atom. Halos are similar to {#.stars~}, except halos may also be used for automatically displaying which atoms are currently selected. (This option is enabled using {#.selectionHalos~}.) The radius can be set using the same options as for {#.spacefill~}, but the actual radius of the halo will always be from 4 to 10 pixels larger then the nominal radius.  ','0|1','','')
newCmd('',' ','','','Turn halos on or off. The radius will change with the current spacefill setting.','*1','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns halo rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','Resets the radius of the halo to the default {#.spacefill~} radius.','*4','reset','')


newCmd('.help',' ','','*v+11.0 -- new;','Opens a new browser window to the interactive help page. See also {#.set(misc)~set helpPath}. Currently for the applet only; the query will be searched as an exact, complete phrase (applet only).','1','query','')

newCmd('.hbonds',' ','','*v+12.0 -- full support for hydrogen bonds;bond;hbondc;hbond;calc','Hydrogen bonds can be turned on or off, given custom widths in Angstroms, or colored (see {#.color(bond object)~color hbonds} and {#.set(bond styles)~set HBONDS} ). In addition, Jmol 12.0 allows calculation of actual hydrogen bonds (when hydrogen atoms are present in the selected set) or "pseudo" hydrogen bonds if hydrogen atoms are not present.','0|1','','')
newCmd('',' ','','','','*1','._on_off{"ON"}','')
newCmd('',' ','','','','*2','._hbond_width_angstroms','')

newCmd('',' ','','*v+12.0;calc','Calculates hydrogen bonds involving atoms currently selected and displays them. See {#.calculate~calculate HBONDS} for details.','*4','CALCULATE','')
newCmd('',' ','','*v+12.0;calc','Deletes hydrogen bonds between atoms in the currently selected set of atoms. Both atoms of a given hydrogen bond need to be selected. ','*5','DELETE','')
newCmd('.hide',' ','','*v+12.0 -- hide/display bonds;dhide','The opposite of {#.display~}. Hides atoms and associated structures (bonds, halos, stars, cartoons, etc.). <b>Hide</b> is similar to {#.select~} or {#.restrict~} in its syntax. Unlike  <b>restrict</b>, though, <b>hide</b> is completely reversible using <b>hide none</b>. (<b>Restrict</b> acts by setting the "size" of the object to 0; <b>hide</b> leaves the size the same, but just hides the object until unhidden. Group-based structures such as {#.cartoon~cartoons} and traces are hidden whenever their lead atom (the one that determines their position in space) is hidden. Hidden atoms can be selected with <b>select hidden</b> or deselected with <b>select not hidden</b>. Atoms can be added to the hidden set using <b>hide hidden or ...</b> or removed from the hidden set using <b>hide hidden and not ....</b>. In addition, starting with Jmol 12.2, bonds can be hidden or displayed.','0|1','','')
newCmd('',' ','','','Hides a set of atoms. Displays atoms and associated structures (bonds, halos, stars, cartoons, etc.) and hides all others. Starting with Jmol 12.2, the optional parameters ADD and REMOVE prior to the atom expression allow quick alternatives to "hidden or..." and "hidden and not..." In addition, the keyword GROUP prior to the atom expression provides a quick alternative to "within(group, ...)".','1','._atom_expression','')
newCmd('',' ','','','Hides a set of bonds.','2','[{{&#046;&#046;&#046;}}]','')
newCmd('',' ','','','Hides all bonds.','3','BONDS','')
newCmd('.history',' ','','*v+11.0 -- new;','The <b>history</b> command turns command history recording on and off. Each time it is invoked, the command history list is cleared. See also {#.set(files and scripts)~set history} and {#.show~show history}.','1','._on_off{"ON"}','')

newCmd('.hover',' ','','*v+11.0 -- adds hoverCallback and multiline hover;label','Turns on and off pop-up labels that appear when the user "hovers" the mouse over the atom or a point associated with a {#.draw~} object. <b>hover off</b> disables the hover message; <b>hover on</b> enables it again. Any other parameter or string is used as the label, which can contain atom property fields such as %a or %U. To change the delay time prior to the label appearing, use <b>set hoverDelay x.x</b> where <b>x.x</b> is the delay time in seconds. Setting this time to 0.001 seconds effectively makes the hover label appear immediately. See also  {#.label~}. Multiple lines can be indicated using &#124; (vertical bar). In the Jmol applet version 11.0, even with <b>hover OFF</b>, the hover message can be sent to a JavaScript function on the applet\'s page using {#.set_hoverCallback~}.','1|2','._on_off_string','')

newCmd('.if','11.7.40','','*v+11.4;math;flow','Jmol scripting allows for both the standard <b>if/else/elseif</b> statement set as well as the standard Java/JavaScript-like inline <b>(ifThis ? thenThis : elseThis)</b> construct.','1|2','','')
newCmd('','11.7.40','','*v+11.4;math;flow','The <b>if</b> statement takes a logical expression of variables and evaluates it either as TRUE or FALSE. Tests include a wide variety of flags that can be set using the {#.set~set} command as well as any {#.Jmolmath~user-defined variables}. The standard commands <b>elseif</b> (or <b>else if</b>) and <b>else</b> are supported. <b>If</b> blocks are defined by matched pairs of braces:<br /><br /><br />if (doSpacefill) {{<br />&nbsp;&nbsp;spacefill only<br />}} elseif (doWireframe) {{<br />&nbsp;&nbsp;wireframe only<br />}} else {{<br />&nbsp;&nbsp;wireframe 0.15;spacefill 20%<br />}}<br /> <br />These constructs are compatible with {#.goto~}. However, using <b>goto</b> to jump into the middle of an <b>if</b>,  <b>while</b>, or  <b>for</b> block is considered bad form and is not recommended. <br /> <br />Note that with the Jmol application multiline scripts are not allowed on the script window command line. Either separate commands by semicolon and introduce them all on the same line, or store the script in a file and run it using the {#.script~} command. This is not an issue with the applet.<br /><br /> <b>if/else/endif</b> syntax is also supported. If/else/endif syntax can be entered on a single line without unnecessary semicolons: <br/><br/>&nbsp;&nbsp;&nbsp;&nbsp;if (this) print "this" else print "that"<br/>&nbsp;&nbsp;&nbsp;&nbsp;if (this) {{ print "this" }} else {{ print "that" }} <br/>&nbsp;&nbsp;&nbsp;&nbsp;if (this); print "this"; else; print "that"; endif;<br/><br/>are all identical.','TEXT','if/else/elseif','')
newCmd('','11.7.42','','*v+11.8;','You can use a similar construct to Java\'s/JavaScript\'s <b>(ifTrue ? thisValue : orThisTrue ? thatValue : otherValue)</b>. Jmol accepts this format and also allows for a syntax that uses semicolons: <br /><br />&nbsp;&nbsp;&nbsp;&nbsp;xxx = if(ifTrue; thisValue; thatValue)<br /><br />For example, <br /><br />&nbsp;&nbsp;&nbsp;&nbsp;set echo top left;echo @{{(_modelnumber > 1.1 ? "previous" : "next")}}<br /><br /> or, equivalently, <br /><br />&nbsp;&nbsp;&nbsp;&nbsp;set echo top left;echo @{{if(_modelnumber > 1.1;"previous" ; "next")}}<br /><br />. Used in conjunction with inline FOR, the inline IF provides a powerful and concise way to work with molecular data.','TEXT','in-line IF','')



newCmd('.initialize',' ','','*v+11.2 -- new;load;rreset','The <b>initialize</b> command is a specialized script command only intended for use within a state script created using the {#.write~write state} or {#.save~save state} commands. It deletes user variables and resets most settings to original Jmol values in an anticipation of those settings being set in the state. It does not do a full reset to Jmol\'s initial state, however. If that capability is desired, one should save the initial state and then {#.restore~} it. Settings specifically NOT set include: allowGestures, allowKeyStrokes, allowModelkit, allowMultiTouch, debugScript, defaultDirectory, disablePopupMenu, legacyAutoBonding, messageStyleChime, useScriptQueue, useArcBall.','0|1','','')
newCmd('.invertSelected','11.1.13','','*v+11.2 -- new;anim','This command allows inversion of the selected atoms through a point or across a plane. With no parameters, the model is inverted through its geometric center.','0|1','','')
newCmd('','11.1.13','','*v+11.2;','Following the POINT keyword may be any expression that can be evaluated to give a point, including {{x y z}} (cartesian or fractional), a draw object identifier, $xxx, or an atom expression in parentheses, which will be evaluated as an average position.','*1','POINT','point_definition')
newCmd('','11.1.13','','*v+11.2;','inverts the selected atoms across a plane defined using any valid {#.plane expressions~plane expression}. ','*2','PLANE','plane_express')
newCmd('','11.1.13','','*v+11.2;','Specifies that the coordinates are the  Miller indices of the plane to be used for the inversion.','*3','HKL','{{h k l}}')
newCmd('','12.0.RC1','','*v+12.0;','Carries out a rotation of 180 degrees around a center about a line connecting the center and the geometric center of any connected atoms not in the list to invert. The results is the standard organic chemist\'s "stereochemical inversion" at that point -- which does not invert stereochemistry along the branches. See also {#.set picking~set picking invertStereo}','*4','STEREO','{{center}}','{{atomsToInvert}}','')
newCmd('.isosurface',' ','','*v+12.0 -- adds <b>"=XXXX"</b> for retrieving data directly from the Uppsala electron density server, <b>LATTICE {{a b c}}</b> for renderning multiple copies of an isosurface, and <b>MEP functionType</b> to support mapping of molecular lipophilic potential;iso','<img style="clear:right" src="img/straightness.png" width="250" height="250" alt="" />Jmol can generate a large variety of objects using the method of isosurfaces. Many of these surfaces can be color-mapped. Two Jmol commands (<b>isosurface</b> and {#.mo~}) render isosurfaces. The <b>isosurface</b> command itself provides ultimate control over these surface renderings. Before using the <b>isosurface</b> command, if you are interested in rendering molecular orbitals, you should first take a look at the {#.mo~} command. <br /><br />The general syntax of the <b>isosurface</b> command is as follows:  <br />  <br /><b>isosurface ID [object id] [construction/mapping parameters] [surface object] [additional mapping-only parameters] MAP [color mapping dataset] [display options] </b>.<br /><br />The <b>isosurface</b> command is complex, and while the order of parameters is flexible <i>within these groups</i>, parameters in different groups should not be swapped in terms of order within the command, or the result is unpredictable or will cause an error message to be displayed. While the id is optional, it is highly recommended. Specifically for the options ON, OFF, and DELETE, the id may include a wildcard character (asterisk). Thus, <b>isosurface pl* on</b> turns on all isosurface objects having an ID starting with "pl". <br /> <br /> Data for these surfaces can be from several sources (see below). In addition, atom-based data from a variety of sources can be mapped onto an isosurface. <br /><br />The isosurface itself represents the points in space where scalar values cross a specified "cutoff" value. Inside the surface, values are greater or equal to a specified positive cutoff or less than or equal to a specified negative cutoff. The default cutoff depends upon the type of object displayed. Note that positive and negative surfaces may be created separately, or, using the SIGN keyword, they can be generated together. <br /><br />Color mapping of one object onto another is a simple as listing both an object and a dataset within the same isosurface command. Several keywords affecting the mapping are allowed. The default color scheme uses a red-->orange-->yellow-->green-->blue rainbow, where red represents minimum values and blue represents maximum values, but several other schemes are available (see below). Mapped data can be expressed as a series of contour lines.','0|1|2','','')

newCmd('',' ','','*v+11.6;','The optional identifier allows you to refer back to this isosurface later for turning the surface on or off, deleting the surface, or changing display options. It must be either the first parameter or the second, just after DELETE. If the identifier is missing, behavior depends upon version. The ID keyword is optional but recommended, because then one can use any name for the ID; otherwise the name must not be one of the many Jmol command or keyword names. Leaving off the ID when creating an isosurface <i>replaces</i> the current isosurface with the new one.','TEXT','ID [object id]','')
newCmd('',' ','','*v+11.0;','[||<b>ADDHYDROGENS</b>|For a solvent or sasurface object, accounts for missing hydrogens on carbon atoms only just prior to generating the surface. A simple sp3 model is used, and for that reason this option is not recommended. Instead, use {#.set_pdbaddhydrogens~set pdbAddHydrogens} to load a PDB file and automatically add hydrogen atoms to ATOM and HETATM records. ||<b>IGNORE {{atom expression}}</b>|Specifies a set of atoms to completely ignore when generating a solvent-related surface (<b>SOLVENT</b> or <b>SASURFACE</b>). Typically these might be solvent molecules or atoms: <b>IGNORE {{solvent}}</b>. Note that SASURFACE and SOLVENT surfaces by default ignore solvent molecules; VDW and MOLECULAR do not. ||<b>IONIC radius</b>|Atom radius relative to the ionic radius. Options include: <br/> [|| x.x | specific absolute number of Angstroms for every atom || +/-x.x | offset greater(+) or less(-) than the ionic radius || nn% | percent of the ionic radius || +/-nn% | percent offset from the ionic radius   ||] ||<b>SELECT {{atom_exp}}</b>|Selects a specific subset of the atoms for this surface. Same as <b>select atom_exp; isosurface....</b> except the atoms are selected only within the isosurface command, not the overall script.  ||<b>SELECT {{atom_exp}} ONLY</b>|Selects a specific subset of the atoms for this surface and also applies <b>IGNORE {{not atom_exp}}</b>. (Jmol 12.2) ||<b>SET n</b>|If an isosurface is fragmented, this option allows visualization of only the set of triangles composing one specific surface fragment. ||<b>VDW radius</b>|Atom radius relative to the van der Waals radius. Options include: <br/> [|| x.x | specific absolute number of Angstroms for every atom || +/-x.x | offset greater(+) or less(-) than the van der Waals radius || nn% | percent of the van der Waals radius || +/-nn% | percent offset from the van der Waals radius   ||]  || <b>WITHIN x.xx {{atomExpression or point}}</b>| only creates a surface within the specified distance in Angstroms from the specified atoms or point. ||]','TEXT','[construction/mapping parameters] -- molecular/solvent surfaces','')
newCmd('',' ','','*v+11.0;','[||<b>IGNORE {{atom expression}}</b>|Specifies a set of atoms to completely ignore when generating a molecular orbital, thus showing only selected atomic contributions.||<b>SCALE x.xxx </b>|In the case of molecular orbitals, scales the volume around the orbital. It may be useful in cases where Jmol misjudges the full extent of an orbital, thus truncating it.||<b>SELECT {{atom expression}}</b>|Selects a specific subset of the atoms for this molecular orbital. (Essentially the opposite of IGNORE.)||<b>SIGN</b>|For molecular orbitals derived from cube files, indicates that the data have both positive and negative values and that they should both be displayed.||<b>SQUARED</b> | Data are to be squared prior to surface generation. ||]','TEXT','[construction/mapping parameters] -- molecular orbitals','')
newCmd('',' ','','*v+11.0;','[||<b>ANISOTROPY&nbsp;{{a&nbsp;b&nbsp;c}}</b>|Sets the anisotropic distortion of an object in the x, y, and z directions.||<b>CACHE</b> | Caches the isosurface in memory and replaces it with  its JVXL equivalent. The <b>CACHE</b> option along with isosurface creation or alone instructs Jmol to immediately create JVXL data for the specified surface and to load that data instead. The surface remains in memory and can be used again, even after a new file is loaded using cache://isosurface_<ID> where <ID> is the isosurface ID such as "isosurface1". The command ISOSURFACE CACHE alone will cache the current isosurface. If the cache is no longer needed, then RESET CACHE will release the memory used to hold the JVXL data for the isosurface. The result should be essentially equivalent to the original command. (It is recommended that the original be a relatively simple command, because not all nuances of an isosurface may be stored in the JVXL data.) Note that <b>this option is not compatible with saving the state as an SPT file.</b> Instead, use {#.write~WRITE PNGJ} or {#.write~WRITE JMOL}, in which case the cached isosurface will be saved in JVXL format within the PNGJ or JMOL zip directory. (Jmol 13.0) ||<b>CAP</b> | See <b>SLAB</b>.  The distinction between SLAB and CAP is that SLAB leaves the isosurface open at the boundaries, whereas CAP closes it off. ||<b>CENTER {{x y z}}</b>|Centers an atomic orbital, sphere, ellipsoid, or lobe at a specific point in molecular space. In the case of crystal structures, these may be {#.fractional coordinates~}.||<b>CENTER (atom_exp)</b>|Centers an atomic orbital, sphere, ellipsoid, or lobe on a certain atom or at the geometric center of this set of atoms.||<b>CENTER $object</b>|Centers an atomic orbital, sphere, ellipsoid, or lobe on a certain drawn object such as a point, line, or plane.||<b>ECCENTRICITY <br />{{cx cy cz f_ab}}</b>|Sets the eccentricity of a sphere, ellipsoid, atomic orbital, or lobe. The first three numbers define the "principal" axis; the fourth parameter defines the ratio of the other two perpendicular axes to this main axis.||<b>PHASE "type"</b>|Indicates that an orbital or other object is to be colored based on the position of the voxel in space. With an atomic orbital and no parameters, indicates regions of (+) and (-) oribital value with different colors. Valid types include "x", "y", "z", "xy", "xz", "yz", "z2", and "x2-y2".||<b>SCALE x.xxx </b>|In the case of objects for which eccentricity can apply (spheres, ellipsoids, atomic orbitals, lobes, and, starting with Jmol 13.0, volumetric data), scales the object (default 1.0).||<b>SLAB [plane definition]</b><br /><b>SLAB BOUNDBOX</b><br /><b>SLAB UNITCELL</b><br /><b>SLAB within x.y {{point}}</b>| [Jmol 12.0] clean slabbing of an isosurface based on a plane definition such as x=3 or {{1 0 0 -3}} or the currently defined unitcell or boundbox. A negative sign prior to a plane definition indicates "the opposite-facing plane". The distinction between SLAB and CAP is that SLAB leaves the isosurface open at the boundaries, whereas CAP closes it off. Smooth spherical slabbing based on distance x.y from a point is also possible. A negative distance implies "not within" and creates a circular hole at this position. (Jmol 12.2)||]','TEXT','[construction/mapping parameters] -- general shapes and volumetric (cube file) data','')
newCmd('',' ','','*v+11.8;','[||<b>ANGSTROMS</b>|For a cube file or user-defined function f(x,y), indicates that the volumetric definitions are in Angstroms instead of Bohr (default).||<b>BLOCKDATA</b>|Indicates that data for surfaces in multiple-surface CUBE files are in sequential blocks rather than the Gaussian standard of being interspersed, where all data values for a coordinate point are listed together.||<b> BOUNDBOX </b>| Specifies that the surface to be generated must be within the currently defined {#.boundbox~}. (Jmol 12.0)||<b> BOUNDBOX {{atomExpression or point}} {{atomExpression or point}}</b>| Specifies that the surface to be generated must be within a volume defined by the specified two diagonal end points. (Jmol 12.0)||<b> COLOR DENSITY</b>| Aallowing rendering of the actual grid of numbers (volume rendering) of the data rather than an actual isosurface. With <b>CUTOFF 0.0</b>, this setting delivers the entire set of data points. It is recommended that the BOUNDBOX parameter be used with a relatively small boundbox setting or the WITHIN parameter is used in order to not have an out-of-memory condition resulting from this option. For example: <b>boundbox scale 1.2 {{tyr}};isosurface color density cutoff 1.6 boundbox  "3hyd_map.ccp4.gz" mesh nofill</b> If a range of values is desired, use <b>CUTOFF [min, max]</b> (Jmol 12.2) to indicate the minimum and the maximum values to be included; for example: <b>isosurface color density cutoff [-5,10] within 4.0 {{*}} "apbs.dx"</b>. || <b> CUTOFF x.xxx</b>|Sets the cutoff value defining an isosurface. Typically, smaller values correspond to a larger object. Cutoffs can be either positive or negative. In the case of a molecular orbital, a positive number indicates to use both positive and negative cutoffs. Adding an explicit "+" sign before the number indicates that only the positive part of the surface is desired. (See also SIGMA, below.)|| <b> CUTOFF [min,max]</b>|see COLOR DENSITY, above||<b>DEBUG</b>|Produces voluminous detail for program debugging.||<b>DOWNSAMPLE n<b>|"Downsampling" is the process by which an image or other data set is selectively sampled, producing a result that is grainier and has less resolution than the original data set. This option for the <b>isosurface</b> command indicates that only a portion of the data for a file should be used for the surface, creating an isosurface with lower resolution but with the full range of the original data. <b>n</b> is an integer -- <b>downSample 2</b> uses only every other data point; <b>downSample 3</b> takes only every third data point, etc. Supported for cube-based data sets only.  ||<b>FIXED/MODELBASED</b>|Sets whether the surface generated is to be associated with the fixed window -- and thus appear with all frames/models -- or is to be associated with the currently displayed model (the default).||<b>FULLPLANE</b>| for PLANE objects, indicates that color mapping should be extended to complete the plane. (Jmol 12.0)|| <b>GRIDPOINTS</b>|Adds the specific gridpoints used by the "marching cubes" algorithm for the calculation of the isosurface. Primarily for discussion and debugging of the isosurface algorithm.|| <b>INSIDEOUT</b>|For certain datasets it sometimes happens that the surface rendered appears inside-out (dark on the outside and bright on the inside). If this is the case, add <b>INSIDEOUT</b> to the isosurface command prior to specification of the file to load. Jmol will reverse the sense of what is inside and what is outside the surface. This flag only affects rendering in Jmol, not export to PovRay. ||<b>MODEL n</b>|Sets the identity of the model with which this isosurface is to be associated. (Defaults to the currently displayed model.)||<b>RESOLUTION x.x</b>|Sets the resolution of a constructed isosurface three-dimensional grid of "voxels", in points per Angstrom. Does not apply to CUBE or JVXL files, which have a resolution defined by internal file parameters.||<b>SIGN c1 c2</b>|Indicates that cube data have both positive and negative values, and that they should both be displayed using the value given for CUTOFF and its negative. The two colors to use may be given optionally.||<b>SIGMA x.xx</b> | Sets the cutoff based on the root mean square deviation of the data. (Currently supported only for the MRC/CCP4 file format; Jmol 12.0.) || <b>SQUARED</b> | Data are to be squared prior to surface generation.  ||<b>WITHIN x.xx {{atomExpression or point}}</b>| only creates the portion of the surface within the specified distance in Angstroms of the specified atoms or point. (Jmol 12.0)||<b>WITHIN x.xx {{atomExpression or point}}</b>| only creates a surface within the specified distance in Angstroms from the specified atoms or point. ||<b>WITHIN -x.xx {{atomExpression or point}}</b>| a negative value for the WITHIN distance indicates <b>not</b> within x.xx Angstroms of the specified atoms or point. (Jmol 12.2)||]','TEXT','[construction/mapping parameters] -- general file loading','')
newCmd('','11.9.7','','*v+11.0;','Jmol can color isosurfaces in a wide variety of ways, from simple single colors to mapped contours based on a second data set. See also the {#.colorother~color ISOSURFACE} command for additional options and changing the color of an isosurface after creation. [||<b>COLOR &lt;color&gt;</b>|Colors an isosurface the specified color name or value, such as <b>[xFF0000]</b>. ||<b>COLOR "c1 c2 c3..."</b>|Where c1, c2, etc. are color names or values, such as <b>red</b> or <b>[xFF0000]</b> within double quotes, this option allows specification of descrete colors to be used for contour mapping. ||<b>COLOR c1 TO c2 n| (also <b>color isosurface ......</b>) colors an isosurface with a range of <b>n</b> colors from color c1 to color c2. For example, <b>color isosurface red to white 30</b>. ||<b>COLOR RANGE <br />x.xxx y.yyy</b>|(or <i>color absolute</i>) Indicates to color the specified range of value from one end to the other end of the color scheme. If numbers are not included or COLOR RANGE ALL is specified, then the minimum and maximum data values in the file are used.||<b>COLORSCHEME "xxx" </b>|Sets the color scheme to one of "roygb" (default rainbow), "bw" (black/white, Jmol 12.0), "bwr" (blue-white-red), "rwb" (red-white-blue), "wb" (white/black, Jmol 12.0), "low" (red-green), or "high" (green-blue). An optional parameter TRANSLUCENT prior to the color scheme name creates a gradient of translucency from transparent to opaque across the color scheme [Jmol 12.0]. An additional scheme "sets" colors the isosurface different colors for different surface fragments (for example, internal cavities).||<b>CONTOUR n</b>|Specifies to display the object as a set of contour lines. Then number of lines is optional; 9 contour lines are shown by default. Using the CONTOUR keyword sets the default display to be CONTOURLINES NOFILL.||<b>CONTOUR -n</b>|With a negative number, specifies for a plane or f(x,y) object the specific single contour to depict.||<b>CONTOUR DISCRETE</b><br /> [a,b,c,d,e,...]|Sets the contour levels to discrete values. In addition, see the COLOR option, above, for the discrete coloring option. [Jmol 12.0]||<b>CONTOUR INCREMENT <br />{{from,to,step}}</b> |Sets the contour values starting with the <b>from</b> value to the <b>to</b> value in increments of <b>step</b>. In addition, see the COLOR option, above, for the discrete coloring option. [Jmol 12.0]||<b>REVERSECOLOR</b>|For colorschemes in particular, the REVERSECOLOR option switches the direction of the color scheme. This parameter should be given just prior to the COLORSCHEME parameter.||<b>SCALE3D x.x</b>|generates a 3D plot of the desired scale from a mapped plane. It can be introduced either with the original definition of the isosurface or later, after the plane has been created and displayed. Negative numbers invert the graph (forming valleys instead of mountains); 0 removes the 3D scaling. Note that this rendering can be combined with CONTOUR to form a 3D "topo map". (Jmol 12.0)||]','TEXT','[color and contour options]','')
newCmd('',' ','','*v+11.0;','Several isosurface options relate specifically to molecular or solvent-related surfaces.  <br /><br />[||<b>CAVITY<br />cr er</b>| Renders a depiction of a molecular cavity. The optional parameters cr and er determine the overall properties of the cavity. cr (cavity radius, default 1.2) sets the radius in Angstroms of the "probe" molecule that would fit into the cavity. er (envelope radius, default 10) sets the radius of the probe used to define the outer limits of the molecule. Smaller numbers for the cavity radius lead to more detailed cavities; smaller numbers for the envelope radius lead to cavities that are more internal and extend less toward the outer edge of the molecule. Qualifier <b>INTERIOR CAVITY</b> creates isosurfaces only for cavities that do not extend to the outer surface of the molecule. Qualifier <b>POCKET CAVITY</b> creates isosurfaces only for pockets that extend to the outer surface of the model, showing them more as troughs or open "pockets". Qualifiers <b>MINSET n</b> and <b>MAXSET n</b> prior to CAVITY set limits to the number of faces involved. For example, <b>isosurface MINSET 1000 POCKET CAVITY</b> will only create isosurface cavities that involve at least 1000 triangular faces. ||<b>MEP</b>|Depicts the molecular electrostatic potential, as calculated by SUM(q_i/r_i), where q_i is the partial charge on atom i (as found in the loaded model file) and r_i is the distance from atom i to a particular point in space. The molecular electrostatic potential is not typically displayed itself. Rather, it is usually mapped onto a molecular surface. For example: <b>isosurface resolution 6 SOLVENT map MEP</b> produces a smooth surface at the van der Waals distance around the molecule colored by the molecular electrostatic potential.||<b>MOLECULAR</b>|Same as <b>SOLVENT 1.4</b>, but solvent moleules are not ignored.||<b >SASURFACE radius</b >|Depicts the "solvent-accessible" surface based on the currently selected atom set. This surface is described by the <i >center</i > of the solvent probe as it rolls along the surface. It is larger than the "molecular" surface. The radius is optional. Solvent molecules are ignored by default. Starting with Jmol 12.0, if either the <b>VDW</b> or the <b>IONIC</b> keywords are present, <i>sasurface 0</i> is assumed, but solvent is not ignored.||<b >SOLVENT radius</b >|Depicts the "solvent-excluded" or "molecular" surface around the currently selected atoms (or the entire model if no atoms are selected). If only a subset of the atoms is selected, then only the corresponding subset of the molecular surface is depicted. This surface is defined as the surface formed by rolling a spherical solvent "probe" around the molecule at the distance of the van der Waals radii of the atoms. The specification of the radius of this probe is optional; its default setting is determined by the <b >set radius</b > command (Jmol default 1.2). Solvent molecules are ignored by default.||]','TEXT','[surface object] -- molecular/solvent surfaces','')
newCmd('',' ','','*v+11.0;','Both atomic orbitals and molecular orbitals can be displayed in Jmol. Note that molecular orbitals in particular can be generated more effectively using the {#.mo~} command rather than the <b>isosurface</b> command. Alternatively, the "LCAO cartoon" option creates the sort of dumbbell-shaped cartoonish orbitals seen in textbooks in discussion of pi bonding and hybridization. <br /><br />[|| <b >ATOMICORBITAL <br />n l m Zeff</b >|The Schroedinger solution to the hydrogen atom wavefunction. The three quantum numbers n, l, and m, must be such that abs(m) &lt;= l &lt; n. For solutions with imaginary roots, the two m values simply designate the two real linear combinations of these imaginary solutions. The optional effective nuclear charge, Zeff, determines the overall size of the orbital (default is 6, carbon). Add the keyword PHASE for a two-color orbital, which can be colored using the SIGN keyword followed by two color names or values.||<b >ATOMICORBITAL <br />n l m Zeff POINTS nPoints radius seed</b >|Creates a "pointilist" probability-based visualization of an atomic orbital using the specified number of points and optional radius extent and random seed. The radius parameter, if present, must be a decimal number. ||<b >LCAOCARTOON <br />"type" (atom_exp)</b >|Draws a lobe or p orbital (two lobes) centered at the FIRST atom of the specified expression. (Typically this would be an expression for a single, specific atom, such as <b >atomno=3</b >). See the {#.lcaoCartoon~lcaoCartoon} command for a discussion of the possible types of LCAO cartoons (as well as a simpler way to create them).||<b >LOBE <br />{{cx cy cz f_ab}}</b >|Draws a single tear-drop shaped object (an xy-compressed center lobe of a dz2 orbital) anywhere at any size in any direction with any amount of distortion. The first three numbers define the axis of the lobe; the fourth parameter defines its eccentricity  -- the ratio of the other two perpendicular axes to this main axis.||<b >MO n</b >|Denotes the n-th molecular orbital described in the file that is currently loaded. Adjusting the CUTOFF to suit the situation is recommended. Molecular orbitals are automatically bicolor; color them one specific color using COLOR and just one color name or value. RESOLUTION can be used to good effect to increase or decrease the precision of the rendering. <b >Note that only the atom-based orbitals for the currently selected atoms are utilized</b >. (Although, if no atoms are selected, all atomic orbitals are used in the calculation.) Thus, one can selectively see how atomic orbitals from each atom in the molecule are contributing to any given molecular orbital.||<b>MO HOMO/LUMO +/-n</b>|Selects for display a molecular orbital based on energy-proximity to the highest-occupied or lowest-unoccupied molecular orbital. For example, <b>isosurface MO HOMO</b> or <b>isosurface MO LUMO +3</b>.|| <b>MO ... POINTS nPoints seed</b>| Produces a "pointilist" probability-based visualization of a molecular orbital using the specified number of points and an optional random seed. ||]','TEXT','[surface object] -- atomic and molecular orbitals','')
newCmd('',' ','','*v+11.0;','There are several general shapes that can be created as "isosurfaces". These include: <br /><br />[||<b >ELLIPSOID <br />{{cx cy cz f_ab}}</b >|Draws an ellipsoid having a single unique axis. The first three numbers define the "principal" axis; the fourth parameter defines the eccentricity of the ellipsoid -- the ratio of the other two perpendicular axes to this main axis.||<b>HKL {{h k l}}</b>|Creates a plane through a crystal based on the Miller indices <i>hkl</i>. Adding <b>map molecular</b> creates a slice through the crystal highlighting atomic positions.||<b>PLANE</b>|Indicates that what is desired is not really an isosurface but rather a planar slice through the data set. Using COLOR RANGE, the range of mapped values can be changed. The range -0.008 0.008 is recommended for molecular orbitals. Planes, like other surface objects, can be mapped or left unmapped and just colored. Planes are designated using one of the methods for {#.plane expressions~}. If a BOUNDBOX keyword is given prior to PLANE (see above), then the plane will be created within that box.||<b >SPHERE radius</b >|Draws a sphere of the given radius in Angstroms.||]','TEXT','[surface object] -- general shapes','')
newCmd('',' ','','*v+11.8;','Isosurfaces can be created in Jmol using external file-based "volume" or "polygon" data. Note that any of the formats that are volume data (APBS, CUBE, Chem3D, DSN6/OMAP, MRC/CCP4, XPLOR/CNS) can all be used either to generate the surface itself or (after the MAP keyword) to color a surface.<br /><br /><b>FILE "filename" n</b> can be used in either case to depict the n-th isosurface from the specified file. Empty quotes indicate that the loaded structure file already has surface data present (CUBE files, for example, contain coordinates), and that that data should be used for construction of the surface. Thus, for example, <b >load test.cube.gz;isosurface FILE ""</b > will load the coordinates from the GZIPped cube file, then display its first isosurface. A default directory can be set using {#.set (misc)~set defaultDirectory}, and for applets a proxy server can be set for non-host file loading using {#.set (misc)~set appletProxy}. The keyword FILE is not required. The optional number "n" specifies which volume or surface should be used in the case of a CUBE file with multiple orbitals with multiple surfaces. All files read by Jmol may be gzip-compressed.','TEXT','[surface object] -- file-derived isosurfaces','')
newCmd('',' ','','*v+11.8;','Current volume file formats supported include:<br />[|| APBS | {http://apbs.sourceforge.net~APBS DX} volume data files ||  CASTEP | {http://www.castep.org/~CASTEP} density files||  CUBE|{http://www.gaussian.com~Gaussian} {http://www.nersc.gov/nusers/resources/software/apps/chemistry/gaussian/g98/00000430.htm~cube} format volume data files. Units of Bohr are assumed unless the keyword ANGSTROMS precedes the FILE keyword. || Chem3D | {http://www.adeptscience.co.uk/download/dldcat/24/0/All/Chem3D.html~chem3d} 3dxml structure files may contain volume data ||  DSN6/OMAP | binary files created by the {http://eds.bmc.uu.se/eds/~Uppsala Electron Density Server} (Jmol 12.0)  These files can be read directly from the server into Jmol (application and signed applet) using an "=" sign and the PDB ID: <b>isosurface =XXXX</b>. Supporting this are the global variables {#.set_edsUrlCutoff~set edsUrlCutoff} and {#.set_edsUrlFormat~edsUrlFormat}, which set the method of getting cutoff and file from electron density server. The BOUNDBOX isosurface option is automatically applied in this case. ||GRASP/DELPHI| {http://www.msg.ucsf.edu/local/programs/grasp/html/Appendix%20A.html~GRASP} grid files.||Jaguar PLT | {https://www.schrodinger.com/products/14/7/~Jaguar} plt orbital files || NCI|Jmol 12.2 adds support for displaying results of NCIPLOT "noncovalent interactions" calculations.    See <i>"Revealing Noncovalent Interactions", Erin R. Johnson, Shahar Keinan, Paula Mori-Sanchez, Julia Contreras-Garcia, Aron J. Cohen, and Weitao Yang, J. Am. Chem. Soc., 2010, 132, 6498-6506</i> and <i>"NCIPLOT: A Program for Plotting Noncovalent Interaction Regions" Julia Contreras-Garcia, Erin R. Johnson, Shahar Keinan, Robin Chaudret, Jean-Philip Piquemal, David N. Beratan, and Weitao Yang,   J. of Chemical Theory and Computation, 2011, 7, 625-632</i>. For consistency with the mapping colors of these articles, the specific color scheme shown below is recommended. Without reading any files, the <b>NCI</b> keyword specifies to carry out an NCI <i>promolecular</i> calculation. However, if the output files from {http://www.chem.duke.edu/&tilde;yang/Software/softwareNCI.html~NCIPLOT} are available, they can be used as well, for example: <b>isosurface parameters [0 0 0 0 0 0.01] NCI dens.cube</b>. Note that only the density file is required; Jmol can generate the reduced density gradient data from that density file alone. Any CUBEGEN or other format density file that Jmol can read can be used here. (The 0.01 parameter is required only when reading the density file created by NCIPLOT, because that file is scaled up by a factor of 100 relative to standard density files.  If using both NCIPLOT output files, then the command is simply: <b>isosurface parameters [0.5] grad.cube MAP dens.cube fullylit</b>.Parameters are [cutoff, p1, p2, p3, p4, p5], described below. Zero for any parameter specifies to use the default value. [||p1 | -2 to 2|0(all, default), 1(intramolecular), 2(intermolecular), -1(intramolecular NCIPLOT grad/dens filtering), -2(intermolecular NCIPLOT grad/dens filtering).||p2 |rhoMin   |(min rho cutoff to remove very low density points)||p3 |rhoPlot  |(cutoff used to remove covalent density, default 0.07 for promolecular, 0.05 for SCF)||p4 |rhoParam |(fraction of total rho that defines intramolecular, default 0.95)||p5 |dataScaling |(default 1, but set to 0.01 to read back in NCIPLOT -dens.cube file)||] For more information, see {misc/iso-nci.txt~}.    ||  MRC/CCP4 |  binary {http://ami.scripps.edu/software/mrctools/mrc_specification.php~AMI MAP} files have the advantage that they contain information about the root mean square deviation for the data set, allowing use of the SIGMA keyword (Jmol 12.0). In some cases the keyword <b>MRC</b> just precede the FILE keyword in order to force MRC format reading because the file does not have the expected identifying characters M A P in the proper location.||  NFF | {http://paulbourke.net/dataformats/nff/nff1.html~neutral file format}  for electron microscopy data exported from IMOD using the ISOSURFACE command || UBHD || {http://web.archive.org/web/20100701074459/http://adrik.bchs.uh.edu/uhbd.html~University of Houston Brownian Dynamics} files.|| XPLOR/CNS |  {http://cci.lbl.gov/@TILDErwgk/shortcuts/htdocs/current/python/iotbx.xplor.map.html~XPLOR MAP} files ||]','TEXT','[surface object] -- volume file-derived isosurfaces','')
newCmd('',' ','','*v+11.8;','Current surface file formats supported include:<br />[|| EFVET | {http://ef-site.hgc.jp/eF-site/about.html~<i>e</i>F-site EFVET} format surface files. Indicating a number from 0-5 after the file name specifies which type of data to map onto the surface: [||0 | color indicated in file||1 | electrostatic_potential||2 | hydrophobicity||3 | temperature_factor||4 | minimum_curvature||5 | maximum_curvature ||] ||  JVXL | {misc/JVXL-format.pdf~Jmol Voxel} format files are highly compressed surface files created by Jmol using the {#.write~} command. The JVXL format allows for saving the data for a selection of a specific+I452 surface cutoff from a volume file in a way that can be delivered very efficiently over the web and loaded very quickly into Jmol. ||  KINEMAGE | Jmol 12.2 reads contact-point Kinemage files created by {http://molprobity.biochem.duke.edu~MolProbity} See {misc/kinemage.txt~} for details.|| MSMS | Jmol can read an isosurface from a set of {http://mgltools.scripps.edu/packages/MSMS~MSMS} output files. Both the .vert and the .face files must be in the same directory, or only dots will be generated. The keyword MSMS must precede the FILE keyword -- <b>MSMS FILE "xxx.vert"</b> -- because these files do not contain identifying information.||  OBJ | {http://autodesk.com~AutoDesk Wavefront Advanced Visualizer} {http://www.eg-models.de/formats/Format_Obj.html~OBJ} surface data files. (See also {http://www.fileformat.info/format/wavefrontobj/~Wavefront Object format.} In this case the keyword OBJ must precede the FILE keyword. Jmol uses the <b>v</b>ertex and <b>f</b>ace records of this file in order to create a set of polygons that are colored using the name of the <b>g</b>roup record, which is assumed to have the hexidecimal format <b>g kRRGGBB</b>. Note that Jmol does not read material (.mtl) files and so instead relies on this simpler method of assigning colors. Polygons are limited to triangles and quadrilaterals. An optional number after the file name --<b>OBJ FILE "filename" n</b> --  specifies a group from the OBJ file. <b>isosurface OBJ "sample.obj" 3</b>, for instance, reads only the third group of faces. ||  PMESH | Jmol can read both ASCII and binary pmesh surface formats. A "pmesh" is a surface data set consisting of a set of vertices and a set of polygons defining the "facets" of the surface. Polygons are limited to triangles and quadrilaterals. See also <b>PMESH INLINE</b>, below. File formats readable by Jmol include:[||<b>numeric pmesh</b>|This relatively simple format can be found in {misc/10x10pmesh.txt~10x10pmesh.txt}. Jmol reads this file in free format -- values simply need to be separated by some sort of white space.<pre>100\n3.0000 3.0000 1.0000\n2.3333 3.0000 1.0000\n...(98 more like this)\n81\n5\n0\n10\n11\n1\n0\n...(80 more sets like this)\n</pre><ul> <li>The first line defines the number of grid points defining the surface (integer, n)</li><li>The next n lines define the Cartesian coordinates of each of the grid points (n lines of x, y, z floating point data points)</li><li>The next line specifies the number of polygons, m, to be drawn (81 in this case).</li><li>The next m sets of numbers, one number per line, define the polygons. In each set, the first number, p, specifies the number of points in each set. Currently this number must be either 4 (for triangles) or 5 (for quadrilaterals). The next  p numbers specify indexes into the list of data points (starting with 0). The first and last of these numbers must be identical in order to "close" the polygon.</li></ul>||<b>binary pmesh</b>|A more compact binary pmesh format is described in {misc/pmesh.bin.txt~pmesh.bin.txt}. In this case the PMESH keyword is optional, as Jmol can discern the data type from the first four bytes of the file contents. The format has the following specification:<pre>\n 4 bytes: P M \\1 \\0 \n 4 bytes: (int) 1 (test for bigEndian)\n 4 bytes: (int) vertexCount\n 4 bytes: (int) polygonCount\n 64 bytes: reserved\n ---then for each vertex:\n 12 bytes: (float) x,(float) y,(float) z\n ---then for each polyhedron, \n    the number of vertices (from 1 to 4) \n    followed by the index of the vertex \n    in the vertex list, starting with 0:\n    [(int)nVertices,(int)v1,(int)v2,...,(int)vn]\n</pre>||] ||] ','TEXT','[surface object] -- surface data-derived isosurfaces','')
newCmd('',' ','','*v+11.8;','<img style="clear:right" src="img/fxyz.png" alt="" />Several additional sources of surfaces besides external files are available. For <b>functionXY</b> and <b>functionXYZ</b>, the extent and resolution of the grid can be set using the keywords ORIGIN, STEPS, and POINTS. For example, <b>isosurface origin {{-5 -5 -5}} steps {{0.2 0.2 0.2}} points {{50 50 50}} functionxy = "x * y * cos(x * 100)"</b> or <b>isosurface cutoff 2.0 origin {{-5 -5 -5}} steps {{0.2 0.2 0.2}} points {{50 50 50}} functionxyz = "x * y * cos(z*100)" map functionxy = "x * y"</b> (shown here on the right. [||<b>FUNCTIONXY = "..."</b>| graphs a function of x and y with a default range in x and y from -10 to 10 every 0.25 Angstroms. For example, <b>isosurface functionXY = "x * y"||<b >FUNCTIONXY <br />"functionName" <br />{{originXYZ}} <br />{{ni xi yi zi}}<br /> {{nj xj yj zj}}<br /> {{nk xk yk zk}} </b >|The FUNCTIONXY keyword specifies that the Z-value at a given X,Y coordinate will be provided via a JavaScript function in the case of the applet or via a JmolStatusListener call in an application or, if the function name begins with <b>file:</b>, the numbers will be read from that file (relative to the current directory). The parameters mirror the parameters in the header of a CUBE file. Units of ANGSTROMS are assumed. Thus, we require an origin of the voxel space and for each nominal direction x, y, and z, the number of grid points and the direction of the unit vector along that edge. These four quantities must be in braces. In the case of the Jmol applet, there are three reading options. [||ni>0 and nj>0|functionName(app, ix, iy)| one point is returned for each call to the function.||ni&lt;0 and nj>0 | functionName(app, ni, nj)|an entire set of z values separated by white space will be returned as a string. Data are read from low to high coordinates, with X changing in the outer loop (slowly) and Y changing in the inner loop: a11, a12, a13,...,a21, a22, a23,..., etc. ||ni&lt;0 and nj&lt;0|functionName(app, ni, nj, Fxy)| the Fxy[-ni][-nj] array will be filled with z values in the case of the applet or functionXY(functionName, ni, nj, Fxy) will be called in the case of the application. ||] <br />If the functionName starts with the string "file:", then Z value data will be loaded from a file instead of a JavaScript function in the order described above for reading data from a returned string.||<b>FUNCTIONXYZ = "..."</b>| Graphs an isosurface through a block of data defaulting to -10 to 10 Angstroms range in x, y, and z with step every 0.25 Angstroms. For example: FUNCTIONXYZ = "x * x + y + z" </b>. (Jmol 12.2)|| <b >FUNCTIONXYZ <br />"functionName" <br />{{originXYZ}} <br />{{ni xi yi zi}}<br /> {{nj xj yj zj}}<br /> {{nk xk yk zk}} </b >|  Similar to FUNCTIONXY, except that in this case the X,Y,Z cube data will be provided in the order (for x = 1 to ni)(for y = 1 to nj)(for z = 1 to nk). The first two options listed for FUNCTIONXY are not available, and the signs of ni and nj are ignored. The data structure Fxyz[ni][nj][nk] must be filled with data by the function call functionName(app, ni, nj, nk, Fxyz) in the case of the applet or functionXYZ(functionName, ni, nj, nz, Fxyz) in the case of the application.||<b>INLINE @varName</b>| The isosurface data may be in a variable already. For example: <b>x = load("mydata.dat"); isosurface INLINE @x</b> first loads the data into the variable x, then displays the isosurface from that data, possibly giving the opportunity to peek at the data first. It is advisable to reset the variable after use to improve performance, however note that the state will only be preserved if the value of the variable is left unchanged. (Jmol 12.0) ||<b>PMESH INLINE "pmesh-data"</b>| The INLINE keyword can be used with PMESH to directly create a small isosurface. Numeric pmesh data may be specified "in-line" without reference to a separate file. This is particularly useful for pmesh objects with few vertices. Note that the {#.draw~} command also can be used for this purpose.||]','TEXT','[surface object] -- additional surface sources','')
newCmd('',' ','','*v+12.0;','If any parameters spefically relate to the mapping of the surface, not the generation of it, then they can come after the specification of the surface object. Keywords such as COLOR RANGE, CONTOUR, DEBUG, FIXED, FULLPLANE, MODELBASED, MAP, REVERSECOLOR, SCALE3D, and SELECT (when it relates to the color mapping) fall into this category. ','TEXT','[additional mapping-only parameters]','')
newCmd('',' ','','*v+11.0;','Except for SPHERE, ELLIPSOID, LOBE, and LCAOCARTOON, which have no CUBE-file equivalent, all the other surface types (including FUNCTIONXY) can be used as CUBE-type data sets in order to color map another surface, because they all involve the intermediate generation of voxel data within Jmol. Used with PLANE as a surface object, a slice through a set of data can be color-contoured.The keyword MAP is optional, recommended for readability. Starting with Jmol 12.2, the {#setisosurfacekey~set isosurfacekey} option displays a vertical color key that can be hovered over with the mouse to display the values associated with mapped colors and contours. Starting with Jmol 12.2, existing isosurfaces can be mapped or remapped. To do this, simply start the <b>isosurface</b> command with the MAP keyword (with optional ID). The keyword set MAP SQUARED indicates that the values should be squared prior to mapping . The Jmol parameter {#.set(misc)~isosurfacePropertySmoothing} (default TRUE) determines whether the property is smoothed out over the isourface or assigned specifically to the nearest atom. <br /><br />Atom based-properties can be mapped onto a surface using one of the following options.  Starting with Jmol 12.0, if <b>MEP</b> or <b>MEP functionType</b> is followed by <b>PROPERTY xxx</b> or <b>property_xxx</b> or <b>variable...</b>, that data will be used instead of partialCharge data.  <br />[||<b>MEP</b>|molecular electrostatic potential, using partial charge data within the file or assigned to atoms using the <b>{{...}}.partialCharge = ...</b> or {#.datasettingatomproperties~data} command syntax. A standard Coulomb function is used (1/d). ||<b>MEP functionType</b>|(Jmol 12.0) allows setting the function used for the mapping as for {http://www.symyx.com/support/developer/chime/developer_tools/chimerasmol.jsp#set_mep_mlp_function~Chime}, where functionType is the number 0, 1, 2, or 3: [|| 0|1/d|Coulomb\'s law distance function (same as rasmol potential distance function)||1|e^(-d/2)|Gaillard, P., Carrupt, P.A., Testa, B. and Boudon, A., J.Comput.Aided Mol.Des. 8, 83-96 (1994)||2|1/(1+d)|Audry, E.; Dubost, J. P.; Colleter, J. C.; Dallet, P. A new approach to structure-activity relations: the "molecular lipophilicity potential". Eur. J. Med. Chem. 1986, 21, 71-72||3|e^(-d)|Fauchere, J. L.; Quarendon, P.; Kaetterer, L. Estimating and representing hydrophobicity potential. J. Mol. Graphics 1988, 6, 203-206.||] These additional functions thus allow Jmol to use <b>isosurface...MAP MEP</b> to visualize molecular lipophilic potential (MLP) as well. ||<b>PROPERTY xxx</b> | where xxx is an atom-based property value such as partialCharge or temperature or vanderwaals. Jmol 13.2 adds COLOR as an option to apply atom-based color to an isosurface, as commonly done in PyMOL. || <b>property_xxxx</b> | The linking underscore signifies that the property was provided separately using the DATA command and is not model-file based. A previous SELECT ...; DATA "property_xxxx"....end "property_xxxx" or, if the data are from a variable, SELECT...; DATA "property_xxxx @x", must have already been issued. Note that when data is created in this way, only the selected atoms are assigned data values. Atoms thus selected need not be contiguous, but the data will be read from the variable or DATA command line contiguously, disregarding spaces, tabs, line ends, and any string that would not evaluate to a number. This allows for targeting just a specific set of atoms for an isosurface and associated data. Starting with Jmol 12.2, after the property name the keyword <b>WITHIN</b> and a distance in Angstroms can be added to speed the calculation. The default for <b>set isosurfacePropertySmoothing FALSE</b> is 5.0 Angstroms. ||<b>VARIABLE x</b>| The property is in a varible named "x", possibly from a command such as <b>x = load("mydata.txt")</b>. In this case, the variable must contain a value for every atom in the model, even if only a subset of the atoms is being used for the surface. ||] ','TEXT','MAP [color mapping dataset]','')
newCmd('',' ','','*v+11.0;','Display options are indicated in the following table. These may be given at the end of the definition of the surface or in a later command having just these keywords and the identifier (or "ALL") of the desired isosurface.  <br />[||<b>CONTOURLINES/NOCONTOURLINES</b>| turns on and off contour lines generated using the CONTOUR option (see bove). ||<b>DOTS/NODOTS</b>| Display dots at each mesh vertex point.  ||<b>FILL/NOFILL</b>| Display the isosurface as a smoothly filled surface.||<b>FRONTONLY/NOTFRONTONLY</b>| Display only the front half of the surface. This can be useful when the options <b>mesh nofill</b> are used. ||<b>FRONTLIT /BACKLIT /FULLYLIT</b>| In some cases the isosurface may appear flat. Using BACKLIT will fix this; FULLYLIT makes exterior and interior surfaces bright; FRONTLIT is the default setting. ||<b>MESH/NOMESH</b>| Display a mesh of lines intersecting at the vertexes used to construct the isosurface. ||<b>OFFSET {{x y z}}</b>| Offset the isosurface by the specified absolute amount. (Jmol 12.0)||<b>ROTATE quaternion/NONE|</b> Rotate the isosurface by a given amount around a given axis. Rotations are about {{0 0 0}} and are cumulative; ROTATE NONE returns the isosurface to its original orientation. For example: <b>isosurface s1 ROTATE @{{quaternion({{1 0 0}}, 90)}}</b> (Jmol 12.2) ||<b>ON/OFF</b>| Turn the isosurface ON or OFF, but do not delete it. ||<b>OPAQUE/TRANSLUCENT n</b>| Display the isosurface as an opaque or translucent object. Several translucent options are available; see the {#.color~} command. ||<b>TRIANGLES/NOTRIANGLES</b>| Display separated triangles (primarily for debugging purposes).||]','TEXT','[display options] ','')
newCmd('',' ','','*v+11.8;','Calculates the area of the current isosurface and stores that value in the <b>isosurfaceArea</b> variable. The AREA parameter may also accompany any other parameters in the construction of an isosurface. The result is an array unless the optional specific surface set has been specified., in which case it is a decimal number. Sets are ordered from largest to smallest. SET 0 gives the sum of all areas. SET -1 returns the calculation to report the full array.','*21','AREA','SET','(integer)','')

newCmd('','12.1.48','','*v+12.2;','Connects an isosurface to a specific atom so that if that atom is moved or hidden, the isosurface moves or is hidden with it. Use <b>isosurface connect {{none}}</b> to disconnect an isosurface from its atom. ','*21','CONNECT','{{atom expression}}')
newCmd('',' ','','*v+11.0;','Deletes all isosurfaces. ','*31','DELETE','')
newCmd('','12.0.RC9','','*v+12.0;','Jmol 12.0 adds the capability to duplicate isosurface areas based on the unit cell lattice. This is a rendering option, so it can be applied any time after an isosurface is created. It is best done with packed unit cells and slabbed isosurfaces. For example: <b>load quartz.cif {{1 1 1}}; isosurface slab unitcell vdw; isosurface lattice {{3 3 3}}</b> creates an isosurface that spans nine unit cells. The LATTICE keyword may also appear after the MAP keyword, indicating that the mapped data is periodic and should be mapped for multiple unit cells.','*32','LATTICE','{{a b c}}')
newCmd('','11.1.28','','*v+11.2;','Lists all isosurfaces','*39','LIST','')
newCmd('','','','*v+11.8;','When an isosurface consists of more than one set of surfaces (perhaps from <b>isosurface internal cavity</b>), specifying a SET displays just that single subset of the overall isosurface. SET 0 returns the display to all sets. Sets are ordered from largest to smallest.','*410','SET','(integer)')
newCmd('','12.1.50','','*v+12.2;','Isosurface <b>SLAB</b> provides an extensive mechanism allowing for the trimming of an isosurface using a wide variety of methods. Isosurfaces can be "slabbed" based on planes, spheres, bounding boxes, unit cells, proximity to a set of atoms, a range of mapped values, or a dynamic z-plane relative to the user (as in Rasmol slabbing of models, but with isosurfaces instead). The section removed can be removed completely or, using <b>isosurface SLAB TRANSLUCENT</b>, left as a translucent "ghost" surface. An isosurface may be slabbed any number of times in any number of ways. The slabbing is reversible using <b>isosurface SLAB NONE</b> for general slabbing and <b>isosurface SLAB OFF</b> for dynamic z-plane slabbing. Options include: [|| {#.[plane expression]~(plane expression)} | a plane. Note that plane definitions starting with "-" switch sides of the plane ||BOUNDBOX | within the currently defined {#.boundbox~bounding box} ||UNITCELL | within the currently defined {#.unitcell~unit cell} ||WITHIN x.y {{atom expression}} | within the given distance of the specified atom set ||WITHIN x.y (point) | within the given distance of the specified point (a spherical slab) ||WITHIN RANGE x.x y.y | within the mapped value range x.x and y.y.  ||]','*411','SLAB','(parameters)')


newCmd('',' ','','*v+11.8;','Calculates the volume of the current isosurface and stores that value in the <b>isosurfaceVolume</b> variable. The VOLUME parameter may also accompany any other parameters in the construction of an isosurface. The result is an array unless the optional specific surface set has been specified., in which case it is a decimal number. Sets are ordered from largest to smallest. SET 0 gives the sum of all volumes. SET -1 returns the calculation to report the full array.','*42','VOLUME','SET','(integer)','')
newCmd('.javascript','11.1.28','','*v+11.2 -- new;script;javascript','Executes the specified javascript command. Applet only; may be disallowed by setting <b>_jmol.noEval = true</b> within the web page. Note that double quotes or no quotes are required; single quotes are not appropriate in this context.','1|2','&quot;javascript commands&quot;','')


newCmd('.label',' ','','*v+11.8 -- extends atom options with %[…];label;latom','Turns on and off atom labels based on a previous selection. If a string is given, it is used as the label. See also  {#.hover~} and {#.echo~}. Additional options include setting the {#.font~}, {#.color~color}, {#.background~background}, x- and y-{#.set_labeloffset~offsets}, and the text {#.set_labelalignment~alignment} (left, center, or right). Default settings for these characteristics can be set by first issuing <b>select none</b>, so that no real label is set. Note that commands such as <b>cartoons ONLY</b> with the ONLY keyword turn off labels. To use ONLY and preserve labels, simply assign a variable to the labels to be kept and then restore them -- for example, <b>Var x = {{*}}.labels; cartoons ONLY; {{*}}.labels = x</b>.','1|2','','')
newCmd('','','','*v+11.8;','You can specify atom parameters two different ways. You can use the older notation consisting of a percent sign followed by a single character, or you can use a new notation that is more flexible and easier to remember consisting of a percent sign followed by a keyword in brackets. For instance, you can use <b>select *.ca;label "%n%R"</b> or <b>select *.ca;label "%[group]%[resno]"</b>. See {#.atomproperties~atom properties} for a detailed list of atom properties that can be included in labels. In addition, starting with Jmol 12.2, you can read string data into labels from a {#.data~data "mydata"} command using <b>label %{{mydata}}</b>. If data of name "mydata" is not found, the model\'s auxiliaryInfo will be checked. For example, with  P2N files, which add "altName" information for an atom, one can then use <b>label %{{altName}}</b> to label atoms using that alternative name.','TEXT','Atom label parameters','')
newCmd('','','','*v+11.8;','The sign of the digit x in %x.y determines the column formatting, and the digit y determines decimal places for decimal numbers or maximum number of characters for strings. In addition, if y is negative, for numbers, the number is expressed in scientific notation, and for strings, the string is truncated from the right instead of from the left. In the table below are two examples, one numeric and one a string. Leading and trailing periods represent space characters.  [|| <b>%x.y</b> | <b>123.4</b> | <b>"tetrahedral"</b> || %10.3 | ...123.400| .......tet || %-10.3 | 123.400... | tet....... ||  %10.-3 | ..1.234E+2 | .......ral || %-10.-3 |  1.234E+2.. | ral.......  ||] Note that both x and y are independently optional. <b>%6[temperature]</b> would be the unrounded temperature value placed in a 6-character field (and allowed to run over); <b>%.3[shape]</b> would be the first three letters of the the shape of the atom.   ','TEXT','Column formatting and scientific notation','')
newCmd('','','','*v+11.8;','You can use labels for more than just labeling atoms. Using the .label() function you can create variables and save them to files or deliver them to a webpage using JavaScript, thus producing customized output of your choice. For example, the commands<br /><br />x = {{*.ca}}.label("%5[group] %7[resno] %10.2phi %10.2psi");<br />write VAR x "phipsi.xls" <br /><br />creates a file that lists phi and psi Ramachandran angles for a protein. (A similar output can be obtained using the {#.write~write Ramachandran} command.) From a web page, the following JavaScript retrieves bond information for the first silicon atom from an applet: <br /><br />var x = jmolEvaluate(\'print {{_Si}}[1].bonds.label("%6a1 %6a2 %ORDER %4.2LENGTH")\')<br /><br />Note that the {#.write~} command can be used with the signed applet from the {#.console~} to save data on a local drive even if the applet is from another source, such as {http://www.rcsb.org~}. ','TEXT','Label applications','')
newCmd('','','','','Each label is associated with a specific atom center (a label is an atom property). Labels for all atoms that are {#.select~selected} are affected by any label command. You can position a label relative to the center of its associated atom using -{#.set_labeloffset~set labelOffset}, and align the text using {#.set_labelalignment~set labelAlignment} (LEFT, CENTER, or RIGHT). If you want to position text at the average position of a group of atoms (or drawn objects, or an arbitrary x,y,z position), use {#.echo~} instead. For example: <b>set echo pt1 {{20.0 30.0 23.0}};echo "HERE";</b> or <b>set echo group1 {{[ala]120}};echo ALA;</b>.','TEXT','Label positioning','')
newCmd('','','','*v+11.8;','The basic commands for turning labels on and off are <b>label xxx</b>, where xxx is the text or format for the label, <b>label ON</b>, <b>label OFF</b>, and <b>label TOGGLE</b>. ','TEXT','Setting Labels and Turning Them On and Off','')
newCmd('','','','*v+11.8;','<b>Labels on</b> delivers a default label to the currently selected atoms. The default for this label (<b>%[identify]</b>) depends upon the file type and whether more than one file is loaded:<br /> [|| single model, non-PDB data| <b>%[atomName]  #%[atomNo]</b>||mutiple model, non-PDB data| <b>%[atomName]/%[model  #%[atomNo]</b>. Note that the model will be displayed in "file.model" notation. For example, <b>C12/3.2 #10</b> would be an atom in the third model of the second file loaded.|| single model, PDB data| <b>%[group]%[sequence]:%[chain].%[atomName]%%%[altloc]  #%[atomNo]</b>. (Not all residues will show the chain or altloc, in which case the preceding colon or percent sign, respectively, is not displayed. For example: <b>[MET]1:C.N #608</b>.)||mutiple model, PDB data| <b>%[group]%[sequence]:%[chain].%[atomName]%%%[altloc]/%[model]  #%[atomNo]</b>||]<br /><br />Note that <b>labels off</b> deletes the current formatting, and <b>labels on</b> always restores the default label. ','2','._on_off{"ON"}','')
newCmd('','','','','The label command followed by the desired text in quotes creates and displays that label for the currently selected atoms. The quotes are recommended but not necessary if no special characters such as semicolon, braces, or pound sign are present. Use &#124; (vertical bar) or <b>\\n</b> as a line separator for multiline labels. A limited amount of HTML-like formatting can be used: [||&lt;sub&gt; &lt;/sub&gt;|subscript||&lt;sup&gt; &lt;/sup&gt;|superscript||&lt;color COLOR&gt; &lt;/color&gt;|font color; for example <b>label "%i &lt;color red&gt;%a&lt;/color&gt;</b> or <b>label "%i &lt;color #ff0000&gt;%a&lt;/color&gt;"</b>. ||]Single quotes do not count as quotes in this context, but the text may be a simple variable: <b>x = "testing";label @x</b>.','1','&quot;text&quot;','')
newCmd('','11.3.52','','*v+11.4;','This does not work. See {#.set_LabelToggle~set labelToggle}.','2','TOGGLE','(atom expression)')
newCmd('.lcaoCartoon',' ','','*v+11.4;iso','<img src="img/lcao.jpg" height="100" width="100"   alt="LCAO Cartoons"  />The <b>lcaoCartoon</b>  command displays cartoonish atomic p and hybrid sp, sp2, sp3, sp3d, and sp3d2 orbitals like those commonly seen in textbooks in discussions of the method of linear combination of atomic orbitals. <br /><br />Any number of the following options may be strung together in the same command. The {#.isosurface~} command can also be used for the creation of LCAO cartoons. Selections for scale, color, and translucency are "persistent" -- carrying over from one command to the next -- and thus need only to be given once per model loading if they are not to be changed. Note that these options must be given prior to or along with the actual command creating the cartoon; they do not act on lcaocartoons that have already been made. The CREATE keyword and its associated orbital type, optionally with the added keyword MOLECULAR, must be the very last keyword when multiple keywords are involved.','1|2','','')
newCmd('',' ','','*v+11.0;','Turn on/off the selected LCAO cartoon.','*1','._on_off{"ON"}','')
newCmd('','','','*v+11.0;','Creates a new LCAO cartoon (a tear-shaped lobe) of the given type at the currently selected atoms. Of the selected atoms, only atoms with compatible sigma hybridization are used. The CREATE keyword is optional. Other options include LONEPAIR (two small dots) and RADICAL (one small dot) instead of CREATE . Valid types include are shown below. In addition to those listed, adding a "-" sign prior to the designation -- "-sp2" for example -- denotes the position 180 degrees rotated from the described position. For sp2, sp3, sp3d, and sp3d2 hybridizations, starting with Jmol 12.0, the order a-f starts with bonds present in the model in order of atomIndex and continues to unoccupied locations. (Note: Previous versions of Jmol were inconsistent in terms of this ordering.)<br /><br />[||"cpk"| [Jmol 12.0] Creates a sphere at the current spacefill radius. The reason this could be useful is that such spheres, though associated with an atom, can be slabbed and capped like an isosurface. This allows for a useful "unit cell only" rendering of spacefill models, for example, using <b>spacefill ionic;lcaocartoon scale 1.0 CAP unitcell "cpk";spacefill off</b>. (We turn the spacefill off -- it was just to provide the reference sizes.)||"lp"|a lobe at the "lone pair" position at an AX3E center such as the N atom in NH3, with three bonds in sp3 sigma hybridization, or the sp2-hybridized lone pair in an AX2E bent center such as the unprotonated N atom in histidine. (For two dots, use LONEPAIR instead of CREATE.) ||"lpa", "lpb"|a lobe at one of the two sp3 sigma-hybridized lone pair positions at an AX2E2 bent center, such as that in H2O. (For two dots, use LONEPAIR instead of CREATE.)||"px" "py" "pz"|Standard p orbitals at an sp or sp2 sigma-hybridized center. Both lobes "a" and "b" are produced unless LONEPAIR or RADICAL is being used instead of CREATE.||"s"|standard spherical s orbital at any center.||"spa" "spb"|the two sigma bonding/nonbonding positions around a linear sp sigma-hybridized center.||"sp2a" "sp2b" "sp2c"|the three sigma bonding/nonbonding positions around a trigonal planar or bent center.||"sp3a" "sp3b" "sp3c" "sp3d"|the four sp3 sigma-hybridized positions at any center with a geometry consistent with a tetrahedral electron configuration (bent, trigonal pyramidal, tetrahedral).||"sp3da" "sp3db" "sp3dc" "sp3dd" "sp3de"|the five sp3d sigma-hybridized positions at any center that has at least three bonds and is consistent with a trigonal bipyramidal electron configuration (for example, T-shaped or see-saw). (Jmol 12.0) ||"sp3d2a" "sp3d2b" "sp3d2c" "sp3d2d"  "sp3d2e" "sp3d2f"|the six sp3d2 sigma-hybridized positions at any center that has at least three bonds and is consistent with an octahedral electron configuration. (Jmol 12.0)||]','*20','CREATE','&quot;[type]&quot;')
newCmd('','11.8.RC4','','*v+11.8;','Sets two small dots instead of a lobe at the given position.','*21','LONEPAIR','&quot;[type]&quot;')
newCmd('','11.8.RC4','','*v+11.8;','Sets one small dot instead of a lobe at the given position.','*22','RADICAL','&quot;[type]&quot;')
newCmd('',' ','','*v+11.0;','Creates a new LCAO cartoon of the given type using the molecular axes system as reference. Applicable only for "px", "py", "pz", "-px", "-py", and "-pz". ','*3','CREATE','&quot;[type]&quot;','MOLECULAR','')
newCmd('',' ','','*v+11.0;','Colors the orbital one specific color for the lcaocartoon being created with that command or future lcaocartoons.','*4','COLOR','._colorRGB')
newCmd('','11.1.38','','*v+11.2;','Allows for translucent or opaque lobes  for the lcaocartoon being created with that command or future lcaocartoons. For translucency options, see {#.color~}.','*9','TRANSLUCENT or OPAQUE','')
newCmd('',' ','','*v+11.0;','If this is a p orbital, colors the two lobes different colors  for the lcaocartoon being created with that command or future lcaocartoons.','*60','COLOR','._colorRGB','._colorRGB','')
newCmd('',' ','','*v+11.0;','Delete the LCAO cartoons on the currently selected set of atoms.','*62','DELETE','')
newCmd('','11.1.28','','*v+11.2;','Lists all LCAO cartoons','*65','LIST','')
newCmd('',' ','','*v+11.0;','Sets the scale of the LCAO cartoon  for the lcaocartoon being created with that command or future lcaocartoons.','*7','SCALE','(decimal)')
newCmd('',' ','','*v+11.0;','For the already selected atom set, selects what type of orbital for an on/off/delete operation.','*82','SELECT','&quot;[type]&quot;')
newCmd('','11.1.38','','*v+11.2;','Selects a set of atoms. In the absence of this keyword, the previously selected atom set is used.','*81','SELECT','(atom expression)')

newCmd('.[read/write ZIP files]','11.3.48','','*v+11.4;','Jmol can read and write ZIP file collections. In any file reading operation (for example, in the {#.isosurface~}, {#.load~}, or {#.script~} command), Jmol can read a specific file within ZIP or JAR file collections. (The two formats are the same.) Simply indicate the path to the file within the collection using the vertical bar character as a separator: <b>load "myfiles.zip|files|pdb|1crn.pdb"</b>. A path element may be a subdirectory within the ZIP file or the name of a ZIP file contained within one of those directories: <b>isosurface "mysurfaces.zip|isousurfaces|jvxl.zip|surface1.jvxl"</b>. The full directory contents of a ZIP or JAR file can be read out using <b>getProperty fileContents "myfile.zip"</b>, and the contents of specific files within a zip collection can also be read this way as, for example, <b>getProperty fileContents "myfile.zip|JmolManifest"</b>. Note that variables can be assigned these contents using the getProperty() function: <b>var dir = getProperty("fileContents", "myfile.zip|JmolManifest")</b>. The {#.load~} command specifically allows reading of multiple files from a ZIP or JAR collection, and will read files in the order specified in a file with the name "JmolManifest" contained in the compressed file collection. Starting with Jmol 12.0, the command {#.write~write xxx.jmol} or {#.write~write ZIPALL xxx.zip} will create a ZIP file collection that contains a JmolManifest file along with all of the files necessary to recreate the current state. In addition, {#.write~write ZIP xxx.zip} will make a collection that contains all local files necessary, but will simply point to any required remote (http://, ftp://) resource.','1|2','','')


newCmd('.load',' ','','*v+13.0 adds capability to load from PubChem and to read JCAMP-DX (JDX) files (IR, NMR, UV/VIS, GC/MS) and to correlate atoms with peak assignments or molecules and MS fragments;load','Loads the specified file or URL. A wide variety of file types are supported. In general, resolution of file type is based on internal file cues, not the filename or file extension. However, this resolution process can be overridden by specifying a prefix to the file name consisting of the file type followed by two colon characters: <b>load "molpro(xml)::myfile"</b>. Starting with Jmol 12.2, <b>PDB-type files can be loaded with automatic addition of hydrogen atoms and multiple bonds using {#.set_pdbaddhydrogens~SET pdbAddHydrogens}.</b> For PDB, PQR, and mmCIF files that contain HELIX, SHEET, or TURN records provided by the author, those structures are created (though not initially displayed); for PDB, PQR, and mmCIF files that do not contain this information, Jmol will generate secondary structure using DSSP (W. Kabsch and C. Sander, Biopolymers, vol 22, 1983, pp 2577-2637; {http://swift.cmbi.ru.nl/gv/dssp~}) for chains that contain backbone atoms and the method of Levitt and Greer [J.Mol.Biol.(1977) 114, 181-293] for alpha-carbon-only chains. (See also {#.calculate~calculate STRUCTURE}.) <br /><br />Multiple files and a selected model from a multi-model file can be read as well. Files may be Gzipped, and multiple files can be read from compressed ZIP and JAR collections. A default directory can be set using {#.set_defaultDirectory~}, and for applets a proxy server can be set for non-host file loading using {#.set_appletProxy~}. After the filename, the options listed below are available. These options must appear in the order given in the table below. In addition, the load command can take parameters that specify the number of unit cells to generate, the the space group or Jones-Faithful operators to use, and the unit cell dimensions. Using these parameters, the space group and unit cell information within a file can be overridded, and even simple XYZ data files can be turned into crystallographic data sets. The FILTER keyword allows selective loading of atoms as well as construction of the biologically relevant molecule (PDB "BIOMOLECULE" records. See also the headings {#.load APPEND~}, {#.load FILES~},  {#.load MENU~}, {#.load MODELS~}, and {#.load TRAJECTORY~}. Note that with the Jmol application (not the applet) you can also use Edit...Paste to load molecular coordinate from the system clipboard. The same capability for the applet can be had using {#.data~data "model"}. ','1|2','','')
newCmd('.load',' ','','*v+12.2 adds the SUPERCELL option and automatic addition of hydrogen atoms and multiple bonds to PDB files;*v-13.0;load','Loads the specified file or URL. A wide variety of file types are supported. In general, resolution of file type is based on internal file cues, not the filename or file extension. However, this resolution process can be overridden by specifying a prefix to the file name consisting of the file type followed by two colon characters: <b>load "molpro(xml)::myfile"</b>. Starting with Jmol 12.2, <b>PDB-type files can be loaded with automatic addition of hydrogen atoms and multiple bonds using {#.set_pdbaddhydrogens~SET pdbAddHydrogens}.</b> For PDB, PQR, and mmCIF files that contain HELIX, SHEET, or TURN records provided by the author, those structures are created (though not initially displayed); for PDB, PQR, and mmCIF files that do not contain this information, Jmol will generate secondary structure using DSSP (W. Kabsch and C. Sander, Biopolymers, vol 22, 1983, pp 2577-2637; {http://swift.cmbi.ru.nl/gv/dssp~}) for chains that contain backbone atoms and the method of Levitt and Greer [J.Mol.Biol.(1977) 114, 181-293] for alpha-carbon-only chains. (See also {#.calculate~calculate STRUCTURE}.) <br /><br />Multiple files and a selected model from a multi-model file can be read as well. Files may be Gzipped, and multiple files can be read from compressed ZIP and JAR collections. A default directory can be set using {#.set_defaultDirectory~}, and for applets a proxy server can be set for non-host file loading using {#.set_appletProxy~}. After the filename, the options listed below are available. These options must appear in the order given in the table below. In addition, the load command can take parameters that specify the number of unit cells to generate, the the space group or Jones-Faithful operators to use, and the unit cell dimensions. Using these parameters, the space group and unit cell information within a file can be overridded, and even simple XYZ data files can be turned into crystallographic data sets. The FILTER keyword allows selective loading of atoms as well as construction of the biologically relevant molecule (PDB "BIOMOLECULE" records. See also the headings {#.load APPEND~}, {#.load FILES~},  {#.load MENU~}, {#.load MODELS~}, and {#.load TRAJECTORY~}. Note that with the Jmol application (not the applet) you can also use Edit...Paste to load molecular coordinate from the system clipboard. The same capability for the applet can be had using {#.data~data "model"}. ','1|2','','')
newCmd('.load',' ','','*v+12.0 adds SMILES string reading;*v-12.2;load','Loads the specified file or URL. A wide variety of file types are supported. In general, resolution of file type is based on internal file cues, not the filename or file extension. However, this resolution process can be overridden by specifying a prefix to the file name consisting of the file type followed by two colon characters: <b>load "molpro(xml)::myfile"</b>. <br /><br />Multiple files and a selected model from a multi-model file can be read as well. Files may be Gzipped, and multiple files can be read from compressed ZIP and JAR collections. A default directory can be set using {#.set_defaultDirectory~}, and for applets a proxy server can be set for non-host file loading using {#.set_appletProxy~}. After the filename, the options listed below are available. These options must appear in the order given in the table below. In addition, the load command can take parameters that specify the number of unit cells to generate, the the space group or Jones-Faithful operators to use, and the unit cell dimensions. Using these parameters, the space group and unit cell information within a file can be overridded, and even simple XYZ data files can be turned into crystallographic data sets. The FILTER keyword allows selective loading of atoms as well as construction of the biologically relevant molecule (PDB "BIOMOLECULE" records. See also the headings {#.load APPEND~}, {#.load FILES~},  {#.load MENU~}, {#.load MODELS~}, and {#.load TRAJECTORY~}. Note that with the Jmol application (not the applet) you can also use Edit...Paste to load molecular coordinate from the system clipboard. The same capability for the applet can be had using {#.data~data "model"}. ','1|2','','')
newCmd('','','','*v+11.6;','Jmol reads a large number of data formats. Example files can be found at {http://jmol.svn.sourceforge.net/viewvc/jmol/trunk/Jmol-datafiles~}. Supported file types include: <br/><br/>[||$identifier|Starting with Jmol 12.2, the signed applet and the Jmol application can utilized the SMILES-to-3D and database service at the {http://cactus.nci.nih.gov~NIH Cactus server} to load 3D models based simply on {#.Jmol SMARTS/SMILES~SMILES} strings or chemical identifiers such as chemical names, CAS registry numbers, and InChI keys. For example, <b>load $CCC</b> loads the structure of propane, <b>load $C1CCCCC1</b> loads a chair cyclohexane model, and <b>load $tylenol</b> loads a model of acetaminophen. ||:name|Starting with Jmol 13.0, the signed applet and application can load a model directly from PubChem. A name or a PubChem ID may be given. For example, <b>load ":tylenol"</b> or <b>load ":1983"</b> loads a model of acetaminophen. More specifically the syntax <b>load :xxx:yyyy</b>, where xxx is NAME, CAS, CID, or SMILES, specifies the exact sort of compound identifier yyyy that follows the second colon.||=XXXX|The signed applet and application can load a PDB file directly from RCSB. For example, <b>load "=1crn"</b>.||==XXX|Starting with Jmol 12.2, the signed applet and application can load a ligand file directly from RCSB. For example, <b>load "==HEM"</b>.|| ADF|{http://www.scm.com/~Amsterdam Density Functional} output file||AIMS|{https://aimsclub.fhi-berlin.mpg.de/~Ab Initio Molecular Simulation Package}  (input files only) ||AMPAC|{http://www.semichem.com/ampac/~AMPAC} file ||Argus(XML)|{http://www.planaria-software.com/~ArgusLab} AGL file||Chem3D(XML)|CambridgeSoft {http://www.cambridgesoft.com/software/details/?ds=3~Chem3D} C3XML file||CASTEP|{http://www.castep.org~CASTEP} input  , .md  (Jmol 12.2), .geom  (Jmol 12.2), .phonon  (Jmol 12.2), and .magres (Jmol 13.2) file formats. .phonon files are read as trajectories -- see the FILTER specifications, below, to load specific q-point mode sets.||CIF|International Union of Crystallography {http://www.iucr.org/iucr-top/cif/home.html~Crystallographic Information File}, including Macromolecular Crystallographic Information filea (mmCif). NOTE: Jmol 12.2 changes the default for loading CIF files that have GEOM_BOND records to make loading more similar to the way Mercury opens files. Such files are now loaded as full molecules based on the idea that a "molecule" is "symop=1 or connected(symop=1)". The model is displayed WITHOUT a unit cell, and no unit cell information is made available. The standard <b>load xxx.cif {{1 1 1}}</b> is still available if a unit cell is desired.||CML(XML)|{http://cml.sourceforge.net/~Chemical Markup Language} files.  ||CRYSTAL| {http://www.crystal.unito.it/~Crystal09} solid state computation output, including support for 1D (polymer) and 2D (slab) periodicity. This file format creates the atom properties <b>property_spin</b> and <b>property_magneticMoment</b> [Jmol 12.0].  ||CSF|Fujitsu {http://www.computers.us.fujitsu.com/www/products_bioscience.shtml?products/bioscience/cache~Sygress Explorer} (formerly CAChe) chemical structure file, including the reading of ab initio, semiemperical, gaussian, and density functional molecular orbitals||CUBE|Gaussian {http://www.gaussian.com/g_ur/u_cubegen.htm~cubegen} output file||DCD| single precision binary (FORTRAN unformatted) {http://www.ks.uiuc.edu/Research/vmd/plugins/molfile/dcdplugin.html~DCD} trajectory files  (Jmol 12). These files require the COORD keyword and a file giving atom information, such as a PDB file. For example: <b>load trajectory "t.pdb" coord "t.dcd"</b>||DGRID|{http://www.scm.com/News/DGrid.html~DGRID} file reader (Jmol 12). These files are generalized representations of output from a variety of quantum mechanical calculation packages, including especially {http://www.scm.com/Doc/Doc2009.01/ADF/ADFUsersGuide/page430.html~ADF}.||FoldingXYZ|XYZ file created by the {http://fahwiki.net~Folding@home project}||GAMESS|{http://www.msg.chem.iastate.edu/gamess/~General Atomic and Molecular Electronic Structure System} output file||Gaussian|{http://www.gaussian.com/~Gaussian} output file||GhemicalMM|{http://en.wikipedia.org/wiki/Ghemical~Ghemical} molecular mechanics file (MM1GP)||GRO|{http://www.gromacs.org~GROMACS} .gro file format ||GULP|{https://projects.ivec.org/gulp~GULP} file format (Jmol 12.2)||HIN|{http://www.hyper.com/~HyperChem} native file||Jaguar|National Center for Supercomputing Applications Jaguar output file ||JANA2006| Jana2006 M40/M50 files can be read, and incommensurate structures can be displayed.||JCAMP-DX| Starting with Jmol 13.0, Jmol can read specialized JCAMP-DX files containing  ##$MODELS records and automatically sychronize user picking actions based on ##$PEAKS record information.  See {misc/Jmol-JSpecView-specs.pdf~Jmol-JSpecView-specs.pdf}. ||JME|{http://www.molinspiration.com/jme/~Java Molecular Editor} file format (a 2D, not a 3D, format)||MAGRES|{http://www.ccpnc.ac.uk/pmwiki.php/CCPNC/MagresView~Magres} files [Jmol 13.2]||MDTOP, MDCRD|{http://amber.scripts.edu~AMBER} Molecular dynamics topology files and associated coordinate files. ||MOL, MOL2|Symyx (formerly Molecular Design) {http://www.mdli.com/downloads/public/ctfile/ctfile.jsp~structure data files}, including SDF and CTAB V2000 files||MOLDEN|{http://www.cmbi.ru.nl/molden/molden_format.html~Molden} data file||MOLPRO(XML)|{http://www.molpro.net/~Molpro} structure file||Mopac|{http://openmopac.net~OpenMopac} output file (MOPOUT)||MopacGraphF|{http://openmopac.net~OpenMopac} GRAPHF output file (for molecular orbitals)||NWCHEM|Pacific Northwest National Laboratory {http://www.emsl.pnl.gov/docs/nwchem/nwchem.html~NWChem}  output file||Odyssey|WaveFunction {http://www.wavefun.com/products/odyssey/odyssey.html~Odyssey} data file (ODYDATA)||Odyssey(XML)|WaveFunction {http://www.wavefun.com/products/odyssey/odyssey.html~Odyssey} XODYDATA file||PDB|{http://www.wwpdb.org/docs.html~Protein Data Bank} files. PDB reading of X-PLOR using hybrid-36 and NAMD files using hex is also supported. See {http://www.schrodinger.com/AcrobatFile.php?type=supportdocs&type2=&ident=530~} and {http://cci.lbl.gov/cctbx_sources/iotbx/pdb/hybrid_36.py~}. ||PQR|Position/Charge/Radius data file produced by the {http://apbs.sourceforge.net/~Adaptive Poisson-Boltzmann Solver} project||P2N|{http://q4md-forcefieldtools.org/Tutorial/Tutorial-1.php#3~R.E.D. input files}, also reading the AltName field and allowing <b>label %[altName]</b> (Jmol 12.2)||PSI|{http://www.psicode.org/~PSI3} output reader (coordinates only)||PyMOL PSE|{http://pymol.org~PyMOL} session files. Jmol 13.2 will read most PyMOL session files, creating a very close representation of the view as seen in PyMOL 1.6. This ongoing project aims to faithfully reproduce the key features of PyMOL sessions. A representative sampling of how well this works can be found at {http://ispcbck.weizmann.ac.il/a2jmolb/browse~} and {misc/PyMOL2Jmol_browse.pdf~}. FILTER options "nosurface" and "docache" are explained below. ||QCHEM|{http://www.q-chem.com/~Q-Chem} output file||QuantumEspresso|see {http://www.quantum-espresso.org~}; XML files readable in Jmol 12.2.||SHELX|{http://shelx.uni-ac.gwdg.de/SHELX/~SHELX} output file||Spartan|WaveFunction {http://www.wavefun.com/p+I515roducts/spartan.html~Spartan} data file||SMILES|See $smilesString, above. ||SpartanSmol|WaveFunction binary {http://www.wavefun.com/products/spartan.html~Spartan} SMOL data file, including full MacSpartan Spartan directories in ZIP format||V3000|Symyx (formerly Molecular Design) {http://www.mdli.com/downloads/public/ctfile/ctfile.jsp~V3000 Connection Table} (CTAB or SDF) data file||WebMO|{http://www.webmo.net/~WebMO} molecular orbital file reader||VASP|Vienna Ab Initio Simulation Package {http://cms.mpi.univie.a+I502c.at/vasp/~VASP} vasprun.xml files. [Jmol 12.2 adds VASP OUTCAR file reading.] ||Wien2k|{http://www.wien2k.at~Wien2k} data files. WIEN2k performs electronic structure calculations for solids using density functional theory. Using the option <b>spacegroup "none"</b> disregards symmetry information given in the file and simply reads the atom coordinates given in the file, including MULT atom records. For example, <b>load t.struct {{1 1 1}} spacegroup "none"</b> ||XYZ|Minnesota Supercomputer Institute XMol file format. Various extensions to this file format allow reading of the following information separated by whitespace:[||element | x | y | z || element | x | y | z | vibX | vibY | vibZ ||  element | x | y | z | charge | | | | | (Jmol 12)|| element | x | y | z | charge | radius | |  | | (Jmol 12)|| element | x | y | z | charge | vibX | vibY | vibZ | |  (Jmol 12)|| element | x | y | z | charge | vibX | vibY | vibZ | atomNumber|  (Jmol 12)||]  If the charge is an integer, it is read as <b>formalCharge</b>; if it is decimal, then as <b>partialCharge</b>. Any information past x y z is optional, and if missing or uninterpretable as a number (for example, "X" or "--") will be ignored. Starting with Jmol 12.2, the element may be either an atomic symbol or an isotope symbol such as 13C or an atomic number such as 6. ||ZMatrix|The Z-Matrix format allows for molecules to be described by internal coordinates (distances, angles, and dihedrals) rather than in terms of actual Cartesian coordinates. Jmol can read several forms of Z-matrix data, including {http://openmopac.net/manual/gaussian_z.html~MOPAC} and {http://www.gaussian.com/g_tech/g_ur/c_zmat.htm~Gaussian} format as well as a format designed for Jmol that allows also the simple introduction of bond order. Details are given in the {http://jmol.svn.sourceforge.net/viewvc/jmol/trunk/Jmol/src/org/jmol/adapter/readers/simple/ZMatrixReader.java?view=markup~Z-Matrix reader code} itself. The reader supports dummy atoms, atom labels, symbolic variable definitions, Cartesian coordinates or a mix of Cartesian and internal coordinates, alternative distance + two angles rather than distance + angle + dihedral, and specification of bond order. (Jmol 12.3.8). Gaussian and MOPAC file names must be preceded with <b>ZMATRIX::</b> to bypass attempted format identification, as those file formats have no easily recognizable internal characteristics. ||]','TEXT','[File types]','')
newCmd('','11.3.48','','*v+11.4;','Jmol can read specific files within compressed {#.reading ZIP/JAR files~ZIP and JAR collections}. In addition, for the <b>load</b> command specifically, Jmol will look for a file within the collection with the name <b>JmolManifest</b> and follow the directives within it. These directives may be as simple as a list of files to be loaded, one filename per line (which will be read in the order listed). Lines starting with # are comment lines, which may contain any text, but also may contain one or more of the following keywords: <br /><br />[||<b>#EXCEPT_FILES</b>|The list of files specifies files to ignore, not files to load; all other files will be loaded.||<b>#IGNORE_ERRORS</b>|Try to read files, but ignore errors when a file is not recognized as a valid model format that Jmol can read. This option allows easy "mining" of ZIP collections for files that Jmol can recognize, ignoring all others. ||<b>#IGNORE_MANIFEST</b>|Ignore this manifest in its entirety -- simply read all files in the order retrieved by the ZIP file iterator.||]','TEXT','[ZIP/JAR files and JmolManifest]','')
newCmd('','','','','The following options may be indicated after specifying the filename. Each is optional, but if more than one option is indicated, options must be given in the order listed in this table. <br /><br />[||<b>MANIFEST "manifestOptions"</b>|  If the file being loaded is a ZIP or JAR file, Jmol will search for a file in that compressed file collection with the name "JmolManifest" and process it accordingly (see above). However, in the <b>load</b> command, if the keyword <b>MANIFEST</b> and a quoted string follows the filename, then Jmol will use this string as the manifest instead, with lines of the manifest separated by vertical bar "&brvbar;" characters. In this way, standard ZIP collections can be read, and the order of file loading can be specified. For example: <b>load "test.zip" manifest "CH3CL.MOL&brvbar;CH4.MOL"</b> reads only these two files from the ZIP collection, in this order. If the file contains a manifest and that manifest is simply to be ignored, the quoted string should read "IGNORE_MANIFEST".  ||<b> (integer) </b> | Loads only the specified model number (if positive) or vibration number (if negative, starting with Jmol 12.0), skipping the others. For PDB files, the number indicated is the number specified in the MODEL record. (Not supported by all file types.) See also {#.load MODELS~}. ||<b>{{unitCell(s)}}</b> | see <i>Crystallographic Options</i>, below||<b>OFFSET {{x y z}}</b>| Starting with Jmol 12.2, you can offset atoms in a file by a given amount at load time. The offset can be expressed in fractional coordinates, such as {{1/2 1/2 1/2}}, and although primarily intended for CIF file reading, can be applied to any file type. If a space group and/or unit cell are specified, the OFFSET parameter must follow that specification.||<b>FILTER</b> |  The FILTER parameter specifies file-type specific load options (see below). Options should be separated by commas. For atom selection,  more than one specification may be made using a comma between specifications: *:1,*:2, but * and ! may not be mixed within any one type (atom name, group name, or chain ID). Starting with Jmol 12.2, "&brvbar;" indicates OR in a broader context: <b>load 1blu.pdb filter "*.CA &brvbar; HETATM,![HOH]"</b> selects for alpha carbons OR (HETATM and not HOH).||]','TEXT','[General Options]','')
newCmd('','','','','The following crystallographic options may be indicated, starting with the specification of the unit cell. If more than one option is indicated, options must be given in the order listed in this table. <br /><br />[||<b>{{lattice}}</b> | Jmol can read unit cell and symmetry information from selected file types (for example, CIF, PDB, or SHELX). The specific set of unit cells to load can be specified one of two ways -- either using the notation {{i j k}} or the notation {{ijk i\'j\'k\' n}}.   [||<b>{{i j k}}</b> | Loads a block of unit cells between the origin, {{0 0 0}} and the specified unit cell system coordinate. Used alone, {{i j k}} is only for working with files containing both unit cell and space group information (CIF, SHELX, CML, for example). The particular choice {{3 3 3}} is significant, in that it loads 27 unit cells, forming a solid block around a central cell. The unit cell display can then be moved to the origin of this central cell using <b>unitcell {{1 1 1}}</b>, and the display of atoms can be restricted to that center cell using <b>restrict cell=666</b> or <b>restrict cell={{2 2 2}}</b>. Multiple unit cell loading  can be combined with the single-model loading by indicating the model number first, then the number of unit cells: <b>load "myfile.cif" 15 {{3 3 3}}</b>. Quotes are not required. There is no restriction other than memory on the size of i, j, and k (except that all must be positive).  ||<b>{{ijk&nbsp;i\'j\'k\'&nbsp;-1}}</b> | <img src="img/loadunitcell-1.jpg" align="right" />Loads a block of unit cells within the range ijk and i\'j\'k\' (which should include 555) and packs all atoms into the designated set of cells. If atoms are on faces, those atoms will be duplicated at each equivalent face.||<b>{{ijk&nbsp;i\'j\'k\'&nbsp;0}}</b> | <img src="img/loadunitcell0.jpg" align="right" />Loads a block of unit cells within the range ijk and i\'j\'k\' (which should include 555) WITHOUT normalizing the operators. All symmetry-generated atoms are placed based on the exact definition of the symmetry operations found in the file or designated using the <b>spacegroup</b> keyword (see option below). Note, however, that if explicit operations are not provided and therefor must be generated from a spacegroup name, they will be normalized. The list of operations used can be obtained using {#.show~show symmetry}.||<b>{{ijk i\'j\'k\' 1}}</b> | <img src="img/loadunitcell1.jpg" align="right" />Loads a block of unit cells within the range ijk and i\'j\'k\' (which should include 555), normalizing the operators to move the geometric center of the generated set of atoms into cell 555, then applying the lattice translation. Thus, <b>load "filename" {{555 555 1}}</b> is equivalent to <b>load "filename" {{1 1 1}}</b>. For example, <b>load "myfile.cif" {{444 666 1}}</b> loads a block of 27 unit cells, with the geometric center of all units with the bounds of the fractional coordinate range {{-1 -1 -1/}} to {{2 2 2/}}.||] ||<b>PACKED</b>| The keyword <b>PACKED</b> may be used in place of <b>{{555 555 -1}}</b> or after a designated set of cells: <b>load t.struct {{2 2 2}} PACKED</b> in order to load a set of atoms packed into the unit cell.  If atoms are on faces, those atoms will be duplicated at each equivalent face.  ||<b>SUPERCELL {{na nb nc}}</b>| Jmol 12.2 introduces the SUPERCELL option, which allows reassigning the unit cell as any multiple of the actual unit cell. If this option is used, the final loaded symmetry is automatically set to P1, and cell parameters <b>a</b>, <b>b</b>, and <b>c</b> will be modified. A larger lattice can still be indicated, in which case they refer to how many supercells will be loaded. For example: <b>load "quartz.cif" {{2 2 2}} supercell {{5 5 1}}</b> will create a supercell that is 5x5x1 standard unit cells in size and then load a set of eight of those. (Jmol 12.2) ||<b>SUPERCELL "axisFormula"</b>| This supercell option allows creating a supercell using a linear combination of the x, y, and z vectors of the file-based unit cell. If this option is used, the final loaded symmetry is automatically set to P1, and cell parameters <b>a</b>, <b>b</b>, <b>c</b>, <b>alpha</b>, <b>beta</b>, <b>gamma</b> will be modified. For example, <b>load "quartz.cif" packed supercell "2x, x + 2y, z"</b> will load a rectangular unit cell instead of the hexagonal cell indicated in the file.  (Jmol 12.2)||<b>RANGE x.x</b> | Restricts the atoms loaded to those within a given range in angstroms. If x.x is positive, then this range is relative to the entire set of atoms that would be generated using <b>load "filename" {{1 1 1}}</b>, and all atoms within the given distance from ANY atom in that set is loaded; if x.x is negative, then the range is relative to the atoms that would be generated using just <b>load</b> itself (the base x,y,z symmetry set), and the atoms loaded are all those <b>within a box</b> that is the designated distance larger than that needed to just contain the set of atoms. Then, to later select atoms actually within a given distance from the base set, use <b>within</b> with <b>select</b> or <b>display</b>: <b>load 1crn.pdb {{1 1 1}} range -4; display within(4.0, symop=1555)</b>. (This two-step sequence is very efficient.) Note that Jmol will expand the load box by one number in each direction in order to find atoms that are outside of {{i j k}} but still within the designated range: <b>load {{555 555 1}} range 2</b> actually checks all atoms within the 27-cell range 444 through 666.   ||<b>SPACEGROUP "name"</b>| Loads a block of unit cells between the origin, {{0 0 0}} and the specified unit cell system coordinate. If the name is empty quotes, the space group of the currently loaded file will be used (Jmol 12.2). In addition, the symmetry inherent in the file is ignored, and the specified space group is applied. Quotes are required around the space group name. If the space group name itself includes double quotes, use two single quotes or an "escaped double quote" (\\") instead. For example: P 32 2\'\' (single quotes here) or P 32 2\\", not P 32 2". Generally Jmol reads the Jones-Faithful operators from a file, however if the spacegroup name is specified as "ignoreOperators", Jmol will ignore any explict file-based Jones-Faithful operators and instead create the symmetry based on parsing of the space group symbol in the file (Hermann-Mauguin, Hall, or international table number). If the name is a semicolon-separated list of Jones-Faithful operators, such as "x,y,z;x+1/2,y,z", Jmol will ignore any explict file-based operators and instead create the symmetry based on the list provided.  || <b>UNITCELL&nbsp;[a,b,c,alpha,beta, gamma]</b>| Specifies the unit cell to use for models in this file.  If a unit cell is specified in a file, then this option allows overriding that specification. A 0 for b implies polymer (1D) periodicity (and c is ignored), otheriwse a 0 for c indicated slab (2D) periodicity (Jmol 12.0). When both SPACEGROUP and UNITCELL are provided, Jmol can display molecules found in standard Cartesian coordinate files (XYZ, MOL, PDB) as packed unit cells. .|| <b>UNITCELL&nbsp;[ax, ay, az,  bx, by, bz,  cx, cy, cz]</b>|Specifies the unit cell in terms of the XYZ coordinates for the vectors a, b, and c. If bx = by = bz = 0, then cx, cy, and cz are ignored, and polymer (1D) periodicity is implied; otherwise, if cx = cy = cz = 0, then slab (2D) periodicity is implied.  (Jmol 12.2) <b>UNITCELL ""</b> uses the unit cell of the currently loaded file. This allows, for example, adding of atoms to a crystal model using fractional coordinates:(Jmol 12.2) <br /><br />set appendNew false <br />  LOAD DATA "append"<br />  1<br /> testing<br /> Na 0.5 0 0.5 <br /> end "append" {{1 1 1}} spacegroup "" unitcell ""<br />||]','TEXT','[Crystallographic Options]','')
newCmd('',' ','','','The load command by itself reloads the current file.','*10','','')
newCmd('',' ','','','Jmol automatically determines file type based upon the contents of the file. Quotes are recommended. Files containing fractional coordinates are displayed with their unit cell visible. <b>load ""</b> reloads the current file. For files containing multiple models, an optional integer after the file name will load only the specified model. If this number is negative, it refers to a specific vibrational mode in files that contain that information (Jmol 12.0). If this number is 0, it refers to the last model in the set (Jmol 12.0).','*2','&quot;filename&quot;','(integer)')
newCmd('','','','*v+12.2;','An additional way to load a specific set of models using an array, starting  with model 1 for the first. The order of models is not significant. Thus, [1 2 3] is the same as [3 2 1].','*21','&quot;filename&quot;','[i j k l m...]')
newCmd('','','','','File format can be forced by prefixing a filename with "xxxx::" where "xxxx" is a Jmol file type. This should not be necessary in most cases, since Jmol determines file type by scanning the first lines of a file for file-type-specific content. In certain cases, however, where there are extensive comments at the beginning of a file, or a file type (such as MDCRD) does not have any distinguishing characteristics, it is possible for Jmol to fail to discover the file type or to misassign it. In that case, adding "xxxx::" is necessary. ','*31','&quot;filetype::filename&quot;','')
newCmd('',' ','','','Loads the file with the name specified by variable <b>variableName</b>. Jmol 12.0 adds the capability to load a set of files that are defined in an array variable.','*33','@variableName','')
newCmd('','12.0.?','','*v+12.0;','Loads the file data given in quotes. Generally this would be a small molecule in JME or XYZ format.','*40','INLINE','&quot;fileData&quot;')
newCmd('','','','','You can load a model from data contained in a variable. This allows modification of the data prior to display, for example. Quotes are necessary, as without them  -- load @x -- it is the file name that is expected in variable x, not the file contents. For example: <b>x = load("quartz.cif");load "@x" {{2 2 2}};reset x</b>. Note that to save memory, it is a good idea to clear the variable using <b>reset x</b> just after loading. An interesting aspect of this option is that it allows data from a remote file to be saved in the state. This means that if you use <b>write state "somefile.spt"</b>, then that single state script will contain the full file data in a DATA statement. It will be "transportable" and no additional model file will be required. (Jmol 12.2, though, adds the JMOL file format for the {#write~} command, which is even better.)','*41','&quot;@variableName&quot;','')
newCmd('','11.1.22','','','You can load PDB files directly from {http://www.rcsb.org~} or another server of your choice. (This depends upon the setting of {#.set(files and scripts)~set loadFormat}). Simply preface the four-letter PDB id code with "=", and Jmol will load the file. Starting with Jmol 12.0, adding "AS ." ("as" with a period) you can save that file automatically in the default directory (as xxxx.pdb.gz), and using, for example, <b>load =1crn AS "myfile.pdb"</b>, you can save it to some other local file name. Starting with Jmol 12.0, the default is to transfer the file in g-zipped format; If you add ".pdb" to the expression -- <b>load =1crn.pdb</b>, for example -- then Jmol will transfer and save the uncompressed PDB file. ','*43','=XXXX','')
newCmd('','11.1.22','','','Starting with Jmol 12.2, you can load PDB ligand files directly from {http://www.rcsb.org~} or another server of your choice. (This depends upon the setting of {#.set(files and scripts)~set loadLigandFormat}). Simply preface the three-letter PDB id code with "==", and Jmol will load the file. ','*441','==XXX','')
newCmd('','13.1.13','','','Starting with Jmol 13.2, you can load files from a named database. For example, <b>load =mp/1</b> loads a file from {http://www.materialsproject.org~}. Additional options for XX include ligand, nci, nmr, pdb, and pubchem.','*442','=XXX/','')
newCmd('','12.3.23','','','Starting with Jmol 13.0, you can load models from PubChem.  Preface the compound name with colon.  Quotes are only required if nonalphanumeric characters are present in the name.','*443','&quot;:chemical name&quot;','')
newCmd('','12.0.RC15','','','Starting with Jmol 12.0, you can load SMILES strings, and Jmol will turn them into 3D models using the {http://cactus.nci.nih.gov~NIH Cactus server}. As for reading files from any source outside your domain,  you will have to use the signed applet or Jmol application to do this. These files can be saved as MOL files using {#.write~write xxx.mol} or <b>load $xxxx AS "myfile.mol"</b>, and if the conformation is not to your liking, switching to {#.set modelkitMode~} or using {#.set picking~set picking dragMinimize} you can quickly adjust the model to the desired conformation. Quotation marks should be used for names that include the space character: <b>load "$ethyl acetate"</b>.','*45','$smilesString','')
newCmd('','12.0.RC15','','','The service at {http://cactus.nci.nih.gov~} is not limited to SMILES strings. The database includes several million known compounds, which are accessible by CAS registry number, InChI codes, and chemical name as well. For example, <b>load $57-88-5</b> loads a model of chlolesterol, and <b>load $taxol as "taxol.mol"</b> loads a model of taxol and saves it on your hard drive. Starting 4/2011, this service taps into the {http://opsin.ch.cam.ac.uk~OPSIN} service of Cambridge University to return structures from IUPAC or IUPAC-like names. ','*46','$identifier','')
newCmd('','12.0.RC15','','','An alternativ+I440e to the $ syntax for loading SMILES strings.','*47','SMILES','&quot;smilesString&quot;')
newCmd('',' ','','','An optional keyword, {#.loadappend~APPEND}, {#.loaddata~DATA}, {#.loadfiles~FILES}, {#.loadmenu~MENU}, {#.loadmodels~MODELS}, or {#.loadtrajectory~TRAJECTORY}, may be supplied prior to the quoted filename. Other keywords are ignored. (Jmol does not use the Chime-style keyword to specify "file format". Rather, file format can be forced by prefixing a filename with "xxxx::" where "xxxx" is a Jmol file type. However, this should not be necessary in most cases, since Jmol determines file type by scanning the first lines of a file for file-type-specific content. (In certain cases, where there are extensive comments at the beginning of a file, it is possible for Jmol to fail to discover the file type or to misassign it. In that case, xxxx:: should be used.)','*11','keyword','&quot;filename&quot;')
newCmd('','','','*v+11.6;','For individual file types, it is possible to filter the data in the file as it is loaded. The FILTER keyword followed by a quoted string allows for this. The FILTER keyword must be the last keyword in the LOAD command. Specific filters include: [||<b>file type</b> | <b>filter</b> |<b>description</b>||all|<b>FILTER "center"</b>|centers all models in a file based on the first model\'s position. (Jmol 13.0)||all|<b>FILTER "NAME=..." | loads only those models with a name that contains a specified set of characters. (Jmol 13.0) ||all|<b>FILTER "reverseModels"</b>| delivers the models in reverse order (Jmol 13.0). Useful for intrinsic reaction coordinate (IRC) animations.||many|<b>FILTER "noMO"</b>| skips the reading of molecular orbitals for any file that contains them.||many|<b>FILTER "noVIB"</b>| skips the reading of vibrations for any file that contains them.|| CASTEP | <b>FILTER "q=..."</b>| for .phonon files, <b>FILTER "q=n"</b> loads the nth q-point frequency modes. "q=all" reads all phonon modes. For files with multiple extrapolations to q=0, "q=0.1", "q=0.2", etc... loads a specific extrapolation. Q-points can also be specified in terms of fractions, "q=(1/2 1/2 1/2)", or decimals, "q=(0.333 0.111 0.333)". If no filter is specified, but the SUPERCELL keyword is used, Jmol will attempt to find the appropriate matching q-point data for that supercell. || CIF | <b>FILTER "ASSEMBLY n"</b> | loads a specific CIF assembly. For example, <b>load =1vif.cif filter "ASSEMBLY 1"</b>. Specific label_asym_id can be selected as well: <b>load =1vif.cif filter "ASSEMBLY 1;$A"</b> (just label_asym_id A); <b>load =1vif.cif filter "ASSEMBLY 1;!$C"</b> (just label_asym_id values not C). (Jmol 13.2) ||CIF | <b>FILTER "molecular"</b> | instructs Jmol to attempt to complete molecules using either the bonding information in the CIF file or standard van der Waals radii. If this filter is not present and no lattice is indicated either, if bonding information is present in the file, Jmol will use that information to automatically complete molecules. (Jmol 12.2) ||msCIF, JANA2006| <b>FILTER "MODAVERAGE"</b>|  do not read modulation||msCIF, JANA2006| <b>FILTER "MODAXES=xyz"</b>|   read modulation only along specific axes -- x, y, and/or z||CIF, JANA2006| <b>FILTER "NOSYM"</b>|  do not apply symmetry||CIF, GROMACS, MDTOP, PDB, PQR | <b>FILTER "[XXX], .XXX, :X, %X"</b>|  to specify inclusion or exclusion of specific residue types, atom types,  chains, or alternative locations. The prefix "!" indicates NOT. Within each of these four sets, multiple selections are treated as "OR" without "!" and "AND" when "!" is present. With % (alternative location), atoms with no alternative location marking are ALWAYS loaded regardless of this filter. For example, <b>load "1sva.pdb" FILTER "*.CA,%B"</b> loads only alpha carbons that correspond to the conformation involving alternative location B; <b>FILTER "![HOH]"</b> filters out water molecules.|| CRYSTAL | <b>FILTER "INPUT"</b>| load input coordinates only.|| CRYSTAL | <b>FILTER "CONV"</b>| load conventional, not primitive cells.||CSF, SPARTAN|<b>FILTER "noORIENT"</b>| prevents application of the rotation matrix found in the file as the default rotation. (Note: Prior to Jmol 12.0, the orientation saved in the file, which was the way the model was left by the user when the file was saved in Spartan, was not read. Setting this filter returns Jmol to that earlier setting.) ||GAMESS (US) | <b>CHARGE=LOW</b>| indicates to load Lowden charges rather than Mulliken charges.|| GAMESS, GAUSSIAN, GenNBO, Jaguar, NWChem, PSI, and QCHEM | <b>FILTER "xxx"</b>| where "xxx" is a  word on the line Jmol uses to identify a molecular orbital. This allows selective loading of specific types of molecular orbitals --  such as "alpha", "beta", or "NBO" -- for any of these file types, "POPULATION" or "EDMISTON" or "PIPEK" for GAMESS. ("NBO" refers to orbitals generated with the NBO 5.0 option <b>AONBO=P</b>; "EIGEN" or "!NBO" will skip loading of these orbitals.) ||GAMESS, Gaussian, QChem, NWChem| <b>FILTER "NBOCHARGES"</b> | specifies to use atomic partial charges from the NBO calculation. Note that this flag is independent of the <b>NBO</b> filter option. To use both, separate them with a comma: <b>load "myfile.out" FILTER "NBO,nboCharges"</b>.||  JME | <b>FILTER "noMIN"</b>| loads the 2D file, adjusts atoms in preparation for minimization, but does no minimization.|| MOL | <b>FILTER "2D"</b>| indicates to consider the file a 2D file and to apply a automatic hydrogen addition and 2D-to-3D conversion immediately upon loading. "2D-noMIN" does the hydrogen addition but no minimization. || PDB | <b>FILTER "BIOMOLECULE n"</b>| where n is a number &gt; 0 indicating which biomolecule to load. In addition, <b>#&lt;n</b> or any number of <b># n</b> or <b>!#n</b> can be indicated in order to load just a specific subset of biomolecular transformations related to the specified biomolecule. Adding "NOSYMMETRY" along with BIOMOLECULE, indicates that the symmetry transformations in the REMARK 350 BIOMT records should be ignored.|| PDB | <b>FILTER "CONF n"</b> | Loads only a specific configuration based on alternative locations. "CONF 1" loads only atoms in the first configuration -- usually %A and % ("undesignated")||PDB| <b>FILTER "TYPE i,n"</b> |loads customized column data into the atomType property of the atoms. The two numeric parameters are 1-based column number and number of columns. For example: <b>load xxx.pdb filter "TYPE 73,4"</b>.  ||PDB| <b>FILTER "TYPE i,n=XXX"</b>| loads customized column data into the atomType property of the atoms, selecting only those atoms starting with the specified characters in the specified field.||P2N| <b>FILTER "altName"</b>| Use the altName field of the P2N file for a+I542tom names.||PYMOL| <b>FILTER "nosurface"<b>|load a PyMOL file without surface creation, which can be slow. ||PYMOL|<b>FILTER "doCACHE"</b>|caches surface data, creating a JVXL equivalent, specifically useful for JSmol. When using the "doCACHE" option (in the Jmol Java application), file loading should be followed immediately with <b>WRITE PNGJ "xxx.PNG"</b>. The PNGJ file can then be loaded much more efficiently than the PSE file on a web page using JSmol/HTML5.  ||SPARTAN|<b>FILTER "ESPCHARGES"</b>| loads ESP charges rather than Mulliken charges; these are better for displaying electrostatic maps using <b>isosourface molecular map MEP</b>.||]','*6','&quot;filename&quot;','FILTER','&quot;filter specification&quot;','')
newCmd('','11.9.20','','*v+12.0;','Loads a remote file and then also saves it locally. For use with the Jmol application and signed applet only. (Jmol 12.0)','*32','&quot;remoteFilename&quot;','AS','&quot;localFileName&quot;','')
newCmd('.load APPEND','','','*v+11.2;','Adding APPEND to the load command appends a file or a set of files to the current model set without replacing the current model.  Load parameters FILTER , MANIFEST , MODELS , TRAJECTORY , or unit cell/symmetry specification may be included with the APPEND keyword. By default, a new frame is created for each model added. For example:<br /><br />load "myfile.pdb";load APPEND "myligand.xyz";frame *;display 1.1,2.1<br /><br />If <b>set appendNew  false</b> has been issued and only one model is in the appended file, then the file model is added to the currently displayed model, creating no additional frames. Multiple files may be appended if the specified file is a ZIP file. In that case, specifying the manifest in the command after the filename is an option. ','1|2','','')
newCmd('.load [property]','','','*v+11.8;vib','Adding the keyword OCCUPANCY, PARTIALCHARGE, TEMPERATURE, VIBRATION, or XYZ to the load command instructs Jmol to load only that sort of information from a file.<br />[||<b>load OCCUPANCY</b>| Occupancy data are in the form of integers between 0 and 255. Numbers less than 0 are saved as 0; numbers greater than 255 are saved as 255.||<b>load PARTIALCHARGE</b>| Partial charge data are in the form of decimals roughly in the range +/-3.4 x 10<sup>38</sup>.||<b>load TEMPERATURE</b>| Temperature (B-Factor) data are saved as decimals with 0.01 precision within the range -327.68 to 327.67. || <b>load VIBRATION</b> | Vibrational data are in the form of vectors. ||<b>load XYZ</b> | Loads only the coordinates, replacing the current coordinates of an already-loaded model with those in the file to be read. ||]<br /><br /> The data are applied to the currently selected set of atoms based solely on atom position. All standard load parameters are accepted, although many will be ignored, however whereas the default for a normal LOAD operation is to load all files, the default is to read data only from the first model in a multi-model file (or the specific model indicated with an integer after the file name). For each "atom" position and vector that is read, Jmol applies the data to all selected atoms having a unit cell normalized position within {#.set (misc)~loadAtomDataTolerance} (default 0.01) Angstroms of the position read from the file. If the file being loaded contains embedded Jmol script commands, those commands will be processed after the application of the data. For example, <b>load "myfile.struct" {{5 5 1}} PACKED; select _O; set loadAtomDataTolerance -0.2; load VIBRATION "vibs.xyz" 3</b> first loads a set of unit cells from myfile.struct, then applies only to the oxygen atoms the third vibration set found in vibs.xyz. Oxygen atoms in all unit cells will be given data even though the data in vibs.xyz might only be for one unit cell. ','1|2','','')
newCmd('.load DATA','12.0.RC24','','*v+12.0;','Starting with Jmol 12.0, the <b>DATA "model..."</b> and <b>DATA "append..."</b> commands are deprecated in favor of <b>load DATA...</b>. All of the parameter options for the <b>load</b> command are thus also available for the {#.data~} command.  ','1|2','','')
newCmd('.load FILES','','','*v+11.2;','Adding FILES to the load command indicates that a set of files are to be loaded . Filenames must be quoted. Each model encountered is put into a new "frame." No additional parameters other than COORD are allowed (see below). ','1|2','','')
newCmd('','','','','For multiple file loading, all parameters after the first must appear in quotes. Each model is loaded into a new frame, and frames are addressed using a decimal notation -- 1.1 for the first model in the first file, 1.2 for the second model in the first file, 2.1 for the first model in the second file, etc. For example, <b>select (oxygen) and (1.3, 1.4)</b> selects all oxygens in the third or fourth model of the first file loaded. See also {#.set(misc)~set backgroundModel}. The special command <b>load MENU</b> allows loading of a custom Jmol popup menu. ','*1','&quot;filename1&quot;','&quot;filename2&quot;','....','')
newCmd('.load MENU','','','*v+11.4;','The special command <b>load MENU</b> allows loading of a custom Jmol popup menu. ','1|2','','')
newCmd('','','','','File format is that written by Jmol using the {#.show~show MENU} or {#.write~write MENU} command. See examples at {misc/Jmol.mnu~misc/Jmol.mnu} and {misc/test.mnu~misc/test.mnu}. Set the label for a menu to <b>null</b> to remove a menu item or submenu.','*1','&quot;menufile&quot;','')

newCmd('.load MODELS','','','*v+11.8;','The MODELS keyword allows loading of a specific subset of models from a file that contains a model collection. Two syntaxes are available. Note that model numbers refer to the sequence of models encountered in the file, starting with 0, and typically do NOT correspond to the model numbers indicated in PDB MODEL records. Once the models are loaded, they can be accessed using the file.model decimal notation as 1.1, 1.2, 1.3, etc.','1|2','','')
newCmd('','','','','In the first syntax, first and last models along with a "stride" (step) are specified. Models are then read using an equivalent of <b>for (i = first; i <= last; i = i + stride)</b>. Numbers start with 0 for the first model; -1 for <b>last</b> indicates "to the end of the file." For example, <b>load MODELS {{0 10 2}} "..."</b> would load six models, models 0, 2, 4, 6, 8, and 10.','*1','{{first last stride}}','&quot;filename&quot;')
newCmd('','','','','In the second syntax, a list of the specified files to load is given in the Jmol "bitset" syntax. This syntax uses braces within parentheses. Specific models are listed, starting with 0 for the first model. Ranges of models are indicated using colons. For example, <b>load MODELS ({{0 2 4:6}}) "..."</b> loads five models -- models 0, 2, 4, 5, and 6. <b>load MODELS ({{1}}) "..."</b> loads the <i>second</i> model only.','*2','({{i j k:l m...}})','&quot;filename&quot;')
newCmd('','','','*v+12.2;','An additional way to load a specific set of models using an array, starting  with model 1 for the first.','*3','&quot;filename&quot;','[i j k l m...]')
newCmd('.load TRAJECTORY','','','*v+11.4;','Loads a file as a "trajectory", meaning a single model with a series of atom coordinates. (Similar to the sort of multiple-frame animations that the Chime plug-in was able to load.) Loading a file as a trajectory saves substantially on memory requirements, since there is only one set of atoms and bonds. The defining restriction for trajectories is that only one frame can be viewed at a time. Each trajectory is loaded into its own frame as though it were a distinct model, and frames are accessed as usual using the {#.frame~} or {#.model~} command.  In addition, any reference to a specific trajectory, such as <b>select 1.3</b>, switches to that set of coordinates, and {#.measure~Measurements} and the position of {#.cartoon~cartoons} are automatically recalculated when a new trajectory is displayed. Changes to colors in one trajectory effect the same change for all trajectories, since there is really only one model, just different atom positions. Multiple files may be loaded as independent trajectories using APPEND TRAJECTORY in place of simply TRAJECTORY. Jmol 12.0 allows reading of PDB files that contain no MODEL line and are instead simply concatentated versions of the same atoms as trajectories using the TRAJECTORY keyword. Each model must start with atom number 1 for this to work. Starting with Jmol 12.2, the TRAJECTORY keyword also applies to vibrations.','1|2','','')
newCmd('','','','*v+11.4;','Loads the specified file as a trajectory.','*1','&quot;filename&quot;','')
newCmd('','','','*v+11.8;','Loads the specified subset of models from the file as a trajectory. See {#.load MODELS~} for details.','*2','{{first last stride}} or ({{i j k:l m...}})','&quot;filename&quot;')
newCmd('','','','*v+11.8;','The presence of a COORD keyword indicates that the trajectory is to be built from a set of files that includes an AMBER molecular dynamics topology file and associated coordinate files. FILTER is optional, but recommended. For example, <b>FILTER "![WAT]"</b> prevents loading of water molecules. Any number of coordinate files may be specified. Coordinates are loaded as a set of trajectory steps. At least one COORD keyword must be given, and each specification of first, last, and stride must be preceded by a COORD keyword. {{first last stride}} or ({{i j k:l m...}}) is optional, and if not provided defaults to {{0 -1 1}}, which is interpreted as "all trajectory steps from each file." See {#.load MODELS~} for documentation on {{first last stride}} and ({{i j k:l m...}}).','*3','&quot;filename&quot;','FILTER','&quot;filter specification&quot;','COORD {{first last stride}} or ({{i j k:l m...}}) mdcrd::crdfile1')

newCmd('.log','11.9.24','','*v+12.0 -- new;','Logs data to a file. Jmol 12.0 adds a new command specifically for the signed applet and the application. The LOG command works the same as {#.print~} but records the information in a log file. If the string includes "$NOW$", those characters are replaced with the current date and tims. For example: <b>log "$NOW$ " + _version</b>. The string "$CLEAR$" clears the log file. The file to log to must first be designated using <b>set logFile "someName"</b>. This name will be prepended with "JmolLog_" and must not contain any directory path. The file will always be created in the Jar file directory. Note that logging is not possible with the web-based version, even with the signed applet, but signed applet or application running locally can log to a file. In addition to explicit use of the LOG command, two settings, <b>set logCommands</b> and <b>set logGestures</b> allow automatic tracking of commands and gestures (swipe, pinch, zoom, spin) to the designated log file.','1|2','','')


newCmd('.loop',' ','','*v+11.0 -- changes how scripts are interrupted;pause','Causes the script to restart at the beginning, with an optional time delay. See also {#.goto~}.','1','.on','')
newCmd('',' ','','','','1','._time_delay','')
newCmd('.mapProperty',' ','','*v+12.0 -- new;','The mapProperty command allow copying of an atomic property from one set of atoms to another. The operation involves identifying two sets of atoms and associated properties and also a common "key" property such as atomno or resno. If no key is given, atomno is assumed. A shortcut allows quick transfer of atom selection.','0|1','','')
newCmd('',' ','','','For each atom Y in atomExpression2 that matches an atom X in atomExpression1 based on propertyKey, Y.property2 is made to equal X.property1. Property2 must be settable. For example, <b>mapProperty {{1.1}}.temperature {{2.1}}.property_t atomno; color {{2.1}} property_t "rwb" </b>, which would color atoms in model 2.1 (perhaps a {#.plot~}) based on temperature values for model 1.1. ','*1','{{atomExpression1}}.property1','{{atomExpression2}}.property2','propertyKey','')
newCmd('',' ','','','This form of the mapProperty command is a shortcut for <b>mapProperty {{selected}}.selected {{atomExpression}}.selected propertyKey</b>. For example, <b>mapProperty SELECTED {{2.1}}</b> selects atoms in model 2.1 that match atomno with atoms that are already selected.','*2','SELECTED','{{atomExpression}}','propertyKey','')
newCmd('.measure',' ','','*v+11.0 -- adds several new capabilities;','Renders a measurement between the specified atoms. See also {#.set (measure)~}. Two general syntaxes are available. In the older syntax, a series of two to four atom numbers are given, and the appropriate measure (distance, angle, or dihedral angle) is then displayed. The newer, more general syntax is as follows:  <br />  <br />measure RANGE &lt;minValue> &lt;maxValue> ALL|ALLCONNECTED|DELETE (&lt;atom expression>) (&lt;atom expression>) ... <br />  <br />Using this syntax one can specify a <i>set</i> of measurements to define all at once. Note that these sets are embedded {#.atom expressions~atom expressions} that must be enclosed in parentheses. If neither ALL nor ALLCONNECTED is present, only the <i>first matching</i> atom in the entire model set (all frames, so probably the first frame) is matched in each atom expression. When ALL or ALLCONNECTED is specified, all matching criteria in all frames are generated, thus allowing for "animated" measures. In general, this syntax restricts measurements to within the same model. However, measures can also be between two atoms in different frames (different models) as long as each atom expression evaluates to a single, specific atom. (To specify a particular atom in a particular model, use "AND */n", where n is the model number, "ATOMNO=3" by itself, for example, will indicate the third atom in each model/frame, but "ATOMNO=3 and */6" specifies only atom 3 in model 6). If a measurement is made between atoms in different models, both models must be displayed in order for the measurement to appear. A simple way to display two specific models is to use {#.display~display */i or */j}, where i and j are two model numbers. <br /> <br />Thus, for example, <b>measure (*) (*)</b> will measure nothing, because both expressions will simply match the first atom in the first frame. <b>measure allconnected (*/3) (*/3) (*/3)</b> will measure every angle associated with bonds for model 3; <b>measure allconnected (*) (*)</b> will measure every bonded distance in every loaded model; <b>measure all (*) (*)</b> measures all possible interatomic distances in all models (not recommended!). <br /><br />For the applet, using {#.getProperty~getProperty measurementInfo} will then deliver full information relating to all measurements.<br/><br/>The format of the label that accompanies the measurement line can be set for previously created measurements using <b>measure "n:labelFormat"</b>, described below. In addition, the default format (affecting only future-created labels) for measurement labels can be set using {#.set_defaultdistancelabel~set defaultDistanceLabel},  {#.set_defaultanglelabel~set defaultAngleLabel}, and  {#.set_defaulttorsionlabel~set defaultTorsionLabel}. <br /><br />Starting in Jmol 13.2, you can specify units for a specific label using "//" at the end of the format followed by a unit designation such as A or nm. Special units are available for NMR data arising from Magres files: "2://dc_hz" or "2://dc_khz" or "2://khz" for calculated dipolar constant in Hz or kHz and "2://isc_hz" or "2://hz" for calculated J-coupling constants. These units are also available for the measure() function as, for example, <b>x = measure({{a}},{{b}},"isc_hz")</b>.','0|1|2|3|4','','')
newCmd('',' ','','','Turns on and off the distance, angle, dihedral measurement labels and measurement lines. (To turn off just the labels, use {#.set measurement~set measurement OFF} ','*11','._on_off{"ON"}','')
newCmd('','','','*v+11.0;','Changes all previously defined measurement labels of a given type (n = 2, 3, or 4) to the indicated format. The default label is "%VALUE %UNITS" for all types. Starting with Jmol 13.0, specifically for distances, the suffix //A will use angstroms for the default label for distance measurements. Other suffixes include //nm, //pm, and //vdw.  Also available is %#(percent number-sign), which gives the 1-based number of the measurement. Atom information can be included as for {#.label~labels}, adding 1 or 2 to the format code to indicate which atom. So, for example, <b>measure "2:%a1 -- %a2 distance = %0.0VALUE"</b> delivers the two atom names along with the value of the measurement rounded to the nearest integer with no units indicated. Starting with Jmol 12.2, specific units for the measurements can be given after two slashes: <b>measure "2:%a1 -- %a2 distance = %0.1VALUE //A"</b> would display the distance rounded to the nearest 0.1 Angstrom. ','*14','&quot;n:labelFormat&quot;','')
newCmd('',' ','','','Two atoms specify a distance measurement with an optionally given format. This syntax selects the first atom in the currently visible frame set.','*2','(integer)','(integer)','&quot;labelFormat&quot;','')
newCmd('',' ','','','Three atoms specify an angle measurement. The format is optional. This syntax selects the first atom in the currently visible frame set.','*3','(integer)','(integer)','(integer)','&quot;labelFormat&quot;')
newCmd('',' ','','','Four atoms specify a dihedral angle measurement. The format is optional. This syntax selects the first atom in the currently visible frame set.','*4','(integer)','(integer)','(integer)','(integer)  &quot;labelFormat&quot;')
newCmd('',' ','','*v+11.0;','Show the distance, angle, or dihedral angle formed by the FIRST atom in each atom expression. The format is optional. ','*15','(two to four atom expressions, each in parentheses)','&quot;labelFormat&quot;')
newCmd('',' ','','*v+11.0;','Show the distance, angle, or dihedral angle formed by ALL atoms in the first expression with ALL atoms of each additional atom expression. The format is optional. ','*6','ALL','(two to four atom expressions)','&quot;labelFormat&quot;','')
newCmd('',' ','','*v+11.0;','Show the distance, angle, or dihedral angle formed by ALL atoms in the first expression with ALL atoms of each additional atom expression, provided they form a connected set. The format is optional. ','*7','ALLCONNECTED','(two to four atom expressions)','&quot;labelFormat&quot;','')
newCmd('',' ','','*v+11.0;','Deletes all measurements.','*81','DELETE','')
newCmd('',' ','','*v+11.0;','Deletes a specific measurement, in order of their creation, starting with 1.','*82','DELETE','(integer)')
newCmd('',' ','','*v+11.0;','Deletes all matching distance, angle, or dihedral angle measurements that are currently defined based on the atom expressions.','*83','DELETE','(two to four atom expressions)')
newCmd('','12.1.26','','*v+12.2;','Lists all measurements. (Same as {#.show~show measurements}.)','*84','LIST','')
newCmd('',' ','','*v+11.0;','Adding RANGE and two decimal numbers modifies the above commands to limit the measurements created or deleted to only those within this specific range of values in Angstroms (distance) or degrees (angles). The word "RANGE" itself is optional but recommended.','*90','RANGE','(decimal)','(decimal)','ALL|ALLCONNECTED|DELETE (two to four atom expressions, each in parentheses)')
newCmd('','11.9.12','','*v+12.0;ticks','Creates a measure line with ticks along it. There are three levels of ticks - major, minor, and "subminor." Only the major ticks have labels. Which of these tick levels are displayed and the distance between ticks depends upon the parameter that takes the form of a point. The optional keyword FORMAT allows formating of the labels for the major ticks. These are based on an array of strings given after the FORMAT keyword. If the array is shorter than the number of ticks, the formats in the array are repeated. Following that, the optional keyword SCALE allows setting the scale either for each axis direction independently {{scaleX, scaleY, scaleZ}} or overall (as a decimal number). An optional keyword FIRST allows setting of the initial value of the measure. Finally, two points must be indicated.  ','*63','TICKS X|Y|Z','{{major,minor,subminor}} FORMAT ["%0.2f", ...] ','SCALE {{scaleX, scaleY, scaleZ}} | x.xx','FIRST x.xx {{point1}} {{point2}}')


newCmd('.menu','13.0.RC4','','*v+13.0 -- new;','Accepted only as the sole contents of a script, the <b>menu</b> command pops up the context menu for the user as though they had right-clicked the applet or clicked on the Jmol logo.','1','','')

newCmd('.meshribbon',' ','structure','disp','A mesh ribbon is similar to a strand, but is more the quality of a loosely woven fabric. ','0|1','','')
newCmd('',' ','','','','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns meshribbon rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','',' Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._mesh_ribbon_radius','')
newCmd('.message',' ','','*v+11.0 -- NEW;math','Sends a string of text to the messageCallback function (for the applet). The {#.set_MessageCallback~} command can be used to set this JavaScript function. The message is also entered into the scriptStatus queue. Variable values and math expressions can be included in messages and echos. These use the following syntax: <b>message my variable = @{{variablename}}</b>, provided a variable with that name exits. Note that the braces are required -- <i>message my variable = @x</i> will simply deliver "my variable = @x". The @{{...}} syntax is also supported in this context: <b>message the fraction 3 / 2 is @{{ 3 / 2 }}</b>. Basically, any expression that can appear in a {#.print~} command can appear in @{{...}} in a message or echo. For example: <b>message @{{ {{_N and connected(2,_H)}}.size }} NH2 groups are present</b>. If in an echo that is displayed, these variables are updated dynamically. Since the Jmol application can be run "headless" -- with no display -- using the -ions set of flags, you can use a designed message command to deliver model information to other programs, however the {#.print~} command has a simpler syntax and is more flexible.','1','(string)','')

newCmd('.minimize',' ','','*v+13.0 -- replaces the default UFF calculation with MMFF94, where applicable.;','Minimizes the structure using the MMFF94 force field (Jmol 13.0; T. A. Halgren,  J. Comp. Chem. 5 & 6, 616-641 (1996) or an adapted UFF force field (Rappe, A. K., et. al.; J. Am. Chem. Soc. (1992) 114(25) p. 10024-10035). Starting with Jmol 13.0 the default force field is MMFF94, but the UFF force field will automatically be used if atoms are not fo types recognized by MMFF94 or if <b>set forceField "UFF"</b> is issued. Generally the minimization runs in its own thread. If it is desired to wait until the process has completed prior to continuing, use <b>set useMinimizationThread false</b>. You  can retrieve a detailed report of the minimization energy using the "high debug" setting and then either <b>show minimization</b> or <b>x = script("show minimization");print x</b>:<br /><br /><b>set loglevel 6<br />minimize steps 0<br />x = script("show minimization")<br />set debug off<br />print x','0','','')
newCmd('','','','*v+11.6;','Carries out a minimization using the default convergence criterion specified by {#.set(misc)~set minimizationCriterion} or until the maximum number of steps specified by {#.set(misc)~set minimizationSteps} is reached. If {#.set(misc)~set minimizationRefresh} is TRUE, then the model refreshes after each minimization step, producing a sort of animated minimization.','*1','','')
newCmd('','11.9.19','','*v+12.0;','First adds hydrogen atoms, then minimizes the structure.','*2','ADDHYDROGENS','')
newCmd('','','','','Stops a running minimization.','*3','CANCEL','')
newCmd('','11.5.24','','','Stops a running minimization and clears all constraints ','*4','CLEAR','')
newCmd('','11.5.24','','*v+11.6;','Clears any current constraints.','*5','CONSTRAINT','CLEAR')
newCmd('','11.5.24','','*v+11.6;','Constrains the distance, angle, or dihedral angle invoving two to four atoms. If a given atom expression involves more than one atom, only the first atom in the set is made part of the constraint.','*6','CONSTRAINT','(two to four atom expressions)','(decimal)','')
newCmd('','11.5.23','','*v+11.6;','Overrides the setting of {#.set_minimizationCriterion~minimizationCriterion} for this minimization.','*7','CRITERION','')
newCmd('','11.5.24','','*v+11.6;','Do no minimization -- just calculate the energy','*8','ENERGY','')
newCmd('','11.5.24','','*v+11.6;','Specifies an atom set to keep in fixed position during this minimization or future minimizations.','*9','FIX','._atom_expression')
newCmd('','','','','Minimizes only the specified atom set. (Any previous or additional setting of FIX is respected.)','*91','SELECT','._atom_expression')
newCmd('','','','*v+11.6;','Overrides the setting of {#.set(misc)~minimizationSteps} for this calculation. ','*92','STEPS','(integer)')
newCmd('','','','','Same as CANCEL.','*93','STOP','')


newCmd('.mo',' ','','*v+11.2 -- MO SLAB, MO POINTS;iso','<img src="img/mo.png"  alt="benzene HOMO (Spartan)"  />The <b>MO</b> (molecular orbital) command displays molecular orbitals contained in a variety of file formats. One simply loads the file, then selects which orbital to display with <b>MO n</b> where "n" is the number of the molecular orbital to display. (With some file formats, you may need to use the {.#frame~frame or model} command to call up the specific model having the MOs.) Option changes take effect immediately with the currently displayed orbital and stay in effect for later MO commands until a new file is loaded. <br /><br />Note that ONE molecular orbital is allowed per model (that is, per frame). The {#.isosurface~} command can be used for more advanced molecular orbital display options or for displaying several planes or orbitals simultaneously. Default rendering prior to Jmol 11.2 is FILL; starting with Jmol 11.2 MESH NOFILL FRONTONLY. Where indicated below, starting with Jmol 13.2, adding the SQUARED keyword generates the linear combination of all degenerate orbitals if one is indicated (by a number, or HOMO, LUMO, NEXT, or PREV) or the specified set of orbitals in the case of a linear combination, squaring the wave functions prior to adding. In general, this results in showing the contribution of this set of orbitals to the overall total electron density and the symmetry of that density.','1|2','','')
newCmd('',' ','','*v+11.0;','Turn on/off the molecular orbital.','*10','._on_off{"ON"}','')
newCmd('',' ','','*v+11.0;','Selects the specific molecular orbital to dispay, starting with the number 1. Starting with Jmol 12.2, if this number is negative, the colors of the phases are reversed.','*310','(integer)','[SQUARED]')
newCmd('','13.1.9','','*v+11.0;','Calculates the sum of the squares of all occupied wave functions, effectively showing overall electron density.','*311','DENSITY','')
newCmd('',' ','','*v+12.2;','Displays a linear combination of molecular orbitals based on a set of coefficients and orbital numbers, for example: <b>mo [0.5, 20, 0.5, 21]</b>. Coefficients will automatically be normalized such that SUM<sub>i</sub>(c<sub>i</sub>*c<sub>i</sub>) = 1.00. If variables are involved, one needs @{{...}}. Thus, <b>load c6h6.smol; for (f = 0; f <= 10; f++) {{ mo @{{[f/10.0, 20, (10-f)/10.0, 21]}}; refresh }}</b>.(Jmol 12.2) ','*32','[c1 n1,c2 n2, c3 n3,...]','[SQUARED]')

newCmd('',' ','','*v+11.2;','Selects the highest doubly- or singly-occupied molecular orbital, optionally an orbital +/-n from this orbital. (Only applicable to data sets that have orbital occupancies indicated in the file.) ','*61','HOMO','[+/-n]','[SQUARED]','')
newCmd('',' ','','*v+11.2;','Selects the lowest unoccupied molecular orbital, optionally an orbital +/-n from this orbital. (Only applicable to data sets that have orbital occupancies indicated in the file.)','*62','LUMO','[+/-n]','[SQUARED]','')
newCmd('','12.0.RC6','','*v+12.0;','Specifies the model for this command; must be followed by an orbital specification such as <b>HOMO</b> or <b>12</b>.','*64','MODEL','n or x.y')
newCmd('',' ','','*v+11.0;','Displays the next MO in the file using the same characteristics as the currently displayed one.','*65','NEXT','[SQUARED]')
newCmd('',' ','','*v+11.0;','Displays the previous MO in the file using the same characteristics as the currently displayed one.','*85','PREVIOUS','[SQUARED]')
newCmd('',' ','','*v+11.0;','Goes back to full orbital display rather than a planar slice.','*68','NOPLANE','')
newCmd('',' ','','*v+11.0;','Indicates that what is desired is a color-mapped planar slice through the orbital using one of the ways of expressing a {#.plane expressions~plane}. ','*80','PLANE','plane_expression')
newCmd('','12.1.48','','*v+12.2;','Produces a "pointilist" probability-based visualization of a molecular orbital using the specified number of points and an optional random seed. ','*81','POINTS','numberOfPoints','randomSeed','')
newCmd('',' ','','*v+11.0;','Colors the orbital one specific color','*40','COLOR','._colorRGB')
newCmd('',' ','','*v+11.0;','Colors regions where the wave function is less than zero the first color; regions where the wavefunction is greater than zero the second color.','*51','COLOR','._colorRGB','._colorRGB','')
newCmd('',' ','','*v+11.0;','Delete the molecular orbital.','*60','DELETE','')
newCmd('',' ','','*v+11.0;','Sets the cutoff value for the isosurface that defines the orbital. This number may be dependent upon the computational package used to generate the orbitals. Values in the range 0.005 - 0.05 may need to be experimented with in order to get the best display. Values closer to zero lead to surfaces further from the atoms (larger orbitals).  Both positive and negative cutoffs are allowed. A positive number indicates to use both positive and negative cutoffs. Adding an explicit "+" sign before the number indicates that only the positive part of the surface is desired.','*55','CUTOFF','(decimal)')

newCmd('',' ','','*v+11.0;','Sets the resolution of the isosurface rendering in "points per Angstrom". Higher resolution leads to smoother surfaces and more detail but carries the penalty of slower surface generation. Typical values for molecular orbitals are 4-10 points per Angstrom.','*92','RESOLUTION','(decimal)')
newCmd('','12.1.50','','*v+12.2;','MO <b>SLAB</b> provides an extensive mechanism allowing for the trimming of a molecular orbital using a wide variety of methods. MOs can be "slabbed" based on planes, spheres, bounding boxes, unit cells, proximity to a set of atoms, within a range of values, or a dynamic z-plane relative to the user. The section removed can be removed completely or, using <b>MO SLAB TRANSLUCENT</b>, left as a translucent "ghost" surface. An MO may be slabbed any number of times in any number of ways. The slabbing is reversible using <b>MO SLAB NONE</b> for general slabbing, and using <b>MO SLAB OFF</b> for dynamic z-plane slabbing. Options include: [|| {#.[plane expression]~(plane expression)} | a plane. Note that plane definitions starting with "-" switch sides of the plane ||BOUNDBOX | within the currently defined {#.boundbox~bounding box} ||UNITCELL | within the currently defined {#.unitcell~unit cell} ||WITHIN x.y {{atom expression}} | within the given distance of the specified atom set ||WITHIN x.y (point) | within the given distance of the specified point (a spherical slab) ||WITHIN RANGE x.x y.y | within the value range x.x and y.y.  ||]','*93','SLAB','(parameters)')
newCmd('',' ','','*v+11.0;','Sets the format of the orbital title appearing in the upper left corner of the applet. Special format characters include: <br/>[|| %E|energy||%F|filename || %I | molecular orbital number || %M | model number || %N | total number of molecular orbitals || %O|occupancy ||%S|symmetry||%U|energy units||  &nbsp; &nbsp; &#124; &nbsp;  &nbsp;| (vertical bar) new line|| &nbsp; &nbsp;? &nbsp; &nbsp;|(at the beginning of a line) indicates to disregard line if no data for that line are present in the file||]<br/> If a formatted item is not indicated in the file, then it is left blank. The default title is "%F &#124; Model %M  MO %I/%N &#124; Energy = %E %U &#124; ?Symmetry = %S &#124; ?Occupancy = %O". The command <b>MO titleFormat ""</b> may be used to show no title. ','*95','TITLEFORMAT','&quot;format&quot;')

newCmd('.modulation','13.3.4','','*v+13.4;','The MODULATION command, introduced in Jmol 13.4, allows visualization of {http://www.ccp14.ac.uk/ccp/web-mirrors/crys-methods/Project/modulate/modulate.html~incommensurately modulated strutures} that have been read from CIF or JANA2006 files. ','1|1','','')
newCmd('','13.3.4','','*v+13.4;','Turns modulation on','*1','ON','')
newCmd('','13.3.4','','*v+13.4;','Turns modulation off (displaying the average structure)','*2','OFF','')
newCmd('','13.3.4','','*v+13.4;','Specifies the offset parameter "t"','*3','t','')
newCmd('','13.3.4','','*v+13.4;','Animates a once-through sequence of t from t1 to t2.','*4','t1','t2')
newCmd('',' ','','*v+13.4;','Sets the modulation play-back speed in frames per second. May be less than 1.','*5','FPS','(decimal)')



newCmd('.move',' ','','anim;camera','The move command provides powerful animation capabilities. It effectively changes the camera position, allowing you to specify rotations, zooming, and translations to be performed in a specified period of time. xRot, yRot, and zRot are rotations about the cartesian axes in degrees. Zoom specifies a zoom factor, xTrans, yTrans, and zTrans are translations in the range -100 to 100. If you do not know what slab is, just put in a zero. see the slab command for more information.  This command has been superceded by the <b>moveTo</b> command.','11','._drotx','._droty','._drotz','._dzoom;+._dtransx;+._dtransy;+._dtransz;+._dSlab;+._floatSecondsTotal;+._move_fps{30};+._move_maxAccel{5}')
newCmd('.moveto',' ','','*v+11.2 -- adds navigation capability;anim;nav;camera','<img src="img/moveto.gif"  alt="" />The moveto command rotates the molecule to a predefined orientation, as though the camera were moving smoothly to a new position. Two formats can be used. In each, the first (optional) parameter specifies the number of seconds during which the molecule should rotate smoothly from the current orientation to the new orientation.  A 0 for this first parameter specifies an instantaneous reorientation. Starting with Jmol 12.0, if the axis is {{0 0 0}} and the degrees are 0, then the molecule is not reoriented during the operation.  In conjunction with "show orientation" this command allows reading and restoring specific user-specified orientations.','1|2','','')
newCmd('',' ','','*v+11.0;','A simple use of moveTo just has six optional directions. ','*3','timeSeconds','FRONT|BACK|LEFT|RIGHT|TOP|BOTTOM')
newCmd('','','','*v+11.2;','In the second option, the second parameter is a coordinate {{x, y, z}} defining the axis relative to the default orientation about which the molecule should be rotated. The third parameter is the counterclockwise (right-hand) rotation in degrees about this axis. "moveto 0 {{0 0 1}} 0" rotates the model to the default orientation (equivalent to "reset"). If the angle parameter is 0 but any one of x, y, or z is nonzero, then no reorientation occurs (because the axis has been specified, but the rotation is 0 degrees). Following these parameters is the zoom setting in percent, the X- and Y-positions of the rotation center on the screen, as percent of width and height, respectively. The actual molecular coordinate of the rotation center along with the rotation radius (which determines the magnification associated with <b>ZOOM 100</b>) are next. The final parameters define the navigation center molecular coordinate, its X- and Y- position on the screen in percent, and the depth of the navigation point in percent of model depth (100 = front, 0 = rear). In conjunction with "show/save/restore orientation" this command allows reading and restoring specific user-specified orientations.','*4','timeSeconds','{{x y z}}','degrees','zoomPercent;+transX;+transY;+{{x y z}};+rotationRadius;+navigationCenter;+navTransX;+navTransY;+navDepth')
newCmd('','11.1.36','','*v+11.2;','If the zoom setting prior to translation positions is 0, and an atom expression is used for the point, then the moveTo can be designed to automatically zoom to the scale that would fill the screen with that set of atoms. The optional zoom adjustment is in the form +n, -n, *n, or /n, as for {#.zoomTo~}.','*6','timeSeconds','{{x y z}}','degrees','0;+transX;+transY;+(atom expression);+0;+zoomAdjustment;+navigationCenter;+navTransX;+navTransY;+navDepth')
newCmd('','11.1.36','','*v+11.2;','If no translation is involved, then there is also no need for the zoom setting of 0 prior to the atom expression.','*7','timeSeconds','{{x y z}}','degrees','(atom expression);+0;+zoomAdjustment;+navigationCenter;+navTransX;+navTransY;+navDepth')
newCmd('','11.9.8','','*v+11.2;','Reorients the model to the orientation specified by the quaternion given in {{x y z w}} format. ','*80','timeSeconds','QUATERNION','{{ quaternion }}','')
newCmd('','11.9.8','','*v+11.2;','Reorients the model to the orientation specified by the quaternion specified by the atom expression and the current setting of <b>quaternionFrame</b>.','*81','timeSeconds','QUATERNION','{{ atom expression }}','')
newCmd('','11.9.8','','*v+11.2;','Reorients the model to the orientation specified by a molecular frame quaternion, as, for example, retrieved by <b>q = quaternion({{resno=30}})</b> and then used in <b> moveto quaternion MOLECULAR @q</b>. Note that because this quaternion refers to the rotation required to transform the reference frame to the specified molecular frame, in order to "move to" this orientation, one must use its inverse  (bring the molecular frame to the reference frame, <b>q_orientation = !q_molecular</b>). The keyword MOLECULAR simply applies this inversion. ','*82','timeSeconds','QUATERNION','MOLECULAR','{{ quaternion }}')




newCmd('','11.9.24','','*v+11.2;','Stops an asychronous moveto operation that has been started after setting <b>set  waitForMoveTo FALSE</b>. [Jmol 12.0]','*8','STOP','')
newCmd('.navigate','11.1.1','','*v+11.2 -- new;nav','The <b>navigate</b> command allows exploring of the model from the perspective of an airplane capable of flying through the model. In each case, an optional time in seconds determines the  time to reach the objective. If no time is given, the change occurs over two seconds. (In the case of PATH, this is the time to each point along the path, not the total path.) Subcommands can be grouped in the same command by separating them with "/", for example: <b>navigate 2 DEPTH 30 / 5 ROTATE 180 / DEPTH 20 / TRANSLATE X 10</b>.','0|1','','')
newCmd('','11.1.1','','*v+11.2;','Bring the observer to the specifed molecular coordinate, which may be fractional in the case of crystal structures.','*11','timeSeconds','CENTER','{{x y z}}','')
newCmd('','11.1.1','','*v+11.2;','Bring the observer to the geometric center of the set of atoms specified in parentheses.','*12','timeSeconds','CENTER','{{ atom expression }}','')
newCmd('','11.1.1','','*v+11.2;','Bring the observer to the geometric center of the points corresponding to the indicated {#.draw} object.','*13','timeSeconds','CENTER','$object','')
newCmd('','11.1.1','','*v+11.2;','Bring the observer forward or backward to the depth indicated as a percent from back (0) to front (100) of the model. Values can be negative, in which case until a rotate command is given, only screen background will be seen. Values greater than 100 put the observer outside the model. ','*20','timeSeconds','DEPTH','percent','')
newCmd('','11.1.1','','*v+11.2;','Follows the Hermite path given by the specified {#.draw~} object (such as a line, curve, or arrow). This allows easy dynamic development of paths using a predefined draw object followed by <b>set picking draw</b> and <b>show draw</b>. ','*32','timeSeconds','PATH','$object','')



newCmd('','11.1.1','','*v+11.2;','Follows a Hermite path defined by a set of nodes expressed in terms of any combination of atom sets, draw objects, or coordinates.','*33','timeSeconds','PATH','(any combination of coordinates, atom expressions, and objects)','')
newCmd('','11.1.1','','*v+11.2;','Rotates around the X axis at the navigation center.','*42','timeSeconds','ROTATE','X','degrees')
newCmd('','11.1.1','','*v+11.2;','Rotates around the Y axis at the navigation center.','*44','timeSeconds','ROTATE','Y','degrees')
newCmd('','11.1.1','','*v+11.2;','Rotates around the Z axis at the navigation center.','*46','timeSeconds','ROTATE','Z','degrees')

newCmd('','11.1.1','','*v+11.2;','Navigates along the trace of a protein or nucleic acid. The "ride" can be changed depending upon the settings of {#.set(structure)~set sheetsmoothing} and {#.set(structure)~set tracealpha}.','*61','timeSeconds','TRACE','(atom expression)','')
newCmd('','11.1.1','','*v+11.2;','Translates the navigation screen offset to the specified positions expressed as percent of width (X) and height (Y) of the applet/application window.','*62','timeSeconds','TRANSLATE','x.xx','y.yy')
newCmd('','11.1.1','','*v+11.2;','Translates the navigation screen offset horizontally by the specified percent of applet/application window width.','*64','timeSeconds','TRANSLATE','X','x.xx')
newCmd('','11.1.1','','*v+11.2;','Translates the navigation screen offset vertically by the specified percent of applet/application window height.','*66','timeSeconds','TRANSLATE','Y','y.yy')
newCmd('','11.1.1','','*v+11.2;','Translates the navigation screen offset to the screen position corresponding to the given molecular coordinate.','*68','timeSeconds','TRANSLATE','{{x y z}}','')
newCmd('','11.1.1','','*v+11.2;','Translates the navigation screen offset to the screen position corresponding to the geometric center of the specified atoms.','*70','timeSeconds','TRANSLATE','(atom expression)','')
newCmd('','11.1.1','','*v+11.2;','Translates the navigation screen offset to the screen position corresponding to the center of the vertices of the specified draw object.','*72','timeSeconds','TRANSLATE','$object','')


newCmd('.parallel/process','','','*v+12.0;','Jmol 12.0 introduces parallel processing for Jmol. Jmol 12 will be able to use multiple processors on a multiple-CPU machine. Basically what you can do is to tell Jmol which statements in a script you want to run in parallel, and it will do that. The way this is done is to create a function using the keyword <b>PARALLEL</b> in place of the keyword <b>function</b>. Within that block of code, any group of commands surrounded by <b>PROCESS{{ }}</b> will be collected and run in parallel just before Jmol returns from the function. Any commands NOT within these sets will be run BEFORE any <b>PROCESS</b> commands. For example:[||parallel twoIsosurfaces(model1, model2) {{<br>&nbsp;&nbsp;var x = 1<br>&nbsp;&nbsp;process {{<br>&nbsp;&nbsp; isosurface s1 model @model1 molecular; color isosurface red<br>&nbsp;}}<br>&nbsp;&nbsp;process {{ <br>&nbsp;&nbsp; isosurface s2 model @model2 molecular; color isosurface green \n  }}<br>&nbsp; x = 2<br>}}<br>&nbsp;&nbsp;<br>load files "1crn.pdb" "1blu.pdb"<br>twoIsosurfaces("1.1", "2.1")<br>frame *||]In this case, the variable x will be 2 BEFORE the isosurfaces are created. See also {examples-11/data/multi-mo.txt~multi-mo.txt} and {examples-11/data/multi-surface.txt~multi-surface.txt}, and {examples-11/data/multiProcessTest.txt~multiProcessTest.txt}. You can selectively turn on and off the use of multiprocessors using {#.set_multiprocessor~set multiProcessor}. If this setting cannot be set true, then it means you do not have a multiprocessor machine. Not all processes will work; currently the only implemented parallel processes are for isosurfaces and molecular orbitals. The parallel capability of Jmol should be considered experimental at this time.','1|1','','')

newCmd('.pause',' ','','*v+11.8 -- adds stepping capability;pause','Pauses script execution until {#.resume~}, {#.step~}, {#.quit~}, or {#.exit~} is issued. Any text on the command line after <b>pause</b> is send to the user as a message when the script pauses. During the paused condition, commands can be entered from the Jmol application console, and they will be executed. The next command to be executed can also be checked by issuing <b>?</b> from the console.','1','message','')

newCmd('.[plane expressions]',' ','','*v+11.2 -- broader general use of planes;iso;atoms','Several commands in Jmol accept parameters that define planes. The commands {#.depth~},  {#.draw~}, {#.isosurface~},  {#.mo~}, and  {#.slab~} all have the PLANE option, and {#.depth~},  {#.draw~} {#.isosurface~}, and  {#.slab~} also have the HKL option for defining planes using Miller indices. In addition, {#.select~}, {#.restrict~}, {#.display~}, and {#.hide~} as well as many other commands accept {#.atomexpressions~atom expressions}, and these may include one or more WITHIN() function. The method of describing planes in Jmol is defined below:[||<b>hkl</b>|Miller indices are simply indicated by enclosing them in brackets, usually as fractions: {{h k l}}. For example, <br /><br /><b>isosurface hkl {{0 1/2 1/2}}</b><br /><br />. ||<b>plane</b>|A plane is defined in Jmol in one of five ways: (a) listing three points, (b) giving the parametric equation ax + by + cz + d = 0, (c) referencing an already-defined  {#.draw~} or {#.isosurface~} object that describes a plane, (d) using the strings "xy", "xz", or "yz", or (e) using a simple single-variable equation such as "x=3" or "y=-2". Points may be embedded atom expressions (that is, atom expressions surrounded by parentheses), for which the center is used, Cartesian coordinates expressed as {{x,y,z}}, or crystallographic fractional coordinates indicated using "/" in at least one coordinate: {{0,1/2,1/2}}. Commas are optional. Any combination of these three point types may be used. For example: <br /><br /><b>display within(0,plane,@{{plane({{atomno=3}}, {{0 0 0}}, {{0 1/2 1/2}})}})</b><br /><br /> The parametric equation ax + by + cz + d = 0 is expressed as {{a b c d}}. <br /><br />Planes based on <b>draw</b> and <b>isosurface</b> objects are first defined with an ID indicated, for example: <br /><br /><b>draw plane1 (atomno=1) (atomno=2) (atomno=3)</b> <br /><br /> After that, the reference <b>$plane1</b> can be used anywhere a plane expression is required. For instance,  <br /><br /><b>select within(0,plane, $plane1)</b>  <br /><br />or  <br /><br /><b>isosurface PLANE $plane1</b> <br /><br /> These objects can be defined and then hidden from the user using <b>draw off</b> or <b>isosurface off</b>.  For selection purposes the plane is considered of infinite extent even if it only shows up as a small triangle or quadrilateral. ||] ','0','','')

newCmd('.plot','12.0.RC23','','*v+12.0 replaces quaternion and ramachandran commands, and adds new property options;plot, draw','Jmol can create a variety of Ramachandran plots and quaternion maps. In addition,  Jmol 12.0 introduces the capability to quickly generate simple plots relating two or three atom properties. A new frame is created. In the case of property and Ramachandran plots, this frame has its own independent orientation; in the case of quaternion maps, rotation of the map is synchronized with rotation of the model. Only one data frame may be visible at a time.  Also related to this command for quaternions and Ramachandran plots is the {#.draw~} command, which allows depicting of these measures on the model itself. ','0|1','','')
newCmd('','','','plot','Creates the quaternion representation of the protein or nucleic acid in the three dimensions <i>not</i> given by the specified axis.','*21','QUATERNION','w, x, y, or z')
newCmd('','','','plot','Creates the quaternion difference representation of the protein or nucleic acid in "w" projection. "a", for <i>absolute</i>, produces a visualization of q[i] / q[i-1], which defines the helical axis of a helix or beta-pleated sheet; "r", for <i>relative</i>, produces a visualization of q[i-1] \\ q[i], which defines the relative rotation of the (i)th residue from the perspective of the (i-1)th residue, defining the pitch of the helix. The default visualization for "r difference" (utilizing <b>set quaternionFrame "c"</b>) for a typical protein is an ellipse having a major axis tilted at 36 degress from the quaternion X axis and minor axis along the quaternion Z axis. This ellipse represents a composite rotation from one amino acid to the next involving a rotation about the q[i] frame Z axis (perpendicular to the N-CA-C plane)  of approximately 180 - 110 degrees ((180-110)/2 = 36 degrees), followed by a rotation about the q[i] frame X axis (CA-C bond) of 180 + psi[i-1] + phi[i] degrees.','*22','QUATERNION','a,r','DIFFERENCE','')
newCmd('','','','plot','Creates a visualization of dq[i+1] / dq[i], where <b>dq</b> is defined as q[i+1] / q[i] for "a" or q[i] \\ q[i+1] for "r". In the case of "r difference2", for most amino acids this is a point relatively close to the quaternion x axis. ','*23','QUATERNION','a,r','DIFFERENCE2','')
newCmd('','','','plot','Creates a Ramachandran plot for a the currently displayed protein model.  Starting with Jmol 13.0, values of PHI and PSI are available directly from the points involved using, for example, <b>averageHelixPsi = {{2.1 and helix}}.psi</b>.','*31','RAMACHANDRAN','')
newCmd('','','','plot','Creates a "relative" Ramachandran plot for a the currently displayed protein model with the third axis being theta, a measure of "straightness" as calculated using a Ramachandran angle approximation: straightness[i] = 1-acos(abs(cos(theta[i] / 2)))/90, where i refers to a residue and theta is the angle associated with the relative quaternion second derivative dq[i]\\dq[i-1]. Theta can be approximated using Ramachandran angles for "C" straightness simply as theta =approx= psi[i] - psi[i-1] + phi[i+1] - phi[i] (Hanson and Kohler, unpublished results) and for "P" straightness as follows:  If deltaPhi = phi[i+1] - phi[i], deltaPsi = psi[i+1] - psi[i], and alpha = 70<sup>o</sup>, then cos(theta/2) =approx= cos(deltaPsi/2)cos(deltaPhi/2) - sin(alpha)sin(deltaPsi/2)sin(deltaPhi/2). (Hanson and Braun, unpublished results.) Residues close to the phi-psi plane have low values of theta and thus high values of straightness, meaning they are in a region of relatively high structural regularity --- usually helices or sheets.','*32','RAMACHANDRAN','r')
newCmd('','','','plot','Creates a 2D plot relating two atom properties for the currently selected atom set. For example: <b>select *;plot properties atomno temperature</b>. Additional optional parameters MIN {{x y z}} and MAX {{x y z}} may follow. Setting these values sets the scale of the graph within Jmol and truncates data values outside of this range. (The z value in this case is ignored.)','*11','PROPERTIES','property1','property2','')
newCmd('','','','plot','Creates a 3D plot relating two atom properties for the currently selected atom set. For example: <b>select *.CA;plot properties phi psi resno</b>. Additional optional parameters MIN {{x y z}} and MAX {{x y z}} may follow. Setting these values sets the scale of the graph within Jmol and truncates data values outside of this range.','*12','PROPERTIES','property1','property2','property3')


newCmd('.pmesh',' ','','*v+11.8 merges pmesh and isosurface commands;iso','The pmesh command is deprecated. See {#.isosurface~} for information about the pmesh file format options.','0|1|2','','')

newCmd('.polyhedra',' ','polyhedra1,polyhedra2','*v+11.0 -- expanded capability, new algorithm;iso','Jmol will form a wide variety of polyhedral structures. The <b>polyhedra</b> command can be used for either construction of new polyhedra or modification of existing polyhedra.','0|1|2','','')
newCmd('',' ','','','Used for construction of new polyhedra, parameters fall into five subsets:<br /><br /><b>polyhedra [number of vertices] [basis] [selection sets] [display options]</b><br /><br />In order to construct polyhedra, at least one of the first two of these subsets must be present -- the number of vertices or the basis. In addition, some display options (ON, OFF, DELETE -- see below) are incompatible with polyhedron construction.','TEXT','Polyhedron construction','')
newCmd('',' ','','','Polyhedra involve a central atom and from 3 to 20 outer "vertex" atoms. The number of required vertex atoms is specified by an integer (<b>polyhedra 6</b>). More than one number can be specified, separated by a comma or space character (<b>polyhedra 4,6</b>). If no number of vertices is indicated, any number from 3 to 20 is assumed.','TEXT','[number of vertices]','')
newCmd('',' ','','','Polyhedra can be formed based either upon the number of atoms bonded to the central atom (<b>polyhedra 4 BONDS</b>) or upon the number of atoms within a specified distance from the central atom (<b>polyhedra 4,6 RADIUS 2.0</b>). (The keyword "RADIUS" is optional, but if it is absent the radius must be indicated as a decimal number, not an integer, so as to distinguish it from a vertex count.) If no radius is indicated, BONDS is assumed. If both BONDS and RADIUS are specified, then polyhedra of the given type(s) are constructed only to connected atoms that are within the specified radius of the central atom.','TEXT','[basis]','')
newCmd('',' ','','','Potential polyhedra centers and vertex atoms are specified in the form of a standard Jmol embedded {#.atom expressions~atom expression}, such as  <b>{{titanium}}</b> or <b>{{atomno&lt;12 and not nitrogen}}</b> and as such must be specified in parentheses. The first set specifies the centers; the second set specifies the vertex atoms. The optional keyword TO can preceed the vertex set for clarity, as in <b>polyhedra 4,6 BONDS {{titanium}} TO {{oxygen or nitrogen}}</b>. If no atom sets are designated, the assumption is "{{selected}} TO {{*}}". A single designation indicates centers, with "TO {{*}}" implied. If only TO and an atom set is specified, then the centers are taken as the currently selected set of atoms.','TEXT','[selection sets]','')
newCmd('',' ','','','Three sets of display options can be included when constructing polyhedra.<ul><li>Polyhedra can be displayed either as traditional "FLAT" faces (the default) or as "COLLAPSED" faces, which display more clearly the central atom. An empirically adjustable parameter, <b>distanceFactor</b> can be set to a higher value to include more faces in a polyhedron if some do not form. Its default value is 1.85. For collapsed polyhedra, the distance in angstroms from the central atom to the point of collapse can be specified using <b>faceCenterOffset=x.x</b> where x.x is a distance in Angstroms. </li><li>Polyhedra can be displayed either with no edges (NOEDGES, the default), or with a thick line on all edges (EDGES), or on just the front edges (FRONTEDGES). The front-edge-only option is only meaninful when the polyhedra are translucent (see below). </li><li>Polyhedra can be colored by indicating a valid color (e.g. <b>red</b> or <b>yellow</b>) or a hexadecimal RGB color triple (e.g. [xFF0000] or [xFFFF00]). The keywords TRANSLUCENT and OPAQUE can also be used. Alternatively, after polyhedra are created they can be colored using the {#.color (model object)~color polyhedra} command.</li></ul>','TEXT','[display options]','')
newCmd('',' ','','','The <b>polyhedra</b> command can also be used to modify already-constructed polyhedra. Used in this way, the command can take only one atom set expression, and it must not indicate the number of vertices or the basis. If the atom set expression is omitted, "(selected)" is assumed. <br /><br />To turn on, turn off, or delete selected polyhedra, use <b>polyhedra ON</b>, <b>polyhedra OFF</b>, or <b>polyhedra DELETE</b>, respectively. Any of the display options (FLAT, COLLAPSED, EDGES, FRONTEDGES, or NOEDGES) can also be similarly modified. Colors or translucency, however, cannot be applied this way. To color a set of polyhedra that are already formed or to make them translucent, first select the set of centers of the polyhedra to modify, then use the {#.color (model object)~color polyhedra} command.','TEXT','Polyhedron modification','')
newCmd('.print','11.3.17','','*v+11.4;','Prints a {#.Jmolmath~Jmol math} expression to the console window, status message callback function, and jmolScriptWait return value. ','0|1','[math expression]','')


newCmd('.prompt','12.0.RC26','','*v+12.0 -- new;','The <b>prompt</b> command pauses a script until the user presses OK or ESCAPE. See also the {#.functionsx=fyfunctions~prompt()} function.','1|2','','')
newCmd('','','','','If no parameter is given, displays a trace of the script stack leading to this command. This may be useful for debugging script files. ','*1','','')
newCmd('','','','','Displays the message and waits for the user to press OK.','*2','&quot;message&quot;','')

newCmd('.quaternion','','','*v+12.0 (deprecated);plot','In Jmol 12.0, this command has been superceded by the {#.plot~plot QUATERNION} command.','0|1','','')
newCmd('.quit',' ','','*v+11.0 -- new;pause;quit','When the {#.set_scriptQueue~} is turned on, each script waits for the previous to complete. When a LOOP command is involved and the script queue is enabled, the only way to interrupt the looping script is with another script. So, to account for this issue, the roles of <b>quit</b> and {#.exit~} have been expanded. Either <b>quit</b> or <b>exit</b> at the very beginning of a script command halts any previous still-running script. Processing then continues with the second command on the line. Anywhere else in the command, <b>quit</b> and <b>exit</b> abort that script.','0','','')
newCmd('.ramachandran','','','*v+12.0 (deprecated);plot','In Jmol 12.0, this command has been superceded by the {#.plot~plot RAMACHANDRAN} command.','0|1','','')
newCmd('.redo','12.2.RC6','','*v+12.2 -- new;','(application only) same as the Redo console button -- reloads the last undone state.','1|2','','')
newCmd('.redoMove','12.2.RC6','','*v+12.2 -- new;','same as CTRL-SHIFT-Z/CTRL-Y -- redoes changes to atom position when in modelKit mode.','1|2','','')
newCmd('.refresh',' ','','rreset',' Forces a screen repaint during script execution. (Unnecessary, and thus deprecated.)','0','','')
newCmd('.reset',' ','','*v+11.2 -- adds variable reset;rreset','Resets all model orientation or the value of a variable','0|1','','')
newCmd('',' ','','*v+11.2;','Resets all models to their original position: zoom 100; center; translate x 0; translate y 0. Spinning is not affected (see RESET SPIN, below). Note that if the {#.invertSelected~}, {#.rotateSelected~}, {#.translateSelected~} commands have been given, these changes are not reset, because these changes are recorded in the underlying coordinate set. If it is desired to save and restore atom positions after moving atoms relative to each other, the following commands may be issued: select *;translateSelected {{0 0 0}};save state;..(move atoms here)...;restore state.','*1','','')
newCmd('','11.3.29','','*v+11.2;','Resets all aromatic bonds to the type AROMATIC if they are AROMATICSINGLE or AROMATICDOUBLE. Used in conjunction with {#.connect~} and {#.calculate~calculate aromatic} .','*2','AROMATIC','')
newCmd('','','','*v+12.2;','Resets the _errorMessage variable.','*3','ERROR','')
newCmd('','11.3.23','','*v+11.4;math','Deletes all user-defined functions.','*4','FUNCTIONS','')
newCmd('','12.1.10','','*v+12.2;','Does a standard RESET, but also turns off spinning.','*5','SPIN','')
newCmd('','','','*v+12.2;','Resets PDB HELIX, SHEET, and TURN designations to their default file settings after, for example, using <b>calculate STRUCTURE DSSP</b>.','*6','STRUCTURE','')
newCmd('','12.1.10','','*v+12.2;','Resets the default van der Waals radii to standard Jmol values.','*7','VDW','')
newCmd('',' ','','*v+11.2;math','Resets the variable with the given name to the "unset" state.','*8','variableName','')
newCmd('','11.3.23','','*v+11.4;math','Deletes all user-defined variables.','*9','ALL','')
newCmd('.restore',' ','','*v+11.0 -- NEW;rreset','Restores information saved using the {#.save~} command.','0|1','','')
newCmd('',' ','','*v+11.0;','Restores bonding information changed via the {#.connect~} command; if no save name is given, then the last-saved information is restored.','*1','BONDS','saveName')
newCmd('',' ','','*v+11.0;','Restores a previously saved orientation. If no save name is given, then no time in seconds can be given, and the last-saved orientation is restored immediately. If a save name is given, then the number of seconds over which the restoration should be done may be given as well.','*3','ORIENTATION','saveName','timeSeconds','')
newCmd('',' ','','*v+11.0;','Restores a previously saved selection. If no save name is given the last-saved selection is restored. If no saved selection of this name is found, then the action is the same as <b>select none</b>.','*5','SELECTION','saveName')
newCmd('',' ','','*v+11.0;','Retores a previously saved state of the applet. Some of the more complex objects, such as dipoles, isosurfaces, lcaoCartoons, and molecular orbitals are not saved.','*7','STATE','saveName')
newCmd('','12.0.18','','*v+12.0;','Retores a previously saved secondary structure. saveName is optional; without it, this command restores the author-designated structure after use of the {#.calculate~calculate structure} command.  ','*8','STRUCTURE','saveName')
newCmd('.restrict',' ','','dhide','Selects the atoms identified by the {#.atom expressions~atom expression} and deletes all characteristics of all atoms and bonds that are outside the selection set by setting those characteristics "off". For wireframe and backbone, <b>restrict</b> unconditionally follows {#.set_bondmode~set bondmode OR}, ignoring the current bondmode setting, thus removing any bonds or backbone rods involving any atoms not in the restricted set. The <b>restrict</b> command is a holdover from RasMol, kept in Jmol for compatibility only. The  {#.display~} command is far more flexible, as it preserves the shape characteristics (size, color, translucency) of the hidden atoms rather than simply deleting the shapes.  ','0|1','','')
newCmd('',' ','','','Restricts to all atoms;possibly not H atoms.','*1','{"ALL"}','')
newCmd('',' ','','','Restricts atoms based on an {#.atom expressions~atom expression}.','*2','._atom_expression','')
newCmd('',' ','','*v+12.0;','Restricts atoms based on an {#.atom expressions~atom expression} while respecting the setting of <b>bondmode</b>. (Jmol 12.0)','*3','BONDS','._atom_expression')
newCmd('.resume',' ','','*v+11.0 -- NEW;pause','Resumes script execution after a {#.pause~}.','1','','')
newCmd('.return',' ','','*v+11.0 -- NEW;flow','Returns from a {#.functions~function} or a script being run with the {#.script~} command. In the case of a function return, may include an optional return value. ','1','returnValue','')
newCmd('.ribbon',' ','structure','disp','Ribbons offer a representation the protein backbone or nucleic acid helix using a flat band. For proteins, control points are chosen to be the center of the peptide bond, and the ribbon is drawn in the direction of the carbonyl oxygen (thus roughly defining the peptide planes). For nucleic acids, the control points are the midpoints between adjacent backbone phosphorus atoms, and the ribbon is drawn in the direction of the C6 carbon. A hermite curve is used.','0|1','','')
newCmd('',' ','','','','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns ribbon rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','Normally, ribbons vary in width according to the amino acid atom positions. This command sets the width of the ribbon to be a connstant value (a decimal, in Angstroms).  Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._ribbon_radius','')
newCmd('.rocket',' ','structure','*v+11.0 -- adds cartoonRocket alternative;disp','Creates a crude "rocket" cartoon.  See also {#.cartoon~} in association with {#.set (structure)~set cartoonRockets} for a more precise cartoon with rockets. The {#.set(structure)~set rocketBarrels} option removes the arrow heads from rockets. ','0|1','','')
newCmd('',' ','','','','*11','._on_off{"ON"}','')
newCmd('',' ','','',' Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._rocket_radius','')
newCmd('',' ','','*v+11.6;','Turns rocket rendering on and all other rendering (including labels) off.','*12','ONLY','')


newCmd('.rotate',' ','quaternions','*v+12.0 -- adds COMPARE, HELIX, SYMOP, TRANSLATE options;rspin;camera','The <b>rotate</b> and <b>spin</b> commands allow single angle rotation or continued rotation (spinning) in one of two specific frames, effectively changing the "camera" position. These frames include the standard "fixed" applet window frame and the internal "molecular" model frame (the axes in the data file). Coordinate specifications for the rotation can be given several ways, including absolute coordinate positions set in braces as {{x, y, z}}, the geometric center of a subset of atoms in the model specified in an atom set in parentheses, for example <b>{{atomno &lt; 10 and not oxygen}}</b>, or the geometric center of a previously defined draw object, indicated by the drawing object name preceded by a dollar sign (for example, <b>$line1</b>). In addition, rotation may be specified to be around six standard axes (x, y, z, -x, -y, and -z)  as well as several other options. The keyword MOLECULAR may be necessary for explicit model rotation about the internal molecular axes. Defaults are allowed. The default axis for spinining or rotating is the y (vertical) axis; the default angle of rotation is 10 degrees; the default rotation rate is a slow spin of 10 degrees per second. Options may be given in any order and include: [|| &nbsp;| By itself, the <b>rotate</b> command rotates 10 degrees around the y (vertical) axis in a counterclockwise fashion.|| <b>x</b>, <b>y</b>, <b>z</b>, <b>-x</b>, <b>-y</b>, <b>-z</b>|  Rotation about one of the six standard axes. The implied axes are the axes of the window, not the molecule.||<b>(decimal)</b> | one or two decimal numbers can be given. The first decimal number  indicates for a simple rotation the number of degrees for a rotation, either positive (right-hand rotation) or negative (left-hand rotation). For a spin, this first number, if it is the only number given, is taken as the rate of spin. For the TRANSLATION option, this is the number of Angstroms total for the translation; a second decimal number parameter implies the SPIN option and if positive indicates the rate of spinning in degrees per second (or, for the TRANSLATION option, linear movement in angstroms per second). If the second parameter is opposite the sign of the first number, the spinning/translating is carried out over that many seconds. || <b>{{atom expression or point}} {{atom expression or point}}</b> | Rotation about an axis pointing from the first point to the second in right-hand direction. For example, <b>rotate {{0 0 0}} {{1 0 0}} 10</b> is the same as <b>rotate x 10</b>, and <b>spin {{atomno=1}} {{atomno=2}}</b> spins the model around the axis connecting these two atoms. || <b>$drawID</b> | the two atoms desired for rotation can be defined based on a {#.draw~drawn} object such as a line or vector by specifying the ID of that drawn object. || <b>{{3x3 matrix}}</b> | A 3x3 matrix can be used to specify the axis and angle of a rotation; MOLECULAR is implied. || <b>{{4x4 matrix}}</b> | A 4x4 matrix can be used to specify a rotation axis and angle along with a translation; MOLECULAR is implied. || <b>AXISANGLE {{x y z}}</b> |  Rotation about an axis defined as a vector from the origin {{0,0,0}} to the specified {{x,y,z}} coordinate.  || <b>AXISANGLE {{x y z theta}}</b> | Rotation about an axis through {{0 0 0}} and {{x y z}} by the specified number of degrees.    ||<b>HELIX</b> | For any operation involving a translation (4x4 matrix, COMPARE, SYMOP, or TRANSLATE options) carries out the rotation about the optimum helical path. || <b>INTERNAL</b> | Same as <b>MOLECULAR</b>. || <b>MOLECULAR</b> | In the case of simple axes such as <b>x</b> or <b>y</b>, indicates that the axis is defined in terms of molecular coordinates. || <b>QUATERNION {{x y z w}}</b> | Rotation about an axis defined by the specified quaternion in the form of a four-vector. For example, to reset the model to the original orientation without changing the zoom, one can use <b>rotate QUATERNION @{{ !quaternion(script("show rotation")) }}</b>. The keyword QUATERNION is optional.|| <b>SELECTED</b> | Rotate only the currently selected atoms. || <b>SPIN</b> | Provide an animation of the rotation. Implied when two decimals -- a number of degrees and a rate -- are given.|| <b>TRANSLATE</b> | Add a translation to the animation (providing a screw-like motion) MOLECULAR and SELECTED are implied. If only one point is specified, then this command allows for an animation of a movement of the selected set of atoms along the vector connecting {{0 0 0}} and that point. In that case, the extent is in Angstroms and rate is in AngstromsPerSecond; otherwise, as for other rotations, extent is in degrees and rate is in degreesPerSecond. ||]','0|1','','')
newCmd('','13.3.4','','*v+13.4;rspin','Rotates the model to a calculated "best" orientation for viewing, such that the model\'s containing box is thinnest along the line of site (z), and widest horizontally (x).  With ROTATE SELECTED BEST, atom positions are actually changed; otherwise only the viewpoint is changed.','*1','BEST','')
newCmd('','13.1.17','','*v+13.2;','A special rotate BRANCH option takes as its parameter a list of n dihedrals with starting and target angles. All dihedrals are rotated concurrently over nSeconds time. The array list has 6n elements, a sequence of  for each dihedral of i1, i2, i3, i4, v1, v2, where i1, i2, i3, and i4 are atom indices involved in a dihedral, in order, and v1 and v2 are the initial and final dihedral angles for that set of atoms. See also the  {#.compare_bonds~compare BONDS} option, which uses this BRANCH option internally.','*3','BRANCH','@list','nSeconds','')
newCmd('','','','','Rotates the atoms that are in the molecular branch containing {{atom 2}} but not {{atom 1}} about the axis connecting these two atoms. For example, to spin a methyl group having carbon atom C12 about its connected atom by 360 degrees at a rate of 30 degrees per second, one might use <b>spin {{_C and connected(C12)}} {{C12}} 360 30</b>.','*2','BRANCH','{{atom1}}','{{atom2}}','')
newCmd('','','','','A variation of the {#.compare~compare} command. For atoms, determines the best-fit correlation between atoms in atomSet1 and atoms in atomSet2 and then rotates and/or translates atomSet1 to align with atomSet2. The two atom sets must have the same number of atoms and must have a direct 1:1correlations of atoms. (For a more general comparison, see the {#.compare~} command. An array of points, such as @{{ {{2.1}}.xyz.all}} may take the place of the second atom set. This allows, for example, saving of a set of coordinates with <b>tempCoords = {{2.1}}.xyz.all</b> and restoring of those coordinates with <b>ROTATE COMPARE {{2.1}} @{{tempCoords}}</b> some time after a selected atom rotation. ','*4','COMPARE','{{atomset}}','{{atomset or coordinates}}','')
newCmd('','','','','Carry out the symmetry operation number <b>n</b> (or, if <b>n</b> is a negative number, the reverse of that operation) on the currently selected atoms. ','*5','SYMOP','n')

newCmd('.rotateSelected','11.1.13','','*v+11.2 -- new;anim','This command takes all the parameters of {#.rotate~} but only carries out the operation on the currenly selected atoms. In addition, if the keyword SPIN is used, the command starts only the atoms spinning. Note that in its current implementation, this rotate DOES slightly modify atom coordinates; any measurements made after rotating may be off by a fraction of a percent, and continued spinning for a long time may introduced significant errors in relative distances and angles.','0|1','','')
newCmd('.save',' ','','*v+11.0 -- NEW;rreset','Saves a variety of sorts of information for later {#.restore~restoring} either in the current model or a different model.','0|1','','')
newCmd('',' ','','*v+11.0;','Saves bonding information, including atom connections, color, width, and visibility. A save name is optional. ','*1','BONDS','saveName')
newCmd('',' ','','*v+11.0;','Saves a full set of orientation information, including center and position of rotation.  A save name is optional but recommended, as it allows restoring of the orientation over a specified number of seconds.','*3','ORIENTATION','saveName')
newCmd('',' ','','*v+11.0;','Saves the current set of selected atoms for restoring later, either for the currently loaded model or for another model. (If used for only the current model, without a <b>load</b> command between <b>save</b> and <b>restore</b>, <b>save/restore SELECTION xxx</b> works the same as {#.define~define xxx selected/select xxx}.)  Note that when more than one model are loaded, the set of selected atoms is for the entire set of loaded models, not just the currently displayed one. Applying a restore to a completely different model should be done with care or it may not have the intended results. ','*5','SELECTION','saveName')
newCmd('',' ','','*v+11.0;','Temporarily saves the exact current state of the applet or application in memory for later restoring. Note that starting with Jmol 12.0, from the application or signed applet, using the {#.write~} command, you can save the state to a file on your computer in the form of a Jmol state SPT text file or a state-imbedded PNG or JPG file (which, though appearing as a high-resolution image, can also be "drag-dropped" back into Jmol to restore the state exactly, provided necessary files are in the directory of the image file.) You can also write JMOL files, which are ZIP files that contain all necessary files that have been loaded. ','*7','STATE','saveName')
newCmd('',' ','','*v+12.0;','Temporarily saves the structural state of the model, including definitions of HELIX, SHEET, etc.; can be used before the {#.structure~} or {#.calculate~calculate STRUCTURE} commands to restore the previous structural assignments. ','*8','STRUCTURE','saveName')
newCmd('.script',' ','','*v+13.0 -- adds arguments to running scripts from spt files;script;javascript','Loads and executes the specified script file/url. The hash/pound/sharp character (#) character marks a comment to the end of the line or a semicolon. The semicolon character (;) separates multiple statements on the same line. A script file may load another script file, up to 10 deep. Within the script, for any files indicated, prepending the file name with <b>$SCRIPT_PATH$</b> indicates to use the path of script file, not the current path, to find the file. If the command <b>CD $SCRIPT_PATH$</b> is given in a script, then all file references within that script and any scripts that script references will be relative to the path of that script.<br /><br />Starting with Jmol 13.0, arguments can be passed to a script using the syntax:<br /><br /><b>xxx.spt(argument1, argument2,...)</b><br /><br />where the <b>script</b> command word is not given, and the arguments are listed in parentheses, like a function. Within the script, these arguments are in the <b>_arguments</b> variable.','0|1','','')
newCmd('.script',' ','','*v+12.0 -- adds LOCALPATH and REMOTEPATH options.;*v-13.0;script;javascript','Loads and executes the specified script file/url. The hash/pound/sharp character (#) character marks a comment to the end of the line or a semicolon. The semicolon character (;) separates multiple statements on the same line. A script file may load another script file, up to 10 deep. Within the script, for any files indicated, prepending the file name with <b>$SCRIPT_PATH$</b> indicates to use the path of script file, not the current path, to find the file. If the command <b>CD $SCRIPT_PATH$</b> is given in a script, then all file references within that script and any scripts that script references will be relative to the path of that script.<br /><br />Starting with Jmol 13.0, arguments can be passed to a script using the syntax:<br /><br /><b>xxx.spt(argument1, argument2,...)</b><br /><br />where the <b>script</b> command word is not given, and the arguments are listed in parentheses, like a function. Within the script, these arguments are in the <b>_arguments</b> variable.','0|1','','')

newCmd('','11.9.2','','*v+12.0;','When reading a script created with {#.write~write STATE}, the LOCALPATH keyword instructs Jmol to strip all paths beginning with "file:/" down to the indicated path. So, for example, <b>script LOCALPATH "" "myfile.spt"</b> indicates that all local files referenced in the state script should be read from the current default directory. LOCALPATH can be used with scripts other than state scripts created by Jmol. The mechanism is simply looking for instances of /*file*/"some_file_name". If this construction is found in any script read, the replacement will be made.   (Jmol 12.0)','*12','LOCALPATH','&quot;path&quot;','._script_filename','')
newCmd('','11.9.2','','*v+12.0;','When reading a script created with {#.write~write STATE}, the REMOTEPATH keyword instructs Jmol to strip all paths beginning with "http:", "https:", or "ftp:" down to the indicated path. So, for example, <b>script REMOTEPATH "data" "myfile.spt"</b> indicates that all remote files referenced in the state script should be read from the subdirectory "data/" within the current default directory.  REMOTEPATH can be used with scripts other than state scripts created by Jmol. The mechanism is simply looking for instances of /*file*/"some_file_name". If this construction is found in any script read, the replacement will be made. (Jmol 12.0) ','*13','REMOTEPATH','&quot;path&quot;','._script_filename','')
newCmd('',' ','','','Loads and executes the specified script file/url. A hash/pound/sharp character (#) character marks a comment to the end of the line or a semicolon. A semicolon character (;) separates multiple statements on the same line. A script file may load another script file, up to 10 deep. Starting with Jmol 12.0, the command word SCRIPT is not necessary if the file is quoted or is of the simple format xxxxx.yyy.','*11','._script_filename','')
newCmd('','11.1.9','','*v+11.2;','Just checks the file for proper syntax and the presence of files.','*2','._script_filename','CHECK')
newCmd('','11.1.9','','*v+11.2;','Executes command n of the designated script file.','*31','._script_filename','COMMAND','n','')
newCmd('','11.1.9','','*v+11.2;','Executes commands n through m of the designated script file. The second number may be ommitted to indicate "to the end of the file": <b>script "myfile.spt" COMMANDS 10 - </b>','*32','._script_filename','COMMANDS','n - m','')
newCmd('','11.1.9','','*v+11.2;','Executes line n of the designated script file. ','*41','._script_filename','LINE','n','')
newCmd('','11.1.9','','*v+11.2;','Executes lines n through m of the designated script file. The second number may be ommitted to indicate "to the end of the file": <b>script "myfile.spt" LINES 10 - </b>','*42','._script_filename','LINES','n - m','')
newCmd('','11.3.33','','*v+11.6;','Runs the script command result of the {#.jmolmath~Jmol math expression} in one or more applets. The math expression may be a simple single variable name or quoted string or a more complex expression. If the math expression is a quoted string starting with <b>script</b> such as <b>"script dothis.spt"</b>, then the indicated script file will be run in each applet. The applet name can be any of the following.<br />[||&nbsp;&nbsp;&nbsp;*|all applets on the same web page as the originating applet (the script will run last in the originating applet)||&nbsp;&nbsp;&nbsp;>|all OTHER applets||&nbsp;&nbsp;.(period)|just this applet||&nbsp;&nbsp;name|the applet named <b>name</b> or, if that does not exist, <b>jmolApplet</b>name. Note that the function getProperty("appletInfo.registry") provides a list of applets on <i>all pages</i>, not just the same page. These other applets may be targeted if their full name (which includes a unique number extension) is given.||"name1,name2,..."|all applets matching the specified names. Note that quotes ARE required in this case.||] <br />','*5','APPLET','appletName','@{{Jmol math expression}','')
newCmd('','11.3.33','','*v+11.4;','Runs the script command result of the {#.jmolmath~Jmol math expression} rather than from a file. The math expression may be a simple single variable name or a more complex expression. For example, <b>var bgColor="red";script INLINE @{{"background " + bgColor}}</b> or <b>var s = "[arg]";script INLINE @{{"select " + s}}</b>.','*6','INLINE','@{{Jmol math expression}')
newCmd('',' ','','','Applet only: Evaluates the return of the indicated JavaScript function as the script to be executed. Execution is blocked if the web page parameter _jmol.noEval = true. Note that this is different from the {#.javascript~} command, which simply evaluates the specified JavaScript command. Here the function is evaluated on the embedding page, and the return from that function, which is presumed to be Jmol script, is evaluated <i>within Jmol</i>.','1','javascript:functionCall()','')

newCmd('.select',' ','','*v+12.2 -- adds ADD, REMOVE, and GROUP keywords;dhide','Selects the atoms identified by the expression. If no expression is given then all atoms are selected. The optional parameters ADD and REMOVE prior to the atom expression allow quick alternatives to "selected or..." and "selected and not...", respectively. In addition, the keyword GROUP prior to the atom expression provides a quick alternative to "within(group, ...)". ','0|1','','')
newCmd('.select',' ','','*v+11.6 -- adds extended atom property selection;*v-12.2;dhide','Selects the atoms identified by the expression. If no expression is given then all atoms are selected. A second property expression can be added for more general selection of atoms based on atom properties. ','0|1','','')
newCmd('',' ','','','Selects all atoms (possibly not H atoms).','*1','{"ALL"}','')
newCmd('',' ','','','Selects atoms based on an {#.atom expressions~atom expression}. To select atoms specific to a specific model when more than one model is present, use "/n" where "n" is the model number. For example, to select all atoms of model 3, use <b>select */3</b>. ','*2','._atom_expression','')

newCmd('','11.5.40','','','The <b>select</b> command allows for full utilization of Jmol math for atom selection. If the first parameter is an atom expression, such as {{*}} or {{10-30}}, a second parameter can be included. This second parameter must evaluate to a TRUE/FALSE expression involving the <i>individual</i> atoms of the atom expression. Parentheses around the property expression are optional. For example: <b>select {{*.ca}} (atomY < atomX)</b> selects for all alpha carbons for which their X coordinate is less than their Y coordinate. Note that in this context, "x", "y", and "z" are variable names, not coordinates. Use "atomX", "atomY", and "atomZ" if you need to refer to atom coordinates. The variable <b>_x</b> is assigned to the individual atom being tested. This variable can be used in nested select() functions within the select command to represent the atom being tested. For example, <b>select {{*.ca}} (phi < select(y; {{*.ca}}; y.resno = _x.resno + 1).phi))</b> selects alpha carbons of amino acid residues that have phi angles less than that of the phi angle of the next amino acid in the chain.','*4','._atom_expression','(property expression)')
newCmd('.selectionHalos',' ','','*v+11.0 -- NEW;','When ON, Jmol displays {#.halos~} around atoms when they are selected. The radius of the halo is always from 4 to 10 pixels larger than the current setting for {#.spacefill~} or the current setting of halo radius using the {#.halos~} command. The color of any specific halo is determined as described for  {#.color selectionHalos~}.','1','._on_off{"ON"}','')
newCmd('.set','11.1.6','','*v+11.2 greatly expands variable capability;math','Jmol allows a wide range of settings to be changed using the SET command -- see the categories below for details. In most cases the SET command is no longer necessary -- any simple value you can set with SET can be set simply using an assignment of a value to a variable name: <br /><br />strandCount = 6<br /><br /> instead of <b>SET strandCount 6</b>. However, it is recommended that you use SET for all {#.jmol parameters~Jmol parameters}, just as a way of clearly indicating in scripts that you are setting a Jmol parameter and not a user variable. In all cases below, "ON" and "TRUE" are equivalent, and "OFF" and "FALSE" are equivalent. Different words may be used simply because they seem more appropriate for a particular parameter. ','0|1','','')
newCmd('','11.3.56','','*v+11.4;','<b>set</b> by itself lists all Jmol parameters that can be set and all Jmol read-only variables (starting with underscore), along with their current values.','*1','','')
newCmd('','11.3.56','','*v+11.4;','<b>set</b> followed by characters ending in question mark lists all Jmol parameters starting with those characters and all Jmol read-only variables starting with underscore and those characters.','*2','xxx?','')

newCmd('.set (bond styles)',' ','setbonds','','This group of commands sets the appearance of various optional bond effects for the model.','2|3','','')

newCmd('',' ','','*v+11.0;bond','In some file formats (.mol files, for example) the connection data may indicate the bond type--single, double, triple, or quadruple. Use <b>set multipleBonds OFF</b> when you want all bonds to appear as single bonds. ','*67','showMultipleBonds','ON')
newCmd('',' ','','bond','The default Jmol condition. When script commands affect a set of atoms, BOTH atoms must be in the set for the bonds between them to also be affected. Similarly affects the display of backbone units in the selection of protein residues and nucleic acid bases.','*20','bondMode','AND')
newCmd('',' ','','','When script commands affect a set of atoms, EITHER atom may be in the set for the bonds also to be affected. Similarly affects the display of backbone units in the selection of protein residues and nucleic acid bases.','*21','','OR')
newCmd('',' ','','*v+11.4;bond','Setting this parameter TRUE is equivalent to <b>set bondMode OR</b>; it can be tested using Jmol math.','*22','bondModeOr','FALSE')
newCmd('',' ','','*v+11.4;bond','Sets the default bond radius in milliAngstroms. Immediately applies this to all bonds in all models.','*23','bondRadiusMilliAngstroms','(integer)')
newCmd('',' ','','*v+11.4;bond;hbond','When autobonding, the value of bondTolerance is added to the two bond radii of atoms being tested for a bond. A larger bondTolerance allows atoms that are further apart than the sum of their listed radii to still be bonded. This parameter should be adjusted prior to file loading for proper maintaining of the Jmol state.','*25','bondTolerance','(decimal)')
newCmd('','12.0.RC3','','*v+12.0;bond;hbond','For PDB files, generally Jmol will use a hydrogen bond calculation based on RasMol; for other file types, Jmol uses its own calculation. Setting this parameter FALSE causes hydrogen bonds to be calculated for all models using the Jmol calculation and involving {#.hbond~set hbondsAngleMinimum} and {#.hbond~set hbondsDistanceMaximum}.','*52','hbondsRasmol','TRUE')
newCmd('',' ','','*v+11.4;bond;hbond','Setting this parameter TRUE causes hydrogen bonds to be displayed as solid lines rather than dotted lines.','*53','hbondsSolid','FALSE')
newCmd('',' ','','*v+11.4;bond;hbond','Hydrogen bonds between protein amino acid residues or nucleic acid base pairs are displayed as lines. These lines can be displayed whether or not the H atoms are present in the file, and can be drawn either between the two non-hydrogen atoms involved in the bond (O or N, typically, the default) or, alternatively, between the two backbone alpha-carbon atoms, depending upon the desired effect.','*55','hbondsBackbone','FALSE')
newCmd('',' ','','*v+11.0;bond','Sets the minimum bond distance for autobonding. Should be set prior to file loading for proper maintainence of the Jmol state.','*60','minBondDistance','(decimal)')
newCmd('','12.1.11','','*v+12.2;bond','Sets the radius of the multiple bond lines relative to the standard setting.','*63','multipleBondRadiusFactor','0')
newCmd('','12.1.11','','*v+12.2;bond','When changed from its default value of -1, sets the distance between the lines in a multiple bond to a specific value and also instructs Jmol to draw multiple bonds in the plane of the atoms involved. (The default rendering in Jmol is to draw multiple bonds in such a way as they are always visible as multiple bonds.)','*65','multipleBondSpacing','-1')
newCmd('','12.1.46','','*v+12.2;bond','Displays partial bonds as dots, not dashes.','*66','partialDots','FALSE')
newCmd('',' ','','bond','Sulfur-sulfur bonds in cysteine bridges of proteins are displayed as lines. These lines can either be between the two sidechain sulfur atoms (the default) or between the two backbone alpha-carbon atoms, depending upon the desired effect.','*70','ssbonds','BACKBONE or SIDECHAIN')
newCmd('',' ','','*v+11.2;bond','Setting this parameter TRUE is an alternative method of setting disulfide bonds to backbone; it can be tested using Jmol math.','*71','ssBondsBackbone','FALSE')
newCmd('',' ','','bond','Sets the overall scale of all displayed dipole vectors.','*4','dipoleScale','.(-10.0 to 10.0)')

newCmd('.set (callback)','11.2 adds "jmolscript" in addition to just "script"','','*v+11.0 -- NEW mechanism for setting callback functions;javascript','Jmol allows for dynamic JavaScript callback function definition. You can specify the functions to receive callbacks, and you can change the functions at any time. Specific parameters sent to these functions are discussed more fully at {http://jmol.sourceforge.net/jslibrary/#jmolSetCallback~}. To turn off callbacks of a given type, specify "NONE". The function name must be present in JavaScript on the page containing the applet. Specifying "alert" will send the message to the user via a JavaScript alert. Note that quotation marks are required around the function name. If the filename starts with "jmolscript:" then instead of JavaScript being run, Jmol executes the Jmol script that follows. For example, <b>set hoverCallback "jmolscript:script hover.spt"</b> executes the Jmol script "script hover.spt" when an atom is hovered over. The code in the file hover.spt can then respond to that hovering action without the need for JavaScript. ','1|2','','')
newCmd('','11.1.45','','*v+11.0;','Sends a message indicating a change of {#.frame~}. For compatibility with Chime, the second parameter of this function returns a number ONE LESS than the actual frame number. This function returns nine parameters. The product of the last two parameters indicates the true direction of animation. <br /><br />function animFrameCallback(app, frameNo, fileNo, modelNo, firstNo, lastNo, isAnimationRunning, animationDirection, currentDirection)<br />[||app|String|The name of the applet ||frameNo|int|The current frame number (0-based) ||fileNo|int|The current file number (1-based) ||modelNo|int|The current model number within the current file (1-based) ||firstNo|int|The first frame of the animation range, expressed as fileNo*1000000+modelNo ||lastNo|int|The last frame of the animation range, expressed as fileNo*1000000+modelNo ||isAnimationRunning|int|0 (animation is off) or 1 (animation is on) ||animationDirection|int|1 (animation direction +1) or -1 (animation direction -1) ||currentDirection|int|1 (forward) or -1 (reverse)||]','*1','AnimFrameCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.0;','Sends a message indicating what atoms is being hovered over, indendently of whether {#.hover~} is ON or OFF.','*3','HoverCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.6;','Sends a message each time the {#.echo~} command is executed. If an EchoCallback function is not defined, these messages go to the MessageCallback function.','*21','EchoCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.6;javascript','Generally the Jmol applet is allowed to use the eval() function of the host page JavaScript using the {#.javascript~} command (unless execution of JavaScript by the applet has been specifically disallowed by setting _jmol.noEval = true prior to the jmolApplet() call). The setting of <b>evalCallback</b> function must be made prior to jmolAppletCall() using <b>jmolSetCallback("evalCallback","someFunctionName")</b>. It cannot be set using <b>set evalCallback</b>.  The callback sends the JavaScript to be evaluated back to the web page for evaluation rather than initiating that evaluation within Jmol. This could be important for the signed applet in order to isolate threads or for debugging applet calls to eval(). It is used in {Jmol Protein Explorer~http://chemapps.stolaf.edu/pe/protexpl}. ','*22','EvalCallback','')
newCmd('',' ','','*v+11.6;','Sends messages indicating the status of script execution. Line-by-line script commands are sent if one has <b>set debugScript TRUE</b>. If a ScriptCallback function is not defined, these messages go to the MessageCallback function.','*94','ScriptCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.6;','Sends a message indicating the status of measurements made by the user. If a MeasureCallback function is not defined, these messages go to the MessageCallback function.','*5','MeasureCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.6;','Sends a message that indicates the status of a currently running {#.minimize~minimization}.','*7','MinimizationCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.0;','Sends a message each time a file is {#.load~loaded}.','*4','LoadStructCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.0;','Sends a wide variety of messages during script execution.','*6','MessageCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.0;','Sends a message that depends upon the current status of {#.set picking~}.','*8','PickCallback','&quot;function name&quot;')
newCmd('','11.1.29','','*v+11.2;','Sends a message indicating changes of the window size.','*91','ResizeCallback','&quot;function name&quot;')
newCmd('','13.3.5','','*v+13.4;','Sends a message each time a structure is modified. Parameters include mode, atomIndex and modelIndex, where mode is one of: [|| 1 | assignAtom element/charge change start || -1 | assignAtom element/charge change end || 2 | assignBond start || -2 | assignBond end || 3 | assignAtom position start || -3 | assignAtom position end || 4 | delete atom begin || -4 delete atom end || 5 | delete model begin || -5 | delete model end ||]','*98','StructureModifiedCallback','&quot;function name&quot;')
newCmd('',' ','','*v+11.6;','The SyncCallback method allows a JavaScript function to intercept and modify or cancel an applet-applet {#.sync~} message. If the called function returns "" or 0, the synchronization is canceled; any other string is substituted for the script and sent to the other currently synchronized applets.','*98','SyncCallback','&quot;function name&quot;')
newCmd('.set (debugging)',' ','','*v+11.0 introduces an adjustable logging level and showScript option;','&nbsp;','0|1','','')
newCmd('',' ','','*v+12.0;','Turning this parameter ON sets <b>debugScript</b> ON and <b>logLevel</b> to 5; setting it OFF returns these settings to their standard values (OFF and 4, respectively).','*1','debug','OFF')
newCmd('',' ','','','Turning this parameter ON enables debugging (going to a JavaScript message {#.set (callback)~callback}).','*2','debugScript','OFF')
newCmd('','11.5.4','','*v+11.6;','Sets the maximum delay that scripts will use, primarily used for testing scripts. The default setting of this parameter, 0, disables maximum delay checking.','*31','delayMaximumMs','0')
newCmd('','11.7.10','','*v+11.8;','A debugging parameter.','*32','fontCaching','TRUE')
newCmd('','','','*v+11.4;','Primarily for debugging complex scripts. Sets the level of depth in scripts for which the command {#.history~} is recorded. A setting of 0 (the default) indicates that commands in scripts run using the {#.script~} command should not be recorded. A setting of 1 indicates that such commands should be recorded for script commands given at the top level. A setting of "n" indicates that all commands executed from script commands through level "n" should be recorded.  A setting of -1 turns off all history recording.','*33','historyLevel','(integer)')
newCmd('',' ','','*v+11.0;','You may also set the amount of logging sent to the Java console (as opposed to the Jmol {#.console~}). The default level is 4, "information, warnngs, and errors". Levels include: <br />[||0|no messages whatsoever||1|fatal errors only||2|all errors||3|all warnings and errors||4|information, warnings, and errors||5|full debugging||]','*4','logLevel','(0 - 5)')
newCmd('',' ','','*v+11.0;','Turning this parameter ON causes Jmol to show the script commands from a script file as they are executed. A slight difference may be observed when showScript is set in comparison to normal operation in that when showScript is set, an automatic refresh is executed after every command. ','*51','showScript','OFF')
newCmd('',' ','','*v+11.0;','Shows the script commands from a script file as they are executed and pauses the specified number of milliseconds between commands. Setting the delay to 0 turns script showing off.','*52','showScript','milliseconds')
newCmd('','11.1.13','','*v+11.2;','Sets the maximum script depth (how many scripts have been called) for which certain messages should appear in the console and be reported back via callbacks. A value of 0 (default) displays messages only from the topmost level -- as, for example, commands entered from the console; a value of 1 displays messages from the top level and messages due to commands such as "script t.spt" but not from scripts that are called from t.spt; etc. A value of -1 turns off all reporting. Affected commands include {#.connect~}, {#.select~} (and related commands), and {#.set~}. This parameter is particularly useful when scripts utilize {#.goto~message/goto} to control program flow.','*4','scriptReportingLevel','(integer)')
newCmd('.set (antialiasing)','11.3.43','','*v+11.4 -- NEW high-resolution rendering;','Antialiasing is the smoothing of jagged lines and sharp boundaries in an image. Jmol independently antialiases the display and image creation (using the {#.write~} or {#.capture~} commands). If the display is antialiased, the option also exists to antialias translucent objects or not. Memory requirements are doubled and rendering performance will be diminished when antialiasDisplay is turned on.','2|2','','')
newCmd('',' ','','*v+11.4;','Turning this parameter ON results in smoothing of the model rendering in the window.','*1','antialiasDisplay','OFF')
newCmd('',' ','','*v+11.4;','Turning this parameter OFF removes smoothing of the translucent components of the window in addition to the opaque components when antialiasDisplay is turrned on.','*2','antialiasTranslucent','ON')
newCmd('','','','*v+11.4;','Turning this parameter OFF disables smoothing of the images created using the {#.write~} command.','*3','antialiasImages','ON')
newCmd('.set userColorScheme','11.3.13','','*v+11.4 -- NEW customizable user color schemes;color','Sets a custom color scheme. Color values can be names such as "red" or "blue" or hexadecimal numbers as [xRRGGBB] or {{r g b}} triples in the form of 3D points. The scheme can then be referred to as "user"; it\'s reverse as "resu". ','*3','colorName','colorName','…','')
newCmd('.set echo',' ','','*v+11.0 introduces movable echo text and 3D echos, more label control, version 11.6 adds images;label','Sets positioning and visibility parameters associated with text written to the screen using the {#.echo~} command. In all <b>set echo</b> commands, an optional id may be given. If this id is not given, the previously indicated id is assumed. <b>set echo IMAGE</b> allows images to be displayed instead of text.  Characteristics of the text that are settable include: [||background color | Includes translucency. See {#.background~background ECHO}. ||depth | The default Z-position for echoed text and images is in front of the model. However, this can be set to any {#.depth~} in relation to the model, from 0 (rear) to 100 (front). See below. ||font color | Includes translucency. See {#.color~color ECHO}. ||font size, face, style, and scaling factor | See {#.font~font ECHO}.||ID "xxx" | Starting with Jmol 12.2, the optional <b>ID</b> keyword introduces an echo ID that is in quotes or from evaluation, such as <b>set echo ID @{{"test" + i}}</b> in order to distinguish it from echo text, which would be indicated with <b>set echo "text here"</b>. Three special echo IDs, TOP, MIDDLE, and BOTTOM are defined for quick placement of text. These three special echos are always defined along with one of three predefined positions and text alignments: LEFT, CENTER, or RIGHT. Thus we have <b>set echo TOP LEFT</b>, <b>set echo TOP RIGHT</b>, etc. Note that you cannot have two separate echos both with the ID TOP. Thus, after <b>set echo top left;echo  OK</b>, if you issue <b>set echo top right</b>, the text "OK" originally in the top left corner will jump to the top right corner. Note that TOP is an echo label, not an echo position. If you need two echos, one on the top left and one on the top right, it is best to use your own label id -- <b>set echo myTopLeft 0 100%; echo "on the left"; set eho myTopRight 100% 100%; echo "on the right"</b>. The difference in this case between <b>set echo TOP RIGHT</b> and <b>set echo myTopRight 100% 100%</b> is that in the latter case, if you want to right-align multi-line text, you also need to issue <b>set echo myTopRight right</b>. With <b>set echo TOP RIGHT</b>, the text is both positioned on the right and right-aligned.||IMAGE |  You can also place JPEG, PNG, or GIF images at either a 2D screen position or a 3D molecular position. See below. ||model | Echo text can be associated with a specific model, with visibility controlled by the {#.frame~} command. See below. ||scaling | Using the command {#.set misc~set fontScaling TRUE} 3D-positioned images and text along with atom labels can be made to scale when {#.zoom~} is applied. The current zoom setting is used to fix a "scaling factor" used for the font. However the {#.font~} command can be used to set a different reference scaling factor. The absolute scale of an image independent of this 3D zoom scaling, can be set using <b>set echo scale</b>, below.||text alignment| Text in echos is generally left-aligned. You can use <b>set echo myecho CENTER</b> to center text at the given position of that echo or <b>set echo myecho RIGHT</b> to have it right-aligned. Use <b>set echo myecho LEFT</b> to return the text to the default left alignment. Note that the special echo IDs TOP, MIDDLE, and BOTTOM are automatically text-aligned based on their required horizontal postion keyword -- LEFT, RIGHT, or CENTER. If you need center-aligned text that is positioned in the top right corner, for example, use your own ID: <b>set echo myecho 100% 100%;set echo myecho CENTER</b>. <b>Set echo myecho CENTER</b> also aligns multi-line text and images vertically.||visibility | Echos are deleted using <b>set echco off</b>. However, you can selectively display or hide echos using either the <b>set echo HIDDEN/DISPLAYED</b> command or the {#.hide~} and {#.display~} commands with $name, where name is the name given an echo in the <b>set echo</b> command (including the special echos TOP, MIDDLE, and BOTTOM using $top, $middle, and $bottom, respectively). ||x and y position within the window | A "2D" echo. A block of text can be displayed anywhere within the applet or application window. An automatic adjustment is made to prevent text from going outside of the bounds of the window. See below. ||x, y, and z position within the model itself | A 3D echo. This position can be defined as a point in space, the coordinate of an atom, the average coordinate of a set of atoms, the position of a {#.draw~drawn point}, or the average position of a drawn object. See below. ||]<br /><br /> Default settings for font, color, and background can be set by first issuing <b>set echo none</b>, so that no real echo is set, and then issuing the desired {#.font~}, {#.color~}, and {#.background~} commands. In addition, any changes to these settings for any specific echo text become the new default for future text echos that are created for the same model.<br /><br />','1|2|3','','')
newCmd('',' ','','*v+11.0;','Selects the horizontal text alignment (LEFT, CENTER, or RIGHT) for a user-positioned text echo. For the specific echos with ID <b>TOP</b>, <b>MIDDLE</b>, and <b>BOTTOM</b>, the horizontal alignment is also relative to the window. Any text already displayed in this position is immediately realigned. The CENTER option also centers the text box (or image) vertically at the 2D or 3D point defined for this echo.','*32','id','._echo_horizontal{left}')
newCmd('',' ','','*v+11.0;','Sets a custom echo position with the given name, where (0,0) is the bottom left corner. The next-issued {#.echo~} command will write text to this position. Any text already displayed is immediately moved. If the text would flow out of the applet window, it is repositioned just inside the boundary.','*33','id','x-position','y-position','')
newCmd('',' ','','*v+11.0;','Sets echo position based on a percentage of the applet window size. (One or the other or both of x and y can have percent.) Note that <b>set echo myecho 0% 100%</b> is not quite the same as <b>set echo TOP LEFT</b>; <b>set echo myecho 100% 0%</b> is not quite the same as <b>set echo BOTTOM RIGHT</b>. The difference is that TOP LEFT and BOTTOM RIGHT automatically justify multiline text as well. Using percentages allows you to place text anywhere within the window and independently justify the text using <b>set echo myecho LEFT</b>, <b>set echo myecho RIGHT</b>, <b>set echo myecho CENTER</b>.','*340','id','x%','y%','')
newCmd('',' ','','*v+11.6;','Sets a custom echo position with the given name, where [0 0] is the bottom left corner. The next-issued {#.echo~} command will write text to this position. Any text already displayed is immediately moved. If the text would flow out of the applet window, it is repositioned just inside the boundary.','*341','id','[x y]')
newCmd('',' ','','*v+11.6;','Sets echo position based on a percentage of the applet window size. Note that <b>set echo myecho [0 100%]</b> is not quite the same as <b>set echo TOP LEFT</b>; <b>set echo myecho [100 0%]</b> is not quite the same as <b>set echo BOTTOM RIGHT</b>. The difference is that TOP LEFT and BOTTOM RIGHT automatically justify multiline text as well. Using percentages allows you to place text anywhere within the window and independently justify the text using <b>set echo myecho LEFT</b>, <b>set echo myecho RIGHT</b>, or <b>set echo myecho CENTER</b>.','*342','id','[x y %]')
newCmd('',' ','','*v+11.0;','Sets echo position in three dimensions, based on molecular coordinates (which may be fractional). Braces are required.','*35','id','{{x y z}}')
newCmd('',' ','','*v+11.0;','Sets echo position in three dimensions, based on the geometric center of the specified atoms. Braces are required. "OR" should be used rather than ","  in the atom expression unless additional parentheses within the braces are also provided.','*36','id','{{','._atom_expression','}')
newCmd('',' ','','*v+11.6;','Specifies a Jmol script that will be run when the echo is clicked.','*404','id','SCRIPT','&quot;script&quot;','')
newCmd('',' ','','*v+11.6;','By default, when multiple models are present, all echo text is visible regardless of the model currently displayable using the {#.frame~} command. The <b>set echo ... MODEL</b> command associates an echo with a specific model, allowing the text or image to appear and disappear when different models are displayed using the {#.frame~} command.','*402','id','MODEL','(model number)','')
newCmd('',' ','','*v+11.6;','Reads the specified JPG, GIF, or PNG file and displays that image instead of text. All 2D and 3D options are available. Images can be scaled using <b>set echo scale</b> (below).','*401','id','IMAGE','&quot;file name&quot;','')
newCmd('',' ','','*v+11.6;','Sets the depth of the echoed text or image in percent of the model diameter (0 rear, 100 front).','*38','id','DEPTH','%z','')
newCmd('','11.6.RC7','','*v+11.6;','Selectively displays an echo that previously has been hidden. The name can be any defined echo, including TOP, MIDDLE, or BOTTOM.','*39','id','DISPLAYED')
newCmd('','','','*v+11.6;','Scales an echo image to the desired absolute amount.','*404','id','SCALE','(decimal)','')
newCmd('','13.1.19','','*v+13.2;','Connects echo text with an atom or point with a line.','*403','id','POINT','._atom_expression_or_coord','')
newCmd('','11.6.RC7','','*v+11.6;','Selectively hides an echo without deleting it. The name can be any defined echo, including TOP, MIDDLE, or BOTTOM.','*42','id','HIDDEN')
newCmd('',' ','','*v+11.0;','Changes  the text of the echo message; same as <b>echo "some text"</b>. ','*70','id','"some text"')
newCmd('',' ','','*v+11.6;','Sets the current echo to be "all echos" for setting font, color, display, hidden, or text of all echos at once.','*43','ALL','')
newCmd('',' ','','','Sets the current echo to be "none" -- The next <b>echo</b> command will be to the console/message callback only. After this command, <b>set echo off</b> will still delete all echos.','*50','NONE','')
newCmd('',' ','','','Turns off (deletes) all echo texts. See also <b>set echo hidden</b>.','*60','OFF','')

newCmd('.set (files and scripts)',' ','','*v+12.2 -- several new settings, including pdbAddHydrogens;','The following commands relate to how files and scripts are loaded and how scripts are executed. ','0|1','','')

newCmd('','11.1.5','','*v+11.2;','When set TRUE (default), Jmol will read scripts that are contained in certain file types and append them to the script set using <b>set defaultLoadScript</b> (below). Embedded scripts are indicated by "jmolscript:" followed by valid Jmol script commands in the following file locations:<br />[||CIF|any line(s)||MOL|line 3||PDB|any REMARK line(s)||XYZ|line 2||] Support for additional file types will be provided upon user request.  ','*10','allowEmbeddedScripts','')
newCmd('','11.1.22','','*v+11.2;','Setting this parameter to FALSE causes Jmol to add atoms to the last model in a file set rather than add a whole new model when {#.load~load APPEND} is used.','*11','appendNew','TRUE')
newCmd('',' ','','*v+11.0;','Sets the URL for a proxy server when {#.load~loading} a file or reading a {#.pmesh~} or {#.isosurface~} or when reading a file-based {#.script~}. A proxy server is a server-side application (typically written using PERL, PHP, or ColdFusion) on the same host as the JAR file that can deliver files from other servers on the internet. Jmol appends "?url=" followed by the URL of the requested data file to the indicated proxy server name. NOte that Java security requires that this call be to the same server that hosts the JAR file. ','*12','appletProxy','&quot;URL&quot;')
newCmd('','11.1.29','','*v+11.2;bond','Turning this parameter ON instructs Jmol, when applying symmetry to atoms, as in "load xxx.cif {{1 1 1}}", to also apply symmetry to the bonds indicated in the file. The flag is useful when  normal Jmol autobonding would not properly connect atoms, but  the model is "molecular" -- the base atom coordinates are correct for whole molecules. The flag should not be used in cases where the application of symmetry operations creates new bonds that  were not present in the original set, as for quartz.cif, where there is only one bond initially, and after applying symmetry new bonds are created that are between atoms that were created using two different symmetry operations. If bonds are not indicated in a file, this flag is ignored, and Jmol uses its autobonding algorithm to create bonds.','*13','applySymmetryToBonds','OFF')
newCmd('','11.7.7','','*v+11.8;load','The MOL2 and MDTOP file formats specifically do not contain enough information within the file to determine the element for all atoms. (The atom types depend upon which force field was employed, and not all force fields allow the direct decoding of element from atom type.) The <b>atomTypes</b> setting allows the user to correlate atom types with specific elements for these specific readers. The syntax is "abcd=>C;efgh=>H;...", indicating that atom type "abcd" is carbon, atom type "efgh" is hydrogen, etc. The matching is case sensitive.','*14','atomTypes','&quot;...&quot;')
newCmd('',' ','','bond','Some file formats that Jmol reads, such as XYZ, do not contain bonding information. In these cases, the default action for Jmol is to generate bonds automatically based on an algorithm. When given prior to loading a model, the <b>set autobond OFF</b> command causes Jmol to not do any automatic bond creation when subsequent models are loaded.','*16','autobond','ON')
newCmd('','11.7.30','','*v+11.8;','Specifically for Spartan and Sygress readers, this setting (default FALSE) sets the default orientation (when the file is loaded or {#.reset~} is issued) to that given in the file. Starting with Jmol 12.0, the setting is ignored, and the default orientation is automatically used unless FILTER "NoOrient" is included.','*17','autoLoadOrientation','FALSE')
newCmd('','11.6.rc15','','*v+11.6;cd','Sets the path to files specifically for the dialog box that pops up when a file name is "?" in a command such as {#.load~} or {#.write~}. This variable is automatically set when the user navigates to a new directory during a file save dialog.','*18','currentLocalPath','&quot;path&quot;')
newCmd('',' ','','*v+11.0;','Issued prior to the {#.data~} command, sets the text that will separate one set of file datat from another, thus allowing the inline loading of multiple file types.','*19','dataSeparator','&quot;separator text&quot;')
newCmd('',' ','','*v+11.0;cd','Sets the default directory to use for reading all files. This will generally be a relative path such as "./data" or "../files". Note that Java security requires that if the applet is run from a hard drive rather than via the internet, all files read must be either in the directory containing the JAR file or in a subdirectory of that directory. If the applet is unsigned and loaded from the internet, then Java security typically requires that all files read are from the host from which the JAR file was read. But see <b>set appletProxy</b>, above. This flag is applicable only to the Jmol applet, not to the Jmol application.','*20','defaultDirectory','&quot;directory path&quot;')
newCmd('','13.1.2','','*v+13.2;','Sets the default script to use for drag-drop and File|Open. The Jmol default is: <b>zap; load SYNC "%FILE";if (%ALLOWCARTOONS && _loadScript == \'\' && defaultLoadScript == \'\' && _filetype == \'Pdb\') {{if ({{(protein or nucleic)&*/1.1}} && {{*/1.1}}[1].groupindex != {{*/1.1}}[0].groupindex){{select protein or nucleic;cartoons only;}}if ({{visible}}){{color structure}}else{{wireframe -0.1}};if (!{{visible}}){{spacefill 23%}};select *}};</b>','*21','defaultDropScript','&quot;script&quot;')
newCmd('',' ','','*v+11.0;','For crystallographic systems, sets the default lattice to be loaded. {{1 1 1}} loads the standard single unit cell (555). {{2 1 1}} loads two unit cells along the <b>i</b> direction, cells 555 and 655. {{2 2 2}} loads a set of eight unit cells; {{3 3 3}} loads a set of 27. This last is probably most useful, because it loads one unit cell surrounded by all 26 cells sharing its 6 faces, 12 edges, and 8 corners.','*3','defaultLattice','{{i j k}}')
newCmd('',' ','','*v+11.0;','Sets a script to run after any file is loaded. The script must be in quotations. If the script itself needs quotation marks, then it should be placed in a file and indicated as follows: <b>set defaultLoadScript "script myscript.scr"</b>.','*41','defaultLoadScript','&quot;script&quot;')
newCmd('','12.0.RC10','','*v+12.0;','Sets the URL to use for obtaining the Uppsala Electron Density server secondary file that gives cut-off/sigma information.','*45','edsUrlCutoff','&quot;url&quot;')
newCmd('','12.0.RC10','','*v+12.0;','Sets the URL to use for obtaining the Uppsala Electron Density server MAP file.','*47','edsUrlFormat','&quot;url&quot;')
newCmd('',' ','','*v+11.0;','<b>set forceAutoBond ON</b> tells Jmol to disregard any bonding information in a file and use its own internal algorithm for determining connectivity. Its effect is for all future file loads until set OFF. This setting is particularly useful for some PDB and mmCIF files that already have a threshold amount of bonding, so that a full set of bonding can be created automatically at load time. This is necessary for proper assignment of secondary structure. ','*51','forceAutoBond','OFF')
newCmd('',' ','','*v+11.0;','Sets the number of lines of command history to record (minimum 2) and turns history recording ON. <b>set history 0</b> turns off the command history feature but does not actually set the number of lines to zero. See also {#.history~} and {#.show~show history}.','*55','history','nLines')
newCmd('',' ','','*v+11.0;cd','The load format, by default "http://www.rcsb.org/pdb/files/%FILE.pdb.gz", allows setting of the URL that will be used when "=" is used in front of a file name in a {#.load~} command, for example: <b>load =1crn</b>. ','*61','loadFormat','&quot;URL&quot;')
newCmd('','12.1.51','','*v+12.2;','The load format, by default "http://www.rcsb.org/pdb/files/ligand/%FILE.cif", allows setting of the URL that will be used when "==" is used in front of a file name in a {#.load~} command, for example: <b>load ==atp</b>. ','*62','loadLigandFormat','&quot;URL&quot;')
newCmd('','12.1.51','','*v+12.2;quit','The load format for reading information such as chemical names, inchikeys, standard inchi, and images from the NIH/NCI CACTVS server, default http://cactus.nci.nih.gov/chemical/structure/%FILE.','*65','nihResolverFormat','&quot;URL&quot;')
newCmd('','12.3.29','','*v+12.2;','Sets the path used in all file reading operations such as LOAD, ISOSURFACE, and SCRIPT, regardless of what is given for the path in those commands. Also disables the WRITE command. This parameter can be used to scripts originally pointing to one directory to point to another.','*67','pathForAllFiles','&quot;path&quot;')
newCmd('','12.1.51','','*v+12.2;calc','The <b>pdbAddHydrogens</b> option is perhaps the most important new feature in Jmol 12.2. This flag instructs Jmol to add both hydrogens and multiple bonding when PDB files are read. Hydrogen atoms are added to all residues, both ATOM and HETATM. Jmol will fetch RCSB ligand CIF files for attaching hydrogens, and the addition is guaranteed to name hydrogens appropriately. In particular, CH<sub>2</sub> groups are named properly, with the proper prochirality. (Not all algorithms do this properly, but Jmol does.) Hydrogen atoms are added to the sidechain nitrogen atom of lysine and to both nitrogen atoms of histidine -- avoiding an analysis of "most probable" hydrogen position. Hydrogen atoms are not added to water. When a state file (SPT JMOL, PNG, or JPG) is created from such a model, Jmol stores the fetched CIF data as part of that state and does not fetch it again when the file is reopened. ','*68','pdbAddHydrogens','FALSE')
newCmd('',' ','','*v+11.0;quit','Turning this parameter OFF disables script queuing. Setting this parameter OFF should never be necessary, but it is provided here as an option. When script queuing is enabled (the default), scripts that are {#.loop~looping} require {#.quit~} or {#.exit~} to be executed in a subsequent script in order to complete. When script queuing is turned off, scripts from different threads may collide and cause unpredictable behavior or crashing of Jmol.','*7','scriptQueue','ON')
newCmd('','12.0.RC3','','*v+12.0;load','This parameter sets the maximum number of atoms for default rendering of the model. Models with this number or fewer atoms will be rendered with the default spacefill rendering and the default bond diameter; models with more than this number of atoms will be displayed by default with <b>spacefill 0; wireframe 1</b> to conserve resources.','*8','smallMoleculeMaxAtoms','40000')
newCmd('','12.1.7','','*v+12.2;load','The source of files when using the "$" or "SMILES" keyword in the {#load~} command.  Defualt: "http://cactus.nci.nih.gov/chemical/structure/%FILE/file?format=sdf&get3d=True"','*9','smilesURLformat','&quot;URL&quot;')
newCmd('.set (highlights)',' ','','*v+11.0 introduces infinitely variable echos;label','This command group allows for annotation and highlighting of atoms in terms of labels and "halos."','2|3','','')


newCmd('',' ','','disp','(deprecated; see {#.selectionHalos~selectionHalos ON/OFF})','*1','display','SELECTED/NORMAL')
newCmd('',' ','','*v+11.0;','See {#.frank~}','*6','frank','')
newCmd('.set (labels)',' ','','*v+11.0 -- several new customizable features;label','This command group sets parameters associated specifically with atom labels. If the atom expression is not indicated, the action is applied to the currently selected atoms. ','1|2','','')
newCmd('','11.5.4','','*v+11.6;','When <b>fontScaling</b> is set ON, any labels created after that are rescaled when the model is {#.zoom~zoomed}.','*3','fontScaling','OFF')
newCmd('',' ','','','Sets the font size for atom labels for the currently selected atoms.','*5','fontSize','._fontsize{8}')
newCmd('','12.1.50','','*v+12.2;','Sets the default label for PDB atoms. Default: "%m%r"','*1','defaultLabelPDB','&quot;labelFormat&quot;')
newCmd('','12.1.50','','*v+12.2;','Sets the default label for non-PDB atoms. Default: "%a"','*2','defaultLabelXYZ','&quot;labelFormat&quot;')
newCmd('',' ','','','Sets the label text alignment within a multi-line label as left-, right-, or center-justified. (For overall label alignment, see <b>set labeloffset</b>, below.) ','*71','labelAlignment','LEFT, RIGHT, or CENTER')
newCmd('',' ','','','Sets the label offset relative to the atom being labeled. A positive number indicates the number of pixels between the atom center and the beginning of the label. A negative number indicates the number of pixels between the atom center and the end of the label, with the entire label to the left of the atom. Zero indicates centered. ','*92','labelOffset','._xoffset','._yoffset','{{ atom expression }}')
newCmd('',' ','','','Sets the label offset relative to the atom being labeled. A positive number indicates the number of pixels between the atom center and the beginning of the label. A negative number indicates the number of pixels between the atom center and the end of the label, with the entire label to the left of the atom. Zero indicates centered. ','*92','labelOffset','[ sx sy sz ]')
newCmd('',' ','','*v+11.0;','Turning this parameter ON instructs Jmol to add lines pointing from the selected atoms to their respective labels, using the color of the label text ({#.color~color label xxxx}).','*93','labelPointer','OFF','{{ atom expression }}','')
newCmd('',' ','','*v+11.0;','Turns on label pointers to selected atoms, drawing them in the color of the label background.','*94','','BACKGROUND','{{ atom expression }}','')
newCmd('',' ','','*v+11.0;','If a selected label is rotated behind an atom, it is hidden by that atom (default). If an atom expression is given, an indicator of ON or OFF must also be given. OFF is the same as "SET LABELFRONT".','*81','labelAtom','._on_off{"ON"}','{{ atom expression }}','')
newCmd('',' ','','*v+11.0;','The selected labels will always appear in front of all atoms.  If an atom expression is given, an indicator of ON or OFF must also be given. OFF is the same as "SET LABELATOM".','*82','labelFront','._on_off{"ON"}','{{ atom expression }}','')
newCmd('',' ','','*v+11.0;','Selected labels appear in an invisible plane just in front of the atoms <i>of their group only</i>. Applicable only to PDB/mmCIF files.  If an atom expression is given, an indicator of ON or OFF must also be given. OFF is the same as "SET LABELATOM".','*84','labelGroup','._on_off{"ON"}','{{ atom expression }}','')
newCmd('','11.3.52','','*v+11.4;','Toggles the labels on or off for the specified set of atoms. ','*96','labelToggle','{{ atom expression }}')
newCmd('.set (language)','11.1.31','','*v+11.2 -- new;','You can set the language for the applet popup menu to a new lanuage just by selecting the desired language from the language submenu.','1|2','','')
newCmd('',' ','','*v+11.2;','The command <b>set language "xx"</b>, where <b>xx</b> is a two-letter abbreviation for a lanuage, allows this to be done programmatically. Supported languages are listed on the popup menu.','*1','language','&quot;[two-letter code]&quot;')
newCmd('',' ','','*v+11.8;','Setting this parameter OFF turns off language translation. This may be important for pages that need to parse messages coming from Jmol in American English. ','*2','languageTranslation','ON')
newCmd('.set (lighting)',' ','lighting','*v+11.0 -- clearer naming;disp','This commands in this group determine the overall lighting effects, size, and rotation for the model. Note that in a multi-applet environment, changing lighting values (ambientPercent, diffusePercent, specular, specularExponent, specularPercent, or specularPower) changes them for ALL applets. Effect on another applet may not appear until that model is rotated by the user or some command is issued to that applet that requires updating the display.','2|3','','')
newCmd('',' ','','*v+11.0;','Sets the amount of "ambient" light filling the shadows created by the apparent light source. An ambientPercent value of 0 creates an effect of a spotlight on a stage; a value of 100 removes the shadow entirely, creating a flat, nonrealistic effect.','*10','ambientPercent','(integer 0 to 100)')
newCmd('','13.1.13','','*v+13.2;','<b>set celShading TRUE</b>  creates a flat effect similar to a hand-drawn figure.','*20','celShading','FALSE')
newCmd('','12.1.7','','*v+12.2;','Same as the {#.depth~} command, but note that the parameter can be read and manipulated with math: <b>depth += 10</b> or <b>if (depth != 0)…</b>.','*21','depth','(integer)')
newCmd('',' ','','*v+11.0;','Sets the amount of "diffuse" light apparently emanating from the spotlight, but not hitting and reflecting off the model directly. Setting the diffusePercent value to 0 turns this effect off; giving the effect of the model in a black-walled room where no light reflection is possible, effectively turning off all but specular reflections.','*25','diffusePercent','(integer 0 to 100)')
newCmd('','11.9.13','','*v+12.0;','Sets the specular strength using a number which is 2^(specularExponent) [Jmol 12.0]','*3','phongExponent','(integer 0 to 1000)')
newCmd('','12.1.7','','*v+12.2;','Same as the {#.slab~} command, but note that the parameter can be read and manipulated with math: <b>slab += 10</b> or <b>if (slab != 100)…</b>.','*50','slab','(integer)')
newCmd('',' ','','','Turns off the specular reflection (the shiny dot on a curved surface).','*51','specular','OFF')
newCmd('',' ','','*v+11.0;','Ranging from 1 to 10, the specular exponent determines the tightness of the specular spot, with 1 being very broad and 10 being very tight. Techincally, log<sub>2</sub>(phongExponent).','*53','specularExponent','(integer 1 to 10)')
newCmd('',' ','','*v+11.0;','Sets the density of dots in the specular reflection, producing the effect of a very dim light (10) to a very bright light (100).','*57','specularPower','(integer 0 to 100)')
newCmd('',' ','','*v+11.0;','Sets the size of the apparent reflection from a light source.','*55','specularPercent','(integer 0 to 100)')
newCmd('','12.1.7','','*v+12.2;','Sets the point where the zShade is fully transparent.  Ignored if set equal to zSlab (in which case the settings for slab and depth are used).','*80','zDepth','(integer)')
newCmd('','12.1.7','','*v+12.2;','Sets the point where the zShade is fully opaque. Ignored if set equal to zDepth (in which case the settings for slab and depth are used)','*90','zSlab','(integer)')
newCmd('',' ','','*v+11.8;','<b>set zshade ON</b> enables a "fog" or "fade" effect, which shades objects based on the distance from the observer such that distant objects fade into the background. Uses the value of the SLAB and DEPTH settings to determine the point of no effect and full effect, respectively (by default 100 and 0). Starting with Jmol 12.2, the settings of SLAB and DEPTH are overridden by ZSLAB and ZDEPTH when ZSLAB is not equal to ZDEPTH. (Both are 0 by default in Jmol 12.2) The effect is enabled regardless of the setting of (<b>slab ON/OFF</b>) and is reset to OFF upon {#.reset~} or the loading of a  new model. ','*85','zShade','OFF')
newCmd('',' ','','*v+12.0;','This parameter can be adjusted (typically 1, 2, or 3) to create a stronger effect of fog when <b>zShade</b> is set ON.','*87','zShadePower','(integer)')
newCmd('.set (perspective)',' ','','*v+11.0 -- adds zoomLarge;disp;camera','This commands in this group determine the overall lighting effects, size, and rotation for the model. Note that in a multi-applet environment, changing lighting values (ambientPercent, diffusePercent, specular, specularExponent, specularPercent, or specularPower) changes them for ALL applets. Effect on another applet may not appear until that model is rotated by the user or some command is issued to that applet that requires updating the display.','2|3','','')
newCmd('','11.1.1','','*v+11.2;','Sets the amount of perspective. The Jmol default is <b>set cameraDepth 3.0</b>, but you can adjust that to your liking. Smaller numbers lead to more extreme perspective distortion; higher numbers minimize the effect of perspective. The scale in the vertical plane midway from front to back of the model is identical to with <b>set perspectiveDepth off</b> at any zoom level; if <b>p</b> the relative distance of a vertical plane from front (0) to back (1) of the model at a zoom of 100, then the scaling factor is (cameraDepth + 0.5) / (cameraDepth + p) for that plane. This gives, for cameraDepth=3.0, 1.17 for p=0, 1.0 for p=0.5 (as for all cameraDepths), and 0.87 for p=1.','*11','cameraDepth','(positive number)')
newCmd('',' ','','*v+11.2;','Turning this parameter OFF turns off perspective depth. OFF is required for proper function of absolute scale (<b>set scaleAngstromsPerInch  nnn</b>). Perspective depth is automatically turned off by Jmol when loading a model having a unit cell and symmetry operators and automatically turned ON otherwise.','*31','perspectiveDepth','ON')
newCmd('',' ','','*v+11.2;','Jmol\'s perspective model involves a linear perspective that is required for navigation mode. Setting this parameter to 10 returns Jmol to the former perspective model used in Jmol 10. When <b>set navigationMode</b> is invoked, the perspective model is automatically set to 11. ','*32','perspectiveModel','11')
newCmd('',' ','','*v+11.2;','Sets the nominal rotation radius for the model effectively setting the size of the model at <b>zoom 100</b>. Normally set to the radius that will contain within the screen the entire model when rotated relative to the default rotation center. ','*4','rotationRadius','(Angstroms)')
newCmd('',' ','','*v+11.2;','Sets the absolute scale of the model by setting the viewing distance from the user to the model in arbitrary units. The actual scale will depend upon the sizes of both the applet window and the screen. ','*43','scaleAngstromsPerInch','._viewing_distance')
newCmd('',' ','','','Setting this parameter to FALSE allows one to set the center of rotation to a new position without also moving that position to the center pixel of the applet window. ','*8','windowCentered','TRUE')
newCmd('',' ','','*v+11.0;','When set FALSE, this parameter disables zooming. Same as {#.zoom~zoom ON/OFF}','*91','zoomEnabled','TRUE')
newCmd('',' ','','*v+13.2;','When this setting is TRUE, Jmol adjusts the zoom only when the height is changed. Adjusting the width only widens the view (as in PyMOL). Setting zoomHeight TRUE disables <b>zoomLarge</b>.','*92','zoomHeight','FALSE')
newCmd('',' ','','*v+11.0;','(Starting with Jmol 13.2, zoomLarge is ignored when zoomHeight is TRUE.) When TRUE (Jmol default setting), Jmol adjusts the zoom based on the LARGER of the two application/applet dimensions, height and width. Setting this parameter FALSE causes zoom to be based on the smaller of the two dimensions, which may be useful for the applet if the user is allowed to resize the applet window. PyMOL NOTE: Pymol uses <b>set zoomHeight TRUE</b>, and when a PSE file is read, Jmol automatically goes to this mode.','*93','zoomLarge','TRUE')

newCmd('.set (navigation)','11.1.2','','*v+11.2 -- new;nav;disp;camera','This commands in this group set various parameters in relation to navigating through the model. See {misc/navigation.pdf~pdf documentation} for details regarding the Jmol perspective/navigation model.','2|3','','')
newCmd('','11.1.3','','*v+11.2;','When navigating, a small square-with-crosshairs pointer usually appears to help guide the user. Setting this parameter TRUE hides that pointer.','*10','hideNavigationPoint','FALSE')
newCmd('','11.7.47','','*v+11.8;','Sets the nominal speed of {#.navigate~navigation}. Default value is 10.','*14','navFPS','(integer)')
newCmd('','11.1.3','','*v+11.2;','An experimental setting that restricts navigation to the inside of a surface. Not intended for general use.','*1','navigateSurface','FALSE')
newCmd('','11.1.3','','*v+11.2;','Sets the depth of the observer navigation point using the same basis as standard slab and depth -- 0 being the back plane of the model; 100 being the front plane of the model. Values greater than 100 represent observation points in front of model.','*2','navigationDepth','(percent)')
newCmd('','11.1.2','','*v+11.2;','When set TRUE, enables user-directed navigation through the model using the keypad arrow keys according to the following table. Setting this parameter TRUE automatically sets the perspectiveStyle to 11, sets zoomEnabled, and sets perspectiveDepth on. [||UP/DOWN arrows|go forward/back off||LEFT/RIGHT arrows|turn left/turn right||ALT UP/DOWN arrows|pitch upward or downward||SHIFT LEFT/RIGHT/UP/DOWN arrows|shift navigation point on the screen and in relation to the model||]','*33','navigationMode','FALSE')
newCmd('','11.1.5','','*v+11.2;','When set TRUE, and the model has a unit cell defined and symmetry has been applied using {#.load~load} {{i j k}}, creates the effect of an limitless navigation. The navigation center is kept always within the unit cell. The more full unit cells that have been loaded, the more realistic the effect. ','*35','navigationPeriodic','FALSE')
newCmd('','11.1.5','','*v+11.2;','Sets the rate of forward/reverse navigation when using the up/down arrow keys. The default value is 5.','*37','navigationSpeed','(integer)')
newCmd('','11.1.5','','*v+11.2;','Sets the depth of the depth of the slab plane relative to the observer navigation point using the same basis as standard slab and depth -- 0 (default) being at the navigation point; positive numbers being toward the user relative to that. ','*4','navigationSlab','(percent)')
newCmd('','11.7.47','','*v+11.8;','Sets the navigation turning rate in the X direction (yaw) in units of 1/50 degrees per frame.','*51','navX','(decimal)')
newCmd('','11.7.47','','*v+11.8;','Sets the navigation turning rate in the Y direction (pitch) in units of 1/50 degrees per frame.','*52','navY','(decimal)')
newCmd('','11.7.47','','*v+11.8;','Sets the navigation rate in the Z direction (forward motion) relative to its standard value of 1.','*53','navZ','(decimal)')
newCmd('','11.1.3','','*v+11.2;','When set TRUE, the navigation cross-hair cursor is shown whenever in navigation mode, not just while the user is pressing keypad keys or scripted navigation is occurring.','*5','showNavigationPointAlways','FALSE')
newCmd('','11.1.2','','*v+11.2;','The visual range is the MINIMUM distance in Angstroms that is viewable by the observer. The default value of 5.0 indicates that with <b>set navigationMode TRUE</b> any object in a plane that is less than 5.0 Angstroms across will be clipped automatically under the presumption that that object is behind the observer. This prevents large objects close to the user from eclipsing smaller objects further from the observer so as to give the illusion of having entered the model itself and now being within it.','*6','visualRange','(angstroms)')

newCmd('.set (structure)','','','*v+11.0 -- several new customizable features;','This command group allows for customization of the rendering of PDB and mmCIF secondary structures. The default is <b>set cartoonRockets OFF; set ribbonAspectRatio 16; set hermiteLevel 0; set ribbonBorder 0; set sheetSmoothing 1; set strands 5; set traceAlpha ON</b>.','2|3','','')
newCmd('','13.1.16','','*v+13.2;','Setting this parameter ON sets the display of RasMol {#.cartoon~cartoons} to be a mixture of three-dimensional sheet cartoons and alpha helix {#.rocket~rockets}. It is not a precise as a cartoon with respect to helices but more precise than rockets in relation to beta-pleated sheets.','*41','cartoonLadders','OFF')
newCmd('',' ','','*v+12.0;','Setting this parameter ON displays nucleic acid bases as triangles that highlight the sugar edge (red), Watson-Crick edge (green), and Hoogsteen edge (blue). See Nasalean L, Strombaugh J, Zirbel CL, and Leontis NB in {http://books.google.com/books?hl=en&lr=&id=se5JVEqO11AC&oi=fnd&pg=PR11&dq=Non-Protein+Coding+RNAs&ots=3uTkn7m3DA&sig=6LzQREmSdSoZ6yNrQ15zjYREFNE#v=onepage&q&f=false~<b>Non-Protein Coding RNAs</b>}, Nils G. Walter, Sarah A. Woodson, Robert T. Batey, Eds., Chapter 1, p 6.','*40','cartoonBaseEdges','OFF')
newCmd('',' ','','*v+11.0;','Setting this parameter ON sets the display of RasMol {#.cartoon~cartoons} to be a mixture of three-dimensional sheet cartoons and alpha helix {#.rocket~rockets}. It is not a precise as a cartoon with respect to helices but more precise than rockets in relation to beta-pleated sheets.','*42','cartoonRockets','OFF')
newCmd('',' ','','*v+12.0;','Setting this parameter FALSE tells Jmol to not use DSSP for the calculation of secondary structure either upon loading a structure with no HELIX or SHEET information or when using the command {#.calculate~calculate STRUCTURE}.','*45','defaultStructureDSSP','TRUE')
newCmd('',' ','','*v+12.0;','Standard DSSP secondary structure analysis disregards all backbone amide hydrogen atoms that might be in a file and instead calculates its own based on the direction of the C=O vector. Setting this parameter FALSE instructs Jmol to not calculate these file-based positions and to instead use the positions of the hydrogen atoms that are in the file. ','*46','dsspCalculateHydrogenAlways','TRUE')
newCmd('',' ','','*v+11.0;','Determines the amount of curve smoothing used in rendering protein and nucleic acid secondary structures involving {#.cartoon~}, {#.ribbbon~}, {#.rocket~}, and {#.trace~}. <b>set hermiteLevel 0</b> uses a Jmol 10.2 method of rendering these structures, with rope-like traces and paper-thin ribbons and cartoons. Positive values produce more rounded but slower to render shapes, but only when the model is not in motion. Negative numbers produce the same, but also while rotating. A value of 4 or -4 might be a reasonable compromise between look and performance. Starting with Jmol 13, hermite levels 6 or higher (-6 or lower) produce a ribbon cross-section in the shape of an ellipse. Use {#.set_ribbonaspectratio~set ribbonAspectRatio 4} rather than the default value of 16 for a more rounded ellipse.','*50','hermiteLevel','(integer, -8 to 8)')
newCmd('','11.3?','','*v+11.4;','When set ON, and hermiteLevel is set, draws high-resolution hermite curves even if the diameter is small. Otherwise, small-diameter traces are shown in a faster-rendering fashion.','*53','highResolution','OFF')
newCmd('',' ','','*v+11.0;','Sets the thickness of the ribbons in ribbon and cartoon renderings in terms of the width:height aspect ratio; only enabled in conjunction with <b>set hermiteLevel</b> to a non-zero value. <b>set ribbonAspectRatio 0</b> turns off this feature; 8-16 is recommended; higher positive numbers leading to thinner ribbons.','*67','ribbonAspectRatio','(integer)')
newCmd('',' ','','','Turning this parameter ON adds a slight border for ribbons.','*68','ribbonBorder','OFF')
newCmd('','11.3.38','','*v+11.4;','Turning this parameter ON removes the arrow heads from {#.rockets~} and {#.cartoon~cartoon rockets}, turning them into simple cylindrical barrels.','*70','rocketBarrels','OFF')
newCmd('',' ','','*v+11.0;','In conjunction with <b>set traceAlpha</b>, this parameter determines the "waviness" of beta-pleated sheets. <b>set sheetSmoothing 0</b> creates wavy sheets, with the ribbon or trace going directly through the alpha carbons. The default is <b>set sheetSmoothing 1</b>, which produces a more averaged, smoother (standard) ripple. Has no effect unless <b>set traceAlpha</b>.','*72','sheetSmoothing ','(0 to 1)')
newCmd('',' ','','*v+11.2;','Sets the number of strands to display for the {#.strand~} and {#.meshribbon~} shapes both, with a maximum of 20.','*91','strandCount','._strand_count')
newCmd('','11.3.52','','*v+11.4;','Sets the number of strands to display for the {#.strand~} shape, with a maximum of 20.','*94','strandCountForStrands','._strand_count')
newCmd('','11.3.52','','*v+11.4;','Sets the number of strands to display for the {#.meshribbon~} shape, with a maximum of 20.','*92','strandCountForMeshRibbon','._strand_count')
newCmd('',' ','','*v+11.0;','When set TRUE (the default), along with <b>set sheetSmoothing 0</b>, directs Jmol to draw {#.trace~traces} directly through all alpha-carbons. The effect is a wider, more standard alpha helix and a wavier beta-pleated sheet than in Jmol 10.2. Setting <b>set sheetSmoothing 1;set traceAlpha ON</b> directs Jmol to create smooth out the beta-pleated sheets but still follow the alpha carbons for other structure types. With <b>set traceAlpha FALSE</b>, Jmol draws traces through the midpoints of the lines connecting adjacent alpha-carbons, producing tighter alpha helices and smoothed beta-pleated sheets. Also used as the basis for drawing cartoons, meshRibbons, ribbons, rockets, and strands. The sequence <b>set traceAlpha off; set sheetSmoothing 1;set hermiteLevel 0</b> is equivalent to the Jmol 10.2 default. Jmol 11.4 has the default settings: <b>set traceAlpha TRUE; set sheetSmoothing 1;set hermiteLevel 0</b>.','*99','traceAlpha','TRUE')
newCmd('','12.1.13','','*v+12.0;','Ideally, PDB files will contain header information relating to helix, sheet, or turn locations. When this information is not present, or when the {#.calculate~calculate structure} command is given, Jmol uses an admittedly crude Ramachandran angle range check to determine structure type. The <b>set STRUCTURE</b> command allows setting those Ramachandran angle ranges for helix, sheet, and turn. [phi-psi range] is a set of angles in groups of four. Each group of four numbers includes the minimum and maximum phi values followed by the minimum and maximum psi values that would be valid for this structure type. For example, <b>set structure HELIX [-160, 0, -100, 45]</b> sets "helix" to be defined as phi >= -160 and phi <= 0 and psi >= -100 and psi <= 45 (the Jmol default). The default setting for sheet (really just a strand, since we are not requiring cross-strand hydrogen bonding) and turn are <b>set SHEET [-180, -10,   70,  180, -180, -45, -180, -130, 140, 180,  90, 180]; set TURN [30, 90, -15, 95]</b>. It is recommended that this command be followed immediately with <b>calculate structure;cartoons only</b>. (Jmol 12.2)','*96','structure','HELIX|SHEET|TURN','[phi-psi ranges]','')





newCmd('.set (visibility)',' ','','*v+11.0 -- several new features;disp;set','This command group turns on or off specific sets of atoms and axes/cell-related options.','2|3','','')


newCmd('',' ','','*v+11.0;','See {#.axes~}.','*1','axes','._axes_type')
newCmd('','','','*v+11.4;','Allows setting and checking of the current axes mode: (0) window, (1) molecular, (2) unitcell ','*20','axesMode','0, 1, or 2')
newCmd('','','','*v+11.0;','See {#.axes~}; this parameter can be set and checked using Jmol math.','*21','axesMolecular','OFF')
newCmd('','11.1','','*v+11.2;','Allows setting of the axes scale; (default 2)','*22','axesScale','(decimal)')
newCmd('','','','*v+11.0;','See {#.axes~}; this parameter can be set and checked using Jmol math.','*23','axesUnitcell','OFF')
newCmd('','','','','See {#.axes~}; this parameter can be set and checked using Jmol math.','*24','axesWindow','ON')
newCmd('','11.1','','*v+11.2;','Sets the background color to the specified value; can be inspected using Jmol math.','*25','backgroundColor','._colorRGB')
newCmd('',' ','','*v+11.0;','See {#.boundbox~}.','*30','boundbox','._axes_type')
newCmd('','','','*v+11.2;','This parameter sets the default boundbox color and also can be read as the current boundbox color.','*32','boundboxColor','&quot;color_name&quot;')
newCmd('','11.3','','*v+11.4;','Sets the default translucent level (0.5 is standard). ','*33','defaultTranslucent','(decimal)')

newCmd('',' ','','*v+11.0;','<b>set disablePopupMenu TRUE</b> disables the pop-up menu.','*42','disablePopupMenu','FALSE')
newCmd('',' ','','*v+11.0;','Turning this parameter FALSE turns off the unit cell parameter list that is included with {#.unitcell~unitcell ON}.','*44','displayCellParameters','TRUE')
newCmd('',' ','','*v+11.0;','Setting this parameter ON displays all colors as grey scale values.','*50','greyScaleRendering','OFF')
newCmd('',' ','','*v+11.0;','Setting this parameter TRUE hides the file name and contents in the pop-up menu starting with the NEXT FILE LOADED.','*51','hideNameInPopUp','FALSE')
newCmd('',' ','','*v+11.0;','Setting this parameter TRUE tells Jmol to hide any atoms whenever they are not selected. <b>set hideNotSelected TRUE</b> immediately hides the currently unselected atoms. When turned off, no immediate action is taken, but future selections no longer hide atoms. The flag is automatically cleared when a new model is loaded or when the picking mode is set to any sort of selection (atom, group, etc.). ','*52','hideNotSelected','FALSE')
newCmd('','13.3.4','','*v+13.4;','This parameter replaces {#.set_wireframerotation~set wireframeRotation} and determines how many compromises are used in rendering a model when it is in motion due to animation or user manipulation. The goal is to provide "slimmed down" versions of of the model in order to allow smooth rotation. Larger values include all of the capabilities enabled by lower values.  [|| value &gt;=  |   enables ||          8          |antialiasDisplay (and thus, all features)||          7         |translucency||          6          |meshes (contact, draw, isosurface, MO, pmesh, lcaocartoon, CGO)||          5          |cartoons, rockets, trace, ribbon||          4          |geosurfaces||          3          |ellipsoids||        2          |wireframe and balls||          1          |none of the above (same as "set wireframeRotation off")||          0          |reserved for "auto"||]','*54','platformSpeed','1 - 8')
newCmd('','11.0','','*v+11.0;','When <b>refreshing</b> is set FALSE, no writing to the screen is done at all. Setting refreshing to TRUE should only be done in cases where there is a desire to not show intermediate results as a valid well-tested script runs. If the script throws an error, and refreshing is FALSE, Jmol may appear to be frozen when in fact it is not.','*60','refreshing','TRUE')
newCmd('',' ','','*v+11.0;','Sets the radius of the solvent "ball" that would run around the structure defining its outline. After <b>set radius</b>, you must (re)issue {#.dots~dots ON} for it to take effect, and the solvent probe option for dots must be set on using <b>set solvent ON</b> (below).','*82','solventProbeRadius','._probe_radius{1.2}')
newCmd('','13.1.19','','*v+13.2;nmr','Sets the chemical shift reference in PPM for converting magnetic susceptibility to chemical shift. XX is a nucleus such  as H or C. For example, <b>set shift_H 30</b>.','*67','shift_XX','(decimal)')
newCmd('',' ','','*v+11.0;','Setting <b>showAxes</b> TRUE displays the axes at their default line width; setting it false hides the axes.','*68','showAxes','FALSE')
newCmd('',' ','','*v+11.0;','Setting <b>showAxes</b> TRUE displays the bounding box as a thin dotted line; setting it false hides the bounding box.','*69','showBoundBox','FALSE')
newCmd('',' ','','*v+11.0;','Setting <b>showFrank</b> TRUE displays the Jmol frank in the lower right-hand corner. Checking this parameter allows checking to see if the frank is on.','*70','showFrank','TRUE')
newCmd('',' ','','*v+11.0;','When both this parameter is ON and {#.selectionHalos~} are ON, Jmol will display selection halos even if atoms are hidden. ','*71','showHiddenSelectionHalos','FALSE')
newCmd('',' ','','*v+11.0;','Turns on and off display of hydrogen atoms.','*72','showHydrogens','TRUE')
newCmd('',' ','','*v+11.0;','deprecated -- see {#.selectionHalos~}','*73','showSelections','FALSE')
newCmd('','11.0','','*v+11.0;','Setting this parameter TRUE is equivalent to {#.unitcell~unitcell ON}; can be tested using Jmol math. ','*76','showUnitcell','FALSE')
newCmd('','11.9.19','','*v+12.0;','Setting this flag TRUE removes whole atoms and bonds when they are partially clipped. (Jmol 12.0)','*771','slabByAtom','FALSE')
newCmd('','11.9.19','','*v+12.0;','Setting this flag TRUE removes whole molecules when they are partially clipped. (Jmol 12.0)','*772','slabByMolecule','FALSE')
newCmd('','11.0','','*v+11.0;','Setting this parameter TRUE is equivalent to {#.slab~slab ON}; can be tested using Jmol math. ','*78','slabEnabled','FALSE')
newCmd('',' ','setsolvent','*v+11.0;','Turning this parameter ON turns on a solvent "probe" that can be displayed using dots. After <b>set solventPrope ON</b>, a subsequent {#.dots~dots ON} shows the surface of the aggregate of selected atoms using dots. This surface is defined by the contact of a spherical probe (representing a solvent molecule) rolled over the surface of the selected atoms. The radius of the probe sphere of 1.4 Angstroms approximates a water molecule. The default radius for Jmol is 1.2 Angstroms, which can be changed using <b>set solventPropbeRadius</b>. Note that no change in display occurs after <b>set solventProbe ON</b> until the next {#.dots~dots ON} command is encountered. Does not effect the function of {#.isosurface~isosurface solvent}, which draws a surface regardless of the setting of this flag. For a detailed discussion of molecular surfaces, see {http://www.netsci.org/Science/Compchem/feature14.html~}.','*80','solventProbe','OFF')
newCmd('',' ','','*v+13.2;color','Jmol by default considers translucent objects truly translucent. When set FALSE, translucent obects are not translucent to each other, creating an unreal but possibly desired effect similar to the default PyMOL setting transparency_mode 1.','*85','translucent','TRUE')
newCmd('',' ','','*v+11.0;','See {#.unitcell~}.','*90','unitcell','')
newCmd('','','','*v+11.2;','This parameter sets the default unit cell color and also can be read as the current unit cell color.','*93','unitCellColor','&quot;color_name&quot;')

newCmd('.set (measure)','','','*v+11.0 -- several new features;','Sets characteristics of the measurement labels and lines. See also {#.measure~}.','0|1','','')
newCmd('','','','*v+11.0;','Sets the format of the labels for {#.measure~distance measurements}.','*20','defaultDistanceLabel','&quot;format&quot;')
newCmd('','','','*v+11.0;','Same as for defaultDistanceLabel, but for angles.  See {#.measure~} for details.','*21','defaultAngleLabel','&quot;format&quot;')
newCmd('','','','*v+11.0;','Same as for defaultDistanceLabel, but for torsions. See {#.measure~} for details.','*22','defaultTorsionLabel','&quot;format&quot;')
newCmd('','','','*v+11.2;','This parameter should be set TRUE if molecules or atoms are being moved relative to one another using {#.translateSelected~} or {#.rotateSelected~} so that the numbers associated with distances and angles updates properly as atoms are moved.','*24','dynamicMeasurements','ON')
newCmd('','','','*v+11.0;','Turning this parameter TRUE right-justifies measurement labels','*40','justifyMeasurements','FALSE')
newCmd('',' ','','*v+11.0;','Turning this parameter OFF turns off measurement LABELS only, leaving the lines. (To turn off both, use <b>set showMeasurements OFF</b>.)','*63','measurementLabels','ON')
newCmd('',' ','','*v+11.0;','Setting this parameter FALSE turns off measurement lines and labels BOTH. (To turn off just the label, use <b>set measurementLabels OFF</b>.)','*77','showMeasurements','TRUE')
newCmd('',' ','','*v+11.4;*v-12.2;','Sets the units for measurement display to be Angstroms, nanometers, or picometers.','*63','measurementUnits','._distance_unit')
newCmd('',' ','','*v+12.2 -- adds VDW;','Sets the units for measurement display to be one of ANGSTROMS, AU (or BOHR), NM (or NANOMETERS), (PM or PICOMETERS), or VDW. The VDW setting is new to Jmol 12.2 and displays distances as percent of the combined van der Waals radii of two measured atoms. Thus, a measurement less than 100% indicates a "clash."','*63','measurementUnits','(distance unit)')
newCmd('',' ','','','Sets the width of the measurement line in angstroms.','*2','measurements','._width_angstroms')
newCmd('',' ','','','Sets the width of the measurement line in pixels.','*3','','._width_pixels')
newCmd('',' ','','','Sets the measurement line to be dotted.','*4','','DOTTED')




newCmd('.set (misc)',' ','      ','*v+13.0 --adds set energyUnits, forceField, isosurfaceKey, meshScale, nmrUrlFormat, vectorSymmetry;anim;misc','In all cases below, "ON" and "TRUE" are equivalent, and "OFF" and "FALSE" are equivalent.','0|1','','')
newCmd('.set (misc)',' ','      ','*v+11.0 -- new options for temperature, measurements, and backgroundmodel;*v-13.0;anim','In all cases below, "ON" and "TRUE" are equivalent, and "OFF" and "FALSE" are equivalent.','0|1','','')
newCmd('',' ','','*v+12.0;','This parameter is primarily intended for a touch-screen context. Setting this parameter TRUE allows single-point gestures to be detected. Currently the only single-point gesture supported is a "swipe" or "flick" created by a motion of the mouse or finger that is continuing along a line at the time the mouse or finger is released. ','*03','allowGestures','FALSE')
newCmd('',' ','','*v+11.8;','Set this parameter TRUE to allow key strokes in the applet or application window to be interpreted as script commands. These keystrokes will be displayed if the {#.set_showKeystrokes~showKeystrokes} setting is TRUE.','*05','allowKeystrokes','FALSE')
newCmd('','12.0.RC15','','*v+12.0;','Set this parameter FALSE to disable {#.set modelkitmode~modelKitMode}.','*071','allowModelKit','TRUE')
newCmd('','12.1.21','','*v+12.2;','Set this parameter TRUE to  allow the moving of selected atoms (not just whole molecules) using ALT-LEFT drag and ALT-SHIFT-LEFT drag.','*075','allowMoveAtoms','FALSE')
newCmd('',' ','','*v+12.0;','Set this parameter FALSE to disable multi-touch gestures (two-finger spread for zoom, two-finger drag for translation) within a multi-touch environment.','*08','allowMultiTouch','TRUE')
newCmd('',' ','','*v+11.2;','When set TRUE, this parameter allows user rotation of the molecule containing the selected atom using the mouse (holding ALT down while dragging). The coordinates of the rotated molecule will be sightly degraded in this process. ','*09','allowRotateSelected','FALSE')
newCmd('',' ','','*v+11.2;anim;misc;image','Same as {#.animation~animation FPS} -- sets the animation delay in frames per second; can be tested, for example, with  "if (animationFPS > 10)…"','*100','animationFps','(integer)')
newCmd('','13.3.6','','*v+13.4;anim;misc;image','Same as {#.animation~animation MODE} -- sets the animation replay mode; must be in quotes: "ONCE", "LOOP", or "PALINDROME"; can be tested, for example, with  "if (animationMode == \'ONCE\')…"','*100','animationMode','&quot;ONCE&quot; or &quot;LOOP&quot; or &quot;PALINDROME&quot;')
newCmd('','11.6.RC13','','*v+11.6;','Setting this parameter FALSE disallows picking of atoms. See also <b>set bondPicking</b> and <b>set drawPicking</b>.','*150','atomPicking','TRUE')
newCmd('','11.1.22','','*v+11.2;','Setting this parameter TRUE adjusts the rate of frame changes in an animation to a lower rate if the rendering cannot keep up with the frame changing.','*11','autoFPS','FALSE')
newCmd('','11.1.20','','*v+11.2;','Sets the color of all axes. (Same as {#.color~color axes …}.)','*120','axesColor','&quot;color_name&quot;')
newCmd('','11.1.20','','*v+11.2;','Sets the color of the X axis.','*121','axis1Color','&quot;color_name&quot;')
newCmd('','11.1.20','','*v+11.2;','Sets the color of the Y axis.','*13','axis2Color','&quot;color_name&quot;')
newCmd('','11.1.20','','*v+11.2;','Sets the color of the Z axis.','*14','axis3Color','&quot;color_name&quot;')
newCmd('',' ','','*v+11.4;','Sets a particular model or animation frame to be fixed in place during an {#.animation~} or when displaying models {#.load~loaded} from a multi-model file. <b>set backgroundModel 0</b> turns this feature off. For a multifile context, the model must be give in "file.model" format. Certain restrictions apply to scripts when a background model is displayed; in that case it may be important to turn the model off and then back on during selected scripting. ','*15','backgroundModel','(integer >= 1) or &quot;file.model&quot;')
newCmd('','11.6.RC13','','*v+11.6;','Setting this parameter TRUE allows picking of bonds. If pickCallback is enabled, a message is sent to the callback function providing detailed information about the bond that was picked.  See also <b>set atomPicking</b> and <b>set drawPicking</b>.','*17','bondPicking','FALSE')
newCmd('',' ','','*v+11.4;','Jmol can be set to read the chain designations in PDB, mmCIF, and related files either with or without case sensitivity. With the default <b>set chainCaseSensitive FALSE</b>, the chain designations are interpreted as case-insensitive. With <b>set chainCaseSensitive ON</b>, the chain designation is read in a case-sensitive manner -- chain "A" is different than chain "a". This supports {http://www.rcsb.org/pdb/docs/format/pdbguide2.2/guide2.2_frame.html~PDB format} model files with more than 26 chains. The default startup up mode is OFF -- chain designation "a" in a SELECT command will refer to chain "A" in a file.','*2','chainCaseSensitive','FALSE')
newCmd('',' ','','*v+11.2;','Sets the default color scheme to be the traditional Rasmol/Chime scheme or the newer, more subtle, Jmol scheme. This command does not actually change the display for an object unless that object is currently being displayed using the default color scheme. See the {http://jmol.sourceforge.net/jscolors/~Jmol Colors page} for default color scheme details.','*31','defaultColorScheme','JMOL or RASMOL')
newCmd('','11.3.17','','*v+11.4;','Sets the default scale for the arrow tip for {#.draw} arrows.','*32','defaultDrawArrowScale','(decimal)')
newCmd('',' ','','*v+11.4;','Sets the overall defaults to be Jmol standard or more similar to Rasmol, including default color scheme, axes orientation, zero-based XYZ numbering, no spacefill, and simple wireframe. Applied immediately; should be used prior to file loading.','*330','defaults','JMOL or RASMOL')
newCmd('','13.1.16','','*v+13.2;','Sets zoomHeight true and measurementUnits Angstroms','*331','defaults','PyMOL')
newCmd('','11.5.12','','*v+11.6;vdw','Sets the default van der Waals radius set for {#.spacefill~} and related commands. For a detailed listing of the values used, see {misc/vdw_comparison.xls~vdw_comparison.xls}. USER refers to a set of data provided by the user using the {#.data~} command.','*34','defaultVDW','JMOL or BABEL or RASMOL or USER')
newCmd('',' ','','*v+11.2;','Setting this parameter TRUEprovides an alternative way to set the default color scheme to RasMol; testable with Jmol math.','*30','colorRasmol','FALSE')
newCmd('','11.5.RC3','','*v+11.6;dots','When Jmol displays {#.dots~}, the density of dots is determined by the scale of the display. At a small scale, Jmol may display as few as 12 dots; as zoom increases, this number increases to 42, then 162, and finally, at high zoom, it becomes 642 dots. The <b>dotDensity</b> setting allows control over how many dots are displayed at the various scale level cutoffs. (The actual calculation does not use zoom; rather, it uses a measure of pixels per micron). Setting <b>dotDensity</b> to -3 results in Jmol <i>always</i> displaying 12 dots; setting it to 3 results in Jmol <i>always</i> displaying 642 dots. Settings in between these values decrease or increase the number of dots relative to the default setting (0). ','*351','dotDensity','-3 to 3')
newCmd('','12.0.RC24','','*v+12.0;dots;isosurface;mo','Sets the size of {#.dots~} and the dots of {#.isosurface~isosurfaces} and {#.mo~Mos}. ','*352','dotScale','(integer)')
newCmd('',' ','','*v+11.0;dots','Setting this paramter TRUE instructs the {#.calculate~calculate surface} command to disregard atoms that are not selected when calculating the position of the surface (which then determines the parameter {#.atomexpressions~surfaceDistance} for each atom). Also tells Jmol to ignore nonselected atoms when creating a {#.dots~dot surface}. <b>set dotsSelectedOnly TRUE</b> would allow, for example, a continuous set of dots around a ligand even if it is in contact with a protein.','*353','dotsSelectedOnly','FALSE')
newCmd('',' ','','*v+11.0;dots','Setting this parameter OFF instructs Jmol to draw complete spheres of {#.dots~} around each selected atom rather than only the dots that would be derived from that atom in a molecular "dot surface."','*354','dotSurface','ON')
newCmd('',' ','','*v+11.4;','When ON, allows the user to move selected atoms by pressing ALT-SHIFT-LEFT and dragging; when combined with <b>set pickingstyle SELECT DRAG</b>, just LEFT-dragging moves the atoms, and the ALT and SHIFT keys are not required.','*40','dragSelected','OFF')
newCmd('','13.3.6','','*v+13.4;draw','Sets the font size for {#.draw~} object text.','*411','drawFontSize','14')
newCmd('',' ','','*v+11.4;draw','When ON, and the user hovers over a point associated with {#.draw~} object, the name of the object is displayed next to the point.','*412','drawHover','OFF')
newCmd('',' ','','*v+11.4;draw','When ON, Jmol reports picking of points associated with {#.draw~} objects to the console, the messageCallback function, and the pickCallback function. In addition, setting drawPicking true enables measurement of distances and angles involving drawn objects such as points, lines, arrows, and planes. Measurements of this type only appear transiently; they are not saved.  See also <b>set atomPicking</b> and <b>set bondPicking</b>.','*42','drawPicking','OFF')
newCmd('','12.3.26','','*v+13.0;','Sets the units of energy for minimization reports in kJ/mol or kcal/mol.','*430','energyUnits','kJ or kcal')
newCmd('','11.3.33','','*v+11.4;','Sets (or allows you to inspect, expand, or limit) the list of export drivers available to Jmol. ','*431','exportDrivers','&quot;driver_list&quot;')
newCmd('','13.1.19','','*v+13.2;','Sets the export scale for VRML and X3D files created with the {#.write~} command. This setting allows variable scale export to 3D printer CAD software. Defaults to 1.0.','*432','exportScale','(decimal)')
newCmd('','12.3.26','','*v+13.0;','Sets the forcefield for minimization to MMFF94 (default) or UFF. If MMFF94 is chosen and any atoms are of a type not recognized by MMFF94, then UFF is used instead.','*441','forcefield','&quot;MMFF&quot; or &quot;UFF&quot;')
newCmd('','','','*v+11.0;','Sets the formal charge for the selected atoms.','*442','formalCharge','(integer)')
newCmd('','12.0.RC27','','*v+12.0;','Sets the meaning of {{1/2 1/2 1/2}} to be relative to the current {#.unitcell~} (TRUE) or not (FALSE, default).','*451','fractionalRelative','FALSE')
newCmd('','11.8.RC2','','*v+11.8;','Sets the step for calculating straightness and for {#draw~draw helix axis}. The default value is 1, but a value of 2, for example, allows calculating straightness and axes for structures based on pairs of residues. ','*452','helixStep','(integer)')
newCmd('',' ','','*v+11.0;','Sets the web page for the {#.help~} command.','*453','helpPath','&quot;URL&quot;')
newCmd('',' ','','*v+11.2;','Sets the length of time in seconds before a hovering action is acknowledged.','*46','hoverDelay','(decimal)')
newCmd('',' ','','*v+11.2;','Sets the label displayed upon hovering.','*471','hoverLabel','(string)')
newCmd('','11.8.RC6','','*v+11.8;','The <b>imageState</b> property, when TRUE, allows Jmol to insert into JPG and PNG files its state. This allows images to serve both as 2D and 3D models.','*472','imageState','ON')
newCmd('',' ','','*v+12.0;','For a multi-touch screen, set this parameter ON if the Jmol applet is running in a browser with the kiosk mode enabled, indicating there are no other applets or applications running. Starting with Jmol 12.2.9 and 12.3.9 this is a one-time setting -- Setting this parameter OFF is the same as setting it ON, and setting it ON disallows file saves other than logging, prompt dialogs, console, popup menu, and the application script editor. ','*481','isKiosk','OFF')
newCmd('','12.2.RC5','','*v+12.2;','Displays a color key relating to the last-generated isosurface. ','*482','isosurfaceKey','OFF')
newCmd('',' ','','*v+11.4;','In the case of an isosurface that is mapped using atom-based property data, the default action is to smooth out the coloring based on an averaging of color weighted by a factor of 1/distance^4 to atoms. Turning this parameter OFF tells Jmol not to do this and instead produce a patchwork mapping that assigns a color based on the property of only the nearest atom. ','*483','isosurfacePropertySmoothing','ON')
newCmd('',' ','','*v+11.4;','When isosurfacePropertySmoothing is set ON, Jmol applies a function to data so as to reduce the harshness of the edges between data assigned to specific atoms. The setting of isosurfacePropertySmoothingPower allows adjustment of that smoothing from 0 (no smoothing) to 10 (strong smoothing). The default value is 7.','*484','isosurfacePropertySmoothingPower','0 to 10 (7)')
newCmd('',' ','','*v+11.4;','Sets the maximum distance away from a point that that an atom can be found when applying atom data using the {#.load [property]~} command. Default 0.01 Angstroms.Applies the data to all atoms in similar unit cells if the data being read . This allows applying the same data to atoms in all unit cells.','*49','loadAtomDataTolerance','(decimal)')
newCmd('',' ','','*v+11.0;','In situations where there are multiple models, typically only one model is displayed at any given time. In that situation, if a user clicks on a pair of atoms to measure a bond distance, then only one measurement is made. Using <b>set measureAllModels ON</b>, when the user makes measurements in any one frame, ALL similar measurements in all models in hidden frames are generated at the same time. <b>set measureAllModels ON</b> thus allows for very efficient measuring and investigation of differences in a bond distance or bond angle or dihedral angle across multiple models.  ','*500','measureAllModels','OFF')
newCmd('','12.3.29','','*v+13.0;','Adjusts the width of the lines in isosurface and molecular orbital MESH display','*5011','meshScale','1')
newCmd('','','','*v+11.6;','Changes the style of message reporting by script callbacks to one similar to Chime. ','*5012','messageStyleChime','FALSE')
newCmd('','','','*v+11.6;','Sets the criterion for stopping a {#.minimize~minimization}. The default value is 0.001; the minimum value is 0.0001.','*502','minimizationCriterion','(decimal)')
newCmd('','','','*v+11.6;','Set this flag FALSE to not display {#.minimize~minimizations} as they occur.','*503','minimizationRefresh','TRUE')
newCmd('','12.0.RC5','','*v+12.0;','Turns off messages reporting energies for the minimization.','*5041','minimizationSilent','FALSE')
newCmd('','','','*v+11.6;','Sets the number of steps after which a {#.minimize~minimization} will stop even if the desired criterion has not been reached. Default value is 100.','*5042','minimizationSteps','(integer)')
newCmd('','12.2.RC6','','*v+12.2;','The minimum number of pixels a click can be away from an atom on the screen before it is considered too far away to be a click on that atom.','*5043','minPixelSelRadius','6')
newCmd('','','','*v+12.0;','Sets the sensitivity of the mouse when dragging.  The default value of 1.0 for set mousedragFactor allows a 180-degrees rotation when the pointer drags across the full window width.','*505','mouseDragFactor','(decimal)')
newCmd('','','','*v+12.0;','Sets the sensitivity of the mouse wheel. The default value of 1.0 for set mousedragFactor allows a 180-degrees rotation when the pointer drags across the full window width.','*506','mouseWheelFactor','(decimal)')
newCmd('','12.0.RC6','','*v+11.6;','Turns  on parallel multiprocessing. If this setting cannot be set true, then it means you do not have a multiprocessor machine. Used in association with the {#.parallelprocess~parallel} and {#.parallel/processs~process} commands.','*508','multiprocessor','FALSE')
newCmd('','12.3.3','','*v+13.0;','Determines the URL used for <b>SHOW NMR</b>.','*453','nmrUrlFormat','&quot;URL&quot;')
newCmd('','','','*v+11.6;','Setting this flag TRUE tells Jmol to keep the header portion of the last-loaded PDB file in memory for later retrieval using getProperty("fileHeader"). If FALSE (default), the header is not saved, and a call to this function is slower, since it requires reloading the PDB file and parsing it for its header.','*511','pdbGetHeader','FALSE')
newCmd('','','','*v+11.6;','Setting this flag TRUE tells Jmol to assume the groups listed for a given chain in a PDB file are in order already and that Jmol should not check for proper inter-group bonding when assigning polymer sequences that form the basis of secondary structure. This flag allows for custom PDB files where the groups of a chain may not be physically adjacent, yet it is still desired to represent them as single structures.','*512','pdbSequential','FALSE')
newCmd('','11.1','','*v+11.2;vdw','The default size of an atom created when a file is loaded as a percent of the atom\'s van der Waal radius (Jmol standard value: 20).','*52','percentVdwAtom','(integer)')
newCmd('','11.5.42','','*v+11.6;','Sets the format of the message sent to the console and callback functions when an atom is clicked.','*54','pickLabel','(string)')
newCmd('','11.1','','*v+11.2;','The rate of spinning that occurs when a user clicks on the end of a line created using {#.draw~} (default: 20). ','*53','pickingSpinRate','(integer)')
newCmd('','11.5.52','','*v+11.2;pg','Sets the tolerance for two atoms being considered identical after application of a rotation. See {#.calculate~calculate pointgroup} for details.','*541','pointGroupDistanceTolerance','(decimal)')
newCmd('','11.5.52','','*v+11.2;pg','Sets the tolerance for two axes being considered identical prior to application of a rotation. See {#.calculate~calculate pointgroup} for details.','*543','pointGroupLinearTolerance','(decimal)')
newCmd('','11.9.23','','*v+12.0;','This option turns off many memory-consuming features of Jmol that are necessary for preserving the state. It can be used in situations where memory is at a premium and there is no desire to write or save the current Jmol state. (Jmol 12.0)','*550','preserveState','TRUE')
newCmd('','','','*v+11.2;','An integer value of 0 or greater. Sets the number of columns to be read for <b>propertyAtomNumberField</b>.','*551','propertyAtomNumberColumnCount','(integer)')
newCmd('','','','*v+11.2;','An integer value of 0 or greater. Sets the column (when <b>propertyAtomNumberColumnCount</b> > 0) or free-format field (otherwise) for the atom numbers in a {#.data~} set. These are the atom numbers designated in a PDB file or numbers starting with 1 otherwise. Setting the field to 0 indicates that there is no such field, and values should be read in sequentially.','*552','propertyAtomNumberField','(integer)')
newCmd('','','','*v+11.2;','Sets the color scheme associated with properties. SchemeName, in quotes, may be one of "roygb" (default rainbow), "bwr" (blue-white-red), "rwb" (red-white-blue),"low" (red-green), "high" (yellow-blue), "bw" (black-white) and "wb" (white-black), but may be any user-defined color scheme. In Jmol 11.4, this parameter was largely replaced by <b>color "schemeName"</b>.','*56','propertyColorScheme','&quot;colorSchemeName&quot;')
newCmd('','','','*v+11.2;','An integer value of 0 or greater. Sets the number of columns to be read for <b>propertyDataField</b>.','*571','propertyDataColumnCount','(integer)')
newCmd('','','','*v+11.2;','An integer value of 0 or greater. Sets the column (when <b>propertyAtomNumberColumnCount</b> > 0) or free-format field (otherwise) for the property data in a {#.data~} set. Setting this value to 0 indicates that values are simply to be read sequentially from the data.','*572','propertyDataField','(integer)')
newCmd('','11.5.38','','*v+11.6;','Specifies the axes used to define the right-handed {#.quaternion~} frame for proteins and nucleic acids .  These frames (xyz axes and origin) define orientations of amino acid residues in proteins and nucleic acid residues in RNA and DNA. The minimum definition of a frame is an origin (only used for {#.draw~draw quaternion}), a point that will define an axis (usually X, but in some cases Y), and a third point that will be used to generate the other two axes. Specifically, if O, A, and B are three points and  V<sub>OA</sub> defines the direction of the X axis, then V<sub>C</sub> = V<sub>OA</sub> x V<sub>OB</sub> defines the direction of the Z axis, and V<sub>C</sub> x V<sub>OA</sub> defines the direction of the Y axis. If V<sub>OB</sub> defines the direction of the Y axis, then V<sub>C</sub>, as above, defines the direction of the Z axis, and V<sub>OB</sub> x V<sub>C</sub> defines the direction of the X axis. Different frames can used for different purposes, ranging from analyzing amino acid side-chain orientations to quantifying the "straightness" of helices and sheets. Most useful are relative and absolute differences of quaternions, which define local helical axes or the axis, angle, and translation required to move from one orientation to another.<br/>[||A| "alternative"| for proteins, same as N (see below)|for nucleic acids, an alternative backbone quaternion with O = P[i], B(Y) = C4\'[i], and A =  C4\'[i-1], which is closely associated with eta and theta dihedral angles. ||B| "backbone"| for proteins,  O = CA[i], A(X) = CA[i+1], B = CA[i-1]| for nucleic acids, O = P[i], A(X) = P[i+1], B = P[i-1] ||C| "carbon"| for proteins,  O = CA, A(X) = C, and B = N, thus defining the orientation of the peptide sidechains|for nucleic acids, same as N, but with the origin shifted to the "center of heavy atoms" for the base (all atoms associated with the base other than H or C4\')||N|"nitrogen"|a frame based on the peptide N atom and useful specifically in the area of solid state NMR spectroscopy. O = N, and if V<sub>C</sub> = V<sub>N-NH</sub>, and V<sub>D</sub> = V<sub>N-CA</sub>, then V<sub>B</sub>(Y) = V<sub>A</sub> x V<sub>D</sub>, M = axisAngle(V<sub>B</sub>, -17 degrees), then V<sub>A</sub> = MV<sub>C</sub>, and V<sub>C</sub> = V<sub>A</sub> x V<sub>B</sub> defines the direction of the Z axis.|for nucleic acids, O = N9(purines) or N1(pyrimidines), B(Y) = O + V<sub>C1\'O</sub>, and A = V<sub>O-C2</sub> (toward the Watson-Crick edge)||P| "peptide/phosphorus" | for proteins, O = C, A(X) = CA, and Y = N[i+1], thus defining the orientation of the peptide plane| for nucleic acids, O = P[i], A(X) = O3\'[i-1], and B = O5\'[i], thus defining the orientation of the phosphorus tetrahedron||Q|"Quine"| defined as follows: O = (C[i] + N[i+1])/2 and if V<sub>X</sub> = V<sub>CA[i]-C[i]</sub> and V<sub>B</sub> = V<sub>N[i+1]-CA[i+1]</sub>, then V<sub>Z</sub> = V<sub>X</sub>  x V<sub>B</sub>, and V<sub>Y</sub>= V<sub>Z</sub> x V<sub>X</sub>. This frame was an early definition of the orientation of the peptide plane and suffers from the fact that these two vectors are very nearly colinear, and can produce Z directions that are misdirected; provided for historical reference only (J. R. Quine, Journal of Molecular Structure: THEOCHEM, Volume 460, Issues 1-3, 26 February 1999, pages 53-66).|not used for nucleic acids||RC and RP|"ramachandran"|These frame selections specify that straightness should be calculated from Ramachandran angle approximations rather than actual quaternionFrames "C" and "P", respectively|no nucleic acid equivalent||X|"experimental"|Jmol development testing|Jmol development testing||]','*58','quaternionFrame','A,B,C,N,P,Q,RC,RP,X')

newCmd('',' ','','*v+11.0;','When this flag is set to TRUE, the range for {#color~} temperature is set to be adjustable based on the selected set of atoms being colored; when false (the defualt), the range of colors is set based on the range of values in the entire model. Starting with Jmol 12.0, this parameter applies ot any property.','*59','rangeSelected','')
newCmd('','12.0.RC4','','*v+12.0;','Sets the number of milliseconds to wait before a timer expires signalling that Jmol should not wait any longer for a repaint operation. In some large memory-intensive operations, it is sometimes advisable to set this parameter to a higher number.','*60','repaintWaitMs','1000')
newCmd('','11.9.13','','*v+12.0;','Generally when Jmol {#.write~writes} the state, helix/sheet/turn data is saved. In some cases this may not be desired. Setting this flag FALSE prevents the saving of this information in the state.','*61','saveProteinStructureState','TRUE')
newCmd('','11.5.52','','*v+11.6;','Generally when selections are made in Jmol all matching atoms in all models are selected, regardless of which model is currently in {#.frame~}. When set FALSE, this flag indicates that the subset of atoms to select should be automatically set to the current model when frame changes occur.','*62','selectAllModels','TRUE')
newCmd('','','','*v+11.2;','When turned OFF, selections do not select HETATM atoms in PDB files','*63','selectHetero','ON')
newCmd('',' ','','*v+11.2;','When turned OFF, selections do not select hydrogen atoms','*64','selectHydrogen','ON')
newCmd('',' ','','*v+11.8;','When set TRUE,  and {#.set_allowKeyStrokes~} is TRUE, key strokes in the applet or application window to be interpreted as script commands and displayed in the bottom left corner of the window.','*65','showKeyStrokes','TRUE')
newCmd('','13.1.6','','*v+13.2;','When set TRUE, reports timings for isosurface creation  in ms in the Jmol console, and in the Java console reports detailed timings for rendering of shapes.','*66','showTiming','FALSE')
newCmd('','12.1.10','','*v+12.2;','Sets a slab range that is independent of zoom. ','*67','slabRange','(decimal)')
newCmd('','11.3.29','','*v+11.2;','Turning the smartAromatic parameter OFF reverts to a Jmol 10 style of drawing aromatic bonds as a paired solid and dashed line when loading subsequent files. (This command has no immediate effect. Use {#.calculate~reset aromatic;calculate aromatic} to see the effect of smartAromatic)','*68','smartAromatic','ON')
newCmd('',' ','','*v+11.2;','determines the number of frames per second between displays of the molecule -- a small number here results in a jerky stepwise rotation. ','*82','spinFps','._spin_fps')
newCmd('',' ','','*v+11.2;','The <b>set spinX</b>,  <b>set spinY</b>, and <b>set spinZ</b> commands allow for the setting of the spin axes -- x, y, and z -- independently as well as the rate of spin. The actual spinning axis is a complex combination of the three settings. No actual spinning occurs until the {#.spin~spin ON} command is issued or the user turns spinning on using the mouse menu. Jmol has a much richer variety of spinning possibilities to this Chime/Rasmol-standard capability, allowing simple spinning around arbitrary molecular- and window-based axes. See the {#.rotate~} command.','*84','spinX','._spin_rate')
newCmd('',' ','','*v+11.2;','see <b>set spinX</b>','*86','spinY','._spin_rate')
newCmd('',' ','','*v+11.2;','see <b>set spinX</b>','*87','spinZ','._spin_rate')
newCmd('',' ','','*v+11.4;','Displays the version of Jmol used to create the most recently run state script (a parameter rather than a read-only variable only because a script has to create it).','*88','stateVersion','(integer)')
newCmd('',' ','','*v+11.2;','When set OFF, this parameter prevents Jmol from recording status messages and reporting them to a requesting web page. When OFF, the JavaScript method jmolScriptWait() cannot return status from the Jmol appet. ','*891','statusReporting','ON')
newCmd('',' ','','*v+11.0;','Same as {#.stereo~stereo (degrees)}, sets the angle for stereo images; can be checked using Jmol math.','*893','stereoDegrees','(decimal)')
newCmd('','11.3.57','','*v+11.4;','For {#.calculate~calculate STRUTS} and the {#.struts~} command, sets the default radius for new struts. (Jmol 12.0)','*8941','strutDefaultRadius','0.3')
newCmd('','11.3.57','','*v+11.4;','For {#.calculate~calculate STRUTS}, sets the maximum length for a new strut. (Jmol 12.0)','*8942','strutLengthMaximum','7')
newCmd('',' ','','*v+11.2;','For {#.calculate~calculate STRUTS}, turn this flag ON to allow more than one strut on a given atom. (Jmol 12.0)','*8943','strutsMultiple','OFF')
newCmd('',' ','','*v+11.0;','For {#.calculate~calculate STRUTS}, sets the minimum spacing (in terms of residues) for new struts. (Jmol 12.0)','*8944','strutSpacing','6')
newCmd('','11.3.57','','*v+11.4;','When {#.sync~} is ON, delivers mouse actions to the target applet.','*894','syncMouse','OFF')
newCmd('','11.3.57','','*v+11.4;','When {#.sync~} is ON, delivers script events to the target applet.','*895','syncScript','OFF')
newCmd('','13.1.6','','*v+13.2;','Reports various timing information to the Java/JavaScript console.','*896','timing','FALSE')
newCmd('','','','*v+12.0;','Setting this flag FALSE disables the <b>undo/redo</b> buttons on the Jmol application console, saving memory.','*901','undo','ON')
newCmd('','11.7.40','','*v+11.8;','Generally the {#.minimize~} command runs in a separate thread. Setting this parameter FALSE instructs Jmol to use the same thread as is being used for commands. This is important in situations where one wants to wait for completion of the minimization before continuing.','*905','useMinimizationThread','ON')
newCmd('',' ','','*v+11.0;','Currently only applicable in the display of unit cell parameters, the <b>useNumberLocalization</b> setting determines whether or not local number formatting will be used (comma as decimal point, for example).','*91','useNumberLocalization','ON')
newCmd('',' ','','*v+11.0;','Sets the scale for vibration vectors.','*92','vectorScale','(decimal)')
newCmd('',' ','','*v+12.2;','Sets the default VDW radius to be based on Jmol values (OpenBabel or Rasmol, depending  upon file type) or the set used by the Duke University {http://molprobity.biochem.duke.edu/~molProbity} site as  per <b>Word, et al, J. Mol. Biol. (1999) 285, 1711-1733</b>: [||  1.0 | {{_H}}||  1.17 | {{_H and connected(_C) and not connected(within(smiles,"[r6]"))}}||  1.75 | {{_C}}||  1.65 | {{_C and connected(3) and connected(_O)}}||  1.55 | {{_N}}||  1.4 | {{_O}}||  1.8 | {{_P}}||  1.8 | {{_S}} ||]','*931','vdw','JMOL or PROBE')
newCmd('','12.3.2','','*v+13.0;','Displays vibration vectors as double-ended arrows.','*932','vectorSymmetry','FALSE')
newCmd('',' ','','*v+11.0;','Sets the default period of vibrations (in seconds). Setting this value to 0 disables vibration modeling.','*933','vibrationPeriod','(decimal)')
newCmd('',' ','','*v+11.0;','Sets the amplitude of vibrations.','*95','vibrationScale','(decimal)')
newCmd('','11.9.24','','*v+12.0;','Setting this parameter OFF allows {#.moveTo~} operations to continue independently while script commands are processing. An ongoing moveTo operation can then be stopped at any time using <b>moveTo STOP</b> [Jmol 12.0]','*96','waitForMoveto','ON')
newCmd('','11.3.55','','*v+11.4;','Turning this parameter ON sets Jmol to not display spacefill and only display bonds as simple lines during user-based model manipulation. ','*97','wireframeRotation','OFF')
newCmd('.set modelKitMode','12.0.RC15','','*v+12.0 -- new modelKitMode;','Jmol 12.0 includes a mode that allows building and modifying models, with minimization. Features include the ability to easily add, drag, and delete atoms, drag and minimize molecules and molecular fragments such as CH<sub>3</sub> groups, rotate bonds, minimize parts of models -- all mouse-driven. A special menu is employed that sits at the top left corner of the Jmol window. (Clicking the Jmol frank still brings up the standard menu.) After <b>set modelKitMode</b>, a {#.zap~} command creates a CH<sub>4</sub> model, and the picking mode goes to <b>set picking assign_C</b>. ','1|2','','')

newCmd('.set picking',' ','','*v+12.0 -- adds set picking measure SEQUENCE;','The <b>set picking</b> command determines the response to clicking of atoms by the user. For those options for which they apply, the keywords <b>MEASURE</b> and <b>SELECT</b> are optional.','2|2','','')
newCmd('',' ','','','Setting this paramter OFF turns off the sending of picking information (atom identifier, atom number, and atomic position) to both to the status line and to the {#.set_pickCallback~PickCallback} function, if defined.','*1','ON','')
newCmd('',' ','','','When the user clicks an atom, that atom is set to be the center of rotation. If windowCentered is true (default), then this atom is smoothly moved to the window center; if not windowCentered, then the atom stays in position, but now all rotation is around it and if perspectiveDepth is set (<b>set windowCentered off;set perspectiveDepth on</b>), the perspective around this atom shifts to indicate that it is now the focus of the perspective. If the same atom is clicked twice, it is zoomed in on by a factor of two. If no atom is clicked, then the model is zoomed out by a factor of two. ','*4','CENTER','')
newCmd('','11.9.21','','','Performs like measuring distances, except bonds are created. (Jmol 12.0)','*50','CONNECT','')
newCmd('','11.9.28','','','Clicking on an atom deletes it. (Jmol 12.0)','*51','DELETEATOM','')
newCmd('','11.9.21','','','Deletes the bond between pairs of clicked atoms (and with <b>set BONDPICKING TRUE</b>, deletes bonds as they are clicked.. (Jmol 12.0)','*52','DELETEBOND','')
newCmd('','12.0.RC15','','','Clicking and dragging an atom moves its position.','*54','DRAGATOM','')
newCmd('','13.1.17','','*v+13.2;','Clicking and dragging an ligand moves its position.','*55','DRAGLIGAND','')
newCmd('','12.0.RC15','','','Clicking and dragging an atom moves its position and then does a minimization of its associated molecule.','*56','DRAGMINIMIZE','')
newCmd('','12.0.RC15','','','Clicking and dragging an atom moves and minimizes its associated molecule. This setting can be used to simulate docking. ','*57','DRAGMINIMIZEMOLECULE','')
newCmd('','12.1.46','','*v+12.2;','Clicking and dragging on any atom of the structure drags the selected atoms only; using SHIFT with the drag rotates the selected atoms about their center. (Jmol 12.2)','*58','DRAGSELECTED','')
newCmd('',' ','','','The vertices of all {#.draw~} objects are highlighted. Holding the SHIFT key down while dragging moves the object to a new location ("shifts" the object); holding ALT down while dragging moves a single vertex to a new location ("alters" the shape of the object). The draw command required to reproduce the object can be seen immediately using the Java console. This new command can also be seen via message callback, or in the Jmol {#.console~} using {#.show~show DRAW}.','*63','DRAW','')
newCmd('',' ','','','Same as <b>set picking ON</b>.','*65','IDENT','')
newCmd('','12.0.RC15','','*v+12.0;','Selecting an atom that is part of a ring after <b>set picking invertStereo</b> will reverse the two non-ring atoms -- actually rotating them 180 degrees, not doing a planar inversion, thus preserving whatever chirality might be attached to them. (See also {#.invertSelected~invertSelected STEREO}.','*67','INVERTSTEREO','')
newCmd('',' ','','','When the user clicks an atom, the label of that atom is toggled on or off. With  Jmol 12.2, pressing SHIFT while dragging a label changes its position relative to the atom.','*69','LABEL','')
newCmd('',' ','','','Same as <b>set picking MEASURE DISTANCE</b> but also displays a distance measurement on the molecule.','*70','MEASURE','')
newCmd('',' ','pickd','','Turns picking on and returns atom identities and distance between two atoms. Three messages are sent to the {#.set_messageCallback~MessageCallback} function, if defined: Atom #1 (after the first click) and then Atom #2 and Distance (after the second click).','*71','MEASURE','DISTANCE')
newCmd('',' ','picka','','Turns picking on and returns atom identities and angle involving three atoms. Four messages are sent to the {#.set (callback)~MessageCallback} function, if defined: Atom #1 (after the first click), Atom #2 (after the second click), and then Atom #3 and Angle (after the third click).','*72','','ANGLE')
newCmd('',' ','pickt','','Turns picking on and returns atom identities and torsion angle (dihedral angle) involving four atoms. Five messages are sent to the {#.set_messageCallback~MessageCallback} function, if defined: Atom #1 (after the first click), Atom #2 (after the second click), Atom #3 (after the third click), and then Atom#4 and Torsion (after the fourth click).','*74','','TORSION')
newCmd('',' ','','*v+12.0;','Turns picking on and returns the 1-letter-code sequence of atoms in bioSMILES format, showing chain and residue range and biomolecular type (p for protein, d for dna, r for rna). For example: <b>//* chain A protein 36 *// ~p~PG //* 37 *//</b>.','*75','','SEQUENCE')
newCmd('',' ','','','User mouse clicking sets the navigation center to the selected position. See {#.navigation}.','*83','NAVIGATION','')
newCmd('',' ','','','When the user clicks an atom, the selection of that atom is toggled on or off.','*841','SELECT','ATOM')
newCmd('',' ','','','When the user clicks an atom, the selection of the chain associated with that atom is toggled on or off. ','*842','','CHAIN')
newCmd('',' ','','','When the user clicks an atom, the selection of the group associated with that atom is toggled on or off.','*843','','GROUP')
newCmd('',' ','','','When the user clicks an atom, the selection is toggled on or off for all VISIBLE atoms with the same element as the selected atom.','*844','','ELEMENT')
newCmd('','11.9.24','','*v+12.0;','When the user clicks an atom, the selection is toggled on or off for all atoms within the same polymer unit (PDB only).','*846','','POLYMER')
newCmd('','11.9.24','','*v+12.0;','When the user clicks an atom, the selection is toggled on or off for all atoms within the same structural unit (helix, sheet, turn -- PDB only).','*89','','STRUCTURE')
newCmd('',' ','','*v+11.0;','When the user clicks an atom, the selection is toggled on or off for all atoms within the same covalenty bonded set as the selected atom.','*845','','MOLECULE')
newCmd('',' ','','*v+11.0;','When the user clicks an atom, the selection is toggled on or off for all VISIBLE atoms with the same crystallographic site as the selected atom.','*85','','SITE')
newCmd('',' ','','*v+11.0;','Turns picking on and sets it so that when the user clicks two atoms in succession, the model starts to spin around the axis defined by those two atoms. A third click on any atom stops rotation. Optionally, a speed of rotation can be included.','*90','SPIN','._spin_fps')
newCmd('',' ','','*v+11.0;','When the user clicks two atoms, a {#.struts~strut} is created between them.','*93','STRUTS','')
newCmd('',' ','','*v+12.0;','When the user clicks two atoms of a crystallographic file, a representation of their symmetry relationship is drawn, and the operator(s) relating these positions is written to the console. ','*94','SYMMETRY','')
newCmd('.set pickingStyle',' ','','*v+11.0 -- NEW;','With <b>set pickingStyle</b> you can change the behavior of Jmol in response to user mouse actions relating to selecting atoms or measuring distances, angles, and torsions. <br /> [||<b>SELECT</b>|Sets the behavior of Jmol to different clicking styles. For the standard Jmol default selection behavior, use <b>set pickingStyle SELECT toggle</b> The actions of these options depend upon the setting of {#.set picking~set picking SELECT}. If picking has not been set to SELECT, then this command has no immediate effect. In the explanations given below, it is presumed that we have <b>set picking SELECT GROUP</b>. The <b>SELECT</b> keyword is recommended but not required.||<b>MEASURE</b>|<b>set pickingStyle MEASURE ON</b> in conjunction with <b>set picking MEASURE</b> displays the measurements on the structure as they are generated by user clicking. If picking has not been set to MEASURE, then this command has no immediate effect. <b>set pickingStyle MEASURE OFF</b> returns to the default Jmol behavior.||]','0|1','','')
newCmd('',' ','','*v+11.0;','left-click toggles that group selected/not selected (Chime style; Jmol default).','*1','SELECT','toggle')
newCmd('',' ','','*v+11.0;','left-click selects just that group<br />shift-left-click toggles the group selected/not selected (Rasmol style).','*2','','selectOrToggle')
newCmd('',' ','','*v+11.0;','User mouse action is according to the following scheme, which is the style used by {http://pfaat.sourceforge.net~PFAAT}. <br />[||left-click |selects just that group, like rasmol||shift-left-click|toggles the group selected/not selected||alt-left-click |appends the group to the current selection||alt-shift-left-click|removes the group from the current selection||left-click off model |executes (<b>select none</b>) ||]','*3','','extendedSelect')
newCmd('','11.9.24','','*v+12.0;','makes the LEFT button a click-and-drag button when associated also with <b>set PICKING select (molecule, group, chain, etc.)</b> and <b>set dragSelected</b>.','*4','','DRAG')
newCmd('',' ','','*v+11.0;','Returns to the Jmol default <b>toggle</b> picking style.','*5','','NONE')
newCmd('',' ','','*v+11.0;','Clicking atoms measures and displays distance, angle, or torsions based on the setting of <b>set picking MEASURE</b>. By default the user sees this information displayed. Setting <b>set pickingStyle measure OFF</b> when <b>set picking MEASURE</b> is set to DISTANCE, ANGLE, or TORSION tells Jmol to stop indicating measurements on the model when the user clicks, even though the distance, angle, or torsion information is still sent to the message queue. ','*6','MEASURE','ON')


newCmd('.show',' ','','*v+13.0 -- adds show chemical, DSSP, NMR;','show sends information about the model to the {#.set (callback)~MessageCallback} function and to the Java or Jmol {#.console~}. If using the application, using <b>Jmol -ionx filename.spt modelfile.xyz > output.txt</b>, you can put a <b>show</b> command in a script file (filename.spt), and have the output directed to output.txt. The command line options used in this example include -i (silent startup), -o (use standard output), -n (no display), -x (exit after running the specified script file). All of the parameters that can be {#.set~} can be shown. They+I635 are not listed here.','0|1','','')
newCmd('.show',' ','','*v+11.6 -- adds show moveto, show rotation, show translation, show atom, show chain, show group, show selected;*v-13.0;','show sends information about the model to the {#.set (callback)~MessageCallback} function and to the Java or Jmol {#.console~}. If using the application, using <b>Jmol -ionx filename.spt modelfile.xyz > output.txt</b>, you can put a <b>show</b> command in a script file (filename.spt), and have the output directed to output.txt. The command line options used in this example include -i (silent startup), -o (use standard output), -n (no display), -x (exit after running the specified script file). All of the parameters that can be {#.set~} can be shown. They+I635 are not listed here.','0|1','','')
newCmd('',' ','','*v+11.0;','Displays the {misc/JVXL-format.pdf~JVXL} file equivalent. (Not available for all object types.)','*4','ISOSURFACE','')
newCmd('','11.5.45','','*v+11.6;','Lists the atom numbers that are currently selected in the order of atom index; duplicates are not listed twice.','*11','ATOM','')
newCmd('','13.3.6','','*v+13.4;','Operates on the currently selected atoms, reporting a quarternion that would rotate them to their {#.rotate_best~best} orientation for viewing. ','*121','BEST','ROTATION')
newCmd('','13.3.6','','*v+13.4;','Operates on the currently selected atoms, reporting the volume and dimensions of the  best box that would contain the selected atoms. ','*122','BEST','BOX')
newCmd('',' ','','centering;boundbox','Delivers the coordinates of the center coordinate and a directional vector defining a box just perfectly enclosing the model. The vector is determined by taking the min and max values for the atom along each cartesian axis. The center is the geometric center of the model, not the default center of rotation (which is the mean atom position). The eight corners of the boundbox are found by adding the center point to the vector, with all possible combinations of +/- component cectors. The length of a side of the boundbox is determined by doubling the appropriate component of the vector. So, for example, the length of the boundbox along the x-axis is (2*vectorX). Units are in Angstroms. Output is in the form <br />boundbox: (centerX, centerY, centerZ) (vectorX, vectorY, vectorZ)','*13','BOUNDBOX','')
newCmd('','13.1.2','','*v+13.2;cache','Lists the contents of the {#.cache~}.','*19','CACHE','')
newCmd('',' ','','centering','Delivers the coordinates of the center of the model. Units are in Angstroms. Output is in the form <br />center: (centerX, centerY, centerZ)','*20','CENTER','')
newCmd('','11.5.45','','*v+11.6;','Lists the chains (A, B, C, etc.) of the atoms that are currently selected in the order of atom index; duplicates are not listed twice.','*21','CHAIN','')
newCmd('','12.1.43','','*v+12.2;','Displays a property of the selected atoms depending upon the selected option obtained from the {http://cactus.nci.nih.gov~NIH Catus server} (Jmol 12.2). Note that in addition to these, we have <b>show SMILES</b> and <b>show DRAWING</b>. [||<b>name</b>|list of known names of the compound, if found in the NIH database||<b>inchi</b>|standard {http://www.iupac.org/inchi~InChI} string||<b>inchikey</b>|standard InChI key||]','*221','CHEMICAL','[option]')
newCmd('','11.3.13','','*v+11.4;color','Displays the list of colors comprising a color scheme, where "name" is, for example, "roygb", "rwb", or "user", "byElement_Jmol", "byElement_Rasmol", "byResidue_Shapely" (cooresponding to <b>color shapely</b>), and "byResidue_Amino" (corresponding to <b>color amino</b>). ','*222','COLORSCHEME','&quot;name&quot;')
newCmd('',' ','','*v+11.0;','Displays the most recently read {#.data~} of the given type. See also {#.getProperty~getProperty data}.','*25','DATA','&quot;type&quot;')
newCmd('','12.1.43','','*v+12.2;','Displays a drawing of the Jmol structure in a new browser window.','*305','DRAWING','')
newCmd('',' ','','','Displays the command required to generate the current {#.draw~} object. Particularly useful after using {#.set picking~set picking DRAW}.','*303','DRAW','')
newCmd('','12.0.18','','*v+12.2;','Displays a DSSP {misc/1gua-dssp-jmol.txt~secondary structure analysis report}, which includes regions of three kinds of helix (3/10, alpha, and pi), turns, sheets, and bridge groups (DSSP codes G, H, I, T, E, and B, respectively). Bends (DSSP "S") are not included. This report can be sent to a file using, for example: <b>var x = script("show DSSP");write var x "dsspReport.txt"</b>.','*31','DSSP','')
newCmd('',' ','','','Delivers the entire contents of the file for the current model. ','*32','FILE','')
newCmd('',' ','','','Delivers the entire contents of the specified file on the server from which the applet was loaded. The filename must be relative to the current page (not necessarily the directory containing the applet) and must be enclosed in quotation marks. If the file is in ZIP format -- JAR, ZIP, JMOL -- or contains ZIP data (PNGJ), only a directory of the contained files is reported. ','*33','FILE','filepath')
newCmd('','11.3.23','','*v+11.4;','Lists all user-defined functions.','*53','FUNCTIONS','')
newCmd('','11.5.45','','*v+11.6;','Lists the groups (ALA, VAL, etc.) of the atoms that are currently selected in the order of atom index; duplicates are not listed twice.','*54','GROUP','')
newCmd('',' ','','*v+11.0;','Shows n lines of command history. If no number is given, <b>show history</b> by itself shows the full set of recorded command lines (100 maximum by default, but this maximum can be changed using the {#.set(files and scripts)~set history} command. History recording can be turned on and off using the {#.history~} command.','*63','HISTORY','n')
newCmd('','12.1.51','','*v+12.2;','Displays lighting details, including  ambientPercent, diffusePercent, specular, specularPercent, specularPower, specularExponent, and zShade','*68','LIGHTING','')
newCmd('',' ','','*v+11.0;','Displays a table of information relating to {#.measure~measurements} that have been made.','*70','MEASUREMENTS','')
newCmd('',' ','','*v+11.4;','Displays the current Jmol menu in a format that can be loaded using {#.load MENU~}.','*71','MENU','')
newCmd('',' ','','*v+11.0;','Displays the {misc/JVXL-format.pdf~JVXL} file equivalent for the current {#.mo~molecular orbital}. If no MO is currently shown, displays the JVXL equivalent for the entire set of molecular orbitals. (This can take some time to construct.)','*72','MO','')
newCmd('',' ','','','Delivers properties associated with the currently loaded model. Output includes information about all of the models in the set. This command is still in development. The exact form and content of the output is subject to change (and suggestion).','*73','MODEL','')
newCmd('','11.5.45','','*v+11.6;','Same as <b>show orientation moveto</b>','*75','MOVETO','')
newCmd('','12.3.3','','*v+13.0;','Links to  {http://www.nmrdb.org/predictor~} (or whatever URL is indicated in <b>set nmrUrlFormat</b> using the SMILES string of the selected atoms. (Jmol 13.0)','*76','NMR','')
newCmd('',' ','','','Delivers the orientation of the model. Output consists of both a "moveTo" command and an alternative sequence of "reset; rotate z; rotate y; rotate z" commands that would result in the current orientation. Thus, this command allows reading and restoring specific user-specified orientations. Optional types include: [||moveto|the moveto command leading to this orientation, with no comments ||rotation|the current rotation, as a {#.quaternion~}||translation|the current translation in percent X and percent Y from center of the window||]','*77','ORIENTATION','[optional type]')
newCmd('',' ','','','Delivers the PDB file header.','*831','PDBHEADER','')
newCmd('','11.5.52','','pg','Displays a table summarizing the point group of a symmetrical or nearly symmetrical model. See {#.calculate~calculate pointgroup} for details of this calculation.','*833','POINTGROUP','')
newCmd('','11.5.39','','*v+11.6;','Reports a listing of currently selected residues:<pre>[GLY]1:A[ARG]2:A<br />[ARG]3:A<br />[ILE]4:A<br />[GLN]5:A<br />[GLY]6:A<br />[GLN]7:A<br />[ARG]8:A<br />[ARG]9:A<br />[GLY]10:A<br />[ARG]11:A<br />[GLY]12:A</pre>','*841','RESIDUES','')
newCmd('','11.5.45','','*v+11.6;','Same as <b>show orientation rotation</b>','*843','ROTATION','')
newCmd('','11.5.39','','*v+11.6;','Reports the sequence of the currently selected residues in the following format: <pre>Model 1<br />Chain A:<br />[GLY]1    [ARG]2    [ARG]3    [ILE]4    [GLN]5    <br />[GLY]6    [GLN]7    [ARG]8    [ARG]9    [GLY]10   <br />[ARG]11   [GLY]12   [THR]13   [SER]14   [THR]15   <br />[PHE]16   [ARG]17   [ALA]18   [PRO]19   [SER]20   <br />[HIS]21   [ARG]22   [TYR]23   [LYS]24   [ALA]25   <br />[ASP]26   [LEU]27   [GLU]28   [HIS]29   [ARG]30   <br />[LYS]31   [VAL]32   [GLU]33   [ASP]34   [GLY]35   <br />[ASP]36   [VAL]37</pre>.','*852','SEQUENCE','')
newCmd('','11.5.45','','*v+11.6;','Lists the default label for all selected atoms, in the order of atom index.','*851','SELECTED','')
newCmd('',' ','','*v+11.2;','Delivers a list of all "set" commands that have been issued, with their values.','*861','SET','')
newCmd('',' ','','*v+12.0;','Delivers a SMILES string for the selected atoms. Assumes full molecules; does not report for biomolecules -- for that, use <b>print {{selected}}.find("SMILES", true)</b>.','*863','SMILES','')
newCmd('',' ','','*v+11.0;','Displays information relating to crystallographic space groups. If a file with crystallographic symmetry is loaded, <b>show spacegroup</b> by itself displays information for that spacegroup. Quotes are required around the space group name. If the name itself includes double quotes, use two single quotes instead. For example: P 32 2\'\', not P 32 2". ','*87','SPACEGROUP','&quot;name&quot;')
newCmd('',' ','','*v+11.0;','Delivers a Jmol script that, when run, will regenerate the state of the Jmol applet or application. The following objects are not yet supported: dipole, isosurface, lcaoCartoon, mo. If a name is given, then the state {#.save~saved} with that name is used; if no name is given, the state described is the current state.','*880','STATE','[optional name]')
newCmd('','12.1.14','','*v+12.2;','Shows only the lines from the current state that contain the specified phrase "xxx"','*881','STATE','/xxx')
newCmd('','12.1.48','','*v+12.2;','Delivers the Jmol state embedded in an image file of the format PNG or JPG created by Jmol (and thus containing the string "**** Jmol Embedded Script ****"). ','*882','STATE','FILE','&quot;fileName&quot;','')
newCmd('',' ','','*v+12.0;','Describes the specified symmetry operation. For example, <b>show symop 3</b> might give "-y,x-y,z-1/3 3-fold screw axis|translation: 0 0 -1/3". If no integer is given, this command delivers the full list of symmetry operations for the current frame.','*891','SYMOP','(integer)')
newCmd('',' ','','*v+12.0;','Describes the symmetry operation relating the two specified points. (If an atom set is given, only the average position of the coordinates is used.) For example, <b>show symop {{molecule=1}} {{molecule=2}}</b> might give "3 -y,x-y,z-1/3 3-fold screw axis|translation: 0 0 -1/3". If more than one symmetry operation relates the positions, then all are described.','*892','SYMOP','._atom_expression_or_coord','._atom_expression_or_coord','')
newCmd('',' ','','*v+11.0;','Displays information relating to the crystallographic symmetry of the model, including space group name and symmetry operations. (Applicable file formats only.)','*895','SYMMETRY','')
newCmd('','11.9.10','','*v+12.0;','Lists all pending {#set(timeout)~timeouts}.','*901','TIMEOUT','')
newCmd('',' ','','*v+11.8;','Reports a trace of function and script calls leading to the current command.','*902','TRACE','')
newCmd('',' ','','','Delivers the current 3x3 transformation matrix (rotation only). ','*903','TRANSFORM','')
newCmd('','11.5.45','','*v+11.6;','Same as <b>show orientation translation</b>','*905','TRANSLATION','')
newCmd('',' ','','*v+11.0;','Displays information relating to the crystallographic unit cell of the model. (Applicable file formats only.)','*91','UNITCELL','')
newCmd('',' ','','*v+11.0;','Opens the data file in a new browser window. (applet only)','*92','URL','')
newCmd('',' ','','*v+11.0;','Opens the specified URL in a new browser window. (applet only)','*93','URL','URL')
newCmd('',' ','','*v+11.8;','Reports a listing of all variables and their values.','*94','VARIABLES','')
newCmd('',' ','','','Delivers the current zoom setting. Output is in the form of the zoom command: "zoom n" where "n" is an integer percent of "normal" zoom.','*95','ZOOM','')
newCmd('',' ','','*v+11.0;','For {#.draw~} objects, displays the command that can be used to generate that object. For {#.isosurface~} objects, displays the {misc/JVXL-format.pdf~JVXL} file equivalent. The $ is required. (Not available for all object types.)','*99','$objectID','')
newCmd('.show (NMR)','13.3.6','','*v+13.4;nmr','Displays a predicted 1H NMR spectrum using JSpecView (Jmol application only). Atoms on the structure are correlated with signals in the spectrum. The spectrum is constructed using a service provided by [http://nmrdb.org~].<br/><img src=img/jsv.png />','0|1','','')
newCmd('.slab','11.1.5','','*v+11.2 adds internal slabbing;slabdepth;disp;misc',' Slab and Depth together control the percentage of the molecule to be displayed based on clipping planes. <b>slab on</b> turns slab/depth on. <b>slab 50</b>  shows the back 50% of the molecule. <b>slab 25</b> show the back 25% of the molecule. <b>depth</b> does the opposite of <b>slab</b>, hiding atoms far from the user. The default settings to see all of the model, then, are <b>slab 100; depth 0</b>.  Atoms appear solid; bonds appear hollow. Slabbing can also be applied "internally" -- that is, based on molecular coordinates, not screen coordinates. Internal slabbing involves defining a plane ax + by + cz + d = 0 as {{a b c d}}, using Miller indices {{h k l}}, or using standard notation such as "x=3" or "xy". CHIME NOTE: The slab/depth effect is equivalent to the RasMol command \'set slabmode solid\', however \'set slabmode [option]\' is not supported.','0|1','','')
newCmd('','','','','Turns slab/depth on or off. Either <b>slab ON</b> or <b>depth ON</b> can be used; in either case both slab and depth are turned on.','*1','._on_off{"ON"}','')
newCmd('',' ','','','Sets the position of the slab/depth plane from 0 (front) to 100 (rear) of the model','*2','._percent_slab','')
newCmd('','11.1.5','','*v+11.2;','Sets the position of the slab/depth plane based on the specified Miller index plane. For slabbing, with positive hkl values, atoms far from {{0 0 0}} are removed. The value NONE removes the slab/depth plane','*32','HKL','{{h k l}} or NONE')
newCmd('','11.1.5','','*v+11.2;','Sets the position of the slab/depth plane based on the specified Miller index plane; reference point is opposite <b>slab hkl</b> or <b>depth hkl</b>.','*33','-HKL','{{h k l}}')
newCmd('','11.1.5','','*v+11.2;','Sets the slab/depth plane based on a plane using the general syntax for {#.plane expressions~}. For example: <b>slab PLANE {{atomno=3}} {{atomno=2}} {{atomno=1}}; slab on</b>. Note that the order of points defines the direction from the plane of atoms to be removed. If using your right hand to define the path from first point to second to third, your right thumb points to the atoms removed by slabbing. (Opposite this for <b>depth</b>.)','*41','PLANE','plane_expression or NONE')
newCmd('','11.1.5','','*v+11.2;','Reverses the sense of the plane defined as above.','*42','-PLANE','plane_expression')
newCmd('','11.1.5','','*v+11.2;','Resets slab and depth to the standard slab 100, depth 0, clears any internal planes, and turns slab on.','*7','RESET','')
newCmd('','11.1.5','','*v+11.2;','Sets the current slab to be internal, so that rotation of the model preserves the current slab from a molecular perspective.','*8','SET','')
newCmd('.spacefill',' ','','*v+11.0 -- adds solvent-accessible surface "+" option;disp','Renders selected atoms as shaded spheres. A boolean value renders the spheres with the van der Waals radius. A decimal value specifies the sphere radius in Angstroms. An integer followed by "%" specifies sphere radius as a percentage of the van der Waals radius. A "+" sign followed by a number specifies to draw the surface that number of Angstroms beyond the van der Waals radius. See also {#.dots~} and {#.isosurface~isosurface sasurface}. Note that the value of the current spacefill setting can be retrieved using the .radius value for the atom. (For example, <b> print {{atomno=3}}.radius</b>.) [Note: Jmol 12.0 changes the default Van der Waals radius to "AUTO" to allow non-PDB files and PDB files with H atoms to load with a slightly different look than PDB files with no H atoms. This brings Jmol\'s default parameter set in line with OpenBabel 2.2 and affects the default rendering also of {#.halos~} and {#.star~stars}.]','0|1','','')
newCmd('',' ','','','Turns spacefill on/off','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns spacefill rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','The default setting, renders atom sizes for non-PDB files or PDB files with H atoms using OpenBabel 2.2 values. ','*21','AUTO','')
newCmd('',' ','','','Specifying a decimal number generates a sphere at a specific radius in Angstroms.  Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*22','[decimal]','')
newCmd('',' ','','','Specifying an integer percent generates a sphere at the specified percentage of the van der Waals radius. This refers to the currently set van der Waals radius set -- Jmol, Babel, Rasmol, or User.','*31','[integer]','%')
newCmd('',' ','','*v+11.6;vdw','Specifying an integer percent generates a sphere at the specified percentage of the van der Waals radius as defined by Jmol 10 constants (OpenBabel 1.0). See {misc/vdw_comparison.xls~vdw_comparison.xls}.','*32','[integer]','%Jmol')
newCmd('',' ','','*v+11.6;vdw','Specifying an integer percent generates a sphere at the specified percentage of the van der Waals radius as defined by OpenBabel 2.2. See {misc/vdw_comparison.xls~vdw_comparison.xls}','*331','[integer]','%Babel')
newCmd('',' ','','*v+12.0;vdw','Specifying an integer percent generates a sphere at the specified percentage of the van der Waals radius as defined by OpenBabel 2.1. See {misc/vdw_comparison.xls~vdw_comparison.xls}','*333','[integer]','%Babel21')
newCmd('',' ','','*v+11.6;vdw','Specifying an integer percent generates a sphere at the specified percentage of the van der Waals radius as defined by Rasmol. See {misc/vdw_comparison.xls~vdw_comparison.xls}','*34','[integer]','%Rasmol')
newCmd('',' ','','*v+11.6;vdw','Specifying an integer percent generates a sphere at the specified percentage of the van der Waals radius as defined by the user using the {#.data~} command.','*35','[integer]','%User')
newCmd('',' ','','','With an explicit plus sign, <b>spacefill</b> draws the surface the given number of angstroms beyond the van der Waals radius. This is the definition of a solvent-accessible surface. Note that <b>spacefill +0</b> is the same as <b>spacefill 100%</b>. More typical would be <b>spacefill +1.2</b>.','*4','+(solvent probe radius)','')
newCmd('',' ','','','Generates a sphere for eah atom according to its crystallographic B-factor. If Uij data has been read from a CIF file, then this value is set to 100 * (U11 * U22 * U33)^0.333','*6','TEMPERATURE','')
newCmd('',' ','','','Generates a sphere for each atom according to an approximation of its ionic radius.','*5','IONIC','')
newCmd('','','','*v+11.6;','Generates a sphere at the radius corresponding to the minimum anisotropic displacement parameter for the selected atoms factored by the given percentage. See also {#.ellipsoid~}.','*7','ADPMIN','n%')
newCmd('','','','*v+11.6;','Generates a sphere at the radius corresponding to the maximum anisotropic displacement parameter for the selected atoms factored by the given percentage. See also {#.ellipsoid~}.','*8','ADPMAX','n%')
newCmd('.spin',' ','','*v+11.0 -- adds molecular frame spin/rotate and fully integrates spin and rotate as one feature;anim;rspin','Starts and stops the molecule spinning around the axis determined by {#.set_spinX~set spinX, set spinY, and set spinZ}, as though the "camera" were changing position. This command was greatly expanded in version 11.0 to include all the possible functions of {#.rotate~}.','0|1','','')
newCmd('',' ','','*v+11.0;','','1','._on_off{"ON"}','')
newCmd('.ssbonds',' ','','bond','Cisteine disulfide bonds can be turned on or off,  colored, and given customized widths in Angstroms.','0|1','','')
newCmd('',' ','','','','*1','._on_off{"ON"}','')
newCmd('',' ','','','','*2','._ssbond_width_angstroms','')
newCmd('',' ','','','','*3','._ssbond_width_rasmol','')
newCmd('.star',' ','','*v+11.6 -- adds new capability;','The <b>star</b> command places a set of crosshairs of a given size in Angstroms on the selected atoms. The default size of the star is, like spacefill, the nominal van der Waals radius for the atom. The default color for the star is that of the atom it is associated with. Use {#.color~color star [colorname]} to then color selected stars the color of your choice. For options, see {#.spacefill~}. ','1|1','','')





newCmd('.step',' ','','*v+11.8 -- NEW ;pause','Within the Jmol application, after a {#.pause~} command, <b>step</b> executes only the next command and then pauses again. To see what the next command is without executing it, enter <b>?</b> as the command. ','0|1','','')
newCmd('.stereo',' ','','*v+11.0 -- adds customizable colors;','Jmol supports two forms of stereo rendering for molecules. In the first form, the two images are placed side by side and rotated so as to appear from slightly different perspectives, creating the illusion of 3D when a practiced user trains one eye on one image and the other eye on the other image. <br /><br /><img src="img/glasses.gif" style="clear:right"  alt="" />A second form of stereo rendering, <b>anaglyphic rendering</b>, nearly superimposes two identical models that are slightly rotated relative to each other. These models are each of a different color (red and one of blue, cyan, or green). The illusion of depth can then be created when the user wears an inexpensive pair of "3D glasses" that have differently colored lenses.  <br /><br />One should experiment with different background colors when using redcyan or redblue stereo rendering. For many users <b>background grey</b> looks better than <b>background white</b> or <b>background black</b>.','1|2','','')
newCmd('',' ','','','Turns side-by-side stereo viewing on. (Note that if this form of stereo viewing is desired, you will probably want to have the applet width twice the applet height.) The default rotation is -5 degrees. Sets the number of degrees of clockwise vertical-axis rotation of the RIGHT-hand image relative to the LEFT-hand image (which itself does not change rotation when stereo viewing is turned on and off). Negative values correspond to cross-eyed viewing, where the left eye is trained on the right image, and the right eye is trained on the left image. Positive values (clockwise rotation) correspond to "wall-eyed" viewing, where the right eye is trained on the right image and the left eye is trained on the left image. Note that <b>stereo 90</b> may be useful, as it shows two views of a model that rotate synchronously, a "front view" on the left and a "right side view" on the right.','*1','._stereo_angle{5}','')
newCmd('',' ','','','Turns side-by-side stereo viewing on with a previously defined rotation or, if no rotation has been defined, a rotation of 5 degrees.','*2','{"ON"}','')
newCmd('',' ','','','Turns stereo viewing off.','*3','OFF','')
newCmd('',' ','','','Turns red/blue anaglyphic rendering on with a specific relative rotation, if desired. The default is 3 degrees.','*4','REDBLUE','._stereo_angle{3}')
newCmd('',' ','','','Turns red/cyan anaglyphic rendering on with a specific relative rotation, if desired. The default is 3 degrees.','*5','REDCYAN','._stereo_angle{3}')
newCmd('',' ','','','Turns red/green anaglyphic rendering on with a specific relative rotation, if desired. The default is 3 degrees.','*6','REDGREEN','._stereo_angle{3}')
newCmd('',' ','','*v+11.0;','Turns custom two-color anaglyphic rendering on with a specific relative rotation, if desired. The default is 3 degrees. The second color is optional. If it is left off, then the second color is the complement of the first. So, for example: <b>stereo red</b> is the same as <b>stereo red cyan</b>. Note that due to the odd way JavaScript designates green (as [x008000] rather than [x00FF00]), <b>stereo red green</b> is NOT the same as <b>stereo redgreen</b>. The equivalent of <b>stereo redgreen</b> is <b>stereo red lime</b>. When experimenting to match a given set of glasses, it is recommended that you use explicit hex codes for the colors:<br /><br /><b>stereo  [xFF0000] [x00FF00]</b><br /><br /> The set of color names used in Jmol is the JavaScript set, given at {http://jmol.sourceforge.net/jscolors/#JavaScript colors~http://jmol.sourceforge.net/jscolors/#JavaScript colors}.','*7','._colorRGB','._colorRGB','._stereo_angle{3}','')
newCmd('.strand',' ','structure','disp','Strands offer a representation the protein backbone or nucleic acid helix using lines. Curvature control points are as described for {#.ribbon~}. ','0|1','','')
newCmd('',' ','','','','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns strand rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','Normally, strands vary in width according to the amino acid atom positions. This command sets the radius of the set of strands to be a constant value (a decimal, in Angstroms).  Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._strand_radius','')
newCmd('.structure',' ','','*v+11.6 -- new;','The <b>structure</b> command allows customized definition of the secondary structure of a protein model as HELIX, SHEET, TURN, or NONE. ','0|1','','')
newCmd('',' ','','','The atom expression is optional and if missing defaults to the currently selected atoms. <b>structure HELIX {{4-10:B}}</b>, for example, sets residues 4-10 on chain B to be represented as a helix. Note that the file must be of PDB format (from a pdb, mmcif, or suitable mol2 file) as well.','*1','HELIX|SHEET|TURN|NONE','(atomExpression)')
newCmd('.struts','11.9.22','','*v+12.0 - new;bond, hbond, measure, struts','In rapid prototyping of protein models, it is sometimes necessary to add short connectors between strands and helixes to provide strength to the plastic model. Jmol 12.0 adds a new shape, STRUT, that creates these supports. These stick-like objects can be added using the {#.connect~CONNECT ... STRUTS} or {#.set picking~SET PICKING STRUTS} commands, and they can be calculated automatically using {#.calculate ~calculate STRUTS}. STRUTS are not measures and they are not covalent bonds. <b>set strutDefaultRadius 0.3</b> sets the default radius for struts. (Their color by default is translucent white.)','0|1','','')
newCmd('',' ','','','Turn struts on or off','*11','._on_off{"ON"}','')
newCmd('',' ','','','Turns struts on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','Show struts with the given cylinder diameter in Angstroms','*2','._radius_angstroms','')
newCmd('',' ','','','Show struts with the given cylinder diameter in Rasmol units (deprecated).','*3','._radius_rasmol','')
newCmd('.subset',' ','','*v+11.0 -- new;dhide','The <b>subset</b> command selects a subset of the atoms that will serve as a basis for all atom expressions until either another subset command is issued or until a new file is loaded. After <b>subset *:C</b>, for example, <b>select *</b> only selects atoms of chain C, <b>restrict none</b> only clears chain C, and <b>restrict *:C</b> does nothing.  By itself, the <b>subset</b> command is the same as <b>subset all</b>. Since all menu-driven commands work through the script processor, realize that any user selections will also be affected by the current subset. ({#.Hover~hover} is still active, though.) "Select All" from the menu, for example, will only select the current subset of atoms. In addition, atoms outside the current subset cannot be clicked on by the user for atom identification or for measurements. Thus, after operations requiring subset have been completed, consider issuing <b>subset all</b> unless you intend to shut off user clicking.','0|1','','')
newCmd('.switch',' ','','*v+12.0 -- new;math;flow','<b>switch</b> and <b>case</b> allow a simplified syntax as an alternative to mulitple {#.if~} statements. The implementation in Jmol is more like the JavaScript equivalent than the Java version. Like JavaScript, CASE values can be expression that involve any variable type rather than just integer constants. Expressions are evaluated from the top down; the DEFAULT keyword may appear along with any other CASE or by itself. As for these other languages, an optional BREAK command is necessary after a case to prevent continuation of the script into the next case. For example: <br /><br />switch(prompt("What sort of rendering would you like?","Spacefill|Wireframe|Ball&amp;Stick",true)) {{<br />case "Spacefill":<br />&nbsp;&nbsp; spacefill only<br />&nbsp;&nbsp;break<br />case "Wireframe":<br />&nbsp;&nbsp; wireframe only<br />&nbsp;&nbsp;break;<br />case "Ball&amp;Stick":<br />&nbsp;&nbsp;wireframe only;wireframe reset;spacefill reset<br />&nbsp;&nbsp;break;<br />}}','0|1','','')
newCmd('.sync','12.2.RC8','','*v+13.0 adds synchronization with JSpecView and socket-based communication (as used for the Molecular Playground);sync','The <b>sync</b> command allows synchronization among applets and applications. [||Applet Synchronization | For applets, an applet can be a driver, a slave, or (like many supervisors) both (a slave driver, GET IT?). Three optional settings are available. With <b>set syncScript ON</b> scripted commands are transmitted; with <b>set syncMouse ON</b> mouse movements are transmitted. These two parameters are independent. If both are off (the default behavior) and sync is ON, then the orientation of the target applet is maintained the same as the driving applet, but script commands are not automatically transmitted.  Details can be seen at {examples-11/sync2.htm~}.||Inter-Application Communication|The <b>sync</b> command with a numerical first parameter allows Jmol to function as a server and/or client over a local internet port. In server mode (specified with a negative port number), Jmol accepts connections and then responds to JSON-encoded messages relating to script commands, mouse actions, touches, or shut-down requests. As a client, Jmol can accept such messages or also send messages to a connected server. Protocol details can be found in {misc/appsync.txt~}.||JSpecView Synchronization|Jmol 13.0 incorporates {http://jspecview.sourceforge.net~JSpecView} into the application and includes special <b>sync</b> capabilities that allow JCAMP-DX files that contain spectra, models, and assignments for IR, NMR, MS, GC, and GC-MS data to be automatically synchronized for user interaction with the data and the model. This capability within the application extends to the applet, in which case a page would load two applets, JmolApplet.jar and JSpecViewApplet.jar. For example, see {http://chemapps.stolaf.edu/jmol/jspecview/test2.htm~http://chemapps.stolaf.edu/jmol/jspecview/test2.htm}. <br /><br /> The format is <b>sync &tilde; "Select: attributes"</b> where <b>attributes</b> are key/value pairs in the form of HTML or XML: key="value". Attributes include <b>file</b>, <b>model</b>, <b>atoms</b>, <b>select</b>, and <b>script</b>. If Jmol does not find a model with ID <b>file#model</b>, it will load that model. <b>atoms</b> is a list of atom numbers separated by commas: 1,2,3, etc., that will be translated into the selection <b>@1 or @2 or @3</b> within Jmol. <b>select</b> can be any valid Jmol selection such as <b>THR</b> or <b>1-30</b>. Jmol automatically adds <b>visible and</b> to the selection created from <b>atom</b> or <b>select</b>. Requires <b>SYNC ON</b>.  See {misc/Jmol-JSpecView-specs.pdf~Jmol-JSpecView-specs.pdf}||]','1|2','','')
newCmd('.sync','12.2.RC8','','*v+12.2 -- application sockets; *v-13.0;sync','The <b>sync</b> command allows synchronization among applets and applications. [||Applet Synchronization | For applets, an applet can be a driver, a slave, or (like many supervisors) both (a slave driver, GET IT?). Three optional settings are available. With <b>set syncScript ON</b> scripted commands are transmitted; with <b>set syncMouse ON</b> mouse movements are transmitted. These two parameters are independent. If both are off (the default behavior) and sync is ON, then the orientation of the target applet is maintained the same as the driving applet, but script commands are not automatically transmitted.  Details can be seen at {examples-11/sync2.htm~}.||Inter-Application Communication|The <b>sync</b> command with a numerical first parameter allows Jmol to function as a server and/or client over a local internet port. In server mode (specified with a negative port number), Jmol accepts connections and then responds to JSON-encoded messages relating to script commands, mouse actions, touches, or shut-down requests. As a client, Jmol can accept such messages or also send messages to a connected server. Protocol details can be found in {misc/appsync.txt~}.||JSpecView Synchronization|Jmol 13.0 incorporates {http://jspecview.sourceforge.net~JSpecView} into the application and includes special <b>sync</b> capabilities that allow JCAMP-DX files that contain spectra, models, and assignments for IR, NMR, MS, GC, and GC-MS data to be automatically synchronized for user interaction with the data and the model. This capability within the application extends to the applet, in which case a page would load two applets, JmolApplet.jar and JSpecViewApplet.jar. For example, see {http://chemapps.stolaf.edu/jmol/jspecview/test2.htm~http://chemapps.stolaf.edu/jmol/jspecview/test2.htm}. <br /><br /> The format is <b>sync &tilde; "Select: attributes"</b> where <b>attributes</b> are key/value pairs in the form of HTML or XML: key="value". Attributes include <b>file</b>, <b>model</b>, <b>atoms</b>, <b>select</b>, and <b>script</b>. If Jmol does not find a model with ID <b>file#model</b>, it will load that model. <b>atoms</b> is a list of atom numbers separated by commas: 1,2,3, etc., that will be translated into the selection <b>@1 or @2 or @3</b> within Jmol. <b>select</b> can be any valid Jmol selection such as <b>THR</b> or <b>1-30</b>. Jmol automatically adds <b>visible and</b> to the selection created from <b>atom</b> or <b>select</b>. Requires <b>SYNC ON</b>.  See {misc/Jmol-JSpecView-specs.pdf~Jmol-JSpecView-specs.pdf}||]','1|2','','')
newCmd('.sync','11.3.9','','*v+11.4 -- new;*v-12.2;','The <b>sync</b> command allows synchronization among applets and for inter-application communication.  An applet can be a driver, a slave, or (like many supervisors) both (a slave driver, GET IT?). Three optional settings are available. With <b>set syncScript ON</b> scripted commands are transmitted; with <b>set syncMouse ON</b> mouse movements are transmitted. These two parameters are independent. If both are off (the default behavior) and sync is ON, then the orientation of the target applet is maintained the same as the driving applet, but script commands are not automatically transmitted.  Details can be seen at {examples-11/sync2.htm~}.','1|2','','')
newCmd('','11.3.9','','*v+11.4;','Turns on synchronization for just this applet (.), all applets except this one (>), all applets (*), a specific applet on this page (appletId), or a specific applet within a specific synchronization set (appletId[syncId]). All applets selected will drive orientations of all others.','*1','&PER;|>|*|appletId|appletId[syncId]','ON')
newCmd('','11.3.9','','*v+11.4;','Same as <b>sync ON</b>, but the selected applets will be set to receive commands only, not send them.','*2','','SLAVE')
newCmd('','11.3.9','','*v+11.4;','Turns off synchronization for the designated applets.','*3','','OFF')
newCmd('','11.3.9','','*v+11.4;','Sends the command (or scripted sequence of commands) to the desigated applets. Quotation marks ARE required.','*4','','&quot;command&quot;')
newCmd('','12.2.RC8','','*v+12.2;','(application only) Indicating a negative port number initializes Jmol as a server over the specified port. Also sets <b>sync on; sync slave</b>.','*5','-(integer)','')
newCmd('','12.2.RC8','','*v+12.2;','(application only)  Acting as a server, sends a command to its own instance (primarily for testing).','*6','-(integer)','command')
newCmd('','12.2.RC8','','*v+12.2;','(application only)  Sends a Jmol command over the specified port, establishing the connection if it does not exist.','*7','(integer)','command')
newCmd('','12.2.RC8','','*v+12.2;','(application only)  Sends a JSON message to a server.  (See {misc/appsync.txt~}.)','*8','(integer)','JSON message')

newCmd('.timeout','11.9.24','','*v+12.0 - new;','Sets a script to be executed after a given time. (Execution can be checked using <b>set debugscript</b>. The commands as they are executed will appear in the {#.console~}.) See also {#show~show timeout}. [Jmol 12.0] ','2|3','','')
newCmd('','','','*v+12.0;','By itself, the <b>timeout</b> command simply displays a list of pending timeouts.','*1','','')
newCmd('','','','*v+12.0;','Turns off all pending timeouts or &lt;no timeouts set&gt;','*2','OFF','')
newCmd('','','','*v+12.0;','Initiates a timeout. A decimal delay time (for example, <b>3.0</b>) is taken to be seconds; integer time indicates milliseconds. If the delay  time is negative, then the timeout will repeat  until a new file is loaded or a ZAP command is given. The ID keyword is optional.','*3','ID &quot;id&quot;','time(s or ms)','&quot;command&quot;','')
newCmd('','','','','Turns off the named timeout if pending. The ID keyword is optional.','*4','ID &quot;id&quot;','OFF')


newCmd('.trace',' ','structure','disp','A "trace" is a smooth hermite curve through the same control points used by {#.ribbons~}.','0|1','','')
newCmd('',' ','','','','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns trace rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','','*2','._trace_radius','')

newCmd('.translate',' ','','*v+11.8 -- new options;anim;camera','Moves the molecule along the specified window-based axis, X, Y, or Z. ','1|2','','')
newCmd('','','','*v+11.8;','Moves the center of rotation to the specified value. A value of 100 will move the molecule completely out of the window. The value represents the percentage of the display window, and 0 is the window center. A value of 50 will move the center of the molecule to the edge of the window. Positive values are to the right of center for X and below center for Y. ','*1','X or Y','(decimal)')
newCmd('','','','*v+11.8;','Moves the center of rotation by the specified percent of the window width (X) or height (Y). ','*2','X or Y','(decimal)','%','')
newCmd('','','','*v+11.8;','Moves the center of rotation by the specified number of angstroms or nanometers at the midpoint depth of the model. . Positive values shift the model to the right for X and down for Y.','*3','X or Y','(decimal)','NM or ANGSTROMS','')
newCmd('','','','*v+11.8;','Adjusts the zoom setting such that the specified percent change in field of view is effected. Positive values give the illusion of moving the model toward the user. Values that would magnify the model to the extent that less than one angstrom spans the entire window are ignored.','*4','Z','(decimal)','%','')
newCmd('','','','*v+11.8;','Adjusts the zoom setting such that the specified number of angstroms or nanometers are removed from the field of view (positive value, magnification) or added to the field of view (negative value, giving the illusion of zooming out) at the midpoint depth of the model, as defined by the dimension from which zoom is referenced (based on <b>set zoomLarge</b> or <b>set zoomHeight</b>). Values that would magnify the model to the extent that less than one angstrom spans the entire window are ignored.','*5','Z','(decimal)','NM or ANGSTROMS','')


newCmd('.translateSelected',' ','','*v+11.0 -- new;anim','Translates a specifed subset of atoms. All of the options for {#.translate~} are available for <b>translateSelected</b>, which then operate only on the currently selected atoms. ','1|2','','')
newCmd('','','','','Moves the indicated atoms along the specified vector, in Angstroms. {#.fractional coordinates~Fractional coordinates} are allowed and are indicated with a "/" anywhere in any of the three coordinates. If the atom set is not included, the currently selected atom set is used. ','*1','{{x y z}}','{{atomSet}}')
newCmd('.try','','','*v+12.0 - new;flow','Surrounding a set of Jmol script with try{{...}} and following it by catch(e){{...}} allows the catching of errors within larger blocks of script. The action is the same as in JavaScript. For example, in the following, the first model is never lost, and no error message is printed in the applet window.  Any variables created with the VAR keyword are local to the try or catch block.<br /><br />load quartz.cif<br />refresh<br />while(true) {{ <br />&nbsp;&nbsp; try {{<br />&nbsp;&nbsp;&nbsp;&nbsp;  load files "caffeine.xyz" "1d.pdb"<br />&nbsp;&nbsp; }} catch(e) {{<br />&nbsp;&nbsp;&nbsp;&nbsp;  prompt @{{"oops -- " + e}}<br />&nbsp;&nbsp;&nbsp;&nbsp; break;<br /> &nbsp;&nbsp;}}<br />&nbsp;&nbsp; prompt "File loaded successfully"<br />&nbsp;&nbsp; break;<br />}}','1|2','','')
newCmd('.catch','','','*v+12.0 - new;flow','See {#.try~}.','1|2','','')


newCmd('.unbind',' ','','*v+12.0;bind','Removes the tie between a mouse action and a Jmol action or user script.','1|2','','')
newCmd('',' ','','*v+12.0;','By itself, unbind returns Jmol to its default mouse action binding.','*1','','')
newCmd('','','','*v+12.0;','Removes all bindings to the specified mouse action.','*2','._mouse_action','')
newCmd('','','','*v+12.0;','Removes all bindings to the specified jmol action','*3','._jmol_action','')
newCmd('','','','*v+12.0;','Removes all bindings to the specified user script','*4','&quot;script&quot;','')
newCmd('','','','*v+12.0;','Removes the specified mouse action for the specified Jmol action.','*5','._mouse_action','._jmol_action')
newCmd('','','','*v+12.0;','Removes the specified mouse action for the specified script.','*6','._mouse_action','&quot;script&quot;')
newCmd('.undo','12.2.RC6','','*v+12.2 -- new;','(application only) same as the Undo console button -- reloads the previous state','1|2','','')
newCmd('.undoMove','12.2.RC6','','*v+12.2 -- new;','same as CTRL-Z -- undoes changes to atom position when in modelKit mode.','1|2','','')

newCmd('.unitcell',' ','','*v+11.0 -- new;axes','Turns on or off the unit cell for crystal structures, and determines its line style and line width (as a decimal number, in Angstroms).','1|2','','')
newCmd('',' ','','*v+11.0;','Turns the unit cell on or off','*1','._on_off{"ON"}','')
newCmd('',' ','','*v+11.0;','Sets the unit cell line diameter in Angstroms. ','*21','(decimal)','')
newCmd('','13.1.19','','*v+13.2;','Sets the unit cell to be centered on a given atom or coordinate. The coordinate is assumed to be in fractional coordinates already, for example {{0.5 0.5 0.5}}.','*22','CENTER','._atom_expression_or_coord')
newCmd('',' ','','*v+11.0;','Sets the axes style to a thin dotted line.','*3','DOTTED','')
newCmd('',' ','','*v+11.0;','Sets the origin of the unit cell to be the specified coordinate (with braces). No matter how the coorinate is written (with or without "/"), it is interpreted as a fractional coordinate. For example, <b>unitcell {{0.5 0.5 0.5}}</b> sets the origin of the unit cell to {{1/2, 1/2, 1/2}}. This change is for display purposes only -- the actual "{{0 0 0}}" point remains where it was initially.','*4','{{i j k}}','')
newCmd('','12.2.RC6','','*v+12.2;','Creates a block of unit cells, possibly scaled, using the "555" cell naming system. A negative scale, such as -0.5, fills the  specified range of unit cells with unit cells of the specified size (allowing, for instance, filling of a set of supercells with unit cells based on a primitive smaller unit cell). Currently only supports the same fractional scale in each direction. ','*5','{{nnn yyy scale}}','')
newCmd('','11.9.12','','*v+12.0;ticks','Sets the parameters for ticks along the a, b, and c edges of the unit cell. An optional specific axis (X, Y, or Z) can be indicated. There are three levels of ticks - major, minor, and "subminor." Only the major ticks have labels. Which of these tick levels are displayed and the distance between ticks depends upon the parameter that takes the form of a point. This point may be in fractional form, {{1/2 0 0}}. The optional keyword FORMAT allows formating of the labels for the major ticks. These are based on an array of strings given after the FORMAT keyword. If the array is shorter than the number of ticks, the formats in the array are repeated. ','*63','TICKS X|Y|Z','{{major,minor,subminor}} FORMAT ["%0.2f", ...] ')


newCmd('.unset','12.1.32','','*v+12.2;rreset','Same as {#.reset~}; just looks better for variables. Good to use if just a small part of a large array or hashtable has been assigned to another variable and  the rest can be discarded.','1|2','','')
newCmd('.var',' ','','flow','The <b>var</b> keyword is used with variable definitions to localize them to a specific script file or {#.functions~function}. Starting with Jmol 12.0, <b>var</b> variables within {#.for~}, {#.while~}, {#.try~}, and {#.try~catch} blocks or any code surrounded by {{ }} except  {#.if~} and {#.switch~} blocks.','1|2','','')
newCmd('.vector',' ','','disp','Draws vectors arising from vibrational mode data. Operates on whichever atoms have been selected. See also {#.color (atom object)~}','1|2','','')
newCmd('',' ','','','Turns vibration vector display on and off','*1','._on_off{"ON"}','')
newCmd('',' ','','','Sets the diameter of the vector','*2','._diameter_pixels','')
newCmd('',' ','','','Sets the diameter of the vector','*3','._radius_angstroms','')
newCmd('',' ','','','Adjusts the scale of the vector independently of the vibration motion.','*4','SCALE','._vector_scale')
newCmd('.vibration',' ','','*v+11.0 adds vibration period command;vib','Provided information is present in the file (xyz format with columns 6-8 indicating dx, dy, and dz, or Gaussian harmonic frequency output), turns on and off display of vibration animation and allows setting of the time period for the vibration (in seconds) and the scale of the motion relative to the default (1). ','1|2','','')
newCmd('',' ','','','Turns vibration on with a 1-second time period or turns vibration off.','*1','._on_off{"ON"}','')
newCmd('',' ','','','Sets the time period for one full vibration in seconds and turns vibration on. ','*2','._time_period','')
newCmd('',' ','','*v+11.0;','Sets the time period for one full vibration in seconds, but does not turn vibration on','*3','PERIOD','._time_period')
newCmd('',' ','','','Sets the scale of the vibration motion independently of the vector length.','*4','SCALE','._vibration_scale')
newCmd('.while','','','*v+11.4 -- new flow control options;math;flow','<b>while</b>, like {#.if~} evaluates a logical expression, looping until the expression is no longer TRUE:<br /><br />while (x > 0 && x < 10) {{<br />&nbsp;&nbsp;print x<br />&nbsp;&nbsp;x = x - 2<br />}}<br /><br />The looping can be discontinued using either {#.break~} or {#.break~continue}. Any variables created with the VAR keyword are local to the <b>while</b> block.','1|2','','')

newCmd('.wireframe',' ','','disp;bond','Wireframe refers to the bonds drawn between the atoms. A boolean value of ON draws the selected bonds as lines. Alternatively, a numeric value may be used to specify the radius of the bonds. A decimal value such as 0.2 or 0.4 specifies Angstroms. The <b>wireframe</b> command operates on bonds either BETWEEN ANY TWO atoms in a previously selected atom set (having previously {#.setbondstyles~set bondmode AND}) or TO ANY atoms in a previously selected atom set (having previously {#.setbondstyles~set bondmode OR}). Note that the selected atoms must already be connected (based on information in the file, Jmol\'s autobonding algorithm, or from use of the {#.connect~} or {#.bondorder~} command) in order to show any effect. CHIME NOTE: Rasmol and Chime will take <b>wireframe 0</b> as a one-pixel-width bond (equal to <b>wireframe</b> and hence to <b>wireframe on</b>), while Jmol will interpret it as <b>wireframe off</b>.','0|1','','')
newCmd('',' ','','','Turn wireframe on or off','*11','._on_off{"ON"}','')
newCmd('',' ','','*v+11.6;','Turns wireframe rendering on and all other rendering (including labels) off.','*12','ONLY','')
newCmd('',' ','','','Show wireframe with the given cylinder diameter in Angstroms.  Starting with Jmol 12.0, a negative number also implies <b>ONLY</b>.','*2','._radius_angstroms','')
newCmd('',' ','','','Show wireframe with the given cylinder diameter in Rasmol units (deprecated).','*3','._radius_rasmol','')


newCmd('.write',' ','','*v+13.0 adds fully encapsulated PNG files ("PNGJ" format);clip','(application or signed applet only). Writes information to a file or to the system clipboard (or, starting in Jmol 13.0, POSTing to a server using a URL "http://..."  for the file name. Options include: an image of the application screen, the script history, the current model state in the form of a script, the current molecular orbital in the form of a JVXL file, or the current isosurface in the form of a JVXL file. If the filename is simple (no spaces) and the type is clearly in the extension as, for example, <b>write test.spt</b>, then Jmol will deduce the type from the filename and create the proper file. For the clipboard, simply specify CLIPBOARD instead of a file name. For files created with zip formats (PNGJ, JMOL, ZIP), the command <b>show FILE "&lt;filename&gt;"<b> will list the files contained in the ZIP files.','0|1','','')
newCmd('.write',' ','','*v+12.0 adds LOCALPATH and REMOTEPATH options for write STATE;*v-13.0;clip','(application or signed applet only). Writes information to a file or to the system clipboard (or, starting in Jmol 13.0, POSTing to a server using a URL "http://..."  for the file name. Options include: an image of the application screen, the script history, the current model state in the form of a script, the current molecular orbital in the form of a JVXL file, or the current isosurface in the form of a JVXL file. If the filename is simple (no spaces) and the type is clearly in the extension as, for example, <b>write test.spt</b>, then Jmol will deduce the type from the filename and create the proper file. For the clipboard, simply specify CLIPBOARD instead of a file name. For files created with zip formats (PNGJ, JMOL, ZIP), the command <b>show FILE "&lt;filename&gt;"<b> will list the files contained in the ZIP files.','0|1','','')
newCmd('.write (export)',' ','','*v+11.8 adds IDTF, VRML, and X3D options;','Jmol models can be exported to several formats that can be read by external rendering programs. ','0|1','','')

newCmd('','11.8.RC5','','*v+11.8;','Exports the current scene as an [http://u3d.svn.sourceforge.net/viewvc/u3d/trunk/Docs/IntermediateFormat/IDTF%20Format%20Description.pdf~Intermediate File] that can be easily converted to a [http://en.wikipedia.org/wiki/Universal_3D~Universal 3D] file for incorporation into 3D-PDF documents. The conversion to U3D is done external to Jmol using the (Windows-only) program [misc/idtf.zip~]. (This is a copy from the Bin/Win32/Release directory found in the zip file downloadable from [http://sourceforge.net/projects/u3d/]; source code for that C++ program is available at this site as well.) Once Jmol has created the IDTF file using, for example, <b>write IDTF "myfile.idtf"</b>, simply execute from a Windows command window <b>IDTFConverter.exe -input myfile.idtf -output myfile.u3d</b>. The U3D file created will be highly compressed and FAR smaller than that created using other means (such as first creating a VRML file and then converting that to the U3D format). ','*1','IDTF','&quot;fileName&quot;')
newCmd('',' ','','*v+11.4;','Exports the current scene as a Maya file. (Basic elements such as atoms and bonds only.)','*2','MAYA','&quot;fileName&quot;')
newCmd('','','','*v+12.2;','Exports the current scene as a WaveFront OBJ file. The file "fileName" is written along with the additional material file, "fileName.mtl". (Jmol 12.2)','*34','OBJ','&quot;fileName&quot;')
newCmd('','','','*v+11.8;','Exports the current scene as POVRAY. The file "fileName" is written along with the additional file, "fileName.ini". (Typically "fileName" would include the ".pov" extension, so these would be of the form xxx.pov and xxx.pov.ini.) Note that the .pov file will embed the Jmol script, so the model can be loaded back into Jmol using [#.script~script "xxx.pov"].','*35','POVRAY','&quot;fileName&quot;')
newCmd('','12.1.26','','*v+12.2;','Creates a {http://www.symyx.com/downloads/public/ctfile/ctfile.pdf~V3000} file.','*41','V3000','&quot;fileName&quot;')
newCmd('',' ','','*v+11.4;','Exports the current scene as VRML.','*42','VRML','&quot;fileName&quot;')
newCmd('','11.8.RC4','','*v+11.8;','Exports the current scene as X3D.','*5','X3D','&quot;fileName&quot;')
newCmd('.write (images and frames)','','','*v+11.0;image','Jmol can write a 2D image of the model and, for JPG and PNG format, can append its state so that a single file can be read either as an image or as a Jmol state. In addition, You can write sequences of frames as a set of JPG files that can then be incorporated by other programs into movie files. Image quality is determined by {.set_antialiasImages~set antialiasImages}, which is distinct from {.set_antialiasDisplay~set antialiasDisplay}. ','0|1','','')
newCmd('','11.7.1','writeframes','*v+11.8;','Allows making of Jmol movies, creating a set of JPG files filename0001.jpg, filename0002.jpg, etc. for all frames in the atom expression. These files can then be combined using an external program of user\'s choice to create a movie from a sequence of JPEG images. Width and height are optional, as is the atom expression.','*111','FRAMES','.atom_expression','width','height &quot;fileName.jpg&quot;')
newCmd('','12.3.28','writeframes','*v+12.4;','Creates a sequence of JPG file frames that can be combined with movie-making software into a movie. The number of vibration cycles, n, determines the number of frames (individual JPG files) that will be created (20 * n).','*112','VIBRATION','n','&quot;fileName&quot;','')
newCmd('',' ','','*v+11.0;','Creates a "snapshot" image of the current display and saves it to disk as a JPG image. <b>width</b> and <b>height</b> are optional and default to the current frame size. The optional integer n is a quality from 1 to 100 (default 100; or 75 prior to Jmol 13.1.15). If <b>set imageState</b> is TRUE (the default setting), then Jmol inserts into the JPG file its state, allowing a single image file to be used both for viewing as a 2D image and reading as a 3D Jmol state using the [#.script~] command or by dragging and dropping in the Jmol application.','*12','IMAGE width height','JPG','n','&quot;fileName&quot;')
newCmd('',' ','','*v+11.0;','Creates a "snapshot" image of the current display and saves it to disk as a JPG or base-64-encoded JPG image. <b>width</b> and <b>height</b> are optional and default to the current frame size. The optional integer n is a quality from 1 to 100 (default 75).','*14','IMAGE width height','JPG64','n','&quot;fileName&quot;')
newCmd('',' ','','*v+11.0;','Creates a "snapshot" image of the current display in PNG  format and saves it to disk. <b>width</b> and <b>height</b> are optional and default to the current frame size. The amount of compression, <b>n</b>, is a number between 0 and 10 (default 2).  If <b>set imageState</b> is TRUE (the default setting), then Jmol appends to the PNG file its state, allowing a single image file to be used both for viewing as a 2D image and reading as a 3D Jmol state using the [#.script~] command or by dragging and dropping in the Jmol application. For this drag/drop operation to be successful, any local files necessary (model, or isosurface, for example) must be in the directory containing the PNG file, and any remote files must still be accessible via a web connection. See also the PNGJ format, below.','*15','IMAGE width height','PNG','n','&quot;fileName&quot;')
newCmd('','12.3.7','','*v+13.0;','Creates a "snapshot" image of the current display in PNG  format, appending to that file the full state and all necessary files (models, isosurfaces, etc.) required to reproduce it in the form of JMOL zip data format. <b>width</b> and <b>height</b> are optional and default to the current frame size. (Jmol 12.3.7)','*16','IMAGE width height','PNGJ','n','&quot;fileName&quot;')
newCmd('','12.2.RC6','','*v+12.2;','Creates a "snapshot" image of the current display in PNG  format, setting the background to be transparent. It is recommended using an unusual background color, such as [x101010], to ensure only the background is made transparent. Also, <b>set antialiasImages FALSE</b> is recommended to avoid ragged edges around the image. <b>width</b> and <b>height</b> are optional and default to the current frame size. The amount of compression, <b>n</b> is a number between 0 and 10 (default 2). ','*17','IMAGE width height','PNGT','n','&quot;fileName&quot;')
newCmd('',' ','','*v+11.0;','Creates a "snapshot" image of the current display in {http://netpbm.sourceforge.net/doc/ppm.html~PPM} format and saves it to disk. <b>width</b> and <b>height</b> are optional and default to the current frame size. ','*18','IMAGE width height','PPM','&quot;fileName&quot;','')
newCmd('.write (info)',' ','','*v+11.0;','The Jmol history, menu, currently defined functions, and variables may all be written to files. In addition, specific properities, quaternion analysis, and ramachandran information can also be saved in PDB format.','0|1','','')
newCmd('','11.3.23','','*v+11.4;','Writes all user-defined functions to a file. ','*12','FUNCTIONS','&quot;fileName&quot;')
newCmd('',' ','','*v+11.0;','Saves the script command {#.history~} to a file.','*20','HISTORY','&quot;fileName&quot;')
newCmd('','11.9.4','','*v+12.0;','Same as ZIPALL (see below). See also <b>write PNGJ</b>, above.','*3','JMOL','&quot;fileName&quot;')
newCmd('',' ','','*v+11.8;','Saves the current Jmol popup menu as a file. See {#.load MENU~}.','*4','MENU','&quot;fileName&quot;')
newCmd('',' ','','*v+12.0;','Writes a file that is of PDB format and contains the quaternion representation of the rotational frame of individual amino acid residues in a protein or base pairs in a nucleic acid. See the {#.plot~plot QUATERNION} command for parameter options.','*5','QUATERNION','. . . . ','&quot;fileName&quot;','')
newCmd('',' ','','*v+12.0;','Writes a file that is of PDB format and contains the specified parameters in the X, Y, and Z fields, possibly scaled, with the scaling indicated in a REMARK record. See the {#.plot~plot PROPERTIES} command for parameter options.','*7','PROPERTIES','. . . . ','&quot;fileName&quot;','')
newCmd('',' ','','*v+12.0;','Creates a file in PDB format for the alpha carbons of a peptide, where the x, y, and z axis values are actually the PHI, PSI, and OMEGA values for each amino acid. An additional optional keyword DRAW writes a script file that would draw PHI and PSI angles on the structure. See the {#.plot~plot RAMACHANDRAN} command for parameter options.','*8','RAMACHANDRAN','. . . . ','&quot;fileName&quot;','')
newCmd('','13.1.2','','*v+13.2;','Writes the specified data directly to a file.','*21','INLINE','&quot;data&quot;','&quot;fileName&quot;','')
newCmd('','11.1.14','','*v+11.2;','Saves the value of a variable to a file.','*9','VAR','[variable name]','&quot;fileName&quot;','')
newCmd('.write (model)','','','*v+11.2;','There are several options for writing model data to a file or saving the state of Jmol so that a model can be reloaded. ','0|1','','')
newCmd('','11.1.13','','*v+11.2;','Writes molecular data to a file determined by the extension (XYZ, XYZRN, XYZVIB, MOL, PDB, SDF, SPT, V2000). The SDF option creates a multi-model {http://www.symyx.com/downloads/public/ctfile/ctfile.pdf~SDfile}. XYZRN, added in Jmol 12.2, creates files that include van der Waals radius and name, for input to the {http://www.scripps.edu/~sanner/html/msms_home.html~MSMS} program. XYZVIB, added in Jmol 12.2, creates a multi-model XYZ file with vibrational vectors added for all atoms having defined vibration vectors. the type keyword may be omitted if the filename has an letter extension matching one of these types. Only the currently selected atoms are saved. The CD (ChemDoodle) format is a JSON format introduced in Jmol 13.0.','*1',' CD|MOL|PDB|SDF|V2000|XYZ|XYZRN|XYZVIB','&quot;fileName&quot;')
newCmd('','11.1.13','','*v+11.2;','Writes a script file explicitly containing the coordinates of the previously selected atoms, whether or not they have been altered using  {#.invertSelected~}, {#.rotateSelected~}, or {#.translateSelected~}. The SPT keyword is optional if the file ends with the extension <b>.spt<b>.','*21','COORDS','SPT','&quot;fileName&quot;','')
newCmd('','11.3.39','','*v+11.4;','Writes the most recent file loaded to the indicated filename. If the model was loaded using a manifest from a ZIP file, then the entire ZIP file is written.','*3','FILE','&quot;fileName&quot;')
newCmd('','11.9.4','','*v+12.0;','Same as ZIPALL (see below). See also <b>write PNGJ</b>, above. The JMOL keyword is optional if the file ends with the extension <b>.jmol<b>.','*4','JMOL','&quot;fileName&quot;')

newCmd('',' ','','*v+11.0;','Saves the current state to a script file. The STATE keyword is optional if the file ends with the extension <b>.spt<b>.','*5','STATE','&quot;fileName&quot;')
newCmd('','11.9.2','','*v+12.0;','Saves the current state to a script file, stripping all local file references to the indicated relative path. (Jmol 12.0)','*6','STATE','LOCALPATH','&quot;path&quot;','&quot;fileName&quot;')
newCmd('','11.9.2','','*v+12.0;','Saves the current state to a script file, stripping all remote file references to the indicated relative path. ','*7','STATE','REMOTEPATH','&quot;path&quot;','&quot;fileName&quot;')
newCmd('','11.9.4','','*v+12.0;','Saves the current state and all necessary LOCAL files to a ZIP file with the given name.   The ZIP keyword is optional if the file ends with the extension <b>.zip<b>.','*8','ZIP','&quot;fileName&quot;')
newCmd('','11.9.4','','*v+12.0;','Saves the current state and all necessary files -- local or remote -- to a ZIP file with the given name.   The ZIPALL keyword is optional if the file ends with the extension <b>.jmol<b>.','*9','ZIPALL','&quot;fileName&quot;')
newCmd('.write (object)',' ','','*v+11.2;','Several Jmol objects can be saved independently.','0|1','','')
newCmd('',' ','','*v+11.2;','Saves the current isosurface in the form of a JVXL file.','*1','ISOSURFACE','&quot;fileName&quot;')
newCmd('','11.9.23','','*v+12.0;','Generates a relatively compact JVXL XML "vertex-only" mesh surface file from an isosurface. A typical command, after creation of an isosurface, would be <b>write MESH t.jvxl</b>. Note that standard JVXL files are considerably smaller, however.','*2','MESH','&quot;fileName&quot;')
newCmd('',' ','','*v+11.2;','Saves the current molecular orbital in the form of a JVXL file.','*3','MO','&quot;fileName&quot;')
newCmd('',' ','','*v+11.6;pg','Writes a tabular summary of the point group for a symmetrical or nearly symmetrical molecule. ','*4','POINTGROUP','&quot;fileName&quot;')
newCmd('',' ','','*v+11.6;pg','Creates a script file containing DRAW commands that will draw the planes and axes of symmetry for a molecule. ','*5','POINTGROUP','DRAW','&quot;fileName&quot;','')










newCmd('.zap',' ','','*v+11.6 -- selective model removal;;load;rreset','Clears the currently loaded molecule. After <b>zap</b> or before any model is loaded, all of the following commands are active and may be used to create nonmolecular visualizations: {#.axes~},  {#.background~},  {#.boundbox~},  {#.center~}*,  {#.centerAt~},  {#.dipole~}*,  {#.draw~}*,  {#.echo~},  {#.font~},  {#.frank~},  {#.isosurface~}*,  {#.move~},  {#.moveto~},  {#.pmesh~},  {#.restore~},  {#.rotate~}*,  {#.save~},  {#.slab~},  {#.spin~}*,  {#.stereo~},  {#.translate~}, and related {#.set~} commands.  <br/><br/>* indicates only aspects not requiring selection of atoms for these commands. The ZAP command can be used to delete selected models.','1','','')
newCmd('',' ','','*v+11.6;','Clears all loaded models.','*1','','')
newCmd('',' ','','*v+11.6;','Clears all models that include any of the selected atoms. Examples include: <b>zap file=1</b>, <b>zap model=1.1</b>, <b>zap atomno=1</b>, <b>zap not visible</b>','*2','(atom expression)','')




newCmd('.zoom',' ','','*v+11.0 -- new capabilities;anim',' Allows enlarging or shrinking of the displayed model, as though the "camera" were moving toward or away from the model. A percentage value specifies the zoom relative to 100, the default value, which in Jmol is calculated so that all atoms are completely visible on the screen through all rotations using the default vanderWaals rendering percentage. The command "zoom OFF" disables mouse-based zooming and zooms to 100. The command "zoom ON" re-enables zooming at the current zoom setting. If the zoom has been turned off, setting the zoom using, for example, "zoom 50," though it sets the "current zoom setting," has no effect until the next "zoom ON" command is given. If an atom expression or (Jmol 13.0) isosurface ID is given, then zoom also sets the center of that object as the rotation center, and if windowCentered is true (the default), that point is moved the center of the screen. ','*1','._on_off{"ON"}','')
newCmd('','','','*v+12.2;','Zooms to the setting that fills the screen with the currently displayed atoms.','*21','0','')
newCmd('','','','*v+12.0;','Zooms IN by a factor of 2 over the course of 1 second.','*15','IN','')
newCmd('','','','*v+12.0;','Zooms OUT by a factor of 2 over the course of 1 second.','*16','OUT','')
newCmd('',' ','','','Zooms to the specified percent zoom','*20','._percent_zoom','')
newCmd('',' ','','*v+11.0;','Sets a new zoom setting and optionally also designates the specified reference atom expression or coordinate or (Jmol 13.0) isosurface as the rotation center. If a reference atom set or isosurface is given, the value is relative to that group of atoms or isosurface.','*3','optional {{atom expression}} or {{x y z}} or $xxx','percent zoom or 0')
newCmd('',' ','','*v+11.0;','Adds or subtracts an absolute amount from the current zoom setting and optionally sets the rotation center.','*4','optional {{atom expression}} or {{x y z}} or $xxx',' + or - delta')
newCmd('',' ','','*v+11.0;','Multiplies or divides the current zoom setting by the indicated factor and optionally sets the rotation center.','*5','optional {{atom expression}} or {{x y z}} or $xxx',' * or / factor')
newCmd('.zoomto',' ','','*v+11.0 -- new;anim;nav','Carries out a smooth transition to the specified zoom setting. Indicating a new rotation center is optional. All capabilities of {#.zoom~} are supported. Jmol 13.0 adds centering on isosurfaces using $xxx.','1|0','','')
newCmd('',' ','','*v+11.0;','By itself, <b>zoomTo</b> smoothly zooms IN by a factor of 2 over the course of 1 second.','*1','','')
newCmd('','','','*v+12.0;','Smoothly zooms IN by a factor of 2 over the course of 1 second.','*21','IN','')
newCmd('','','','*v+11.0;','Smoothly zooms OUT by a factor of 2 over the course of 1 second.','*22','OUT','')
newCmd('','','','*v+11.0;','Smoothly moves the specified atom or coordinate to the center of the window if windowCentered or designates it as the center of rotation if not windowCentered. xTrans and yTrans parameters specify percent translation within the window along X and Y, respectively (Jmol 12.0). All parameters are options. The default time is 1 second; indicating no center position results in simple, smooth zooming.','*3','._floattime','{{atom expression}} or {{x y z}} or $xxx')
newCmd('','','','*v+11.0;','Smoothly transitions to the indicated zoom setting over the course of the specified number of seconds. Optional xTrans and yTrans parameters specify percent translation within the window along X and Y, respectively (Jmol 12.0)','*4','._floattime','{{atom expression}} or {{x y z}} or $xxx','._percent_zoom','xTrans yTrans')
newCmd('','','','*v+11.0;','Adds or subtracts an absolute amount from the current zoom setting over the course of the specified number of seconds. Optional xTrans and yTrans parameters specify percent translation within the window along X and Y, respectively (Jmol 12.0)','*5','._floattime','{{atom expression}} or {{x y z}} or $xxx',' + or - delta','xTrans yTrans')
newCmd('','','','*v+11.0;','Multiplies or divides the current zoom setting by the indicated factor over the course of the specified number of seconds. Optional xTrans and yTrans parameters specify percent translation within the window along X and Y, respectively (Jmol 12.0)','*6','._floattime','{{atom expression}} or {{x y z}} or $xxx',' * or / factor','xTrans yTrans')
newCmd('','11.1.36','','*v+11.2;','Zooms to the setting that fills the screen with the designated atoms. (Requires <b>set perspectiveModel 11</b> for proper operation.) Can include modifiers +n, -n, *n, /n after the 0. Optional xTrans and yTrans parameters specify percent translation within the window along X and Y, respectively (Jmol 12.0)','*7','._floattime','{{atom expression}} or {{x y z}} or $xxx','0','xTrans yTrans')






























































































































































































































































































